{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import dendropy\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/adrienne/Projects/LanguageofLife/GetGTDBGenomes_old\n"
     ]
    }
   ],
   "source": [
    "cd ../GetGTDBGenomes_old"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "bac = dendropy.Tree.get_from_path('bac120_r89.tree',\n",
    "                                           schema='newick',\n",
    "                                           rooting='force-rooted',\n",
    "                                           preserve_underscores=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "arc = dendropy.Tree.get_from_path('ar122_r89.tree',\n",
    "                                           schema='newick',\n",
    "                                           rooting='force-rooted',\n",
    "                                           preserve_underscores=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23458\n",
      "1248\n",
      "24706\n"
     ]
    }
   ],
   "source": [
    "print(len(bac.taxon_namespace))\n",
    "print(len(arc.taxon_namespace))\n",
    "print(len(bac.taxon_namespace)+len(arc.taxon_namespace))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.19004177679256543\n",
      "0.1987179487179487\n",
      "4706\n"
     ]
    }
   ],
   "source": [
    "print(4458/23458)\n",
    "print(248/1248)\n",
    "#I want ~4458 bacterial genomes and ~248 archaeal genomes to be in the valid/test set\n",
    "#~2229 bac and ~124 arc in each of valid and test\n",
    "#that will give me a little over 2k genomes in each set and will be ~80/10/10 split\n",
    "print(4458+248)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[<Node object at 0x11025b748: '100.0' (None)>,\n",
       " <Node object at 0x112fb69b0: '100.0:p__CG03; c__CG03; o__CG03; f__CG03' (None)>]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bac.seed_node.child_nodes()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def count_children(node,count=0):\n",
    "    for child in node.child_nodes():\n",
    "        if child.taxon:\n",
    "            count += 1\n",
    "        count = count_children(child, count=count)\n",
    "    return count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "23458"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_children(bac.seed_node)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_branch_children_counts(node,max_count,min_count):\n",
    "    for child in node.child_nodes():\n",
    "        count=count_children(child,count=0)\n",
    "        percent = count/23458.0\n",
    "        if count < max_count and count > min_count:\n",
    "            print(child,'# children taxa:',count,'; percent=',percent,'; level',child.level())\n",
    "        get_branch_children_counts(child,max_count,min_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Node object at 0x10fe40fd0: '18.0' (None)> # children taxa: 4547 ; percent= 0.19383579162758974 ; level 14\n",
      "<Node object at 0x10fe51080: '14.0' (None)> # children taxa: 2615 ; percent= 0.11147582914144429 ; level 15\n",
      "<Node object at 0x10fe510f0: '8.0' (None)> # children taxa: 2595 ; percent= 0.11062324153806803 ; level 16\n",
      "<Node object at 0x10fe51160: '82.0:p__Firmicutes_A' (None)> # children taxa: 1886 ; percent= 0.08039901099838008 ; level 17\n",
      "<Node object at 0x10fe511d0: '84.0' (None)> # children taxa: 1876 ; percent= 0.07997271719669197 ; level 18\n",
      "<Node object at 0x10fe51240: '64.0' (None)> # children taxa: 1848 ; percent= 0.07877909455196522 ; level 19\n",
      "<Node object at 0x10fe512b0: '32.0:c__Clostridia' (None)> # children taxa: 1836 ; percent= 0.07826754198993946 ; level 20\n",
      "<Node object at 0x10fe51320: '50.0' (None)> # children taxa: 1409 ; percent= 0.06006479665785659 ; level 21\n",
      "<Node object at 0x10fe51390: '26.0' (None)> # children taxa: 1368 ; percent= 0.05831699207093529 ; level 22\n",
      "<Node object at 0x10fe51400: '37.0' (None)> # children taxa: 706 ; percent= 0.030096342399181516 ; level 23\n",
      "<Node object at 0x10fe51470: '21.0' (None)> # children taxa: 635 ; percent= 0.02706965640719584 ; level 24\n",
      "<Node object at 0x10fe514e0: '100.0:o__Oscillospirales' (None)> # children taxa: 511 ; percent= 0.02178361326626311 ; level 25\n",
      "<Node object at 0x10fe51550: '51.0' (None)> # children taxa: 509 ; percent= 0.021698354505925485 ; level 26\n",
      "<Node object at 0x10ff5fd30: '100.0:o__Lachnospirales' (None)> # children taxa: 662 ; percent= 0.02822064967175377 ; level 23\n",
      "<Node object at 0x10ff5fda0: '81.0' (None)> # children taxa: 659 ; percent= 0.028092761531247335 ; level 24\n",
      "<Node object at 0x10ff5fe10: '98.0' (None)> # children taxa: 619 ; percent= 0.026387586324494842 ; level 25\n",
      "<Node object at 0x10ff5fe80: '100.0:f__Lachnospiraceae' (None)> # children taxa: 615 ; percent= 0.026217068803819594 ; level 26\n",
      "<Node object at 0x10ff5fef0: '12.0' (None)> # children taxa: 612 ; percent= 0.026089180663313154 ; level 27\n",
      "<Node object at 0x10ff5ff60: '4.0' (None)> # children taxa: 604 ; percent= 0.025748145621962655 ; level 28\n",
      "<Node object at 0x10ff5ffd0: '3.0' (None)> # children taxa: 573 ; percent= 0.024426634836729473 ; level 29\n",
      "<Node object at 0x10ff62080: '14.0' (None)> # children taxa: 520 ; percent= 0.02216727768778242 ; level 30\n",
      "<Node object at 0x110105e48: '5.0' (None)> # children taxa: 709 ; percent= 0.030224230539687952 ; level 17\n",
      "<Node object at 0x11020bc50: '16.0' (None)> # children taxa: 1932 ; percent= 0.08235996248614545 ; level 15\n",
      "<Node object at 0x11020bcc0: '60.0:p__Firmicutes; c__Bacilli' (None)> # children taxa: 1878 ; percent= 0.08005797595702958 ; level 16\n",
      "<Node object at 0x11020bd30: '57.0' (None)> # children taxa: 1451 ; percent= 0.06185523062494672 ; level 17\n",
      "<Node object at 0x11020bda0: '38.0' (None)> # children taxa: 1351 ; percent= 0.057592292608065476 ; level 18\n",
      "<Node object at 0x11020be10: '92.0:o__Lactobacillales' (None)> # children taxa: 674 ; percent= 0.028732202233779522 ; level 19\n",
      "<Node object at 0x11020be80: '100.0' (None)> # children taxa: 650 ; percent= 0.027709097109728024 ; level 20\n",
      "<Node object at 0x11020bef0: '99.0' (None)> # children taxa: 566 ; percent= 0.02412822917554779 ; level 21\n",
      "<Node object at 0x110330a20: '42.0' (None)> # children taxa: 677 ; percent= 0.028860090374285958 ; level 19\n",
      "<Node object at 0x110330a90: '28.0' (None)> # children taxa: 561 ; percent= 0.023915082274703726 ; level 20\n",
      "<Node object at 0x110330b00: '49.0:o__Bacillales' (None)> # children taxa: 543 ; percent= 0.023147753431665103 ; level 21\n",
      "<Node object at 0x1104f53c8: '85.0' (None)> # children taxa: 675 ; percent= 0.028774831613948334 ; level 14\n",
      "<Node object at 0x1104f5438: '100.0:p__Cyanobacteria' (None)> # children taxa: 654 ; percent= 0.027879614630403275 ; level 15\n",
      "<Node object at 0x1104f54a8: '91.0' (None)> # children taxa: 650 ; percent= 0.027709097109728024 ; level 16\n",
      "<Node object at 0x1104f5518: '100.0:c__Cyanobacteriia' (None)> # children taxa: 597 ; percent= 0.02544973996078097 ; level 17\n",
      "<Node object at 0x1104f5588: '99.0' (None)> # children taxa: 595 ; percent= 0.025364481200443347 ; level 18\n",
      "<Node object at 0x1104f55f8: '98.0' (None)> # children taxa: 592 ; percent= 0.025236593059936908 ; level 19\n",
      "<Node object at 0x1104f5668: '53.0' (None)> # children taxa: 577 ; percent= 0.024597152357404724 ; level 20\n",
      "<Node object at 0x1104f56d8: '33.0' (None)> # children taxa: 576 ; percent= 0.024554522977235912 ; level 21\n",
      "<Node object at 0x1105f00b8: '14.0' (None)> # children taxa: 4698 ; percent= 0.20027282803308039 ; level 13\n",
      "<Node object at 0x1105f0128: '3.0' (None)> # children taxa: 3181 ; percent= 0.13560405831699207 ; level 14\n",
      "<Node object at 0x1105f0198: '97.0:p__Actinobacteriota' (None)> # children taxa: 3118 ; percent= 0.1329184073663569 ; level 15\n",
      "<Node object at 0x1105f0208: '97.0' (None)> # children taxa: 2972 ; percent= 0.1266945178617103 ; level 16\n",
      "<Node object at 0x1105f0278: '100.0' (None)> # children taxa: 2921 ; percent= 0.12452041947310086 ; level 17\n",
      "<Node object at 0x1105f02e8: '69.0' (None)> # children taxa: 2738 ; percent= 0.1167192429022082 ; level 18\n",
      "<Node object at 0x1105f0358: '99.0:c__Actinobacteria' (None)> # children taxa: 2735 ; percent= 0.11659135476170177 ; level 19\n",
      "<Node object at 0x1105f03c8: '100.0' (None)> # children taxa: 2731 ; percent= 0.11642083724102652 ; level 20\n",
      "<Node object at 0x1105f0438: '10.0' (None)> # children taxa: 2685 ; percent= 0.11445988575326115 ; level 21\n",
      "<Node object at 0x1105f04a8: '71.0' (None)> # children taxa: 1603 ; percent= 0.0683348964106062 ; level 22\n",
      "<Node object at 0x1105f0518: '46.0' (None)> # children taxa: 1071 ; percent= 0.04565606616079802 ; level 23\n",
      "<Node object at 0x1105f0588: '91.0' (None)> # children taxa: 967 ; percent= 0.04122261062324154 ; level 24\n",
      "<Node object at 0x1105f05f8: '98.0:o__Mycobacteriales' (None)> # children taxa: 948 ; percent= 0.0404126524000341 ; level 25\n",
      "<Node object at 0x1105f0668: '56.0' (None)> # children taxa: 829 ; percent= 0.03533975615994543 ; level 26\n",
      "<Node object at 0x1105f06d8: '81.0' (None)> # children taxa: 823 ; percent= 0.03508397987893256 ; level 27\n",
      "<Node object at 0x1105f0748: '100.0' (None)> # children taxa: 775 ; percent= 0.03303776963082957 ; level 28\n",
      "<Node object at 0x1105f07b8: '100.0' (None)> # children taxa: 770 ; percent= 0.03282462272998551 ; level 29\n",
      "<Node object at 0x1105f0828: '100.0:f__Mycobacteriaceae' (None)> # children taxa: 611 ; percent= 0.026046551283144342 ; level 30\n",
      "<Node object at 0x1105f0898: '34.0' (None)> # children taxa: 609 ; percent= 0.02596129252280672 ; level 31\n",
      "<Node object at 0x110974668: '59.0' (None)> # children taxa: 532 ; percent= 0.022678830249808168 ; level 23\n",
      "<Node object at 0x1109746d8: '95.0:o__Streptomycetales' (None)> # children taxa: 531 ; percent= 0.022636200869639356 ; level 24\n",
      "<Node object at 0x110974748: '90.0' (None)> # children taxa: 530 ; percent= 0.022593571489470544 ; level 25\n",
      "<Node object at 0x1109747b8: '100.0:f__Streptomycetaceae' (None)> # children taxa: 528 ; percent= 0.02250831272913292 ; level 26\n",
      "<Node object at 0x110974828: '100.0' (None)> # children taxa: 526 ; percent= 0.022423053968795292 ; level 27\n",
      "<Node object at 0x110974898: '99.0' (None)> # children taxa: 525 ; percent= 0.02238042458862648 ; level 28\n",
      "<Node object at 0x110a32f28: '47.0' (None)> # children taxa: 1082 ; percent= 0.04612498934265496 ; level 22\n",
      "<Node object at 0x110a32f98: '91.0:o__Actinomycetales' (None)> # children taxa: 944 ; percent= 0.040242134879358855 ; level 23\n",
      "<Node object at 0x110a35048: '53.0' (None)> # children taxa: 873 ; percent= 0.03721544888737318 ; level 24\n",
      "<Node object at 0x110a350b8: '16.0' (None)> # children taxa: 867 ; percent= 0.0369596726063603 ; level 25\n",
      "<Node object at 0x110a35128: '88.0' (None)> # children taxa: 864 ; percent= 0.03683178446585387 ; level 26\n",
      "<Node object at 0x110a35198: '94.0' (None)> # children taxa: 544 ; percent= 0.023190382811833915 ; level 27\n",
      "<Node object at 0x110c6c278: '10.0' (None)> # children taxa: 1517 ; percent= 0.06466876971608833 ; level 14\n",
      "<Node object at 0x110c6c2e8: '74.0' (None)> # children taxa: 1446 ; percent= 0.06164208372410265 ; level 15\n",
      "<Node object at 0x110c6c358: '100.0:p__Patescibacteria' (None)> # children taxa: 1031 ; percent= 0.04395089095404553 ; level 16\n",
      "<Node object at 0x110c6c3c8: '68.0' (None)> # children taxa: 775 ; percent= 0.03303776963082957 ; level 17\n",
      "<Node object at 0x110c6c438: '24.0' (None)> # children taxa: 680 ; percent= 0.028987978514792394 ; level 18\n",
      "<Node object at 0x110c6c4a8: '98.0' (None)> # children taxa: 613 ; percent= 0.026131810043481966 ; level 19\n",
      "<Node object at 0x110c6c518: '33.0' (None)> # children taxa: 608 ; percent= 0.025918663142637906 ; level 20\n",
      "<Node object at 0x110ec51d0: '1.0' (None)> # children taxa: 885 ; percent= 0.037727001449398924 ; level 10\n",
      "<Node object at 0x110ec5240: '76.0' (None)> # children taxa: 648 ; percent= 0.0276238383493904 ; level 11\n",
      "<Node object at 0x111002a58: '0.0' (None)> # children taxa: 3175 ; percent= 0.1353482820359792 ; level 9\n",
      "<Node object at 0x111002ac8: '2.0' (None)> # children taxa: 3158 ; percent= 0.13462358257310938 ; level 10\n",
      "<Node object at 0x111002b38: '27.0' (None)> # children taxa: 3132 ; percent= 0.13351521868872027 ; level 11\n",
      "<Node object at 0x111002ba8: '20.0' (None)> # children taxa: 3014 ; percent= 0.1284849518288004 ; level 12\n",
      "<Node object at 0x111002c18: '98.0' (None)> # children taxa: 2946 ; percent= 0.12558615397732117 ; level 13\n",
      "<Node object at 0x111002c88: '53.0' (None)> # children taxa: 2940 ; percent= 0.1253303776963083 ; level 14\n",
      "<Node object at 0x111002cf8: '83.0' (None)> # children taxa: 2852 ; percent= 0.12157899224145281 ; level 15\n",
      "<Node object at 0x111002d68: '100.0:p__Bacteroidota' (None)> # children taxa: 2843 ; percent= 0.1211953278199335 ; level 16\n",
      "<Node object at 0x111002dd8: '95.0' (None)> # children taxa: 2756 ; percent= 0.11748657174524682 ; level 17\n",
      "<Node object at 0x111002e48: '100.0' (None)> # children taxa: 2733 ; percent= 0.11650609600136413 ; level 18\n",
      "<Node object at 0x111002eb8: '100.0:c__Bacteroidia' (None)> # children taxa: 2677 ; percent= 0.11411885071191065 ; level 19\n",
      "<Node object at 0x111002f28: '69.0' (None)> # children taxa: 2505 ; percent= 0.10678659732287493 ; level 20\n",
      "<Node object at 0x111002f98: '66.0' (None)> # children taxa: 2254 ; percent= 0.09608662290050303 ; level 21\n",
      "<Node object at 0x111006048: '31.0' (None)> # children taxa: 2236 ; percent= 0.09531929405746441 ; level 22\n",
      "<Node object at 0x1110060b8: '97.0' (None)> # children taxa: 2090 ; percent= 0.08909540455281781 ; level 23\n",
      "<Node object at 0x111006128: '89.0' (None)> # children taxa: 2074 ; percent= 0.0884133344701168 ; level 24\n",
      "<Node object at 0x111006198: '91.0:o__Bacteroidales' (None)> # children taxa: 1029 ; percent= 0.043865632193707906 ; level 25\n",
      "<Node object at 0x111006208: '100.0' (None)> # children taxa: 910 ; percent= 0.038792735953619234 ; level 26\n",
      "<Node object at 0x111006278: '33.0' (None)> # children taxa: 907 ; percent= 0.038664847813112795 ; level 27\n",
      "<Node object at 0x1110062e8: '37.0' (None)> # children taxa: 691 ; percent= 0.02945690169664933 ; level 28\n",
      "<Node object at 0x111006358: '81.0' (None)> # children taxa: 686 ; percent= 0.02924375479580527 ; level 29\n",
      "<Node object at 0x1110063c8: '51.0' (None)> # children taxa: 666 ; percent= 0.028391167192429023 ; level 30\n",
      "<Node object at 0x111006438: '84.0' (None)> # children taxa: 649 ; percent= 0.027666467729559212 ; level 31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Node object at 0x1110064a8: '93.0' (None)> # children taxa: 628 ; percent= 0.026771250746014153 ; level 32\n",
      "<Node object at 0x111006518: '100.0' (None)> # children taxa: 626 ; percent= 0.02668599198567653 ; level 33\n",
      "<Node object at 0x111006588: '69.0' (None)> # children taxa: 564 ; percent= 0.02404297041521016 ; level 34\n",
      "<Node object at 0x1110065f8: '83.0' (None)> # children taxa: 561 ; percent= 0.023915082274703726 ; level 35\n",
      "<Node object at 0x11137a2e8: '88.0:o__Flavobacteriales' (None)> # children taxa: 1045 ; percent= 0.044547702276408904 ; level 25\n",
      "<Node object at 0x11137a358: '99.0' (None)> # children taxa: 932 ; percent= 0.039730582317333105 ; level 26\n",
      "<Node object at 0x11137a3c8: '31.0' (None)> # children taxa: 909 ; percent= 0.038750106573450425 ; level 27\n",
      "<Node object at 0x11137a438: '100.0' (None)> # children taxa: 890 ; percent= 0.03794014835024299 ; level 28\n",
      "<Node object at 0x11137a4a8: '100.0' (None)> # children taxa: 864 ; percent= 0.03683178446585387 ; level 29\n",
      "<Node object at 0x11137a518: '32.0' (None)> # children taxa: 862 ; percent= 0.036746525705516245 ; level 30\n",
      "<Node object at 0x11137a588: '39.0' (None)> # children taxa: 676 ; percent= 0.028817460994117146 ; level 31\n",
      "<Node object at 0x11137a5f8: '100.0:f__Flavobacteriaceae' (None)> # children taxa: 675 ; percent= 0.028774831613948334 ; level 32\n",
      "<Node object at 0x11137a668: '80.0' (None)> # children taxa: 615 ; percent= 0.026217068803819594 ; level 33\n",
      "<Node object at 0x11167e748: '49.0' (None)> # children taxa: 4663 ; percent= 0.19878079972717197 ; level 14\n",
      "<Node object at 0x11167e7b8: '99.0:c__Gammaproteobacteria' (None)> # children taxa: 4645 ; percent= 0.19801347088413335 ; level 15\n",
      "<Node object at 0x11167e828: '9.0' (None)> # children taxa: 4590 ; percent= 0.19566885497484868 ; level 16\n",
      "<Node object at 0x11167e898: '1.0' (None)> # children taxa: 4585 ; percent= 0.1954557080740046 ; level 17\n",
      "<Node object at 0x11167e908: '0.0' (None)> # children taxa: 3267 ; percent= 0.13927018501150992 ; level 18\n",
      "<Node object at 0x11167e978: '11.0' (None)> # children taxa: 3246 ; percent= 0.13837496802796487 ; level 19\n",
      "<Node object at 0x11167e9e8: '14.0' (None)> # children taxa: 2648 ; percent= 0.11288259868701508 ; level 20\n",
      "<Node object at 0x11167ea58: '25.0' (None)> # children taxa: 2591 ; percent= 0.11045272401739278 ; level 21\n",
      "<Node object at 0x11167eac8: '59.0' (None)> # children taxa: 2457 ; percent= 0.10474038707477193 ; level 22\n",
      "<Node object at 0x11167eb38: '90.0:o__Pseudomonadales' (None)> # children taxa: 1270 ; percent= 0.05413931281439168 ; level 23\n",
      "<Node object at 0x11167eba8: '90.0' (None)> # children taxa: 1047 ; percent= 0.04463296103674653 ; level 24\n",
      "<Node object at 0x11167ec18: '42.0' (None)> # children taxa: 800 ; percent= 0.034103504135049874 ; level 25\n",
      "<Node object at 0x11167ec88: '41.0' (None)> # children taxa: 734 ; percent= 0.03128996504390826 ; level 26\n",
      "<Node object at 0x11167ecf8: '94.0' (None)> # children taxa: 526 ; percent= 0.022423053968795292 ; level 27\n",
      "<Node object at 0x11167ed68: '100.0:f__Pseudomonadaceae' (None)> # children taxa: 514 ; percent= 0.021911501406769545 ; level 28\n",
      "<Node object at 0x1118467b8: '65.0' (None)> # children taxa: 1187 ; percent= 0.050601074260380255 ; level 23\n",
      "<Node object at 0x111846828: '61.0:o__Enterobacterales' (None)> # children taxa: 1186 ; percent= 0.05055844488021144 ; level 24\n",
      "<Node object at 0x111846898: '84.0' (None)> # children taxa: 1174 ; percent= 0.050046892318185696 ; level 25\n",
      "<Node object at 0x111846908: '71.0' (None)> # children taxa: 1150 ; percent= 0.049023787194134194 ; level 26\n",
      "<Node object at 0x111846978: '63.0' (None)> # children taxa: 854 ; percent= 0.03640549066416574 ; level 27\n",
      "<Node object at 0x1118469e8: '84.0' (None)> # children taxa: 803 ; percent= 0.03423139227555631 ; level 28\n",
      "<Node object at 0x111846a58: '84.0' (None)> # children taxa: 598 ; percent= 0.025492369340949783 ; level 29\n",
      "<Node object at 0x111846ac8: '78.0:f__Enterobacteriaceae' (None)> # children taxa: 509 ; percent= 0.021698354505925485 ; level 30\n",
      "<Node object at 0x111846b38: '71.0' (None)> # children taxa: 508 ; percent= 0.021655725125756673 ; level 31\n",
      "<Node object at 0x111a35e10: '6.0' (None)> # children taxa: 598 ; percent= 0.025492369340949783 ; level 20\n",
      "<Node object at 0x111b19198: '61.0' (None)> # children taxa: 1318 ; percent= 0.05618552306249467 ; level 18\n",
      "<Node object at 0x111b19208: '51.0' (None)> # children taxa: 1283 ; percent= 0.05469349475658624 ; level 19\n",
      "<Node object at 0x111b19278: '97.0:o__Burkholderiales' (None)> # children taxa: 1273 ; percent= 0.054267200954898114 ; level 20\n",
      "<Node object at 0x111b192e8: '76.0' (None)> # children taxa: 1271 ; percent= 0.05418194219456049 ; level 21\n",
      "<Node object at 0x111b19358: '13.0' (None)> # children taxa: 1265 ; percent= 0.05392616591354762 ; level 22\n",
      "<Node object at 0x111b193c8: '2.0' (None)> # children taxa: 1261 ; percent= 0.05375564839287237 ; level 23\n",
      "<Node object at 0x111b19438: '6.0' (None)> # children taxa: 970 ; percent= 0.041350498763747974 ; level 24\n",
      "<Node object at 0x111b194a8: '7.0' (None)> # children taxa: 964 ; percent= 0.0410947224827351 ; level 25\n",
      "<Node object at 0x111b19518: '22.0' (None)> # children taxa: 951 ; percent= 0.04054054054054054 ; level 26\n",
      "<Node object at 0x111b19588: '99.0:f__Burkholderiaceae' (None)> # children taxa: 856 ; percent= 0.036490749424503366 ; level 27\n",
      "<Node object at 0x111b195f8: '15.0' (None)> # children taxa: 842 ; percent= 0.03589393810214 ; level 28\n",
      "<Node object at 0x111b19668: '15.0' (None)> # children taxa: 825 ; percent= 0.035169238639270184 ; level 29\n",
      "<Node object at 0x111b196d8: '66.0' (None)> # children taxa: 712 ; percent= 0.03035211868019439 ; level 30\n",
      "<Node object at 0x111d0c9b0: '15.0' (None)> # children taxa: 2967 ; percent= 0.12648137096086623 ; level 14\n",
      "<Node object at 0x111d0ca20: '42.0:c__Alphaproteobacteria' (None)> # children taxa: 2955 ; percent= 0.12596981839884047 ; level 15\n",
      "<Node object at 0x111d0ca90: '32.0' (None)> # children taxa: 2947 ; percent= 0.12562878335748998 ; level 16\n",
      "<Node object at 0x111d0cb00: '18.0' (None)> # children taxa: 2929 ; percent= 0.12486145451445135 ; level 17\n",
      "<Node object at 0x111d0cb70: '7.0' (None)> # children taxa: 2857 ; percent= 0.12179213914229688 ; level 18\n",
      "<Node object at 0x111d0cbe0: '2.0' (None)> # children taxa: 2683 ; percent= 0.11437462699292353 ; level 19\n",
      "<Node object at 0x111d0cc50: '0.0' (None)> # children taxa: 2645 ; percent= 0.11275471054650865 ; level 20\n",
      "<Node object at 0x111d0ccc0: '3.0' (None)> # children taxa: 2618 ; percent= 0.11160371728195072 ; level 21\n",
      "<Node object at 0x111d0cd30: '3.0' (None)> # children taxa: 2617 ; percent= 0.1115610879017819 ; level 22\n",
      "<Node object at 0x111d0cda0: '8.0' (None)> # children taxa: 2590 ; percent= 0.11041009463722397 ; level 23\n",
      "<Node object at 0x111d0ce10: '3.0' (None)> # children taxa: 2587 ; percent= 0.11028220649671754 ; level 24\n",
      "<Node object at 0x111d0ce80: '1.0' (None)> # children taxa: 2555 ; percent= 0.10891806633131554 ; level 25\n",
      "<Node object at 0x111d0cef0: '2.0' (None)> # children taxa: 2502 ; percent= 0.10665870918236849 ; level 26\n",
      "<Node object at 0x111d0cf60: '10.0' (None)> # children taxa: 2224 ; percent= 0.09480774149543866 ; level 27\n",
      "<Node object at 0x111d0cfd0: '22.0' (None)> # children taxa: 2206 ; percent= 0.09404041265240004 ; level 28\n",
      "<Node object at 0x111d0f080: '11.0' (None)> # children taxa: 2195 ; percent= 0.0935714894705431 ; level 29\n",
      "<Node object at 0x111d0f0f0: '16.0' (None)> # children taxa: 2191 ; percent= 0.09340097194986785 ; level 30\n",
      "<Node object at 0x111d0f160: '16.0' (None)> # children taxa: 1780 ; percent= 0.07588029670048597 ; level 31\n",
      "<Node object at 0x111d0f1d0: '27.0' (None)> # children taxa: 1779 ; percent= 0.07583766732031716 ; level 32\n",
      "<Node object at 0x111d0f240: '22.0' (None)> # children taxa: 984 ; percent= 0.04194731008611135 ; level 33\n",
      "<Node object at 0x111d0f2b0: '20.0:o__Rhizobiales' (None)> # children taxa: 961 ; percent= 0.04096683434222866 ; level 34\n",
      "<Node object at 0x111d0f320: '37.0' (None)> # children taxa: 937 ; percent= 0.03994372921817717 ; level 35\n",
      "<Node object at 0x111d0f390: '45.0' (None)> # children taxa: 933 ; percent= 0.03977321169750192 ; level 36\n",
      "<Node object at 0x111d0f400: '49.0' (None)> # children taxa: 889 ; percent= 0.03789751897007417 ; level 37\n",
      "<Node object at 0x111d0f470: '48.0' (None)> # children taxa: 888 ; percent= 0.03785488958990536 ; level 38\n",
      "<Node object at 0x111d0f4e0: '48.0' (None)> # children taxa: 576 ; percent= 0.024554522977235912 ; level 39\n",
      "<Node object at 0x111d0f550: '21.0' (None)> # children taxa: 518 ; percent= 0.022082018927444796 ; level 40\n",
      "<Node object at 0x111e71278: '88.0' (None)> # children taxa: 795 ; percent= 0.03389035723420582 ; level 33\n",
      "<Node object at 0x111e712e8: '93.0:o__Rhodobacterales' (None)> # children taxa: 579 ; percent= 0.02468241111774235 ; level 34\n",
      "<Node object at 0x111e71358: '100.0:f__Rhodobacteraceae' (None)> # children taxa: 578 ; percent= 0.024639781737573536 ; level 35\n",
      "<Node object at 0x111e713c8: '89.0' (None)> # children taxa: 577 ; percent= 0.024597152357404724 ; level 36\n",
      "<Node object at 0x111e71438: '89.0' (None)> # children taxa: 575 ; percent= 0.024511893597067097 ; level 37\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Node object at 0x111e714a8: '89.0' (None)> # children taxa: 572 ; percent= 0.02438400545656066 ; level 38\n",
      "<Node object at 0x111e71518: '44.0' (None)> # children taxa: 570 ; percent= 0.024298746696223037 ; level 39\n",
      "<Node object at 0x111e71588: '56.0' (None)> # children taxa: 567 ; percent= 0.0241708585557166 ; level 40\n",
      "<Node object at 0x111e715f8: '44.0' (None)> # children taxa: 561 ; percent= 0.023915082274703726 ; level 41\n",
      "<Node object at 0x111e71668: '66.0' (None)> # children taxa: 551 ; percent= 0.023488788473015602 ; level 42\n",
      "<Node object at 0x111e716d8: '10.0' (None)> # children taxa: 540 ; percent= 0.023019865291158667 ; level 43\n",
      "<Node object at 0x111e71748: '9.0' (None)> # children taxa: 537 ; percent= 0.02289197715065223 ; level 44\n",
      "<Node object at 0x1129be208: '20.0' (None)> # children taxa: 841 ; percent= 0.03585130872197118 ; level 11\n",
      "<Node object at 0x1129be278: '0.0' (None)> # children taxa: 738 ; percent= 0.03146048256458351 ; level 12\n",
      "<Node object at 0x1129be2e8: '1.0' (None)> # children taxa: 525 ; percent= 0.02238042458862648 ; level 13\n"
     ]
    }
   ],
   "source": [
    "get_branch_children_counts(bac.seed_node,max_count=5000,min_count=500)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18933\n",
      "19016\n",
      "4525\n",
      "4442\n",
      "2465 in valid set - p_Firmicutes_A, o_Rhodobacterales\n",
      "1977 in test set - o_Bacteroidales, o_Mycobacteriales\n",
      "1523 in new valid set - o__Actinomycetales, o_Rhodobacterales\n",
      "1990 in new test set - o_Bacteroidales, o__Rhizobiales\n"
     ]
    }
   ],
   "source": [
    "#trying to pick only groups order or higher; class or phylum preferred; 1972\n",
    "#c_Actinobacteria=2735\n",
    "#p_Firmicutes_A=1886\n",
    "#c__Clostridia=1836\n",
    "#o__Oscillospirales=511\n",
    "#p__Cyanobacteria=654\n",
    "#p_Patescibacteria=1031\n",
    "#c_Alphaproteobacteria=2955\n",
    "###o__Actinomycetales=944\n",
    "#o_Rhodobacterales=579\n",
    "#o_Bacteroidales=1029\n",
    "#o__Mycobacteriales=948 - too similar to actinomycetales\n",
    "#o__Flavobacteriales=1045\n",
    "#o__Rhizobiales=961\n",
    "#aiming for 4458\n",
    "\n",
    "2735+1878 \n",
    "4458-2735\n",
    "4458-1886-579 #1900\n",
    "print(23458-(1886+579+1029+1031))\n",
    "print(23458-(1886+579+1029+948)) #firmicutes_A, rhodobacterales, bacteroidales, mycobacteriales\n",
    "print((1886+579+1029+1031)) #firmicutes, rhodobacterales, bacteroidales, patescibacteria\n",
    "print((1886+579+1029+948)) #firmicutes_A, rhodobacterales, bacteroidales, mycobacteriales I like this one\n",
    "print((1886+579),'in valid set - p_Firmicutes_A, o_Rhodobacterales')\n",
    "print((1029+948),'in test set - o_Bacteroidales, o_Mycobacteriales')\n",
    "\n",
    "print((944+579),'in new valid set - o__Actinomycetales, o_Rhodobacterales') \n",
    "print((1029+961),'in new test set - o_Bacteroidales, o__Rhizobiales')\n",
    "\n",
    "#if do same groups but at class level: valid = c__Actinobacteria and c__Alphaproteobacteria; test = c__Bacteroidia and c__Clostridia\n",
    "#replaced rhizobiales with clostridia because rhizobiales are also alphaproteobacteria\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.06164494454788311"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(944+579)/24706"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Node object at 0x1161926a0: '100.0:c__Halobacteria; o__Halobacteriales' (None)> # children taxa: 229 ; percent= 0.009762128058658026 ; level 10\n",
      "<Node object at 0x116122f60: '100.0' (None)> # children taxa: 228 ; percent= 0.009719498678489214 ; level 11\n",
      "<Node object at 0x11674d160: '48.0' (None)> # children taxa: 220 ; percent= 0.009378463637138717 ; level 12\n",
      "<Node object at 0x114a0b438: '42.0' (None)> # children taxa: 214 ; percent= 0.009122687356125841 ; level 13\n",
      "<Node object at 0x115679d68: '43.0' (None)> # children taxa: 199 ; percent= 0.008483246653593656 ; level 14\n",
      "<Node object at 0x1169c80f0: '56.0' (None)> # children taxa: 137 ; percent= 0.005840225083127292 ; level 15\n",
      "<Node object at 0x115261a90: '100.0:f__Haloferacaceae' (None)> # children taxa: 81 ; percent= 0.0034529797936738 ; level 16\n",
      "<Node object at 0x116ae7748: '97.0' (None)> # children taxa: 46 ; percent= 0.001960951487765368 ; level 17\n",
      "<Node object at 0x11494c518: '100.0' (None)> # children taxa: 37 ; percent= 0.0015772870662460567 ; level 18\n",
      "<Node object at 0x11670d550: '99.0' (None)> # children taxa: 33 ; percent= 0.0014067695455708075 ; level 19\n",
      "<Node object at 0x1160b35f8: '100.0' (None)> # children taxa: 32 ; percent= 0.001364140165401995 ; level 20\n",
      "<Node object at 0x115d349e8: '27.0' (None)> # children taxa: 35 ; percent= 0.0014920283059084321 ; level 17\n",
      "<Node object at 0x114545a90: '99.0' (None)> # children taxa: 32 ; percent= 0.001364140165401995 ; level 18\n",
      "<Node object at 0x11664af28: '100.0:f__Haloarculaceae' (None)> # children taxa: 56 ; percent= 0.002387245289453491 ; level 16\n",
      "<Node object at 0x1154d6d68: '99.0' (None)> # children taxa: 39 ; percent= 0.0016625458265836816 ; level 17\n",
      "<Node object at 0x116781828: '90.0' (None)> # children taxa: 35 ; percent= 0.0014920283059084321 ; level 18\n",
      "<Node object at 0x114e11d30: '55.0' (None)> # children taxa: 62 ; percent= 0.0026430215704663653 ; level 15\n",
      "<Node object at 0x114e115c0: '38.0' (None)> # children taxa: 59 ; percent= 0.0025151334299599285 ; level 16\n",
      "<Node object at 0x114e114a8: '50.0' (None)> # children taxa: 57 ; percent= 0.0024298746696223036 ; level 17\n",
      "<Node object at 0x114e119e8: '79.0:f__Natrialbaceae' (None)> # children taxa: 56 ; percent= 0.002387245289453491 ; level 18\n",
      "<Node object at 0x114e11c18: '100.0' (None)> # children taxa: 55 ; percent= 0.002344615909284679 ; level 19\n",
      "<Node object at 0x114e11908: '76.0' (None)> # children taxa: 53 ; percent= 0.0022593571489470544 ; level 20\n",
      "<Node object at 0x116c06080: '30.0' (None)> # children taxa: 51 ; percent= 0.0021740983886094295 ; level 21\n",
      "<Node object at 0x118f52ef0: '100.0' (None)> # children taxa: 46 ; percent= 0.001960951487765368 ; level 22\n",
      "<Node object at 0x118f52128: '53.0' (None)> # children taxa: 43 ; percent= 0.0018330633472589308 ; level 23\n",
      "<Node object at 0x1148efeb8: '5.0' (None)> # children taxa: 35 ; percent= 0.0014920283059084321 ; level 24\n",
      "<Node object at 0x115c1f710: '100.0:c__Methanomicrobia; o__Methanomicrobiales' (None)> # children taxa: 71 ; percent= 0.0030266859919856766 ; level 10\n",
      "<Node object at 0x1178b68d0: '72.0' (None)> # children taxa: 67 ; percent= 0.002856168471310427 ; level 11\n",
      "<Node object at 0x1178b60f0: '48.0' (None)> # children taxa: 60 ; percent= 0.002557762810128741 ; level 12\n",
      "<Node object at 0x114d94320: '90.0' (None)> # children taxa: 80 ; percent= 0.0034103504135049876 ; level 8\n",
      "<Node object at 0x114d94b70: '97.0:c__Methanosarcinia' (None)> # children taxa: 76 ; percent= 0.0032398328928297383 ; level 9\n",
      "<Node object at 0x114d94908: '83.0:o__Methanosarcinales' (None)> # children taxa: 63 ; percent= 0.0026856509506351777 ; level 10\n",
      "<Node object at 0x114d94128: '91.0' (None)> # children taxa: 58 ; percent= 0.002472504049791116 ; level 11\n",
      "<Node object at 0x115f06c18: '53.0' (None)> # children taxa: 50 ; percent= 0.002131469008440617 ; level 12\n",
      "<Node object at 0x115f06550: '100.0:f__Methanosarcinaceae' (None)> # children taxa: 48 ; percent= 0.0020462102481029927 ; level 13\n",
      "<Node object at 0x11654d6d8: '100.0:p__Thermoplasmatota' (None)> # children taxa: 287 ; percent= 0.012234632108449143 ; level 4\n",
      "<Node object at 0x1164a4128: '91.0' (None)> # children taxa: 285 ; percent= 0.012149373348111519 ; level 5\n",
      "<Node object at 0x116ba49e8: '88.0' (None)> # children taxa: 272 ; percent= 0.011595191405916958 ; level 6\n",
      "<Node object at 0x118d82b38: '95.0:c__Poseidoniia' (None)> # children taxa: 205 ; percent= 0.00873902293460653 ; level 7\n",
      "<Node object at 0x118d82da0: '100.0' (None)> # children taxa: 192 ; percent= 0.00818484099241197 ; level 8\n",
      "<Node object at 0x118d82a58: '100.0:o__Poseidoniales' (None)> # children taxa: 191 ; percent= 0.008142211612243159 ; level 9\n",
      "<Node object at 0x118d82cc0: '100.0:f__Thalassoarchaeaceae' (None)> # children taxa: 104 ; percent= 0.004433455537556484 ; level 10\n",
      "<Node object at 0x119367208: '88.0' (None)> # children taxa: 100 ; percent= 0.004262938016881234 ; level 11\n",
      "<Node object at 0x119367320: '100.0' (None)> # children taxa: 94 ; percent= 0.004007161735868361 ; level 12\n",
      "<Node object at 0x119367780: '100.0' (None)> # children taxa: 82 ; percent= 0.0034956091738426124 ; level 13\n",
      "<Node object at 0x119367a58: '99.0' (None)> # children taxa: 73 ; percent= 0.003111944752323301 ; level 14\n",
      "<Node object at 0x119367128: '100.0' (None)> # children taxa: 60 ; percent= 0.002557762810128741 ; level 15\n",
      "<Node object at 0x118d57a58: '85.0' (None)> # children taxa: 51 ; percent= 0.0021740983886094295 ; level 16\n",
      "<Node object at 0x11770c780: '100.0:f__Poseidoniaceae' (None)> # children taxa: 87 ; percent= 0.003708756074686674 ; level 10\n",
      "<Node object at 0x11770c668: '91.0' (None)> # children taxa: 81 ; percent= 0.0034529797936738 ; level 11\n",
      "<Node object at 0x116907048: '100.0' (None)> # children taxa: 78 ; percent= 0.003325091653167363 ; level 12\n",
      "<Node object at 0x1178f5240: '100.0' (None)> # children taxa: 58 ; percent= 0.002472504049791116 ; level 13\n",
      "<Node object at 0x1178f5588: '50.0' (None)> # children taxa: 49 ; percent= 0.002088839628271805 ; level 14\n",
      "<Node object at 0x1178f5da0: '90.0' (None)> # children taxa: 40 ; percent= 0.0017051752067524938 ; level 15\n",
      "<Node object at 0x1191882e8: '86.0:c__Thermoplasmata' (None)> # children taxa: 67 ; percent= 0.002856168471310427 ; level 7\n",
      "<Node object at 0x119188d68: '37.0' (None)> # children taxa: 65 ; percent= 0.0027709097109728026 ; level 8\n",
      "<Node object at 0x116bce080: '79.0' (None)> # children taxa: 44 ; percent= 0.0018756927274277432 ; level 9\n",
      "<Node object at 0x118e5c390: '86.0' (None)> # children taxa: 42 ; percent= 0.0017904339670901184 ; level 10\n",
      "<Node object at 0x118e5ca58: '100.0' (None)> # children taxa: 36 ; percent= 0.0015346576860772445 ; level 11\n",
      "<Node object at 0x118e5c978: '100.0:o__Methanomassiliicoccales' (None)> # children taxa: 32 ; percent= 0.001364140165401995 ; level 12\n",
      "<Node object at 0x118bd23c8: '85.0' (None)> # children taxa: 157 ; percent= 0.006692812686503538 ; level 3\n",
      "<Node object at 0x116bafb70: '86.0:p__Euryarchaeota' (None)> # children taxa: 150 ; percent= 0.006394407025321852 ; level 4\n",
      "<Node object at 0x118af9400: '66.0' (None)> # children taxa: 104 ; percent= 0.004433455537556484 ; level 5\n",
      "<Node object at 0x118af9be0: '79.0' (None)> # children taxa: 102 ; percent= 0.004348196777218859 ; level 6\n",
      "<Node object at 0x118af97f0: '100.0:c__Methanobacteria; o__Methanobacteriales' (None)> # children taxa: 84 ; percent= 0.003580867934180237 ; level 7\n",
      "<Node object at 0x118af9908: '100.0' (None)> # children taxa: 83 ; percent= 0.003538238554011425 ; level 8\n",
      "<Node object at 0x118af9080: '98.0' (None)> # children taxa: 81 ; percent= 0.0034529797936738 ; level 9\n",
      "<Node object at 0x118af95c0: '99.0:f__Methanobacteriaceae' (None)> # children taxa: 77 ; percent= 0.0032824622729985507 ; level 10\n",
      "<Node object at 0x118d73748: '85.0' (None)> # children taxa: 44 ; percent= 0.0018756927274277432 ; level 11\n",
      "<Node object at 0x118d73198: '97.0' (None)> # children taxa: 40 ; percent= 0.0017051752067524938 ; level 12\n",
      "<Node object at 0x118d73828: '96.0' (None)> # children taxa: 39 ; percent= 0.0016625458265836816 ; level 13\n",
      "<Node object at 0x118d73668: '96.0' (None)> # children taxa: 34 ; percent= 0.0014493989257396197 ; level 14\n",
      "<Node object at 0x11935f160: '100.0' (None)> # children taxa: 33 ; percent= 0.0014067695455708075 ; level 11\n",
      "<Node object at 0x1168fef28: '100.0:c__Thermococci' (None)> # children taxa: 46 ; percent= 0.001960951487765368 ; level 5\n",
      "<Node object at 0x1156a5a58: '100.0:o__Thermococcales; f__Thermococcaceae' (None)> # children taxa: 41 ; percent= 0.0017478045869213062 ; level 6\n",
      "<Node object at 0x1156ed630: '58.0' (None)> # children taxa: 39 ; percent= 0.0016625458265836816 ; level 7\n",
      "<Node object at 0x118fb40f0: '62.0' (None)> # children taxa: 38 ; percent= 0.0016199164464148692 ; level 8\n",
      "<Node object at 0x118fb4358: '52.0' (None)> # children taxa: 33 ; percent= 0.0014067695455708075 ; level 9\n",
      "<Node object at 0x118fb4470: '62.0' (None)> # children taxa: 31 ; percent= 0.0013215107852331826 ; level 10\n",
      "<Node object at 0x118dfbe80: '100.0' (None)> # children taxa: 232 ; percent= 0.009890016199164464 ; level 2\n",
      "<Node object at 0x118dfb400: '98.0:p__Crenarchaeota' (None)> # children taxa: 218 ; percent= 0.009293204876801091 ; level 3\n",
      "<Node object at 0x118dfb668: '64.0' (None)> # children taxa: 217 ; percent= 0.00925057549663228 ; level 4\n",
      "<Node object at 0x118dfbf60: '95.0' (None)> # children taxa: 121 ; percent= 0.005158155000426294 ; level 5\n",
      "<Node object at 0x1154b1080: '80.0:c__Nitrososphaeria' (None)> # children taxa: 81 ; percent= 0.0034529797936738 ; level 6\n",
      "<Node object at 0x118aa2898: '87.0' (None)> # children taxa: 72 ; percent= 0.003069315372154489 ; level 7\n",
      "<Node object at 0x118aa2048: '100.0' (None)> # children taxa: 70 ; percent= 0.0029840566118168642 ; level 8\n",
      "<Node object at 0x118aa22e8: '100.0:o__Nitrososphaerales' (None)> # children taxa: 68 ; percent= 0.0028987978514792394 ; level 9\n",
      "<Node object at 0x118aa2a90: '100.0' (None)> # children taxa: 60 ; percent= 0.002557762810128741 ; level 10\n",
      "<Node object at 0x118aa2f28: '100.0' (None)> # children taxa: 57 ; percent= 0.0024298746696223036 ; level 11\n",
      "<Node object at 0x117e53ac8: '100.0:f__Nitrosopumilaceae' (None)> # children taxa: 48 ; percent= 0.0020462102481029927 ; level 12\n",
      "<Node object at 0x117e53ef0: '100.0' (None)> # children taxa: 41 ; percent= 0.0017478045869213062 ; level 13\n",
      "<Node object at 0x117e53390: '100.0' (None)> # children taxa: 36 ; percent= 0.0015346576860772445 ; level 14\n",
      "<Node object at 0x117e53588: '71.0' (None)> # children taxa: 35 ; percent= 0.0014920283059084321 ; level 15\n",
      "<Node object at 0x1193c6550: '93.0:c__Bathyarchaeia' (None)> # children taxa: 40 ; percent= 0.0017051752067524938 ; level 6\n",
      "<Node object at 0x1193c61d0: '83.0' (None)> # children taxa: 39 ; percent= 0.0016625458265836816 ; level 7\n",
      "<Node object at 0x118aef668: '80.0' (None)> # children taxa: 96 ; percent= 0.004092420496205985 ; level 5\n",
      "<Node object at 0x118aef940: '79.0:c__Thermoprotei' (None)> # children taxa: 93 ; percent= 0.003964532355699548 ; level 6\n",
      "<Node object at 0x118aef860: '48.0' (None)> # children taxa: 90 ; percent= 0.003836644215193111 ; level 7\n",
      "<Node object at 0x118aef208: '100.0' (None)> # children taxa: 50 ; percent= 0.002131469008440617 ; level 8\n",
      "<Node object at 0x115958a20: '80.0' (None)> # children taxa: 49 ; percent= 0.002088839628271805 ; level 9\n",
      "<Node object at 0x11955b0f0: '67.0' (None)> # children taxa: 40 ; percent= 0.0017051752067524938 ; level 8\n",
      "<Node object at 0x11858b630: '98.0' (None)> # children taxa: 38 ; percent= 0.0016199164464148692 ; level 9\n",
      "<Node object at 0x11858bcf8: '100.0:o__Thermoproteales' (None)> # children taxa: 33 ; percent= 0.0014067695455708075 ; level 10\n",
      "<Node object at 0x114de2780: '97.0' (None)> # children taxa: 161 ; percent= 0.006863330207178788 ; level 1\n",
      "<Node object at 0x116016630: '80.0' (None)> # children taxa: 151 ; percent= 0.006437036405490664 ; level 2\n",
      "<Node object at 0x116016208: '50.0' (None)> # children taxa: 124 ; percent= 0.005286043140932731 ; level 3\n",
      "<Node object at 0x116016f98: '85.0' (None)> # children taxa: 121 ; percent= 0.005158155000426294 ; level 4\n",
      "<Node object at 0x116016400: '77.0:p__Nanoarchaeota; c__Nanoarchaeia' (None)> # children taxa: 101 ; percent= 0.004305567397050047 ; level 5\n",
      "<Node object at 0x116c65cf8: '67.0' (None)> # children taxa: 97 ; percent= 0.004135049876374797 ; level 6\n",
      "<Node object at 0x116338c50: '74.0' (None)> # children taxa: 93 ; percent= 0.003964532355699548 ; level 7\n",
      "<Node object at 0x1148e3160: '86.0' (None)> # children taxa: 88 ; percent= 0.0037513854548554865 ; level 8\n",
      "<Node object at 0x117d00b38: '76.0' (None)> # children taxa: 55 ; percent= 0.002344615909284679 ; level 9\n",
      "<Node object at 0x117d00f60: '90.0:o__Woesearchaeales' (None)> # children taxa: 42 ; percent= 0.0017904339670901184 ; level 10\n",
      "<Node object at 0x117d005f8: '10.0' (None)> # children taxa: 31 ; percent= 0.0013215107852331826 ; level 11\n",
      "<Node object at 0x11847d780: '100.0:o__Pacearchaeales' (None)> # children taxa: 33 ; percent= 0.0014067695455708075 ; level 9\n",
      "<Node object at 0x119bf4048: '100.0' (None)> # children taxa: 32 ; percent= 0.001364140165401995 ; level 10\n"
     ]
    }
   ],
   "source": [
    "get_branch_children_counts(arc.seed_node,300,30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "994\n",
      "254\n",
      "259\n",
      "238\n",
      "131 in valid set - o_Methanosarcinales and o_Nitrososphaerales\n",
      "107 in test set - c_Thermoplasmata and c_Bathyarchaeia\n",
      "107 in new valid set - c_Thermoplasmata and c_Bathyarchaeia\n",
      "131 in new test set - o_Methanosarcinales and o_Nitrososphaerales\n"
     ]
    }
   ],
   "source": [
    "#f__Haloferacaceae=81\n",
    "#f__Haloarculaceae=56\n",
    "#c__Methanosarcinia=76\n",
    "#o__Methanosarcinales=63\n",
    "#f__Methanosarcinaceae=48\n",
    "#c__Thermoplasmata=67 #acidophiles\n",
    "#o__Nitrososphaerales=68 #ammonia oxidizers etc\n",
    "#o__Methanomicrobiales=71\n",
    "#o__Methanobacteriales=84 #methanobact or thermococci\n",
    "#c__Thermococci=46\n",
    "#c__Bathyarchaeia=40 #bathy or thermoprot or nitro\n",
    "#c__Thermoprotei=93\n",
    "#o__Thermoproteales=33\n",
    "#o__Woesearchaeales=42\n",
    "print(1248-(56+63+67+68))\n",
    "print((56+63+67+68)) #f_haloarculaceae, o_methanosarcinales, c_thermoplasmata, o_nitrososphaerales\n",
    "print((84+40+67+68))\n",
    "print((63+67+68+40)) #methanosarcinales, thermoplasmata, nitrososphaerales, bathyarchaeia\n",
    "print((63+68),'in valid set - o_Methanosarcinales and o_Nitrososphaerales') \n",
    "print((67+40),'in test set - c_Thermoplasmata and c_Bathyarchaeia')\n",
    "\n",
    "#for new arc set - just swap test and valid \n",
    "print((67+40),'in new valid set - c_Thermoplasmata and c_Bathyarchaeia')\n",
    "print((63+68),'in new test set - o_Methanosarcinales and o_Nitrososphaerales') \n",
    "\n",
    "#at class level: valid = c_Thermoplasmata and c_Bathyarchaeia; test = c__Methanosarcinia and c__Nitrososphaeria"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "new valid set size: 1630 0.06597587630535093\n",
      "new test set size: 2121 0.0858495911924229\n"
     ]
    }
   ],
   "source": [
    "print('new valid set size:',(107+1523), (107+1523)/24706)\n",
    "print('new test set size:', (131+1990), (131+1990)/24706)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "#in ['c_Bathyarchaeia','o__Methanosarcinales','c__Thermoplasmata','o__Nitrososphaerales']\n",
    "\n",
    "def get_subset_leaves(node,to_match,taxa=[]):\n",
    "    for child in node.child_nodes():\n",
    "        if child.label:\n",
    "            if len([s for s in to_match if s in child.label])>0:\n",
    "                count=count_children(child,count=0)\n",
    "                print('appending',count,'labels from node', child)\n",
    "                these_taxa = [leaf.taxon.label for leaf in child.leaf_nodes()]\n",
    "                taxa = taxa.extend(these_taxa)\n",
    "        get_subset_leaves(child,to_match,taxa=taxa)\n",
    "    return taxa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 282,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['GB_GCA_002780705.1', 'GB_GCA_002841685.1', 'GB_GCA_002779055.1']"
      ]
     },
     "execution_count": 282,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[leaf.taxon.label for leaf in child.leaf_nodes()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "appending 63 labels from node <Node object at 0x10f76a550: '83.0:o__Methanosarcinales' (None)>\n",
      "appending 68 labels from node <Node object at 0x10e5b28d0: '100.0:o__Nitrososphaerales' (None)>\n",
      "appending 67 labels from node <Node object at 0x10f654908: '86.0:c__Thermoplasmata' (None)>\n",
      "appending 40 labels from node <Node object at 0x10d12f6a0: '93.0:c__Bathyarchaeia' (None)>\n",
      "131 107\n"
     ]
    }
   ],
   "source": [
    "#arc_taxa = get_subset_leaves(arc.seed_node, to_match=['f__Haloarculaceae','o__Methanosarcinales','c__Thermoplasmata','o__Nitrososphaerales'])\n",
    "arc_valid = get_subset_leaves(arc.seed_node, to_match=['o__Methanosarcinales','o__Nitrososphaerales'],taxa=[])\n",
    "arc_test = get_subset_leaves(arc.seed_node, to_match=['c__Thermoplasmata','c__Bathyarchaeia'],taxa=[])\n",
    "print(len(arc_valid),len(arc_test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "appending 944 labels from node <Node object at 0x114ea6fd0: '91.0:o__Actinomycetales' (None)>\n",
      "appending 579 labels from node <Node object at 0x116263320: '93.0:o__Rhodobacterales' (None)>\n",
      "appending 1029 labels from node <Node object at 0x1154791d0: '91.0:o__Bacteroidales' (None)>\n",
      "appending 961 labels from node <Node object at 0x1161002e8: '20.0:o__Rhizobiales' (None)>\n",
      "1523 1990\n"
     ]
    }
   ],
   "source": [
    "#bac_taxa = get_subset_leaves(bac.seed_node, to_match=['p__Firmicutes_A','o__Rhodobacterales','o__Bacteroidales','o__Mycobacteriales'])\n",
    "bac_valid = get_subset_leaves(bac.seed_node, to_match=['p__Firmicutes_A','o__Rhodobacterales'],taxa=[])\n",
    "bac_test = get_subset_leaves(bac.seed_node, to_match=['o__Bacteroidales','o__Mycobacteriales'],taxa=[])\n",
    "print(len(bac_valid),len(bac_test))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 351,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['GB_GCA_900317315.1',\n",
       " 'GB_GCA_900316815.1',\n",
       " 'GB_GCA_900317595.1',\n",
       " 'GB_GCA_900316435.1',\n",
       " 'GB_GCA_900318905.1',\n",
       " 'GB_GCA_900315195.1',\n",
       " 'GB_GCA_900317875.1',\n",
       " 'GB_GCA_900315085.1',\n",
       " 'GB_GCA_900318495.1',\n",
       " 'GB_GCA_003520555.1']"
      ]
     },
     "execution_count": 351,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bac_taxa[0:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ncbi_assembly_name</th>\n",
       "      <th>ncbi_genbank_assembly_accession</th>\n",
       "      <th>ncbi_bioproject</th>\n",
       "      <th>ncbi_biosample</th>\n",
       "      <th>ncbi_assembly_level</th>\n",
       "      <th>ncbi_organism_name</th>\n",
       "      <th>ncbi_refseq_category</th>\n",
       "      <th>ncbi_genome_representation</th>\n",
       "      <th>parsed_genome_accession</th>\n",
       "      <th># assembly_accession</th>\n",
       "      <th>ftp_path</th>\n",
       "      <th>asm_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ASM582v2</td>\n",
       "      <td>GCA_000005825.2</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02603086</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>Bacillus pseudofirmus OF4</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>full</td>\n",
       "      <td>GCF_000005825.2</td>\n",
       "      <td>GCA_000005825.2</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>ASM582v2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ASM668v1</td>\n",
       "      <td>GCA_000006685.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02603994</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>Chlamydia muridarum str. Nigg</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>full</td>\n",
       "      <td>GCF_000006685.1</td>\n",
       "      <td>GCA_000006685.1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>ASM668v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ASM694v2</td>\n",
       "      <td>GCA_000006945.2</td>\n",
       "      <td>PRJNA57799</td>\n",
       "      <td>SAMN02604315</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>Salmonella enterica subsp. enterica serovar Ty...</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>full</td>\n",
       "      <td>GCF_000006945.2</td>\n",
       "      <td>GCA_000006945.2</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>ASM694v2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ASM696v1</td>\n",
       "      <td>GCA_000006965.1</td>\n",
       "      <td>PRJNA57603</td>\n",
       "      <td>SAMEA3283068</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>Sinorhizobium meliloti 1021</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>full</td>\n",
       "      <td>GCF_000006965.1</td>\n",
       "      <td>GCA_000006965.1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>ASM696v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ASM698v1</td>\n",
       "      <td>GCA_000006985.1</td>\n",
       "      <td>PRJNA57897</td>\n",
       "      <td>SAMN02604006</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>Chlorobaculum tepidum TLS</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>full</td>\n",
       "      <td>GCF_000006985.1</td>\n",
       "      <td>GCA_000006985.1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>ASM698v1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  ncbi_assembly_name ncbi_genbank_assembly_accession ncbi_bioproject  \\\n",
       "0           ASM582v2                 GCA_000005825.2     PRJNA224116   \n",
       "1           ASM668v1                 GCA_000006685.1     PRJNA224116   \n",
       "2           ASM694v2                 GCA_000006945.2      PRJNA57799   \n",
       "3           ASM696v1                 GCA_000006965.1      PRJNA57603   \n",
       "4           ASM698v1                 GCA_000006985.1      PRJNA57897   \n",
       "\n",
       "  ncbi_biosample ncbi_assembly_level  \\\n",
       "0   SAMN02603086     Complete Genome   \n",
       "1   SAMN02603994     Complete Genome   \n",
       "2   SAMN02604315     Complete Genome   \n",
       "3   SAMEA3283068     Complete Genome   \n",
       "4   SAMN02604006     Complete Genome   \n",
       "\n",
       "                                  ncbi_organism_name   ncbi_refseq_category  \\\n",
       "0                          Bacillus pseudofirmus OF4  representative genome   \n",
       "1                      Chlamydia muridarum str. Nigg  representative genome   \n",
       "2  Salmonella enterica subsp. enterica serovar Ty...       reference genome   \n",
       "3                        Sinorhizobium meliloti 1021       reference genome   \n",
       "4                          Chlorobaculum tepidum TLS       reference genome   \n",
       "\n",
       "  ncbi_genome_representation parsed_genome_accession # assembly_accession  \\\n",
       "0                       full         GCF_000005825.2      GCA_000005825.2   \n",
       "1                       full         GCF_000006685.1      GCA_000006685.1   \n",
       "2                       full         GCF_000006945.2      GCA_000006945.2   \n",
       "3                       full         GCF_000006965.1      GCA_000006965.1   \n",
       "4                       full         GCF_000006985.1      GCA_000006985.1   \n",
       "\n",
       "                                            ftp_path  asm_name  \n",
       "0  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...  ASM582v2  \n",
       "1  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...  ASM668v1  \n",
       "2  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...  ASM694v2  \n",
       "3  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...  ASM696v1  \n",
       "4  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...  ASM698v1  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ftp = pd.read_csv(\"GTDBrepGenomes_withftp.csv\")\n",
    "ftp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 320,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23458\n",
      "4442\n",
      "4442\n",
      "4442\n",
      "19016\n"
     ]
    }
   ],
   "source": [
    "parsed = pd.DataFrame()\n",
    "start = len(ftp)\n",
    "print(start)\n",
    "for b in bac_taxa:\n",
    "    if b[3:] in list(ftp['ncbi_genbank_assembly_accession']):\n",
    "        parsed = parsed.append(ftp[ftp['ncbi_genbank_assembly_accession']==b[3:]])\n",
    "        ftp.drop(ftp[ftp['ncbi_genbank_assembly_accession']==b[3:]].index,inplace=True)\n",
    "    elif b[3:] in list(ftp['parsed_genome_accession']):\n",
    "        parsed = parsed.append(ftp[ftp['parsed_genome_accession']==b[3:]])\n",
    "        ftp.drop(ftp[ftp['parsed_genome_accession']==b[3:]].index, inplace=True)\n",
    "print(len(parsed))\n",
    "print(len(bac_taxa))\n",
    "print(start-len(ftp))\n",
    "print(len(ftp))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 330,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2392\n",
      "1248\n",
      "['accession', 'ambiguous_bases', 'checkm_completeness', 'checkm_contamination', 'checkm_marker_count', 'checkm_marker_lineage', 'checkm_marker_set_count', 'checkm_strain_heterogeneity', 'coding_bases', 'coding_density', 'contig_count', 'gc_count', 'gc_percentage', 'genome_size', 'gtdb_genome_representative', 'gtdb_representative', 'gtdb_taxonomy', 'gtdb_type_designation', 'gtdb_type_designation_sources', 'gtdb_type_species_of_genus', 'l50_contigs', 'l50_scaffolds', 'longest_contig', 'longest_scaffold', 'lsu_23s_contig_len', 'lsu_23s_count', 'lsu_23s_length', 'lsu_23s_query_id', 'lsu_5s_contig_len', 'lsu_5s_count', 'lsu_5s_length', 'lsu_5s_query_id', 'lsu_silva_23s_blast_align_len', 'lsu_silva_23s_blast_bitscore', 'lsu_silva_23s_blast_evalue', 'lsu_silva_23s_blast_perc_identity', 'lsu_silva_23s_blast_subject_id', 'lsu_silva_23s_taxonomy', 'mean_contig_length', 'mean_scaffold_length', 'mimag_high_quality', 'mimag_low_quality', 'mimag_medium_quality', 'n50_contigs', 'n50_scaffolds', 'ncbi_assembly_level', 'ncbi_assembly_name', 'ncbi_assembly_type', 'ncbi_bioproject', 'ncbi_biosample', 'ncbi_contig_count', 'ncbi_contig_n50', 'ncbi_country', 'ncbi_date', 'ncbi_genbank_assembly_accession', 'ncbi_genome_category', 'ncbi_genome_representation', 'ncbi_isolate', 'ncbi_isolation_source', 'ncbi_lat_lon', 'ncbi_molecule_count', 'ncbi_ncrna_count', 'ncbi_organism_name', 'ncbi_protein_count', 'ncbi_refseq_category', 'ncbi_rrna_count', 'ncbi_scaffold_count', 'ncbi_scaffold_l50', 'ncbi_scaffold_n50', 'ncbi_scaffold_n75', 'ncbi_scaffold_n90', 'ncbi_seq_rel_date', 'ncbi_spanned_gaps', 'ncbi_species_taxid', 'ncbi_ssu_count', 'ncbi_strain_identifiers', 'ncbi_submitter', 'ncbi_taxid', 'ncbi_taxonomy', 'ncbi_taxonomy_unfiltered', 'ncbi_total_gap_length', 'ncbi_total_length', 'ncbi_translation_table', 'ncbi_trna_count', 'ncbi_type_material_designation', 'ncbi_ungapped_length', 'ncbi_unspanned_gaps', 'ncbi_wgs_master', 'protein_count', 'scaffold_count', 'ssu_contig_len', 'ssu_count', 'ssu_gg_blast_align_len', 'ssu_gg_blast_bitscore', 'ssu_gg_blast_evalue', 'ssu_gg_blast_perc_identity', 'ssu_gg_blast_subject_id', 'ssu_gg_taxonomy', 'ssu_length', 'ssu_query_id', 'ssu_silva_blast_align_len', 'ssu_silva_blast_bitscore', 'ssu_silva_blast_evalue', 'ssu_silva_blast_perc_identity', 'ssu_silva_blast_subject_id', 'ssu_silva_taxonomy', 'total_gap_length', 'trna_aa_count', 'trna_count', 'trna_selenocysteine_count']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession</th>\n",
       "      <th>ambiguous_bases</th>\n",
       "      <th>checkm_completeness</th>\n",
       "      <th>checkm_contamination</th>\n",
       "      <th>checkm_marker_count</th>\n",
       "      <th>checkm_marker_lineage</th>\n",
       "      <th>checkm_marker_set_count</th>\n",
       "      <th>checkm_strain_heterogeneity</th>\n",
       "      <th>coding_bases</th>\n",
       "      <th>coding_density</th>\n",
       "      <th>...</th>\n",
       "      <th>ssu_silva_blast_align_len</th>\n",
       "      <th>ssu_silva_blast_bitscore</th>\n",
       "      <th>ssu_silva_blast_evalue</th>\n",
       "      <th>ssu_silva_blast_perc_identity</th>\n",
       "      <th>ssu_silva_blast_subject_id</th>\n",
       "      <th>ssu_silva_taxonomy</th>\n",
       "      <th>total_gap_length</th>\n",
       "      <th>trna_aa_count</th>\n",
       "      <th>trna_count</th>\n",
       "      <th>trna_selenocysteine_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GB_GCA_002503125.1</td>\n",
       "      <td>0</td>\n",
       "      <td>98.40</td>\n",
       "      <td>0.8</td>\n",
       "      <td>188</td>\n",
       "      <td>p__Euryarchaeota (UID3)</td>\n",
       "      <td>125</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1639333</td>\n",
       "      <td>76.576757</td>\n",
       "      <td>...</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>8999</td>\n",
       "      <td>20</td>\n",
       "      <td>41</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GB_GCA_002497145.1</td>\n",
       "      <td>0</td>\n",
       "      <td>77.08</td>\n",
       "      <td>0.8</td>\n",
       "      <td>188</td>\n",
       "      <td>p__Euryarchaeota (UID3)</td>\n",
       "      <td>125</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1514651</td>\n",
       "      <td>93.723880</td>\n",
       "      <td>...</td>\n",
       "      <td>558</td>\n",
       "      <td>725</td>\n",
       "      <td>0</td>\n",
       "      <td>90.143</td>\n",
       "      <td>DQ156348.18592.20063</td>\n",
       "      <td>Archaea;Euryarchaeota;Thermoplasmata;Thermopla...</td>\n",
       "      <td>9871</td>\n",
       "      <td>18</td>\n",
       "      <td>31</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GB_GCA_002503185.1</td>\n",
       "      <td>0</td>\n",
       "      <td>85.68</td>\n",
       "      <td>0.0</td>\n",
       "      <td>188</td>\n",
       "      <td>p__Euryarchaeota (UID3)</td>\n",
       "      <td>125</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1408875</td>\n",
       "      <td>77.368162</td>\n",
       "      <td>...</td>\n",
       "      <td>1439</td>\n",
       "      <td>2531</td>\n",
       "      <td>0</td>\n",
       "      <td>98.402</td>\n",
       "      <td>FJ167436.1.1440</td>\n",
       "      <td>Archaea;Euryarchaeota;Methanobacteria;Methanob...</td>\n",
       "      <td>69581</td>\n",
       "      <td>20</td>\n",
       "      <td>42</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GB_GCA_002506705.1</td>\n",
       "      <td>0</td>\n",
       "      <td>54.21</td>\n",
       "      <td>0.0</td>\n",
       "      <td>188</td>\n",
       "      <td>p__Euryarchaeota (UID3)</td>\n",
       "      <td>125</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1053371</td>\n",
       "      <td>91.473234</td>\n",
       "      <td>...</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>15699</td>\n",
       "      <td>13</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GB_GCA_002497265.1</td>\n",
       "      <td>0</td>\n",
       "      <td>71.22</td>\n",
       "      <td>1.6</td>\n",
       "      <td>188</td>\n",
       "      <td>p__Euryarchaeota (UID3)</td>\n",
       "      <td>125</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1151737</td>\n",
       "      <td>88.512361</td>\n",
       "      <td>...</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>33568</td>\n",
       "      <td>15</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 110 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            accession  ambiguous_bases  checkm_completeness  \\\n",
       "0  GB_GCA_002503125.1                0                98.40   \n",
       "1  GB_GCA_002497145.1                0                77.08   \n",
       "2  GB_GCA_002503185.1                0                85.68   \n",
       "3  GB_GCA_002506705.1                0                54.21   \n",
       "4  GB_GCA_002497265.1                0                71.22   \n",
       "\n",
       "   checkm_contamination  checkm_marker_count    checkm_marker_lineage  \\\n",
       "0                   0.8                  188  p__Euryarchaeota (UID3)   \n",
       "1                   0.8                  188  p__Euryarchaeota (UID3)   \n",
       "2                   0.0                  188  p__Euryarchaeota (UID3)   \n",
       "3                   0.0                  188  p__Euryarchaeota (UID3)   \n",
       "4                   1.6                  188  p__Euryarchaeota (UID3)   \n",
       "\n",
       "   checkm_marker_set_count  checkm_strain_heterogeneity  coding_bases  \\\n",
       "0                      125                          0.0       1639333   \n",
       "1                      125                          0.0       1514651   \n",
       "2                      125                          0.0       1408875   \n",
       "3                      125                          0.0       1053371   \n",
       "4                      125                          0.0       1151737   \n",
       "\n",
       "   coding_density            ...             ssu_silva_blast_align_len  \\\n",
       "0       76.576757            ...                                  none   \n",
       "1       93.723880            ...                                   558   \n",
       "2       77.368162            ...                                  1439   \n",
       "3       91.473234            ...                                  none   \n",
       "4       88.512361            ...                                  none   \n",
       "\n",
       "   ssu_silva_blast_bitscore  ssu_silva_blast_evalue  \\\n",
       "0                      none                    none   \n",
       "1                       725                       0   \n",
       "2                      2531                       0   \n",
       "3                      none                    none   \n",
       "4                      none                    none   \n",
       "\n",
       "   ssu_silva_blast_perc_identity ssu_silva_blast_subject_id  \\\n",
       "0                           none                       none   \n",
       "1                         90.143       DQ156348.18592.20063   \n",
       "2                         98.402            FJ167436.1.1440   \n",
       "3                           none                       none   \n",
       "4                           none                       none   \n",
       "\n",
       "                                  ssu_silva_taxonomy total_gap_length  \\\n",
       "0                                               none             8999   \n",
       "1  Archaea;Euryarchaeota;Thermoplasmata;Thermopla...             9871   \n",
       "2  Archaea;Euryarchaeota;Methanobacteria;Methanob...            69581   \n",
       "3                                               none            15699   \n",
       "4                                               none            33568   \n",
       "\n",
       "  trna_aa_count trna_count trna_selenocysteine_count  \n",
       "0            20         41                         0  \n",
       "1            18         31                         0  \n",
       "2            20         42                         0  \n",
       "3            13         20                         0  \n",
       "4            15         30                         0  \n",
       "\n",
       "[5 rows x 110 columns]"
      ]
     },
     "execution_count": 330,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "arc = pd.read_csv(\"ar122_metadata_r89.tsv\",sep='\\t')\n",
    "print(len(arc))\n",
    "print(len(arc[arc['gtdb_representative']=='t']))\n",
    "print(list(arc.columns))\n",
    "arc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#do the same for archaea\n",
    "\n",
    "start = len(ftp)\n",
    "print(start)\n",
    "for b in bac_taxa:\n",
    "    if b[3:] in list(ftp['ncbi_genbank_assembly_accession']):\n",
    "        parsed = parsed.append(ftp[ftp['ncbi_genbank_assembly_accession']==b[3:]])\n",
    "        ftp.drop(ftp[ftp['ncbi_genbank_assembly_accession']==b[3:]].index,inplace=True)\n",
    "    elif b[3:] in list(ftp['parsed_genome_accession']):\n",
    "        parsed = parsed.append(ftp[ftp['parsed_genome_accession']==b[3:]])\n",
    "        ftp.drop(ftp[ftp['parsed_genome_accession']==b[3:]].index, inplace=True)\n",
    "print(len(parsed))\n",
    "print(len(bac_taxa))\n",
    "print(start-len(ftp))\n",
    "print(len(ftp))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 306,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'GCA_900317315.1'"
      ]
     },
     "execution_count": 306,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bac_taxa[0][3:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 311,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 311,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ftp[ftp['ncbi_genbank_assembly_accession']=='GCA_002493005.1']\n",
    "'GCA_002493005.1' in list(ftp['ncbi_genbank_assembly_accession'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Selecting smaller valid set - was an issue for having a continuous validation set with reasonably sized training chunks held in CPU memory\n",
    "128GB CPU could handle 18000 seqs/file * 2596 valid files, about 80% of the current 22333 seqs/file\n",
    "\n",
    "want ~80% size original valid set - or (2465+131)*0.8 = 2077, ~1972+105 bac+arc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "appending 67 labels from node <Node object at 0x1073f5630: '86.0:c__Thermoplasmata' (None)>\n",
      "appending 40 labels from node <Node object at 0x10888f908: '93.0:c__Bathyarchaeia' (None)>\n",
      "appending 63 labels from node <Node object at 0x107467d68: '83.0:o__Methanosarcinales' (None)>\n",
      "appending 68 labels from node <Node object at 0x1075f43c8: '100.0:o__Nitrososphaerales' (None)>\n",
      "107 131\n"
     ]
    }
   ],
   "source": [
    "#see above for new sets\n",
    "arc_valid = get_subset_leaves(arc.seed_node, to_match=['c__Thermoplasmata','c__Bathyarchaeia'],taxa=[])\n",
    "arc_test = get_subset_leaves(arc.seed_node, to_match=['o__Methanosarcinales','o__Nitrososphaerales'],taxa=[])\n",
    "print(len(arc_valid),len(arc_test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "appending 944 labels from node <Node object at 0x107376eb8: '91.0:o__Actinomycetales' (None)>\n",
      "appending 579 labels from node <Node object at 0x1087b4208: '93.0:o__Rhodobacterales' (None)>\n",
      "appending 1029 labels from node <Node object at 0x10794a0b8: '91.0:o__Bacteroidales' (None)>\n",
      "appending 961 labels from node <Node object at 0x1086521d0: '20.0:o__Rhizobiales' (None)>\n",
      "1523 1990\n"
     ]
    }
   ],
   "source": [
    "#see above for new sets\n",
    "bac_valid = get_subset_leaves(bac.seed_node, to_match=['o__Actinomycetales','o__Rhodobacterales'],taxa=[])\n",
    "bac_test = get_subset_leaves(bac.seed_node, to_match=['o__Bacteroidales','o__Rhizobiales'],taxa=[])\n",
    "print(len(bac_valid),len(bac_test))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['GB_GCA_002503545.1',\n",
       " 'RS_GCF_000300255.2',\n",
       " 'RS_GCF_001481295.1',\n",
       " 'RS_GCF_001560915.1',\n",
       " 'RS_GCF_000350305.1']"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "arc_valid[0:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "23458"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ftp = pd.read_csv(\"GTDBrepGenomes_withftp.csv\")\n",
    "len(ftp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23458\n",
      "1523\n",
      "1523\n",
      "1523\n",
      "21935\n"
     ]
    }
   ],
   "source": [
    "valdf = pd.DataFrame()\n",
    "start = len(ftp)\n",
    "print(start)\n",
    "for b in bac_valid:\n",
    "    if b[3:] in list(ftp['ncbi_genbank_assembly_accession']):\n",
    "        valdf = valdf.append(ftp[ftp['ncbi_genbank_assembly_accession']==b[3:]])\n",
    "        ftp.drop(ftp[ftp['ncbi_genbank_assembly_accession']==b[3:]].index,inplace=True)\n",
    "    elif b[3:] in list(ftp['parsed_genome_accession']):\n",
    "        valdf = valdf.append(ftp[ftp['parsed_genome_accession']==b[3:]])\n",
    "        ftp.drop(ftp[ftp['parsed_genome_accession']==b[3:]].index, inplace=True)\n",
    "print(len(valdf))\n",
    "print(len(bac_valid))\n",
    "print(start-len(ftp))\n",
    "print(len(ftp))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21935\n",
      "1990\n",
      "1990\n",
      "1990\n",
      "19945\n"
     ]
    }
   ],
   "source": [
    "testdf = pd.DataFrame()\n",
    "start = len(ftp)\n",
    "print(start)\n",
    "for b in bac_test:\n",
    "    if b[3:] in list(ftp['ncbi_genbank_assembly_accession']):\n",
    "        testdf = testdf.append(ftp[ftp['ncbi_genbank_assembly_accession']==b[3:]])\n",
    "        ftp.drop(ftp[ftp['ncbi_genbank_assembly_accession']==b[3:]].index,inplace=True)\n",
    "    elif b[3:] in list(ftp['parsed_genome_accession']):\n",
    "        testdf = testdf.append(ftp[ftp['parsed_genome_accession']==b[3:]])\n",
    "        ftp.drop(ftp[ftp['parsed_genome_accession']==b[3:]].index, inplace=True)\n",
    "print(len(testdf))\n",
    "print(len(bac_test))\n",
    "print(start-len(ftp))\n",
    "print(len(ftp))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th># assembly_accession</th>\n",
       "      <th>asm_name</th>\n",
       "      <th>ftp_path</th>\n",
       "      <th>index</th>\n",
       "      <th>ncbi_assembly_level</th>\n",
       "      <th>ncbi_assembly_name</th>\n",
       "      <th>ncbi_bioproject</th>\n",
       "      <th>ncbi_biosample</th>\n",
       "      <th>ncbi_genbank_assembly_accession</th>\n",
       "      <th>ncbi_genome_representation</th>\n",
       "      <th>ncbi_organism_name</th>\n",
       "      <th>ncbi_refseq_category</th>\n",
       "      <th>parsed_genome_accession</th>\n",
       "      <th>srr_accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1243</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA9912</td>\n",
       "      <td>SRR10597655</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1244</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA9915</td>\n",
       "      <td>SRR10597654</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1245</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA9950</td>\n",
       "      <td>SRR10597653</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1246</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA9966</td>\n",
       "      <td>SRR10597652</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1247</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA9989</td>\n",
       "      <td>SRR10597651</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     # assembly_accession asm_name ftp_path index ncbi_assembly_level  \\\n",
       "1243                  NaN      NaN      NaN   NaN                none   \n",
       "1244                  NaN      NaN      NaN   NaN                none   \n",
       "1245                  NaN      NaN      NaN   NaN                none   \n",
       "1246                  NaN      NaN      NaN   NaN                none   \n",
       "1247                  NaN      NaN      NaN   NaN                none   \n",
       "\n",
       "     ncbi_assembly_name ncbi_bioproject ncbi_biosample  \\\n",
       "1243               none            none           none   \n",
       "1244               none            none           none   \n",
       "1245               none            none           none   \n",
       "1246               none            none           none   \n",
       "1247               none            none           none   \n",
       "\n",
       "     ncbi_genbank_assembly_accession ncbi_genome_representation  \\\n",
       "1243                            none                       none   \n",
       "1244                            none                       none   \n",
       "1245                            none                       none   \n",
       "1246                            none                       none   \n",
       "1247                            none                       none   \n",
       "\n",
       "     ncbi_organism_name ncbi_refseq_category parsed_genome_accession  \\\n",
       "1243               none                 none                 UBA9912   \n",
       "1244               none                 none                 UBA9915   \n",
       "1245               none                 none                 UBA9950   \n",
       "1246               none                 none                 UBA9966   \n",
       "1247               none                 none                 UBA9989   \n",
       "\n",
       "     srr_accession  \n",
       "1243   SRR10597655  \n",
       "1244   SRR10597654  \n",
       "1245   SRR10597653  \n",
       "1246   SRR10597652  \n",
       "1247   SRR10597651  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#so now we have valdf, testdf, and ftp is train (for bacteria only so far)\n",
    "#now do same for archaea\n",
    "arcftp = pd.read_csv(\"GTDBrepGenomes_arc_withftpsrr.csv\")\n",
    "arcftp.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'arcftp' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-15-7b82a59bf05d>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0marc_valdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mstart\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marcftp\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstart\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mb\u001b[0m \u001b[0;32min\u001b[0m \u001b[0marc_valid\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mb\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m3\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marcftp\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'ncbi_genbank_assembly_accession'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'arcftp' is not defined"
     ]
    }
   ],
   "source": [
    "arc_valdf = pd.DataFrame()\n",
    "start = len(arcftp)\n",
    "print(start)\n",
    "for b in arc_valid:\n",
    "    if b[3:] in list(arcftp['ncbi_genbank_assembly_accession']):\n",
    "        arc_valdf = arc_valdf.append(arcftp[arcftp['ncbi_genbank_assembly_accession']==b[3:]])\n",
    "        arcftp.drop(arcftp[arcftp['ncbi_genbank_assembly_accession']==b[3:]].index,inplace=True)\n",
    "    elif b[3:] in list(arcftp['parsed_genome_accession']):\n",
    "        arc_valdf = arc_valdf.append(arcftp[arcftp['parsed_genome_accession']==b[3:]])\n",
    "        arcftp.drop(arcftp[arcftp['parsed_genome_accession']==b[3:]].index, inplace=True)\n",
    "    elif b in list(arcftp['parsed_genome_accession']): #for when only have SRR for e.g. UBA9915\n",
    "        arc_valdf = arc_valdf.append(arcftp[arcftp['parsed_genome_accession']==b])\n",
    "        arcftp.drop(arcftp[arcftp['parsed_genome_accession']==b].index, inplace=True)\n",
    "    else:\n",
    "        print(b)\n",
    "print(len(arc_valdf))\n",
    "print(len(arc_valid))\n",
    "print(start-len(arcftp))\n",
    "print(len(arcftp))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1141\n",
      "131\n",
      "131\n",
      "131\n",
      "1010\n"
     ]
    }
   ],
   "source": [
    "arc_testdf = pd.DataFrame()\n",
    "start = len(arcftp)\n",
    "print(start)\n",
    "for b in arc_test:\n",
    "    if b[3:] in list(arcftp['ncbi_genbank_assembly_accession']):\n",
    "        arc_testdf = arc_testdf.append(arcftp[arcftp['ncbi_genbank_assembly_accession']==b[3:]])\n",
    "        arcftp.drop(arcftp[arcftp['ncbi_genbank_assembly_accession']==b[3:]].index,inplace=True)\n",
    "    elif b[3:] in list(arcftp['parsed_genome_accession']):\n",
    "        arc_testdf = arc_testdf.append(arcftp[arcftp['parsed_genome_accession']==b[3:]])\n",
    "        arcftp.drop(arcftp[arcftp['parsed_genome_accession']==b[3:]].index, inplace=True)\n",
    "    elif b in list(arcftp['parsed_genome_accession']): #for when only have SRR for e.g. UBA9915\n",
    "        arc_testdf = arc_testdf.append(arcftp[arcftp['parsed_genome_accession']==b])\n",
    "        arcftp.drop(arcftp[arcftp['parsed_genome_accession']==b].index, inplace=True)\n",
    "    else:\n",
    "        print(b)\n",
    "print(len(arc_testdf))\n",
    "print(len(arc_test))\n",
    "print(start-len(arcftp))\n",
    "print(len(arcftp))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20955 20955\n",
      "1630 1630\n",
      "2121 2121\n"
     ]
    }
   ],
   "source": [
    "#so now arcftp is train, arc_valdf is valid, and arc_testdf is test (for archaea only)\n",
    "#now merge bac and arc train/valid/test sets\n",
    "\n",
    "train = ftp.append(arcftp, sort=True)\n",
    "print(len(ftp)+len(arcftp), len(train))\n",
    "valid = valdf.append(arc_valdf,sort=True)\n",
    "print(len(valdf)+len(arc_valdf),len(valid))\n",
    "test = testdf.append(arc_testdf,sort=True)\n",
    "print(len(testdf)+len(arc_testdf),len(test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [],
   "source": [
    "train.to_csv('TrainNew_GTDBrepwithftp.csv')\n",
    "valid.to_csv('ValidNew_GTDBrepwithftp.csv')\n",
    "test.to_csv('TestNew_GTDBrepwithftp.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [],
   "source": [
    "#extract filenames and save as text file\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th># assembly_accession</th>\n",
       "      <th>asm_name</th>\n",
       "      <th>ftp_path</th>\n",
       "      <th>index</th>\n",
       "      <th>ncbi_assembly_level</th>\n",
       "      <th>ncbi_assembly_name</th>\n",
       "      <th>ncbi_bioproject</th>\n",
       "      <th>ncbi_biosample</th>\n",
       "      <th>ncbi_genbank_assembly_accession</th>\n",
       "      <th>ncbi_genome_representation</th>\n",
       "      <th>ncbi_organism_name</th>\n",
       "      <th>ncbi_refseq_category</th>\n",
       "      <th>parsed_genome_accession</th>\n",
       "      <th>srr_accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000005825.2</td>\n",
       "      <td>ASM582v2</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM582v2</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02603086</td>\n",
       "      <td>GCA_000005825.2</td>\n",
       "      <td>full</td>\n",
       "      <td>Bacillus pseudofirmus OF4</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000005825.2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000006685.1</td>\n",
       "      <td>ASM668v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM668v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02603994</td>\n",
       "      <td>GCA_000006685.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Chlamydia muridarum str. Nigg</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000006685.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000006945.2</td>\n",
       "      <td>ASM694v2</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM694v2</td>\n",
       "      <td>PRJNA57799</td>\n",
       "      <td>SAMN02604315</td>\n",
       "      <td>GCA_000006945.2</td>\n",
       "      <td>full</td>\n",
       "      <td>Salmonella enterica subsp. enterica serovar Ty...</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>GCF_000006945.2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000006985.1</td>\n",
       "      <td>ASM698v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM698v1</td>\n",
       "      <td>PRJNA57897</td>\n",
       "      <td>SAMN02604006</td>\n",
       "      <td>GCA_000006985.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Chlorobaculum tepidum TLS</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>GCF_000006985.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GCA_000007025.1</td>\n",
       "      <td>ASM702v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM702v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02603141</td>\n",
       "      <td>GCA_000007025.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Rickettsia conorii str. Malish 7</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000007025.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  # assembly_accession  asm_name  \\\n",
       "0      GCA_000005825.2  ASM582v2   \n",
       "1      GCA_000006685.1  ASM668v1   \n",
       "2      GCA_000006945.2  ASM694v2   \n",
       "4      GCA_000006985.1  ASM698v1   \n",
       "5      GCA_000007025.1  ASM702v1   \n",
       "\n",
       "                                            ftp_path index  \\\n",
       "0  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...   NaN   \n",
       "1  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...   NaN   \n",
       "2  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...   NaN   \n",
       "4  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...   NaN   \n",
       "5  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...   NaN   \n",
       "\n",
       "  ncbi_assembly_level ncbi_assembly_name ncbi_bioproject ncbi_biosample  \\\n",
       "0     Complete Genome           ASM582v2     PRJNA224116   SAMN02603086   \n",
       "1     Complete Genome           ASM668v1     PRJNA224116   SAMN02603994   \n",
       "2     Complete Genome           ASM694v2      PRJNA57799   SAMN02604315   \n",
       "4     Complete Genome           ASM698v1      PRJNA57897   SAMN02604006   \n",
       "5     Complete Genome           ASM702v1     PRJNA224116   SAMN02603141   \n",
       "\n",
       "  ncbi_genbank_assembly_accession ncbi_genome_representation  \\\n",
       "0                 GCA_000005825.2                       full   \n",
       "1                 GCA_000006685.1                       full   \n",
       "2                 GCA_000006945.2                       full   \n",
       "4                 GCA_000006985.1                       full   \n",
       "5                 GCA_000007025.1                       full   \n",
       "\n",
       "                                  ncbi_organism_name   ncbi_refseq_category  \\\n",
       "0                          Bacillus pseudofirmus OF4  representative genome   \n",
       "1                      Chlamydia muridarum str. Nigg  representative genome   \n",
       "2  Salmonella enterica subsp. enterica serovar Ty...       reference genome   \n",
       "4                          Chlorobaculum tepidum TLS       reference genome   \n",
       "5                   Rickettsia conorii str. Malish 7  representative genome   \n",
       "\n",
       "  parsed_genome_accession srr_accession  \n",
       "0         GCF_000005825.2           NaN  \n",
       "1         GCF_000006685.1           NaN  \n",
       "2         GCF_000006945.2           NaN  \n",
       "4         GCF_000006985.1           NaN  \n",
       "5         GCF_000007025.1           NaN  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th># assembly_accession</th>\n",
       "      <th>asm_name</th>\n",
       "      <th>ftp_path</th>\n",
       "      <th>index</th>\n",
       "      <th>ncbi_assembly_level</th>\n",
       "      <th>ncbi_assembly_name</th>\n",
       "      <th>ncbi_bioproject</th>\n",
       "      <th>ncbi_biosample</th>\n",
       "      <th>ncbi_genbank_assembly_accession</th>\n",
       "      <th>ncbi_genome_representation</th>\n",
       "      <th>ncbi_organism_name</th>\n",
       "      <th>ncbi_refseq_category</th>\n",
       "      <th>parsed_genome_accession</th>\n",
       "      <th>srr_accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1241</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA9640</td>\n",
       "      <td>SRR10597658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1243</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA9912</td>\n",
       "      <td>SRR10597655</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1245</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA9950</td>\n",
       "      <td>SRR10597653</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1246</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA9966</td>\n",
       "      <td>SRR10597652</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1247</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA9989</td>\n",
       "      <td>SRR10597651</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     # assembly_accession asm_name ftp_path index ncbi_assembly_level  \\\n",
       "1241                  NaN      NaN      NaN   NaN                none   \n",
       "1243                  NaN      NaN      NaN   NaN                none   \n",
       "1245                  NaN      NaN      NaN   NaN                none   \n",
       "1246                  NaN      NaN      NaN   NaN                none   \n",
       "1247                  NaN      NaN      NaN   NaN                none   \n",
       "\n",
       "     ncbi_assembly_name ncbi_bioproject ncbi_biosample  \\\n",
       "1241               none            none           none   \n",
       "1243               none            none           none   \n",
       "1245               none            none           none   \n",
       "1246               none            none           none   \n",
       "1247               none            none           none   \n",
       "\n",
       "     ncbi_genbank_assembly_accession ncbi_genome_representation  \\\n",
       "1241                            none                       none   \n",
       "1243                            none                       none   \n",
       "1245                            none                       none   \n",
       "1246                            none                       none   \n",
       "1247                            none                       none   \n",
       "\n",
       "     ncbi_organism_name ncbi_refseq_category parsed_genome_accession  \\\n",
       "1241               none                 none                 UBA9640   \n",
       "1243               none                 none                 UBA9912   \n",
       "1245               none                 none                 UBA9950   \n",
       "1246               none                 none                 UBA9966   \n",
       "1247               none                 none                 UBA9989   \n",
       "\n",
       "     srr_accession  \n",
       "1241   SRR10597658  \n",
       "1243   SRR10597655  \n",
       "1245   SRR10597653  \n",
       "1246   SRR10597652  \n",
       "1247   SRR10597651  "
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th># assembly_accession</th>\n",
       "      <th>asm_name</th>\n",
       "      <th>ftp_path</th>\n",
       "      <th>index</th>\n",
       "      <th>ncbi_assembly_level</th>\n",
       "      <th>ncbi_assembly_name</th>\n",
       "      <th>ncbi_bioproject</th>\n",
       "      <th>ncbi_biosample</th>\n",
       "      <th>ncbi_genbank_assembly_accession</th>\n",
       "      <th>ncbi_genome_representation</th>\n",
       "      <th>ncbi_organism_name</th>\n",
       "      <th>ncbi_refseq_category</th>\n",
       "      <th>parsed_genome_accession</th>\n",
       "      <th>srr_accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000005825.2</td>\n",
       "      <td>ASM582v2</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM582v2</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02603086</td>\n",
       "      <td>GCA_000005825.2</td>\n",
       "      <td>full</td>\n",
       "      <td>Bacillus pseudofirmus OF4</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000005825.2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000006685.1</td>\n",
       "      <td>ASM668v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM668v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02603994</td>\n",
       "      <td>GCA_000006685.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Chlamydia muridarum str. Nigg</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000006685.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000006945.2</td>\n",
       "      <td>ASM694v2</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM694v2</td>\n",
       "      <td>PRJNA57799</td>\n",
       "      <td>SAMN02604315</td>\n",
       "      <td>GCA_000006945.2</td>\n",
       "      <td>full</td>\n",
       "      <td>Salmonella enterica subsp. enterica serovar Ty...</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>GCF_000006945.2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000006985.1</td>\n",
       "      <td>ASM698v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM698v1</td>\n",
       "      <td>PRJNA57897</td>\n",
       "      <td>SAMN02604006</td>\n",
       "      <td>GCA_000006985.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Chlorobaculum tepidum TLS</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>GCF_000006985.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GCA_000007025.1</td>\n",
       "      <td>ASM702v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM702v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02603141</td>\n",
       "      <td>GCA_000007025.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Rickettsia conorii str. Malish 7</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000007025.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GCA_000007085.1</td>\n",
       "      <td>ASM708v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM708v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02603009</td>\n",
       "      <td>GCA_000007085.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Caldanaerobacter subterraneus subsp. tengconge...</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000007085.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GCA_000007145.1</td>\n",
       "      <td>ASM714v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM714v1</td>\n",
       "      <td>PRJNA57887</td>\n",
       "      <td>SAMN02603845</td>\n",
       "      <td>GCA_000007145.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Xanthomonas campestris pv. campestris str. ATC...</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>GCF_000007145.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GCA_000007205.1</td>\n",
       "      <td>ASM720v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM720v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02602979</td>\n",
       "      <td>GCA_000007205.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Chlamydia pneumoniae TW-183</td>\n",
       "      <td>na</td>\n",
       "      <td>GCF_000007205.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GCA_000007325.1</td>\n",
       "      <td>ASM732v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM732v1</td>\n",
       "      <td>PRJNA57885</td>\n",
       "      <td>SAMN02603417</td>\n",
       "      <td>GCA_000007325.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Fusobacterium nucleatum subsp. nucleatum ATCC ...</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>GCF_000007325.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GCA_000007365.1</td>\n",
       "      <td>ASM736v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM736v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02604269</td>\n",
       "      <td>GCA_000007365.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Buchnera aphidicola str. Sg (Schizaphis graminum)</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000007365.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>GCA_000007605.1</td>\n",
       "      <td>ASM760v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM760v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02603993</td>\n",
       "      <td>GCA_000007605.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Chlamydia caviae GPIC</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000007605.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GCA_000007705.1</td>\n",
       "      <td>ASM770v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM770v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02603528</td>\n",
       "      <td>GCA_000007705.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Chromobacterium violaceum ATCC 12472</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000007705.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GCA_000007725.1</td>\n",
       "      <td>ASM772v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM772v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02604289</td>\n",
       "      <td>GCA_000007725.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Buchnera aphidicola str. Bp (Baizongia pistaciae)</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000007725.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GCA_000007825.1</td>\n",
       "      <td>ASM782v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM782v1</td>\n",
       "      <td>PRJNA57975</td>\n",
       "      <td>SAMN02603340</td>\n",
       "      <td>GCA_000007825.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Bacillus cereus ATCC 14579</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>GCF_000007825.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GCA_000007905.1</td>\n",
       "      <td>ASM790v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM790v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02604097</td>\n",
       "      <td>GCA_000007905.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Helicobacter hepaticus ATCC 51449</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000007905.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GCA_000007925.1</td>\n",
       "      <td>ASM792v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM792v1</td>\n",
       "      <td>PRJNA57995</td>\n",
       "      <td>SAMN02603142</td>\n",
       "      <td>GCA_000007925.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Prochlorococcus marinus subsp. marinus str. CC...</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>GCF_000007925.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>GCA_000007985.2</td>\n",
       "      <td>ASM798v2</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM798v2</td>\n",
       "      <td>PRJNA57743</td>\n",
       "      <td>SAMN02603989</td>\n",
       "      <td>GCA_000007985.2</td>\n",
       "      <td>full</td>\n",
       "      <td>Geobacter sulfurreducens PCA</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>GCF_000007985.2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GCA_000008025.1</td>\n",
       "      <td>ASM802v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM802v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02604000</td>\n",
       "      <td>GCA_000008025.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Wolbachia endosymbiont of Drosophila melanogaster</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000008025.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GCA_000008045.1</td>\n",
       "      <td>ASM804v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM804v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02603327</td>\n",
       "      <td>GCA_000008045.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Rickettsia typhi str. Wilmington</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000008045.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GCA_000008185.1</td>\n",
       "      <td>ASM818v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM818v1</td>\n",
       "      <td>PRJNA57583</td>\n",
       "      <td>SAMN02603967</td>\n",
       "      <td>GCA_000008185.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Treponema denticola ATCC 35405</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>GCF_000008185.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GCA_000008205.1</td>\n",
       "      <td>ASM820v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM820v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02603526</td>\n",
       "      <td>GCA_000008205.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Mycoplasma hyopneumoniae J</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000008205.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GCA_000008305.1</td>\n",
       "      <td>ASM830v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM830v1</td>\n",
       "      <td>PRJNA58055</td>\n",
       "      <td>SAMN03081416</td>\n",
       "      <td>GCA_000008305.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Mesoplasma florum L1</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>GCF_000008305.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>GCA_000008365.1</td>\n",
       "      <td>ASM836v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM836v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN03081417</td>\n",
       "      <td>GCA_000008365.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Mycoplasma mobile 163K</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000008365.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GCA_000008385.1</td>\n",
       "      <td>ASM838v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM838v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02603678</td>\n",
       "      <td>GCA_000008385.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Wolbachia endosymbiont strain TRS of Brugia ma...</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000008385.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GCA_000008465.1</td>\n",
       "      <td>ASM846v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM846v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02604184</td>\n",
       "      <td>GCA_000008465.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Idiomarina loihiensis L2TR</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000008465.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GCA_000008485.1</td>\n",
       "      <td>ASM848v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM848v1</td>\n",
       "      <td>PRJNA57609</td>\n",
       "      <td>SAMN02603156</td>\n",
       "      <td>GCA_000008485.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Legionella pneumophila subsp. pneumophila str....</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>GCF_000008485.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>GCA_000008545.1</td>\n",
       "      <td>ASM854v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM854v1</td>\n",
       "      <td>PRJNA57723</td>\n",
       "      <td>SAMN02603986</td>\n",
       "      <td>GCA_000008545.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Thermotoga maritima MSB8</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>GCF_000008545.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GCA_000008565.1</td>\n",
       "      <td>ASM856v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM856v1</td>\n",
       "      <td>PRJNA57665</td>\n",
       "      <td>SAMN02603976</td>\n",
       "      <td>GCA_000008565.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Deinococcus radiodurans R1</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>GCF_000008565.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>GCA_000008625.1</td>\n",
       "      <td>ASM862v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM862v1</td>\n",
       "      <td>PRJNA57765</td>\n",
       "      <td>SAMN02603207</td>\n",
       "      <td>GCA_000008625.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Aquifex aeolicus VF5</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>GCF_000008625.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>GCA_000008685.2</td>\n",
       "      <td>ASM868v2</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM868v2</td>\n",
       "      <td>PRJNA57581</td>\n",
       "      <td>SAMN02603966</td>\n",
       "      <td>GCA_000008685.2</td>\n",
       "      <td>full</td>\n",
       "      <td>Borreliella burgdorferi B31</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>GCF_000008685.2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1145</th>\n",
       "      <td>GCA_900111935.1</td>\n",
       "      <td>IMG-taxon_2617270732_annotated_assembly</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900111935.1</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>IMG-taxon 2617270732 annotated assembly</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN04487937</td>\n",
       "      <td>GCA_900111935.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Halorubrum sodomense</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_900111935.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1146</th>\n",
       "      <td>GCA_900112175.1</td>\n",
       "      <td>IMG-taxon_2617270920_annotated_assembly</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900112175.1</td>\n",
       "      <td>Contig</td>\n",
       "      <td>IMG-taxon 2617270920 annotated assembly</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN04487947</td>\n",
       "      <td>GCA_900112175.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Halogeometricum rufum</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_900112175.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1147</th>\n",
       "      <td>GCA_900112205.1</td>\n",
       "      <td>IMG-taxon_2693429869_annotated_assembly</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900112205.1</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>IMG-taxon 2693429869 annotated assembly</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN05444422</td>\n",
       "      <td>GCA_900112205.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Halobiforma haloterrestris</td>\n",
       "      <td>na</td>\n",
       "      <td>GCF_900112205.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1148</th>\n",
       "      <td>GCA_900113245.1</td>\n",
       "      <td>IMG-taxon_2617270917_annotated_assembly</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900113245.1</td>\n",
       "      <td>Contig</td>\n",
       "      <td>IMG-taxon 2617270917 annotated assembly</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN04488063</td>\n",
       "      <td>GCA_900113245.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Halopelagius inordinatus</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_900113245.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1149</th>\n",
       "      <td>GCA_900114435.1</td>\n",
       "      <td>IMG-taxon_2667527389_annotated_assembly</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900114435.1</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>IMG-taxon 2667527389 annotated assembly</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN05216559</td>\n",
       "      <td>GCA_900114435.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Halomicrobium zhouii</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_900114435.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1150</th>\n",
       "      <td>GCA_900114455.1</td>\n",
       "      <td>IMG-taxon_2619618990_annotated_assembly</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900114455.1</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>IMG-taxon 2619618990 annotated assembly</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN04487950</td>\n",
       "      <td>GCA_900114455.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Halogranum rubrum</td>\n",
       "      <td>na</td>\n",
       "      <td>GCF_900114455.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1151</th>\n",
       "      <td>GCA_900114585.1</td>\n",
       "      <td>IMG-taxon_2593339150_annotated_assembly</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900114585.1</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>IMG-taxon 2593339150 annotated assembly</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02910297</td>\n",
       "      <td>GCA_900114585.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Methanobrevibacter olleyae</td>\n",
       "      <td>na</td>\n",
       "      <td>GCF_900114585.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1153</th>\n",
       "      <td>GCA_900115675.1</td>\n",
       "      <td>IMG-taxon_2663762745_annotated_assembly</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900115675.1</td>\n",
       "      <td>Contig</td>\n",
       "      <td>IMG-taxon 2663762745 annotated assembly</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN05216277</td>\n",
       "      <td>GCA_900115675.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Halolamina pelagica</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_900115675.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154</th>\n",
       "      <td>GCA_900115785.1</td>\n",
       "      <td>IMG-taxon_2617270895_annotated_assembly</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900115785.1</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>IMG-taxon 2617270895 annotated assembly</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN04488124</td>\n",
       "      <td>GCA_900115785.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Halogeometricum limi</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_900115785.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1155</th>\n",
       "      <td>GCA_900116205.1</td>\n",
       "      <td>IMG-taxon_2639762563_annotated_assembly</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900116205.1</td>\n",
       "      <td>Contig</td>\n",
       "      <td>IMG-taxon 2639762563 annotated assembly</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN04488556</td>\n",
       "      <td>GCA_900116205.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Halostagnicola kamekurae</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_900116205.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1156</th>\n",
       "      <td>GCA_900129775.1</td>\n",
       "      <td>IMG-taxon_2695420988_annotated_assembly</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900129775.1</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>IMG-taxon 2695420988 annotated assembly</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN05443636</td>\n",
       "      <td>GCA_900129775.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Halobaculum gomorrense</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_900129775.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1158</th>\n",
       "      <td>GCA_900156425.1</td>\n",
       "      <td>IMG-taxon_2681812914_annotated_assembly</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900156425.1</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>IMG-taxon 2681812914 annotated assembly</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN05421858</td>\n",
       "      <td>GCA_900156425.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Haladaptatus litoreus</td>\n",
       "      <td>na</td>\n",
       "      <td>GCF_900156425.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1159</th>\n",
       "      <td>GCA_900156475.1</td>\n",
       "      <td>IMG-taxon_2681812919_annotated_assembly</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900156475.1</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>IMG-taxon 2681812919 annotated assembly</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN05421752</td>\n",
       "      <td>GCA_900156475.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Natronorubrum thiooxidans</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_900156475.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1163</th>\n",
       "      <td>GCA_900188065.1</td>\n",
       "      <td>IMG-taxon_2724679803_annotated_assembly</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900188065.1</td>\n",
       "      <td>Contig</td>\n",
       "      <td>IMG-taxon 2724679803 annotated assembly</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN06264855</td>\n",
       "      <td>GCA_900188065.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Halorubrum vacuolatum</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_900188065.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1165</th>\n",
       "      <td>GCA_900198835.1</td>\n",
       "      <td>chr_EXT12c</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900198835.1</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>chr_EXT12c</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMEA104149373</td>\n",
       "      <td>GCA_900198835.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Thermococcus sp. EXT12c</td>\n",
       "      <td>na</td>\n",
       "      <td>GCF_900198835.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1167</th>\n",
       "      <td>GCA_900215575.1</td>\n",
       "      <td>IMG-taxon_2728369221_annotated_assembly</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900215575.1</td>\n",
       "      <td>Contig</td>\n",
       "      <td>IMG-taxon 2728369221 annotated assembly</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN06269185</td>\n",
       "      <td>GCA_900215575.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Natronoarchaeum philippinense</td>\n",
       "      <td>na</td>\n",
       "      <td>GCF_900215575.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169</th>\n",
       "      <td>GCA_900313645.1</td>\n",
       "      <td>Rumen_uncultured_genome_RUG823</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900313645.1</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>Rumen uncultured genome RUG823</td>\n",
       "      <td>PRJEB21624</td>\n",
       "      <td>SAMEA104666900</td>\n",
       "      <td>GCA_900313645.1</td>\n",
       "      <td>full</td>\n",
       "      <td>uncultured Methanobrevibacter sp.</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_900313645.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1170</th>\n",
       "      <td>GCA_900314605.1</td>\n",
       "      <td>Rumen_uncultured_genome_RUG012</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900314605.1</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>Rumen uncultured genome RUG012</td>\n",
       "      <td>PRJEB21624</td>\n",
       "      <td>SAMEA104666878</td>\n",
       "      <td>GCA_900314605.1</td>\n",
       "      <td>full</td>\n",
       "      <td>uncultured Methanobrevibacter sp.</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_900314605.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1171</th>\n",
       "      <td>GCA_900314615.1</td>\n",
       "      <td>Rumen_uncultured_genome_RUG018</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900314615.1</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>Rumen uncultured genome RUG018</td>\n",
       "      <td>PRJEB21624</td>\n",
       "      <td>SAMEA104666879</td>\n",
       "      <td>GCA_900314615.1</td>\n",
       "      <td>full</td>\n",
       "      <td>uncultured Methanobrevibacter sp.</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_900314615.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1172</th>\n",
       "      <td>GCA_900314635.1</td>\n",
       "      <td>Rumen_uncultured_genome_RUG020</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900314635.1</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>Rumen uncultured genome RUG020</td>\n",
       "      <td>PRJEB21624</td>\n",
       "      <td>SAMEA104666880</td>\n",
       "      <td>GCA_900314635.1</td>\n",
       "      <td>full</td>\n",
       "      <td>uncultured Methanobrevibacter sp.</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_900314635.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1173</th>\n",
       "      <td>GCA_900314695.1</td>\n",
       "      <td>Rumen_uncultured_genome_RUG031</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900314695.1</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>Rumen uncultured genome RUG031</td>\n",
       "      <td>PRJEB21624</td>\n",
       "      <td>SAMEA104666881</td>\n",
       "      <td>GCA_900314695.1</td>\n",
       "      <td>full</td>\n",
       "      <td>uncultured Methanobrevibacter sp.</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_900314695.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1174</th>\n",
       "      <td>GCA_900316895.1</td>\n",
       "      <td>Rumen_uncultured_genome_RUG236</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900316895.1</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>Rumen uncultured genome RUG236</td>\n",
       "      <td>PRJEB21624</td>\n",
       "      <td>SAMEA104666888</td>\n",
       "      <td>GCA_900316895.1</td>\n",
       "      <td>full</td>\n",
       "      <td>uncultured Methanobrevibacter sp.</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_900316895.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1175</th>\n",
       "      <td>GCA_900317865.1</td>\n",
       "      <td>Rumen_uncultured_genome_RUG338</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900317865.1</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>Rumen uncultured genome RUG338</td>\n",
       "      <td>PRJEB21624</td>\n",
       "      <td>SAMEA104666890</td>\n",
       "      <td>GCA_900317865.1</td>\n",
       "      <td>full</td>\n",
       "      <td>uncultured Methanobrevibacter sp.</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_900317865.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1176</th>\n",
       "      <td>GCA_900318035.1</td>\n",
       "      <td>Rumen_uncultured_genome_RUG344</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900318035.1</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>Rumen uncultured genome RUG344</td>\n",
       "      <td>PRJEB21624</td>\n",
       "      <td>SAMEA104666891</td>\n",
       "      <td>GCA_900318035.1</td>\n",
       "      <td>full</td>\n",
       "      <td>uncultured Methanobrevibacter sp.</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_900318035.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1177</th>\n",
       "      <td>GCA_900319535.1</td>\n",
       "      <td>Rumen_uncultured_genome_RUG492</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900319535.1</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>Rumen uncultured genome RUG492</td>\n",
       "      <td>PRJEB21624</td>\n",
       "      <td>SAMEA104666892</td>\n",
       "      <td>GCA_900319535.1</td>\n",
       "      <td>full</td>\n",
       "      <td>uncultured Methanobrevibacter sp.</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_900319535.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1178</th>\n",
       "      <td>GCA_900319985.1</td>\n",
       "      <td>Rumen_uncultured_genome_RUG545</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900319985.1</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>Rumen uncultured genome RUG545</td>\n",
       "      <td>PRJEB21624</td>\n",
       "      <td>SAMEA104666894</td>\n",
       "      <td>GCA_900319985.1</td>\n",
       "      <td>full</td>\n",
       "      <td>uncultured Methanobrevibacter sp.</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_900319985.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1179</th>\n",
       "      <td>GCA_900320515.1</td>\n",
       "      <td>Rumen_uncultured_genome_RUG593</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900320515.1</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>Rumen uncultured genome RUG593</td>\n",
       "      <td>PRJEB21624</td>\n",
       "      <td>SAMEA104666895</td>\n",
       "      <td>GCA_900320515.1</td>\n",
       "      <td>full</td>\n",
       "      <td>uncultured Methanobrevibacter sp.</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_900320515.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1180</th>\n",
       "      <td>GCA_900320955.1</td>\n",
       "      <td>Rumen_uncultured_genome_RUG648</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900320955.1</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>Rumen uncultured genome RUG648</td>\n",
       "      <td>PRJEB21624</td>\n",
       "      <td>SAMEA104666896</td>\n",
       "      <td>GCA_900320955.1</td>\n",
       "      <td>full</td>\n",
       "      <td>uncultured Methanobrevibacter sp.</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_900320955.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1181</th>\n",
       "      <td>GCA_900321995.1</td>\n",
       "      <td>Rumen_uncultured_genome_RUG748</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900321995.1</td>\n",
       "      <td>Contig</td>\n",
       "      <td>Rumen uncultured genome RUG748</td>\n",
       "      <td>PRJEB21624</td>\n",
       "      <td>SAMEA104666897</td>\n",
       "      <td>GCA_900321995.1</td>\n",
       "      <td>full</td>\n",
       "      <td>uncultured Methanobrevibacter sp.</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_900321995.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1182</th>\n",
       "      <td>GCA_900322125.1</td>\n",
       "      <td>Rumen_uncultured_genome_RUG761</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>GCA_900322125.1</td>\n",
       "      <td>Contig</td>\n",
       "      <td>Rumen uncultured genome RUG761</td>\n",
       "      <td>PRJEB21624</td>\n",
       "      <td>SAMEA104666902</td>\n",
       "      <td>GCA_900322125.1</td>\n",
       "      <td>full</td>\n",
       "      <td>uncultured Methanosphaera sp.</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_900322125.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20904 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     # assembly_accession                                 asm_name  \\\n",
       "0         GCA_000005825.2                                 ASM582v2   \n",
       "1         GCA_000006685.1                                 ASM668v1   \n",
       "2         GCA_000006945.2                                 ASM694v2   \n",
       "4         GCA_000006985.1                                 ASM698v1   \n",
       "5         GCA_000007025.1                                 ASM702v1   \n",
       "6         GCA_000007085.1                                 ASM708v1   \n",
       "8         GCA_000007145.1                                 ASM714v1   \n",
       "9         GCA_000007205.1                                 ASM720v1   \n",
       "10        GCA_000007325.1                                 ASM732v1   \n",
       "11        GCA_000007365.1                                 ASM736v1   \n",
       "13        GCA_000007605.1                                 ASM760v1   \n",
       "15        GCA_000007705.1                                 ASM770v1   \n",
       "16        GCA_000007725.1                                 ASM772v1   \n",
       "17        GCA_000007825.1                                 ASM782v1   \n",
       "18        GCA_000007905.1                                 ASM790v1   \n",
       "19        GCA_000007925.1                                 ASM792v1   \n",
       "20        GCA_000007985.2                                 ASM798v2   \n",
       "21        GCA_000008025.1                                 ASM802v1   \n",
       "22        GCA_000008045.1                                 ASM804v1   \n",
       "23        GCA_000008185.1                                 ASM818v1   \n",
       "24        GCA_000008205.1                                 ASM820v1   \n",
       "25        GCA_000008305.1                                 ASM830v1   \n",
       "26        GCA_000008365.1                                 ASM836v1   \n",
       "27        GCA_000008385.1                                 ASM838v1   \n",
       "28        GCA_000008465.1                                 ASM846v1   \n",
       "29        GCA_000008485.1                                 ASM848v1   \n",
       "30        GCA_000008545.1                                 ASM854v1   \n",
       "31        GCA_000008565.1                                 ASM856v1   \n",
       "32        GCA_000008625.1                                 ASM862v1   \n",
       "33        GCA_000008685.2                                 ASM868v2   \n",
       "...                   ...                                      ...   \n",
       "1145      GCA_900111935.1  IMG-taxon_2617270732_annotated_assembly   \n",
       "1146      GCA_900112175.1  IMG-taxon_2617270920_annotated_assembly   \n",
       "1147      GCA_900112205.1  IMG-taxon_2693429869_annotated_assembly   \n",
       "1148      GCA_900113245.1  IMG-taxon_2617270917_annotated_assembly   \n",
       "1149      GCA_900114435.1  IMG-taxon_2667527389_annotated_assembly   \n",
       "1150      GCA_900114455.1  IMG-taxon_2619618990_annotated_assembly   \n",
       "1151      GCA_900114585.1  IMG-taxon_2593339150_annotated_assembly   \n",
       "1153      GCA_900115675.1  IMG-taxon_2663762745_annotated_assembly   \n",
       "1154      GCA_900115785.1  IMG-taxon_2617270895_annotated_assembly   \n",
       "1155      GCA_900116205.1  IMG-taxon_2639762563_annotated_assembly   \n",
       "1156      GCA_900129775.1  IMG-taxon_2695420988_annotated_assembly   \n",
       "1158      GCA_900156425.1  IMG-taxon_2681812914_annotated_assembly   \n",
       "1159      GCA_900156475.1  IMG-taxon_2681812919_annotated_assembly   \n",
       "1163      GCA_900188065.1  IMG-taxon_2724679803_annotated_assembly   \n",
       "1165      GCA_900198835.1                               chr_EXT12c   \n",
       "1167      GCA_900215575.1  IMG-taxon_2728369221_annotated_assembly   \n",
       "1169      GCA_900313645.1           Rumen_uncultured_genome_RUG823   \n",
       "1170      GCA_900314605.1           Rumen_uncultured_genome_RUG012   \n",
       "1171      GCA_900314615.1           Rumen_uncultured_genome_RUG018   \n",
       "1172      GCA_900314635.1           Rumen_uncultured_genome_RUG020   \n",
       "1173      GCA_900314695.1           Rumen_uncultured_genome_RUG031   \n",
       "1174      GCA_900316895.1           Rumen_uncultured_genome_RUG236   \n",
       "1175      GCA_900317865.1           Rumen_uncultured_genome_RUG338   \n",
       "1176      GCA_900318035.1           Rumen_uncultured_genome_RUG344   \n",
       "1177      GCA_900319535.1           Rumen_uncultured_genome_RUG492   \n",
       "1178      GCA_900319985.1           Rumen_uncultured_genome_RUG545   \n",
       "1179      GCA_900320515.1           Rumen_uncultured_genome_RUG593   \n",
       "1180      GCA_900320955.1           Rumen_uncultured_genome_RUG648   \n",
       "1181      GCA_900321995.1           Rumen_uncultured_genome_RUG748   \n",
       "1182      GCA_900322125.1           Rumen_uncultured_genome_RUG761   \n",
       "\n",
       "                                               ftp_path            index  \\\n",
       "0     ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "1     ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "2     ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "4     ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "5     ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "6     ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "8     ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "9     ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "10    ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "11    ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "13    ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "15    ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "16    ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "17    ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "18    ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "19    ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "20    ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "21    ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "22    ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "23    ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "24    ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "25    ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "26    ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "27    ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "28    ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "29    ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "30    ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "31    ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "32    ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "33    ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...              NaN   \n",
       "...                                                 ...              ...   \n",
       "1145  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900111935.1   \n",
       "1146  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900112175.1   \n",
       "1147  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900112205.1   \n",
       "1148  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900113245.1   \n",
       "1149  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900114435.1   \n",
       "1150  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900114455.1   \n",
       "1151  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900114585.1   \n",
       "1153  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900115675.1   \n",
       "1154  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900115785.1   \n",
       "1155  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900116205.1   \n",
       "1156  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900129775.1   \n",
       "1158  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900156425.1   \n",
       "1159  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900156475.1   \n",
       "1163  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900188065.1   \n",
       "1165  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900198835.1   \n",
       "1167  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900215575.1   \n",
       "1169  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900313645.1   \n",
       "1170  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900314605.1   \n",
       "1171  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900314615.1   \n",
       "1172  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900314635.1   \n",
       "1173  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900314695.1   \n",
       "1174  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900316895.1   \n",
       "1175  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900317865.1   \n",
       "1176  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900318035.1   \n",
       "1177  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900319535.1   \n",
       "1178  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900319985.1   \n",
       "1179  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900320515.1   \n",
       "1180  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900320955.1   \n",
       "1181  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900321995.1   \n",
       "1182  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...  GCA_900322125.1   \n",
       "\n",
       "     ncbi_assembly_level                       ncbi_assembly_name  \\\n",
       "0        Complete Genome                                 ASM582v2   \n",
       "1        Complete Genome                                 ASM668v1   \n",
       "2        Complete Genome                                 ASM694v2   \n",
       "4        Complete Genome                                 ASM698v1   \n",
       "5        Complete Genome                                 ASM702v1   \n",
       "6        Complete Genome                                 ASM708v1   \n",
       "8        Complete Genome                                 ASM714v1   \n",
       "9        Complete Genome                                 ASM720v1   \n",
       "10       Complete Genome                                 ASM732v1   \n",
       "11       Complete Genome                                 ASM736v1   \n",
       "13       Complete Genome                                 ASM760v1   \n",
       "15       Complete Genome                                 ASM770v1   \n",
       "16       Complete Genome                                 ASM772v1   \n",
       "17       Complete Genome                                 ASM782v1   \n",
       "18       Complete Genome                                 ASM790v1   \n",
       "19       Complete Genome                                 ASM792v1   \n",
       "20       Complete Genome                                 ASM798v2   \n",
       "21       Complete Genome                                 ASM802v1   \n",
       "22       Complete Genome                                 ASM804v1   \n",
       "23       Complete Genome                                 ASM818v1   \n",
       "24       Complete Genome                                 ASM820v1   \n",
       "25       Complete Genome                                 ASM830v1   \n",
       "26       Complete Genome                                 ASM836v1   \n",
       "27       Complete Genome                                 ASM838v1   \n",
       "28       Complete Genome                                 ASM846v1   \n",
       "29       Complete Genome                                 ASM848v1   \n",
       "30       Complete Genome                                 ASM854v1   \n",
       "31       Complete Genome                                 ASM856v1   \n",
       "32       Complete Genome                                 ASM862v1   \n",
       "33       Complete Genome                                 ASM868v2   \n",
       "...                  ...                                      ...   \n",
       "1145            Scaffold  IMG-taxon 2617270732 annotated assembly   \n",
       "1146              Contig  IMG-taxon 2617270920 annotated assembly   \n",
       "1147            Scaffold  IMG-taxon 2693429869 annotated assembly   \n",
       "1148              Contig  IMG-taxon 2617270917 annotated assembly   \n",
       "1149            Scaffold  IMG-taxon 2667527389 annotated assembly   \n",
       "1150            Scaffold  IMG-taxon 2619618990 annotated assembly   \n",
       "1151            Scaffold  IMG-taxon 2593339150 annotated assembly   \n",
       "1153              Contig  IMG-taxon 2663762745 annotated assembly   \n",
       "1154            Scaffold  IMG-taxon 2617270895 annotated assembly   \n",
       "1155              Contig  IMG-taxon 2639762563 annotated assembly   \n",
       "1156            Scaffold  IMG-taxon 2695420988 annotated assembly   \n",
       "1158            Scaffold  IMG-taxon 2681812914 annotated assembly   \n",
       "1159            Scaffold  IMG-taxon 2681812919 annotated assembly   \n",
       "1163              Contig  IMG-taxon 2724679803 annotated assembly   \n",
       "1165     Complete Genome                               chr_EXT12c   \n",
       "1167              Contig  IMG-taxon 2728369221 annotated assembly   \n",
       "1169            Scaffold           Rumen uncultured genome RUG823   \n",
       "1170            Scaffold           Rumen uncultured genome RUG012   \n",
       "1171            Scaffold           Rumen uncultured genome RUG018   \n",
       "1172            Scaffold           Rumen uncultured genome RUG020   \n",
       "1173            Scaffold           Rumen uncultured genome RUG031   \n",
       "1174            Scaffold           Rumen uncultured genome RUG236   \n",
       "1175            Scaffold           Rumen uncultured genome RUG338   \n",
       "1176            Scaffold           Rumen uncultured genome RUG344   \n",
       "1177            Scaffold           Rumen uncultured genome RUG492   \n",
       "1178            Scaffold           Rumen uncultured genome RUG545   \n",
       "1179            Scaffold           Rumen uncultured genome RUG593   \n",
       "1180            Scaffold           Rumen uncultured genome RUG648   \n",
       "1181              Contig           Rumen uncultured genome RUG748   \n",
       "1182              Contig           Rumen uncultured genome RUG761   \n",
       "\n",
       "     ncbi_bioproject  ncbi_biosample ncbi_genbank_assembly_accession  \\\n",
       "0        PRJNA224116    SAMN02603086                 GCA_000005825.2   \n",
       "1        PRJNA224116    SAMN02603994                 GCA_000006685.1   \n",
       "2         PRJNA57799    SAMN02604315                 GCA_000006945.2   \n",
       "4         PRJNA57897    SAMN02604006                 GCA_000006985.1   \n",
       "5        PRJNA224116    SAMN02603141                 GCA_000007025.1   \n",
       "6        PRJNA224116    SAMN02603009                 GCA_000007085.1   \n",
       "8         PRJNA57887    SAMN02603845                 GCA_000007145.1   \n",
       "9        PRJNA224116    SAMN02602979                 GCA_000007205.1   \n",
       "10        PRJNA57885    SAMN02603417                 GCA_000007325.1   \n",
       "11       PRJNA224116    SAMN02604269                 GCA_000007365.1   \n",
       "13       PRJNA224116    SAMN02603993                 GCA_000007605.1   \n",
       "15       PRJNA224116    SAMN02603528                 GCA_000007705.1   \n",
       "16       PRJNA224116    SAMN02604289                 GCA_000007725.1   \n",
       "17        PRJNA57975    SAMN02603340                 GCA_000007825.1   \n",
       "18       PRJNA224116    SAMN02604097                 GCA_000007905.1   \n",
       "19        PRJNA57995    SAMN02603142                 GCA_000007925.1   \n",
       "20        PRJNA57743    SAMN02603989                 GCA_000007985.2   \n",
       "21       PRJNA224116    SAMN02604000                 GCA_000008025.1   \n",
       "22       PRJNA224116    SAMN02603327                 GCA_000008045.1   \n",
       "23        PRJNA57583    SAMN02603967                 GCA_000008185.1   \n",
       "24       PRJNA224116    SAMN02603526                 GCA_000008205.1   \n",
       "25        PRJNA58055    SAMN03081416                 GCA_000008305.1   \n",
       "26       PRJNA224116    SAMN03081417                 GCA_000008365.1   \n",
       "27       PRJNA224116    SAMN02603678                 GCA_000008385.1   \n",
       "28       PRJNA224116    SAMN02604184                 GCA_000008465.1   \n",
       "29        PRJNA57609    SAMN02603156                 GCA_000008485.1   \n",
       "30        PRJNA57723    SAMN02603986                 GCA_000008545.1   \n",
       "31        PRJNA57665    SAMN02603976                 GCA_000008565.1   \n",
       "32        PRJNA57765    SAMN02603207                 GCA_000008625.1   \n",
       "33        PRJNA57581    SAMN02603966                 GCA_000008685.2   \n",
       "...              ...             ...                             ...   \n",
       "1145     PRJNA224116    SAMN04487937                 GCA_900111935.1   \n",
       "1146     PRJNA224116    SAMN04487947                 GCA_900112175.1   \n",
       "1147     PRJNA224116    SAMN05444422                 GCA_900112205.1   \n",
       "1148     PRJNA224116    SAMN04488063                 GCA_900113245.1   \n",
       "1149     PRJNA224116    SAMN05216559                 GCA_900114435.1   \n",
       "1150     PRJNA224116    SAMN04487950                 GCA_900114455.1   \n",
       "1151     PRJNA224116    SAMN02910297                 GCA_900114585.1   \n",
       "1153     PRJNA224116    SAMN05216277                 GCA_900115675.1   \n",
       "1154     PRJNA224116    SAMN04488124                 GCA_900115785.1   \n",
       "1155     PRJNA224116    SAMN04488556                 GCA_900116205.1   \n",
       "1156     PRJNA224116    SAMN05443636                 GCA_900129775.1   \n",
       "1158     PRJNA224116    SAMN05421858                 GCA_900156425.1   \n",
       "1159     PRJNA224116    SAMN05421752                 GCA_900156475.1   \n",
       "1163     PRJNA224116    SAMN06264855                 GCA_900188065.1   \n",
       "1165     PRJNA224116  SAMEA104149373                 GCA_900198835.1   \n",
       "1167     PRJNA224116    SAMN06269185                 GCA_900215575.1   \n",
       "1169      PRJEB21624  SAMEA104666900                 GCA_900313645.1   \n",
       "1170      PRJEB21624  SAMEA104666878                 GCA_900314605.1   \n",
       "1171      PRJEB21624  SAMEA104666879                 GCA_900314615.1   \n",
       "1172      PRJEB21624  SAMEA104666880                 GCA_900314635.1   \n",
       "1173      PRJEB21624  SAMEA104666881                 GCA_900314695.1   \n",
       "1174      PRJEB21624  SAMEA104666888                 GCA_900316895.1   \n",
       "1175      PRJEB21624  SAMEA104666890                 GCA_900317865.1   \n",
       "1176      PRJEB21624  SAMEA104666891                 GCA_900318035.1   \n",
       "1177      PRJEB21624  SAMEA104666892                 GCA_900319535.1   \n",
       "1178      PRJEB21624  SAMEA104666894                 GCA_900319985.1   \n",
       "1179      PRJEB21624  SAMEA104666895                 GCA_900320515.1   \n",
       "1180      PRJEB21624  SAMEA104666896                 GCA_900320955.1   \n",
       "1181      PRJEB21624  SAMEA104666897                 GCA_900321995.1   \n",
       "1182      PRJEB21624  SAMEA104666902                 GCA_900322125.1   \n",
       "\n",
       "     ncbi_genome_representation  \\\n",
       "0                          full   \n",
       "1                          full   \n",
       "2                          full   \n",
       "4                          full   \n",
       "5                          full   \n",
       "6                          full   \n",
       "8                          full   \n",
       "9                          full   \n",
       "10                         full   \n",
       "11                         full   \n",
       "13                         full   \n",
       "15                         full   \n",
       "16                         full   \n",
       "17                         full   \n",
       "18                         full   \n",
       "19                         full   \n",
       "20                         full   \n",
       "21                         full   \n",
       "22                         full   \n",
       "23                         full   \n",
       "24                         full   \n",
       "25                         full   \n",
       "26                         full   \n",
       "27                         full   \n",
       "28                         full   \n",
       "29                         full   \n",
       "30                         full   \n",
       "31                         full   \n",
       "32                         full   \n",
       "33                         full   \n",
       "...                         ...   \n",
       "1145                       full   \n",
       "1146                       full   \n",
       "1147                       full   \n",
       "1148                       full   \n",
       "1149                       full   \n",
       "1150                       full   \n",
       "1151                       full   \n",
       "1153                       full   \n",
       "1154                       full   \n",
       "1155                       full   \n",
       "1156                       full   \n",
       "1158                       full   \n",
       "1159                       full   \n",
       "1163                       full   \n",
       "1165                       full   \n",
       "1167                       full   \n",
       "1169                       full   \n",
       "1170                       full   \n",
       "1171                       full   \n",
       "1172                       full   \n",
       "1173                       full   \n",
       "1174                       full   \n",
       "1175                       full   \n",
       "1176                       full   \n",
       "1177                       full   \n",
       "1178                       full   \n",
       "1179                       full   \n",
       "1180                       full   \n",
       "1181                       full   \n",
       "1182                       full   \n",
       "\n",
       "                                     ncbi_organism_name  \\\n",
       "0                             Bacillus pseudofirmus OF4   \n",
       "1                         Chlamydia muridarum str. Nigg   \n",
       "2     Salmonella enterica subsp. enterica serovar Ty...   \n",
       "4                             Chlorobaculum tepidum TLS   \n",
       "5                      Rickettsia conorii str. Malish 7   \n",
       "6     Caldanaerobacter subterraneus subsp. tengconge...   \n",
       "8     Xanthomonas campestris pv. campestris str. ATC...   \n",
       "9                           Chlamydia pneumoniae TW-183   \n",
       "10    Fusobacterium nucleatum subsp. nucleatum ATCC ...   \n",
       "11    Buchnera aphidicola str. Sg (Schizaphis graminum)   \n",
       "13                                Chlamydia caviae GPIC   \n",
       "15                 Chromobacterium violaceum ATCC 12472   \n",
       "16    Buchnera aphidicola str. Bp (Baizongia pistaciae)   \n",
       "17                           Bacillus cereus ATCC 14579   \n",
       "18                    Helicobacter hepaticus ATCC 51449   \n",
       "19    Prochlorococcus marinus subsp. marinus str. CC...   \n",
       "20                         Geobacter sulfurreducens PCA   \n",
       "21    Wolbachia endosymbiont of Drosophila melanogaster   \n",
       "22                     Rickettsia typhi str. Wilmington   \n",
       "23                       Treponema denticola ATCC 35405   \n",
       "24                           Mycoplasma hyopneumoniae J   \n",
       "25                                 Mesoplasma florum L1   \n",
       "26                               Mycoplasma mobile 163K   \n",
       "27    Wolbachia endosymbiont strain TRS of Brugia ma...   \n",
       "28                           Idiomarina loihiensis L2TR   \n",
       "29    Legionella pneumophila subsp. pneumophila str....   \n",
       "30                             Thermotoga maritima MSB8   \n",
       "31                           Deinococcus radiodurans R1   \n",
       "32                                 Aquifex aeolicus VF5   \n",
       "33                          Borreliella burgdorferi B31   \n",
       "...                                                 ...   \n",
       "1145                               Halorubrum sodomense   \n",
       "1146                              Halogeometricum rufum   \n",
       "1147                         Halobiforma haloterrestris   \n",
       "1148                           Halopelagius inordinatus   \n",
       "1149                               Halomicrobium zhouii   \n",
       "1150                                  Halogranum rubrum   \n",
       "1151                         Methanobrevibacter olleyae   \n",
       "1153                                Halolamina pelagica   \n",
       "1154                               Halogeometricum limi   \n",
       "1155                           Halostagnicola kamekurae   \n",
       "1156                             Halobaculum gomorrense   \n",
       "1158                              Haladaptatus litoreus   \n",
       "1159                          Natronorubrum thiooxidans   \n",
       "1163                              Halorubrum vacuolatum   \n",
       "1165                            Thermococcus sp. EXT12c   \n",
       "1167                      Natronoarchaeum philippinense   \n",
       "1169                  uncultured Methanobrevibacter sp.   \n",
       "1170                  uncultured Methanobrevibacter sp.   \n",
       "1171                  uncultured Methanobrevibacter sp.   \n",
       "1172                  uncultured Methanobrevibacter sp.   \n",
       "1173                  uncultured Methanobrevibacter sp.   \n",
       "1174                  uncultured Methanobrevibacter sp.   \n",
       "1175                  uncultured Methanobrevibacter sp.   \n",
       "1176                  uncultured Methanobrevibacter sp.   \n",
       "1177                  uncultured Methanobrevibacter sp.   \n",
       "1178                  uncultured Methanobrevibacter sp.   \n",
       "1179                  uncultured Methanobrevibacter sp.   \n",
       "1180                  uncultured Methanobrevibacter sp.   \n",
       "1181                  uncultured Methanobrevibacter sp.   \n",
       "1182                      uncultured Methanosphaera sp.   \n",
       "\n",
       "       ncbi_refseq_category parsed_genome_accession srr_accession  \n",
       "0     representative genome         GCF_000005825.2           NaN  \n",
       "1     representative genome         GCF_000006685.1           NaN  \n",
       "2          reference genome         GCF_000006945.2           NaN  \n",
       "4          reference genome         GCF_000006985.1           NaN  \n",
       "5     representative genome         GCF_000007025.1           NaN  \n",
       "6     representative genome         GCF_000007085.1           NaN  \n",
       "8          reference genome         GCF_000007145.1           NaN  \n",
       "9                        na         GCF_000007205.1           NaN  \n",
       "10         reference genome         GCF_000007325.1           NaN  \n",
       "11    representative genome         GCF_000007365.1           NaN  \n",
       "13    representative genome         GCF_000007605.1           NaN  \n",
       "15    representative genome         GCF_000007705.1           NaN  \n",
       "16    representative genome         GCF_000007725.1           NaN  \n",
       "17         reference genome         GCF_000007825.1           NaN  \n",
       "18    representative genome         GCF_000007905.1           NaN  \n",
       "19         reference genome         GCF_000007925.1           NaN  \n",
       "20         reference genome         GCF_000007985.2           NaN  \n",
       "21    representative genome         GCF_000008025.1           NaN  \n",
       "22    representative genome         GCF_000008045.1           NaN  \n",
       "23         reference genome         GCF_000008185.1           NaN  \n",
       "24    representative genome         GCF_000008205.1           NaN  \n",
       "25         reference genome         GCF_000008305.1           NaN  \n",
       "26    representative genome         GCF_000008365.1           NaN  \n",
       "27    representative genome         GCF_000008385.1           NaN  \n",
       "28    representative genome         GCF_000008465.1           NaN  \n",
       "29         reference genome         GCF_000008485.1           NaN  \n",
       "30         reference genome         GCF_000008545.1           NaN  \n",
       "31         reference genome         GCF_000008565.1           NaN  \n",
       "32         reference genome         GCF_000008625.1           NaN  \n",
       "33         reference genome         GCF_000008685.2           NaN  \n",
       "...                     ...                     ...           ...  \n",
       "1145  representative genome         GCF_900111935.1           NaN  \n",
       "1146  representative genome         GCF_900112175.1           NaN  \n",
       "1147                     na         GCF_900112205.1           NaN  \n",
       "1148  representative genome         GCF_900113245.1           NaN  \n",
       "1149  representative genome         GCF_900114435.1           NaN  \n",
       "1150                     na         GCF_900114455.1           NaN  \n",
       "1151                     na         GCF_900114585.1           NaN  \n",
       "1153  representative genome         GCF_900115675.1           NaN  \n",
       "1154  representative genome         GCF_900115785.1           NaN  \n",
       "1155  representative genome         GCF_900116205.1           NaN  \n",
       "1156  representative genome         GCF_900129775.1           NaN  \n",
       "1158                     na         GCF_900156425.1           NaN  \n",
       "1159  representative genome         GCF_900156475.1           NaN  \n",
       "1163  representative genome         GCF_900188065.1           NaN  \n",
       "1165                     na         GCF_900198835.1           NaN  \n",
       "1167                     na         GCF_900215575.1           NaN  \n",
       "1169                     na         GCA_900313645.1           NaN  \n",
       "1170                     na         GCA_900314605.1           NaN  \n",
       "1171                     na         GCA_900314615.1           NaN  \n",
       "1172                     na         GCA_900314635.1           NaN  \n",
       "1173                     na         GCA_900314695.1           NaN  \n",
       "1174                     na         GCA_900316895.1           NaN  \n",
       "1175                     na         GCA_900317865.1           NaN  \n",
       "1176                     na         GCA_900318035.1           NaN  \n",
       "1177                     na         GCA_900319535.1           NaN  \n",
       "1178                     na         GCA_900319985.1           NaN  \n",
       "1179                     na         GCA_900320515.1           NaN  \n",
       "1180                     na         GCA_900320955.1           NaN  \n",
       "1181                     na         GCA_900321995.1           NaN  \n",
       "1182                     na         GCA_900322125.1           NaN  \n",
       "\n",
       "[20904 rows x 14 columns]"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train.dropna(subset=['ftp_path'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20955\n",
      "51\n"
     ]
    }
   ],
   "source": [
    "print(len(train))\n",
    "print(len(train[train['ftp_path'].isna()]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th># assembly_accession</th>\n",
       "      <th>asm_name</th>\n",
       "      <th>ftp_path</th>\n",
       "      <th>index</th>\n",
       "      <th>ncbi_assembly_level</th>\n",
       "      <th>ncbi_assembly_name</th>\n",
       "      <th>ncbi_bioproject</th>\n",
       "      <th>ncbi_biosample</th>\n",
       "      <th>ncbi_genbank_assembly_accession</th>\n",
       "      <th>ncbi_genome_representation</th>\n",
       "      <th>ncbi_organism_name</th>\n",
       "      <th>ncbi_refseq_category</th>\n",
       "      <th>parsed_genome_accession</th>\n",
       "      <th>srr_accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1183</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10036</td>\n",
       "      <td>SRR10597631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1184</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10106</td>\n",
       "      <td>SRR10597620</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1185</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10107</td>\n",
       "      <td>SRR10597609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1186</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10117</td>\n",
       "      <td>SRR10597583</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1187</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10136</td>\n",
       "      <td>SRR10597561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1188</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10154</td>\n",
       "      <td>SRR10597694</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1189</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10161</td>\n",
       "      <td>SRR10597683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1190</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10192</td>\n",
       "      <td>SRR10597645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1191</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10194</td>\n",
       "      <td>SRR10597643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1192</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10200</td>\n",
       "      <td>SRR10597641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1193</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10204</td>\n",
       "      <td>SRR10597640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1194</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10205</td>\n",
       "      <td>SRR10597639</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1195</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10207</td>\n",
       "      <td>SRR10597638</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1196</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10210</td>\n",
       "      <td>SRR10597637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1197</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10214</td>\n",
       "      <td>SRR10597636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1198</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10216</td>\n",
       "      <td>SRR10597635</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1199</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10219</td>\n",
       "      <td>SRR10597634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1200</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10242</td>\n",
       "      <td>SRR10597633</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1201</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10516</td>\n",
       "      <td>SRR10597629</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1206</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA11384</td>\n",
       "      <td>SRR10597619</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1207</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA11423</td>\n",
       "      <td>SRR10597618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1208</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA11576</td>\n",
       "      <td>SRR10597617</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1209</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA11803</td>\n",
       "      <td>SRR10597603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1211</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA11892</td>\n",
       "      <td>SRR10597599</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1212</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA11921</td>\n",
       "      <td>SRR10597598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1213</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA11998</td>\n",
       "      <td>SRR10597597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1214</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA12027</td>\n",
       "      <td>SRR10597595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1215</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA12206</td>\n",
       "      <td>SRR10597587</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1216</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA12207</td>\n",
       "      <td>SRR10597586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1217</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA12459</td>\n",
       "      <td>SRR10597579</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1218</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA12570</td>\n",
       "      <td>SRR10597568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1221</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA8160</td>\n",
       "      <td>SRR10597564</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1222</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA8261</td>\n",
       "      <td>SRR10597560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1223</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA8339</td>\n",
       "      <td>SRR10597558</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1225</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA8574</td>\n",
       "      <td>SRR10597604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1226</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA8684</td>\n",
       "      <td>SRR10597699</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1228</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA8725</td>\n",
       "      <td>SRR10597697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1229</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA8734</td>\n",
       "      <td>SRR10597696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1230</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA8915</td>\n",
       "      <td>SRR10597681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1233</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA9212</td>\n",
       "      <td>SRR10597676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1234</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA9293</td>\n",
       "      <td>SRR10597674</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1235</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA9297</td>\n",
       "      <td>SRR10597673</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1237</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA9354</td>\n",
       "      <td>SRR10597667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1238</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA9513</td>\n",
       "      <td>SRR10597665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1239</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA9596</td>\n",
       "      <td>SRR10597662</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1240</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA9638</td>\n",
       "      <td>SRR10597659</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1241</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA9640</td>\n",
       "      <td>SRR10597658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1243</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA9912</td>\n",
       "      <td>SRR10597655</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1245</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA9950</td>\n",
       "      <td>SRR10597653</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1246</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA9966</td>\n",
       "      <td>SRR10597652</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1247</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA9989</td>\n",
       "      <td>SRR10597651</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     # assembly_accession asm_name ftp_path index ncbi_assembly_level  \\\n",
       "1183                  NaN      NaN      NaN   NaN                none   \n",
       "1184                  NaN      NaN      NaN   NaN                none   \n",
       "1185                  NaN      NaN      NaN   NaN                none   \n",
       "1186                  NaN      NaN      NaN   NaN                none   \n",
       "1187                  NaN      NaN      NaN   NaN                none   \n",
       "1188                  NaN      NaN      NaN   NaN                none   \n",
       "1189                  NaN      NaN      NaN   NaN                none   \n",
       "1190                  NaN      NaN      NaN   NaN                none   \n",
       "1191                  NaN      NaN      NaN   NaN                none   \n",
       "1192                  NaN      NaN      NaN   NaN                none   \n",
       "1193                  NaN      NaN      NaN   NaN                none   \n",
       "1194                  NaN      NaN      NaN   NaN                none   \n",
       "1195                  NaN      NaN      NaN   NaN                none   \n",
       "1196                  NaN      NaN      NaN   NaN                none   \n",
       "1197                  NaN      NaN      NaN   NaN                none   \n",
       "1198                  NaN      NaN      NaN   NaN                none   \n",
       "1199                  NaN      NaN      NaN   NaN                none   \n",
       "1200                  NaN      NaN      NaN   NaN                none   \n",
       "1201                  NaN      NaN      NaN   NaN                none   \n",
       "1206                  NaN      NaN      NaN   NaN                none   \n",
       "1207                  NaN      NaN      NaN   NaN                none   \n",
       "1208                  NaN      NaN      NaN   NaN                none   \n",
       "1209                  NaN      NaN      NaN   NaN                none   \n",
       "1211                  NaN      NaN      NaN   NaN                none   \n",
       "1212                  NaN      NaN      NaN   NaN                none   \n",
       "1213                  NaN      NaN      NaN   NaN                none   \n",
       "1214                  NaN      NaN      NaN   NaN                none   \n",
       "1215                  NaN      NaN      NaN   NaN                none   \n",
       "1216                  NaN      NaN      NaN   NaN                none   \n",
       "1217                  NaN      NaN      NaN   NaN                none   \n",
       "1218                  NaN      NaN      NaN   NaN                none   \n",
       "1221                  NaN      NaN      NaN   NaN                none   \n",
       "1222                  NaN      NaN      NaN   NaN                none   \n",
       "1223                  NaN      NaN      NaN   NaN                none   \n",
       "1225                  NaN      NaN      NaN   NaN                none   \n",
       "1226                  NaN      NaN      NaN   NaN                none   \n",
       "1228                  NaN      NaN      NaN   NaN                none   \n",
       "1229                  NaN      NaN      NaN   NaN                none   \n",
       "1230                  NaN      NaN      NaN   NaN                none   \n",
       "1233                  NaN      NaN      NaN   NaN                none   \n",
       "1234                  NaN      NaN      NaN   NaN                none   \n",
       "1235                  NaN      NaN      NaN   NaN                none   \n",
       "1237                  NaN      NaN      NaN   NaN                none   \n",
       "1238                  NaN      NaN      NaN   NaN                none   \n",
       "1239                  NaN      NaN      NaN   NaN                none   \n",
       "1240                  NaN      NaN      NaN   NaN                none   \n",
       "1241                  NaN      NaN      NaN   NaN                none   \n",
       "1243                  NaN      NaN      NaN   NaN                none   \n",
       "1245                  NaN      NaN      NaN   NaN                none   \n",
       "1246                  NaN      NaN      NaN   NaN                none   \n",
       "1247                  NaN      NaN      NaN   NaN                none   \n",
       "\n",
       "     ncbi_assembly_name ncbi_bioproject ncbi_biosample  \\\n",
       "1183               none            none           none   \n",
       "1184               none            none           none   \n",
       "1185               none            none           none   \n",
       "1186               none            none           none   \n",
       "1187               none            none           none   \n",
       "1188               none            none           none   \n",
       "1189               none            none           none   \n",
       "1190               none            none           none   \n",
       "1191               none            none           none   \n",
       "1192               none            none           none   \n",
       "1193               none            none           none   \n",
       "1194               none            none           none   \n",
       "1195               none            none           none   \n",
       "1196               none            none           none   \n",
       "1197               none            none           none   \n",
       "1198               none            none           none   \n",
       "1199               none            none           none   \n",
       "1200               none            none           none   \n",
       "1201               none            none           none   \n",
       "1206               none            none           none   \n",
       "1207               none            none           none   \n",
       "1208               none            none           none   \n",
       "1209               none            none           none   \n",
       "1211               none            none           none   \n",
       "1212               none            none           none   \n",
       "1213               none            none           none   \n",
       "1214               none            none           none   \n",
       "1215               none            none           none   \n",
       "1216               none            none           none   \n",
       "1217               none            none           none   \n",
       "1218               none            none           none   \n",
       "1221               none            none           none   \n",
       "1222               none            none           none   \n",
       "1223               none            none           none   \n",
       "1225               none            none           none   \n",
       "1226               none            none           none   \n",
       "1228               none            none           none   \n",
       "1229               none            none           none   \n",
       "1230               none            none           none   \n",
       "1233               none            none           none   \n",
       "1234               none            none           none   \n",
       "1235               none            none           none   \n",
       "1237               none            none           none   \n",
       "1238               none            none           none   \n",
       "1239               none            none           none   \n",
       "1240               none            none           none   \n",
       "1241               none            none           none   \n",
       "1243               none            none           none   \n",
       "1245               none            none           none   \n",
       "1246               none            none           none   \n",
       "1247               none            none           none   \n",
       "\n",
       "     ncbi_genbank_assembly_accession ncbi_genome_representation  \\\n",
       "1183                            none                       none   \n",
       "1184                            none                       none   \n",
       "1185                            none                       none   \n",
       "1186                            none                       none   \n",
       "1187                            none                       none   \n",
       "1188                            none                       none   \n",
       "1189                            none                       none   \n",
       "1190                            none                       none   \n",
       "1191                            none                       none   \n",
       "1192                            none                       none   \n",
       "1193                            none                       none   \n",
       "1194                            none                       none   \n",
       "1195                            none                       none   \n",
       "1196                            none                       none   \n",
       "1197                            none                       none   \n",
       "1198                            none                       none   \n",
       "1199                            none                       none   \n",
       "1200                            none                       none   \n",
       "1201                            none                       none   \n",
       "1206                            none                       none   \n",
       "1207                            none                       none   \n",
       "1208                            none                       none   \n",
       "1209                            none                       none   \n",
       "1211                            none                       none   \n",
       "1212                            none                       none   \n",
       "1213                            none                       none   \n",
       "1214                            none                       none   \n",
       "1215                            none                       none   \n",
       "1216                            none                       none   \n",
       "1217                            none                       none   \n",
       "1218                            none                       none   \n",
       "1221                            none                       none   \n",
       "1222                            none                       none   \n",
       "1223                            none                       none   \n",
       "1225                            none                       none   \n",
       "1226                            none                       none   \n",
       "1228                            none                       none   \n",
       "1229                            none                       none   \n",
       "1230                            none                       none   \n",
       "1233                            none                       none   \n",
       "1234                            none                       none   \n",
       "1235                            none                       none   \n",
       "1237                            none                       none   \n",
       "1238                            none                       none   \n",
       "1239                            none                       none   \n",
       "1240                            none                       none   \n",
       "1241                            none                       none   \n",
       "1243                            none                       none   \n",
       "1245                            none                       none   \n",
       "1246                            none                       none   \n",
       "1247                            none                       none   \n",
       "\n",
       "     ncbi_organism_name ncbi_refseq_category parsed_genome_accession  \\\n",
       "1183               none                 none                UBA10036   \n",
       "1184               none                 none                UBA10106   \n",
       "1185               none                 none                UBA10107   \n",
       "1186               none                 none                UBA10117   \n",
       "1187               none                 none                UBA10136   \n",
       "1188               none                 none                UBA10154   \n",
       "1189               none                 none                UBA10161   \n",
       "1190               none                 none                UBA10192   \n",
       "1191               none                 none                UBA10194   \n",
       "1192               none                 none                UBA10200   \n",
       "1193               none                 none                UBA10204   \n",
       "1194               none                 none                UBA10205   \n",
       "1195               none                 none                UBA10207   \n",
       "1196               none                 none                UBA10210   \n",
       "1197               none                 none                UBA10214   \n",
       "1198               none                 none                UBA10216   \n",
       "1199               none                 none                UBA10219   \n",
       "1200               none                 none                UBA10242   \n",
       "1201               none                 none                UBA10516   \n",
       "1206               none                 none                UBA11384   \n",
       "1207               none                 none                UBA11423   \n",
       "1208               none                 none                UBA11576   \n",
       "1209               none                 none                UBA11803   \n",
       "1211               none                 none                UBA11892   \n",
       "1212               none                 none                UBA11921   \n",
       "1213               none                 none                UBA11998   \n",
       "1214               none                 none                UBA12027   \n",
       "1215               none                 none                UBA12206   \n",
       "1216               none                 none                UBA12207   \n",
       "1217               none                 none                UBA12459   \n",
       "1218               none                 none                UBA12570   \n",
       "1221               none                 none                 UBA8160   \n",
       "1222               none                 none                 UBA8261   \n",
       "1223               none                 none                 UBA8339   \n",
       "1225               none                 none                 UBA8574   \n",
       "1226               none                 none                 UBA8684   \n",
       "1228               none                 none                 UBA8725   \n",
       "1229               none                 none                 UBA8734   \n",
       "1230               none                 none                 UBA8915   \n",
       "1233               none                 none                 UBA9212   \n",
       "1234               none                 none                 UBA9293   \n",
       "1235               none                 none                 UBA9297   \n",
       "1237               none                 none                 UBA9354   \n",
       "1238               none                 none                 UBA9513   \n",
       "1239               none                 none                 UBA9596   \n",
       "1240               none                 none                 UBA9638   \n",
       "1241               none                 none                 UBA9640   \n",
       "1243               none                 none                 UBA9912   \n",
       "1245               none                 none                 UBA9950   \n",
       "1246               none                 none                 UBA9966   \n",
       "1247               none                 none                 UBA9989   \n",
       "\n",
       "     srr_accession  \n",
       "1183   SRR10597631  \n",
       "1184   SRR10597620  \n",
       "1185   SRR10597609  \n",
       "1186   SRR10597583  \n",
       "1187   SRR10597561  \n",
       "1188   SRR10597694  \n",
       "1189   SRR10597683  \n",
       "1190   SRR10597645  \n",
       "1191   SRR10597643  \n",
       "1192   SRR10597641  \n",
       "1193   SRR10597640  \n",
       "1194   SRR10597639  \n",
       "1195   SRR10597638  \n",
       "1196   SRR10597637  \n",
       "1197   SRR10597636  \n",
       "1198   SRR10597635  \n",
       "1199   SRR10597634  \n",
       "1200   SRR10597633  \n",
       "1201   SRR10597629  \n",
       "1206   SRR10597619  \n",
       "1207   SRR10597618  \n",
       "1208   SRR10597617  \n",
       "1209   SRR10597603  \n",
       "1211   SRR10597599  \n",
       "1212   SRR10597598  \n",
       "1213   SRR10597597  \n",
       "1214   SRR10597595  \n",
       "1215   SRR10597587  \n",
       "1216   SRR10597586  \n",
       "1217   SRR10597579  \n",
       "1218   SRR10597568  \n",
       "1221   SRR10597564  \n",
       "1222   SRR10597560  \n",
       "1223   SRR10597558  \n",
       "1225   SRR10597604  \n",
       "1226   SRR10597699  \n",
       "1228   SRR10597697  \n",
       "1229   SRR10597696  \n",
       "1230   SRR10597681  \n",
       "1233   SRR10597676  \n",
       "1234   SRR10597674  \n",
       "1235   SRR10597673  \n",
       "1237   SRR10597667  \n",
       "1238   SRR10597665  \n",
       "1239   SRR10597662  \n",
       "1240   SRR10597659  \n",
       "1241   SRR10597658  \n",
       "1243   SRR10597655  \n",
       "1245   SRR10597653  \n",
       "1246   SRR10597652  \n",
       "1247   SRR10597651  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train[train['ftp_path'].isna()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_filenames(df):\n",
    "    filenames = []\n",
    "    for ix,row in df.iterrows():\n",
    "        if row['ftp_path']:\n",
    "            fname = str(row['ftp_path']).split('/')[-1]+'.fna'\n",
    "            if fname=='nan.fna':\n",
    "                fname = str(row['srr_accession'])+'.fna'\n",
    "        filenames.append(fname) \n",
    "    return filenames\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "test_filenames = get_filenames(test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2121 2121\n",
      "1630 1630\n",
      "20955 20955\n"
     ]
    }
   ],
   "source": [
    "test_filenames = get_filenames(test)\n",
    "print(len(test), len(test_filenames))\n",
    "valid_filenames = get_filenames(valid)\n",
    "print(len(valid),len(valid_filenames))\n",
    "train_filenames = get_filenames(train)\n",
    "print(len(train),len(train_filenames))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "#save filenames as text files\n",
    "import os\n",
    "def save_filenames(outfile,filenames):\n",
    "    with open(outfile,'w') as f:\n",
    "        for item in filenames:\n",
    "            f.write(\"%s\\n\" % item)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "            \n",
    "save_filenames('trainnew_filenames.txt',train_filenames)\n",
    "save_filenames('validnew_filenames.txt',valid_filenames)\n",
    "save_filenames('testnew_filenames.txt',test_filenames)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Filter valid set to not have too many examples each genera\n",
    "\n",
    "also to reduce validset size"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_subset_leaves(node,to_match,taxa=[]):\n",
    "    for child in node.child_nodes():\n",
    "        if child.label:\n",
    "            if len([s for s in to_match if s in child.label])>0:\n",
    "                count=count_children(child,count=0)\n",
    "                print('appending',count,'labels from node', child)\n",
    "                these_taxa = [leaf.taxon.label for leaf in child.leaf_nodes()]\n",
    "                taxa = taxa.extend(these_taxa)\n",
    "        get_subset_leaves(child,to_match,taxa=taxa)\n",
    "    return taxa\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_leaves(node,match,leaves=[]):\n",
    "    for child in node.child_nodes():\n",
    "        if child.label:\n",
    "            if match in child.label:\n",
    "                leaves = leaves.extend([leaf for leaf in child.leaf_nodes()])\n",
    "        get_leaves(child,match,leaves=leaves)\n",
    "    return leaves\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "67"
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "thermo = get_leaves(arc.seed_node,match='c__Thermoplasmata',leaves=[])\n",
    "len(thermo)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Node object at 0x10db39438: 'None' (<Taxon 0x10db39eb8 'GB_GCA_002503545.1'>)>\n",
      "None\n",
      "'GB_GCA_002503545.1'\n"
     ]
    }
   ],
   "source": [
    "print(thermo[0])\n",
    "print(thermo[0].label)\n",
    "print(thermo[0].taxon)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "def print_children(node,match,max_count=0,min_count=0):\n",
    "    for child in node.child_nodes():\n",
    "        if child.label:\n",
    "            if match in child.label:\n",
    "                get_branch_children_counts(child,max_count,min_count)\n",
    "        print_children(child,match,max_count,min_count)\n",
    "        \n",
    "def get_all_branch_children(node):\n",
    "    for child in node.child_nodes():\n",
    "        count=count_children(child,count=0)\n",
    "        percent = count/23458.0\n",
    "        print(child,'# children taxa:',count,'; level',child.level())\n",
    "        get_all_branch_children(child)\n",
    "\n",
    "def print_node_children_counts(node,match):\n",
    "    for child in node.child_nodes():\n",
    "        if child.label:\n",
    "            if match in child.label:\n",
    "                count=count_children(child,count=0)\n",
    "                percent = count/23458.0\n",
    "                print(child,'# children taxa:',count,'; level',child.level())\n",
    "                get_all_branch_children(child)\n",
    "        print_node_children_counts(child,match)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Node object at 0x10f654908: '86.0:c__Thermoplasmata' (None)> # children taxa: 67 ; level 7\n",
      "<Node object at 0x10e7cdd30: '37.0' (None)> # children taxa: 65 ; level 8\n",
      "<Node object at 0x10e7cdac8: '79.0' (None)> # children taxa: 44 ; level 9\n",
      "<Node object at 0x10e7cd390: '86.0' (None)> # children taxa: 42 ; level 10\n",
      "<Node object at 0x10e7cd4e0: '100.0' (None)> # children taxa: 36 ; level 11\n",
      "<Node object at 0x10e7cd2b0: '100.0:o__Methanomassiliicoccales' (None)> # children taxa: 32 ; level 12\n",
      "<Node object at 0x10e921860: '100.0:f__Methanomethylophilaceae' (None)> # children taxa: 21 ; level 13\n",
      "<Node object at 0x10e921208: '100.0' (None)> # children taxa: 20 ; level 14\n",
      "<Node object at 0x10e921d68: '100.0' (None)> # children taxa: 17 ; level 15\n",
      "<Node object at 0x10e9219b0: '92.0' (None)> # children taxa: 13 ; level 16\n",
      "<Node object at 0x10e921748: '100.0' (None)> # children taxa: 8 ; level 17\n",
      "<Node object at 0x10e921e80: '100.0:g__Methanomethylophilus' (None)> # children taxa: 6 ; level 18\n",
      "<Node object at 0x10e943780: '100.0' (None)> # children taxa: 5 ; level 19\n",
      "<Node object at 0x10e943c18: '53.0' (None)> # children taxa: 4 ; level 20\n",
      "<Node object at 0x10e943978: '100.0' (None)> # children taxa: 3 ; level 21\n",
      "<Node object at 0x10db390f0: '100.0' (None)> # children taxa: 2 ; level 22\n",
      "<Node object at 0x10db39438: 'None' (<Taxon 0x10db39eb8 'GB_GCA_002503545.1'>)> # children taxa: 0 ; level 23\n",
      "<Node object at 0x10db39c88: 'None' (<Taxon 0x10db395f8 'RS_GCF_000300255.2'>)> # children taxa: 0 ; level 23\n",
      "<Node object at 0x10dbb66d8: 'None' (<Taxon 0x10dbb6828 'RS_GCF_001481295.1'>)> # children taxa: 0 ; level 22\n",
      "<Node object at 0x10dbb64e0: 'None' (<Taxon 0x10dbb6eb8 'RS_GCF_001560915.1'>)> # children taxa: 0 ; level 21\n",
      "<Node object at 0x10d2f4400: 'None' (<Taxon 0x10d2f4d30 'RS_GCF_000350305.1'>)> # children taxa: 0 ; level 20\n",
      "<Node object at 0x10d2f4320: 'None' (<Taxon 0x10d2f4940 'GB_GCA_002495325.1'>)> # children taxa: 0 ; level 19\n",
      "<Node object at 0x10e7427f0: '53.0' (None)> # children taxa: 2 ; level 18\n",
      "<Node object at 0x10e742978: 'None' (<Taxon 0x10e7426d8 'GB_GCA_002506905.1'>)> # children taxa: 0 ; level 19\n",
      "<Node object at 0x10e7424e0: 'None' (<Taxon 0x10e742ba8 'RS_GCF_000800805.1'>)> # children taxa: 0 ; level 19\n",
      "<Node object at 0x10d29a390: '100.0' (None)> # children taxa: 5 ; level 17\n",
      "<Node object at 0x10d29afd0: '100.0:g__UBA71' (None)> # children taxa: 3 ; level 18\n",
      "<Node object at 0x10d29add8: '100.0' (None)> # children taxa: 2 ; level 19\n",
      "<Node object at 0x10d29a2b0: 'None' (<Taxon 0x10d29a978 'GB_GCA_002504495.1'>)> # children taxa: 0 ; level 20\n",
      "<Node object at 0x10e887128: 'None' (<Taxon 0x10e887be0 'GB_GCA_002506425.1'>)> # children taxa: 0 ; level 20\n",
      "<Node object at 0x10e887978: 'None' (<Taxon 0x10e887518 'GB_GCA_002504405.1'>)> # children taxa: 0 ; level 19\n",
      "<Node object at 0x10e887cc0: '100.0:g__ISO4-G1' (None)> # children taxa: 2 ; level 18\n",
      "<Node object at 0x10d48ae80: 'None' (<Taxon 0x10d48ab00 'GB_GCA_001563305.1'>)> # children taxa: 0 ; level 19\n",
      "<Node object at 0x10d48a940: 'None' (<Taxon 0x10d48a4e0 'GB_GCA_002506175.1'>)> # children taxa: 0 ; level 19\n",
      "<Node object at 0x10cf08048: '100.0:g__VadinCA11' (None)> # children taxa: 4 ; level 16\n",
      "<Node object at 0x10cf08f60: '100.0' (None)> # children taxa: 3 ; level 17\n",
      "<Node object at 0x10cf08198: '96.0' (None)> # children taxa: 2 ; level 18\n",
      "<Node object at 0x10cf08518: 'None' (<Taxon 0x10d8f6cf8 'GB_GCA_002505345.1'>)> # children taxa: 0 ; level 19\n",
      "<Node object at 0x10d8f61d0: 'None' (<Taxon 0x10d8f66d8 'GB_GCA_002509405.1'>)> # children taxa: 0 ; level 19\n",
      "<Node object at 0x10d8f6e10: 'None' (<Taxon 0x10d7db4e0 'GB_GCA_002498365.1'>)> # children taxa: 0 ; level 18\n",
      "<Node object at 0x10d7db898: 'None' (<Taxon 0x10d7db400 'GB_GCA_002498605.1'>)> # children taxa: 0 ; level 17\n",
      "<Node object at 0x10ec4ecc0: '100.0:g__RumEn-M2' (None)> # children taxa: 3 ; level 15\n",
      "<Node object at 0x10ec4e630: '99.0' (None)> # children taxa: 2 ; level 16\n",
      "<Node object at 0x10ec4e1d0: 'None' (<Taxon 0x10ec4e080 'GB_GCA_001421175.1'>)> # children taxa: 0 ; level 17\n",
      "<Node object at 0x10ec4eac8: 'None' (<Taxon 0x10ec4ee10 'GB_GCA_002506255.1'>)> # children taxa: 0 ; level 17\n",
      "<Node object at 0x10d702470: 'None' (<Taxon 0x10d702828 'GB_GCA_002502965.1'>)> # children taxa: 0 ; level 16\n",
      "<Node object at 0x10d702940: 'None' (<Taxon 0x10e57e208 'UBA9915'>)> # children taxa: 0 ; level 14\n",
      "<Node object at 0x10e57ee48: '99.0' (None)> # children taxa: 11 ; level 13\n",
      "<Node object at 0x10e57ec18: '100.0:f__Methanomassiliicoccaceae' (None)> # children taxa: 7 ; level 14\n",
      "<Node object at 0x10e57e470: '86.0' (None)> # children taxa: 6 ; level 15\n",
      "<Node object at 0x10cd20908: '100.0:g__DTU008' (None)> # children taxa: 4 ; level 16\n",
      "<Node object at 0x10cd206d8: '53.0' (None)> # children taxa: 3 ; level 17\n",
      "<Node object at 0x10cd20128: '98.0' (None)> # children taxa: 2 ; level 18\n",
      "<Node object at 0x10cd20390: 'None' (<Taxon 0x10cd209e8 'GB_GCA_001421185.1'>)> # children taxa: 0 ; level 19\n",
      "<Node object at 0x10cd204e0: 'None' (<Taxon 0x10d3b0668 'GB_GCA_002498285.1'>)> # children taxa: 0 ; level 19\n",
      "<Node object at 0x10d3b0be0: 'None' (<Taxon 0x10d3b0128 'GB_GCA_002504525.1'>)> # children taxa: 0 ; level 18\n",
      "<Node object at 0x10d3b0f28: 'None' (<Taxon 0x10e1ebe80 'GB_GCA_001512965.1'>)> # children taxa: 0 ; level 17\n",
      "<Node object at 0x10e1eb630: '94.0:g__Methanomassiliicoccus' (None)> # children taxa: 2 ; level 16\n",
      "<Node object at 0x10e1ebb00: 'None' (<Taxon 0x10e1eb2b0 'GB_GCA_002508545.1'>)> # children taxa: 0 ; level 17\n",
      "<Node object at 0x10e1eb550: 'None' (<Taxon 0x10e1eb898 'RS_GCF_000308215.1'>)> # children taxa: 0 ; level 17\n",
      "<Node object at 0x10e3aba20: 'None' (<Taxon 0x10e3ab390 'RS_GCF_000404225.1'>)> # children taxa: 0 ; level 15\n",
      "<Node object at 0x10e3abd30: '96.0' (None)> # children taxa: 4 ; level 14\n",
      "<Node object at 0x10e3abb38: '100.0:f__UBA472; g__UBA472' (None)> # children taxa: 2 ; level 15\n",
      "<Node object at 0x10e3ab470: 'None' (<Taxon 0x10ee11550 'GB_GCA_002497075.1'>)> # children taxa: 0 ; level 16\n",
      "<Node object at 0x10ee11208: 'None' (<Taxon 0x10ee11940 'UBA8940'>)> # children taxa: 0 ; level 16\n",
      "<Node object at 0x10ee112e8: '100.0' (None)> # children taxa: 2 ; level 15\n",
      "<Node object at 0x10e984550: 'None' (<Taxon 0x10e9841d0 'GB_GCA_003135935.1'>)> # children taxa: 0 ; level 16\n",
      "<Node object at 0x10e9840f0: 'None' (<Taxon 0x10e984358 'GB_GCA_003153895.1'>)> # children taxa: 0 ; level 16\n",
      "<Node object at 0x10e687a20: '100.0:o__UBA10834; f__UBA10834' (None)> # children taxa: 4 ; level 12\n",
      "<Node object at 0x10e687828: '71.0' (None)> # children taxa: 3 ; level 13\n",
      "<Node object at 0x10e687908: '100.0:g__COMBO-56-21' (None)> # children taxa: 2 ; level 14\n",
      "<Node object at 0x10e6872e8: 'None' (<Taxon 0x10ea1ff98 'GB_GCA_001800675.1'>)> # children taxa: 0 ; level 15\n",
      "<Node object at 0x10ea1fc88: 'None' (<Taxon 0x10ea1f160 'GB_GCA_001800815.1'>)> # children taxa: 0 ; level 15\n",
      "<Node object at 0x10ea1f630: 'None' (<Taxon 0x10e88a860 'UBA9653'>)> # children taxa: 0 ; level 14\n",
      "<Node object at 0x10e88a278: 'None' (<Taxon 0x10e88a160 'UBA10834'>)> # children taxa: 0 ; level 13\n",
      "<Node object at 0x10e88af98: '85.0' (None)> # children taxa: 6 ; level 11\n",
      "<Node object at 0x10e88aba8: '100.0:o__SG8-5; f__SG8-5' (None)> # children taxa: 3 ; level 12\n",
      "<Node object at 0x10d7bc710: '100.0:g__UBA147' (None)> # children taxa: 2 ; level 13\n",
      "<Node object at 0x10d7bccc0: 'None' (<Taxon 0x10d7bc7f0 'GB_GCA_002496385.1'>)> # children taxa: 0 ; level 14\n",
      "<Node object at 0x10d7bc5f8: 'None' (<Taxon 0x10d7bc240 'GB_GCA_002499085.1'>)> # children taxa: 0 ; level 14\n",
      "<Node object at 0x10d45ecc0: 'None' (<Taxon 0x10d45e978 'GB_GCA_001595885.1'>)> # children taxa: 0 ; level 13\n",
      "<Node object at 0x10d45eda0: '100.0:o__RBG-16-68-12; f__RBG-16-68-12; g__RBG-16-68-12' (None)> # children taxa: 3 ; level 12\n",
      "<Node object at 0x10d45e470: '100.0' (None)> # children taxa: 2 ; level 13\n",
      "<Node object at 0x10d45e390: 'None' (<Taxon 0x10d45e6a0 'GB_GCA_001800745.1'>)> # children taxa: 0 ; level 14\n",
      "<Node object at 0x10f7f9908: 'None' (<Taxon 0x10f7f9240 'GB_GCA_001800825.1'>)> # children taxa: 0 ; level 14\n",
      "<Node object at 0x10f7f9dd8: 'None' (<Taxon 0x10e4dc710 'UBA8695'>)> # children taxa: 0 ; level 13\n",
      "<Node object at 0x10e4dc5c0: '100.0:o__UBA184; f__UBA184; g__UBA184' (None)> # children taxa: 2 ; level 10\n",
      "<Node object at 0x10e4dccf8: 'None' (<Taxon 0x10e4dc4e0 'GB_GCA_002495185.1'>)> # children taxa: 0 ; level 11\n",
      "<Node object at 0x10d778320: 'None' (<Taxon 0x10d778208 'GB_GCA_002503985.1'>)> # children taxa: 0 ; level 11\n",
      "<Node object at 0x10d778128: '100.0' (None)> # children taxa: 21 ; level 9\n",
      "<Node object at 0x10d778d68: '100.0:o__Thermoplasmatales' (None)> # children taxa: 17 ; level 10\n",
      "<Node object at 0x10d7785f8: '100.0:f__Thermoplasmataceae' (None)> # children taxa: 16 ; level 11\n",
      "<Node object at 0x10d9105f8: '30.0' (None)> # children taxa: 15 ; level 12\n",
      "<Node object at 0x10d9103c8: '17.0' (None)> # children taxa: 12 ; level 13\n",
      "<Node object at 0x10d910a90: '36.0' (None)> # children taxa: 9 ; level 14\n",
      "<Node object at 0x10d910d68: '100.0' (None)> # children taxa: 5 ; level 15\n",
      "<Node object at 0x10d9107f0: '100.0' (None)> # children taxa: 4 ; level 16\n",
      "<Node object at 0x10d910b70: '100.0:g__Ferroplasma' (None)> # children taxa: 3 ; level 17\n",
      "<Node object at 0x10e5e8048: '100.0' (None)> # children taxa: 2 ; level 18\n",
      "<Node object at 0x10e5e8828: 'None' (<Taxon 0x10e5e84a8 'RS_GCF_000152265.2'>)> # children taxa: 0 ; level 19\n",
      "<Node object at 0x10e5e8a90: 'None' (<Taxon 0x10e5e8eb8 'RS_GCF_002078355.1'>)> # children taxa: 0 ; level 19\n",
      "<Node object at 0x10d13a780: 'None' (<Taxon 0x10d13a860 'GB_GCA_002505185.1'>)> # children taxa: 0 ; level 18\n",
      "<Node object at 0x10d13a470: 'None' (<Taxon 0x10d13ac18 'RS_GCF_001402945.1'>)> # children taxa: 0 ; level 17\n",
      "<Node object at 0x10d13acf8: 'None' (<Taxon 0x10d359160 'RS_GCF_900176435.1'>)> # children taxa: 0 ; level 16\n",
      "<Node object at 0x10d3596d8: '56.0' (None)> # children taxa: 4 ; level 15\n",
      "<Node object at 0x10d359c50: '78.0' (None)> # children taxa: 2 ; level 16\n",
      "<Node object at 0x10d359a58: 'None' (<Taxon 0x10d359470 'GB_GCA_000965745.1'>)> # children taxa: 0 ; level 17\n",
      "<Node object at 0x10d359eb8: 'None' (<Taxon 0x10cebcb70 'GB_GCA_002204705.1'>)> # children taxa: 0 ; level 17\n",
      "<Node object at 0x10cebcc50: '100.0' (None)> # children taxa: 2 ; level 16\n",
      "<Node object at 0x10cebc2b0: 'None' (<Taxon 0x10cebceb8 'GB_GCA_002498845.1'>)> # children taxa: 0 ; level 17\n",
      "<Node object at 0x10cebc390: 'None' (<Taxon 0x10e6cb7f0 'GB_GCA_002503205.1'>)> # children taxa: 0 ; level 17\n",
      "<Node object at 0x10e6cb0f0: '100.0:g__Thermoplasma' (None)> # children taxa: 3 ; level 14\n",
      "<Node object at 0x10e6cb8d0: '100.0' (None)> # children taxa: 2 ; level 15\n",
      "<Node object at 0x10e6cbb38: 'None' (<Taxon 0x10dfdfc88 'RS_GCF_000195915.1'>)> # children taxa: 0 ; level 16\n",
      "<Node object at 0x10dfdf7f0: 'None' (<Taxon 0x10dfdf6d8 'RS_GCF_003205235.1'>)> # children taxa: 0 ; level 16\n",
      "<Node object at 0x10dfdf4e0: 'None' (<Taxon 0x10d050e10 'RS_GCF_000011185.1'>)> # children taxa: 0 ; level 15\n",
      "<Node object at 0x10d050710: '46.0' (None)> # children taxa: 3 ; level 13\n",
      "<Node object at 0x10d050278: '100.0' (None)> # children taxa: 2 ; level 14\n",
      "<Node object at 0x10d0504a8: 'None' (<Taxon 0x10d050f28 'GB_GCA_000496135.1'>)> # children taxa: 0 ; level 15\n",
      "<Node object at 0x10d050390: 'None' (<Taxon 0x10e2e87f0 'GB_GCA_900083515.1'>)> # children taxa: 0 ; level 15\n",
      "<Node object at 0x10e2e8ef0: 'None' (<Taxon 0x10e2e8a58 'GB_GCA_002502705.1'>)> # children taxa: 0 ; level 14\n",
      "<Node object at 0x10e2e80f0: 'None' (<Taxon 0x10ed850f0 'GB_GCA_001316105.1'>)> # children taxa: 0 ; level 12\n",
      "<Node object at 0x10ed85ac8: 'None' (<Taxon 0x10ed85550 'GB_GCA_001856825.1'>)> # children taxa: 0 ; level 11\n",
      "<Node object at 0x10ed859b0: '99.0' (None)> # children taxa: 4 ; level 10\n",
      "<Node object at 0x10ed856a0: '99.0:o__Acidiprofundales' (None)> # children taxa: 3 ; level 11\n",
      "<Node object at 0x10ed851d0: '100.0:f__Aciduliprofundaceae; g__Aciduliprofundum' (None)> # children taxa: 2 ; level 12\n",
      "<Node object at 0x10d90d898: 'None' (<Taxon 0x10d90dda0 'RS_GCF_000025665.1'>)> # children taxa: 0 ; level 13\n",
      "<Node object at 0x10d90d550: 'None' (<Taxon 0x10d90d470 'RS_GCF_000327505.1'>)> # children taxa: 0 ; level 13\n",
      "<Node object at 0x10d90d208: 'None' (<Taxon 0x10d3a5ba8 'GB_GCA_002011395.1'>)> # children taxa: 0 ; level 12\n",
      "<Node object at 0x10d3a5f60: 'None' (<Taxon 0x10d3a5438 'GB_GCA_002878135.1'>)> # children taxa: 0 ; level 11\n",
      "<Node object at 0x10d62e9e8: '100.0' (None)> # children taxa: 2 ; level 8\n",
      "<Node object at 0x10d62ee80: 'None' (<Taxon 0x10d62e208 'GB_GCA_002254385.1'>)> # children taxa: 0 ; level 9\n",
      "<Node object at 0x10d62e908: 'None' (<Taxon 0x10d62e400 'GB_GCA_002254765.1'>)> # children taxa: 0 ; level 9\n"
     ]
    }
   ],
   "source": [
    "print_node_children_counts(arc.seed_node, match='c__Thermoplasmata')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Node object at 0x10d12f6a0: '93.0:c__Bathyarchaeia' (None)> # children taxa: 40 ; level 6\n",
      "<Node object at 0x10d12f940: '83.0' (None)> # children taxa: 39 ; level 7\n",
      "<Node object at 0x10d12fa20: '96.0' (None)> # children taxa: 27 ; level 8\n",
      "<Node object at 0x10d65f128: '99.0' (None)> # children taxa: 21 ; level 9\n",
      "<Node object at 0x10d65ff98: '100.0:o__B26-1' (None)> # children taxa: 20 ; level 10\n",
      "<Node object at 0x10d65f2b0: '95.0' (None)> # children taxa: 19 ; level 11\n",
      "<Node object at 0x10d65fb70: '100.0' (None)> # children taxa: 16 ; level 12\n",
      "<Node object at 0x10d65f390: '100.0:f__UBA233' (None)> # children taxa: 12 ; level 13\n",
      "<Node object at 0x10d65f6a0: '100.0' (None)> # children taxa: 10 ; level 14\n",
      "<Node object at 0x10ce122b0: '100.0' (None)> # children taxa: 8 ; level 15\n",
      "<Node object at 0x10ce12668: '31.0' (None)> # children taxa: 6 ; level 16\n",
      "<Node object at 0x10ce12390: '100.0' (None)> # children taxa: 5 ; level 17\n",
      "<Node object at 0x10ce12048: '100.0' (None)> # children taxa: 4 ; level 18\n",
      "<Node object at 0x10ce12cf8: '100.0' (None)> # children taxa: 2 ; level 19\n",
      "<Node object at 0x10d187cf8: 'None' (<Taxon 0x10d1879b0 'GB_GCA_003141855.1'>)> # children taxa: 0 ; level 20\n",
      "<Node object at 0x10d187ac8: 'None' (<Taxon 0x10d187198 'GB_GCA_003151735.1'>)> # children taxa: 0 ; level 20\n",
      "<Node object at 0x10d187278: '94.0' (None)> # children taxa: 2 ; level 19\n",
      "<Node object at 0x10eb02a20: 'None' (<Taxon 0x10eb025f8 'GB_GCA_003164295.1'>)> # children taxa: 0 ; level 20\n",
      "<Node object at 0x10eb02d30: 'None' (<Taxon 0x10eb024a8 'GB_GCA_003165195.1'>)> # children taxa: 0 ; level 20\n",
      "<Node object at 0x10d21eef0: 'None' (<Taxon 0x10d21e3c8 'GB_GCA_003133625.1'>)> # children taxa: 0 ; level 18\n",
      "<Node object at 0x10d21efd0: 'None' (<Taxon 0x10d21e0f0 'GB_GCA_001273335.1'>)> # children taxa: 0 ; level 17\n",
      "<Node object at 0x10d21ea58: '38.0' (None)> # children taxa: 2 ; level 16\n",
      "<Node object at 0x10d042400: 'None' (<Taxon 0x10d042a90 'GB_GCA_001273385.1'>)> # children taxa: 0 ; level 17\n",
      "<Node object at 0x10d0420f0: 'None' (<Taxon 0x10d042be0 'UBA11852'>)> # children taxa: 0 ; level 17\n",
      "<Node object at 0x10d042978: '60.0' (None)> # children taxa: 2 ; level 15\n",
      "<Node object at 0x10cc309b0: 'None' (<Taxon 0x10cc303c8 'GB_GCA_002010975.1'>)> # children taxa: 0 ; level 16\n",
      "<Node object at 0x10cc30240: 'None' (<Taxon 0x10cc30be0 'GB_GCA_002779555.1'>)> # children taxa: 0 ; level 16\n",
      "<Node object at 0x10e085be0: '100.0:g__UBA233' (None)> # children taxa: 2 ; level 14\n",
      "<Node object at 0x10e085710: 'None' (<Taxon 0x10e085160 'GB_GCA_002010925.1'>)> # children taxa: 0 ; level 15\n",
      "<Node object at 0x10e085828: 'None' (<Taxon 0x10e085d30 'GB_GCA_002507245.1'>)> # children taxa: 0 ; level 15\n",
      "<Node object at 0x10e085a90: '100.0:f__BA1' (None)> # children taxa: 4 ; level 13\n",
      "<Node object at 0x10d27e5f8: '100.0' (None)> # children taxa: 3 ; level 14\n",
      "<Node object at 0x10d27ee10: '100.0' (None)> # children taxa: 2 ; level 15\n",
      "<Node object at 0x10d27e9b0: 'None' (<Taxon 0x10d27e860 'GB_GCA_003096235.1'>)> # children taxa: 0 ; level 16\n",
      "<Node object at 0x10d27e748: 'None' (<Taxon 0x10d27e4e0 'GB_GCA_003096255.1'>)> # children taxa: 0 ; level 16\n",
      "<Node object at 0x10d2b4e48: 'None' (<Taxon 0x10d2b4a90 'GB_GCA_001399795.1'>)> # children taxa: 0 ; level 15\n",
      "<Node object at 0x10d2b4cf8: 'None' (<Taxon 0x10d2b49b0 'GB_GCA_001399805.1'>)> # children taxa: 0 ; level 14\n",
      "<Node object at 0x10d9d0d68: '100.0:f__B26-1' (None)> # children taxa: 3 ; level 12\n",
      "<Node object at 0x10d9d0e48: '100.0:g__B26-1' (None)> # children taxa: 2 ; level 13\n",
      "<Node object at 0x10d9d0828: 'None' (<Taxon 0x10d9d0630 'GB_GCA_001593925.1'>)> # children taxa: 0 ; level 14\n",
      "<Node object at 0x10d9d00b8: 'None' (<Taxon 0x10d9d0710 'GB_GCA_001593935.1'>)> # children taxa: 0 ; level 14\n",
      "<Node object at 0x10e83f2b0: 'None' (<Taxon 0x10e83f940 'GB_GCA_001593845.1'>)> # children taxa: 0 ; level 13\n",
      "<Node object at 0x10e83fa58: 'None' (<Taxon 0x10e83f048 'GB_GCA_002490245.1'>)> # children taxa: 0 ; level 11\n",
      "<Node object at 0x10e83f438: 'None' (<Taxon 0x10e83fc18 'GB_GCA_002255025.1'>)> # children taxa: 0 ; level 10\n",
      "<Node object at 0x10d690ef0: '100.0' (None)> # children taxa: 6 ; level 9\n",
      "<Node object at 0x10d690b70: '100.0:o__TCS64; f__TCS64' (None)> # children taxa: 5 ; level 10\n",
      "<Node object at 0x10d690278: '100.0' (None)> # children taxa: 4 ; level 11\n",
      "<Node object at 0x10d690fd0: '100.0:g__UBA8941' (None)> # children taxa: 2 ; level 12\n",
      "<Node object at 0x10d690048: 'None' (<Taxon 0x10eb602b0 'GB_GCA_001774245.1'>)> # children taxa: 0 ; level 13\n",
      "<Node object at 0x10eb60390: 'None' (<Taxon 0x10eb60c18 'UBA8941'>)> # children taxa: 0 ; level 13\n",
      "<Node object at 0x10eb600b8: '100.0:g__RBG-16-57-9' (None)> # children taxa: 2 ; level 12\n",
      "<Node object at 0x10eb60ef0: 'None' (<Taxon 0x10d1e49e8 'GB_GCA_001775965.1'>)> # children taxa: 0 ; level 13\n",
      "<Node object at 0x10d1e45c0: 'None' (<Taxon 0x10d1e4198 'GB_GCA_001776015.1'>)> # children taxa: 0 ; level 13\n",
      "<Node object at 0x10d1e4d68: 'None' (<Taxon 0x10d1e46a0 'GB_GCA_002726865.1'>)> # children taxa: 0 ; level 11\n",
      "<Node object at 0x10dc00828: 'None' (<Taxon 0x10dc00e80 'GB_GCA_001593875.1'>)> # children taxa: 0 ; level 10\n",
      "<Node object at 0x10dc002b0: '65.0' (None)> # children taxa: 12 ; level 8\n",
      "<Node object at 0x10dc001d0: '84.0' (None)> # children taxa: 9 ; level 9\n",
      "<Node object at 0x10f7e1d68: '100.0:o__40CM-2-53-6' (None)> # children taxa: 8 ; level 10\n",
      "<Node object at 0x10f7e1358: '100.0' (None)> # children taxa: 7 ; level 11\n",
      "<Node object at 0x10f7e19b0: '100.0:f__40CM-2-53-6; g__40CM-2-53-6' (None)> # children taxa: 5 ; level 12\n",
      "<Node object at 0x10f7e1b00: '100.0' (None)> # children taxa: 3 ; level 13\n",
      "<Node object at 0x10f7e1668: '87.0' (None)> # children taxa: 2 ; level 14\n",
      "<Node object at 0x10ceb4940: 'None' (<Taxon 0x10ceb44e0 'GB_GCA_001915065.1'>)> # children taxa: 0 ; level 15\n",
      "<Node object at 0x10ceb4c88: 'None' (<Taxon 0x10ceb43c8 'GB_GCA_002495485.1'>)> # children taxa: 0 ; level 15\n",
      "<Node object at 0x10d30d1d0: 'None' (<Taxon 0x10d30d320 'GB_GCA_001919285.1'>)> # children taxa: 0 ; level 14\n",
      "<Node object at 0x10d30dcf8: '100.0' (None)> # children taxa: 2 ; level 13\n",
      "<Node object at 0x10d30dc18: 'None' (<Taxon 0x10d30d710 'GB_GCA_001918745.1'>)> # children taxa: 0 ; level 14\n",
      "<Node object at 0x10d093358: 'None' (<Taxon 0x10d093e80 'GB_GCA_001918765.1'>)> # children taxa: 0 ; level 14\n",
      "<Node object at 0x10d093160: '100.0' (None)> # children taxa: 2 ; level 12\n",
      "<Node object at 0x10d093240: 'None' (<Taxon 0x10d093c88 'GB_GCA_001768965.1'>)> # children taxa: 0 ; level 13\n",
      "<Node object at 0x10e7d48d0: 'None' (<Taxon 0x10e7d49e8 'GB_GCA_003152955.1'>)> # children taxa: 0 ; level 13\n",
      "<Node object at 0x10e7d4eb8: 'None' (<Taxon 0x10e7d4c50 'GB_GCA_001775955.1'>)> # children taxa: 0 ; level 11\n",
      "<Node object at 0x10cf62898: 'None' (<Taxon 0x10cf62978 'GB_GCA_001775995.1'>)> # children taxa: 0 ; level 10\n",
      "<Node object at 0x10cf62198: '100.0:o__B25; f__B25' (None)> # children taxa: 3 ; level 9\n",
      "<Node object at 0x10cf624a8: '56.0' (None)> # children taxa: 2 ; level 10\n",
      "<Node object at 0x10cf62b38: 'None' (<Taxon 0x10d173320 'GB_GCA_001593855.1'>)> # children taxa: 0 ; level 11\n",
      "<Node object at 0x10d173ba8: 'None' (<Taxon 0x10d173f60 'GB_GCA_002011035.1'>)> # children taxa: 0 ; level 11\n",
      "<Node object at 0x10d173cc0: 'None' (<Taxon 0x10d173ac8 'GB_GCA_002254975.1'>)> # children taxa: 0 ; level 10\n",
      "<Node object at 0x10d3e0c18: 'None' (<Taxon 0x10d3e0d68 'GB_GCA_001593865.1'>)> # children taxa: 0 ; level 7\n"
     ]
    }
   ],
   "source": [
    "print_node_children_counts(arc.seed_node,match='c__Bathyarchaeia')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "import random \n",
    "\n",
    "def get_node(tree,match):\n",
    "    for child in tree.seed_node.child_nodes():\n",
    "        if child.label:\n",
    "            if match in child.label:\n",
    "                matched_node = child\n",
    "        get_node(tree,match)\n",
    "    return matched_node\n",
    "\n",
    "def parse_genera_from_node(node,taxa,genomes_per_genera=1):\n",
    "    for child in node.child_nodes():\n",
    "        #if it's a genus, randomly select one genome from this genus to keep, delete the rest\n",
    "        #(go delete route because there are occasionally unclassified taxa that don't have g__ labels)\n",
    "        if child.label:\n",
    "            if 'g__' in child.label:\n",
    "                leaf_taxa = [leaf.taxon.label for leaf in child.leaf_nodes()]\n",
    "                print(child.label,'# children taxa:',len(leaf_taxa))\n",
    "                to_keep = random.sample(leaf_taxa,k=genomes_per_genera)\n",
    "                to_delete = leaf_taxa\n",
    "                [to_delete.remove(keeper) for keeper in to_keep]\n",
    "                [taxa.remove(delete) for delete in to_delete]\n",
    "                print('length new taxa list:',len(taxa))\n",
    "        parse_genera_from_node(child,taxa,genomes_per_genera)\n",
    "        \n",
    "\n",
    "def parse_genera_from_tree(node,to_match,taxa,genomes_per_genera=1):\n",
    "    for child in node.child_nodes():\n",
    "        if child.label:\n",
    "            if len([s for s in to_match if s in child.label])>0:\n",
    "                print('processing',child.label,'...')\n",
    "                parse_genera_from_node(child,taxa,genomes_per_genera=genomes_per_genera)\n",
    "                print('length new taxa:',len(taxa))\n",
    "        parse_genera_from_tree(child,to_match,taxa,genomes_per_genera=genomes_per_genera)\n",
    "        \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "107\n",
      "1523\n"
     ]
    }
   ],
   "source": [
    "print(len(arc_valid))\n",
    "print(len(bac_valid))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "processing 86.0:c__Thermoplasmata ...\n",
      "100.0:g__Methanomethylophilus # children taxa: 6\n",
      "length new taxa list: 102\n",
      "100.0:g__UBA71 # children taxa: 3\n",
      "length new taxa list: 100\n",
      "100.0:g__ISO4-G1 # children taxa: 2\n",
      "length new taxa list: 99\n",
      "100.0:g__VadinCA11 # children taxa: 4\n",
      "length new taxa list: 96\n",
      "100.0:g__RumEn-M2 # children taxa: 3\n",
      "length new taxa list: 94\n",
      "100.0:g__DTU008 # children taxa: 4\n",
      "length new taxa list: 91\n",
      "94.0:g__Methanomassiliicoccus # children taxa: 2\n",
      "length new taxa list: 90\n",
      "100.0:f__UBA472; g__UBA472 # children taxa: 2\n",
      "length new taxa list: 89\n",
      "100.0:g__COMBO-56-21 # children taxa: 2\n",
      "length new taxa list: 88\n",
      "100.0:g__UBA147 # children taxa: 2\n",
      "length new taxa list: 87\n",
      "100.0:o__RBG-16-68-12; f__RBG-16-68-12; g__RBG-16-68-12 # children taxa: 3\n",
      "length new taxa list: 85\n",
      "100.0:o__UBA184; f__UBA184; g__UBA184 # children taxa: 2\n",
      "length new taxa list: 84\n",
      "100.0:g__Ferroplasma # children taxa: 3\n",
      "length new taxa list: 82\n",
      "100.0:g__Thermoplasma # children taxa: 3\n",
      "length new taxa list: 80\n",
      "100.0:f__Aciduliprofundaceae; g__Aciduliprofundum # children taxa: 2\n",
      "length new taxa list: 79\n",
      "length new taxa: 79\n",
      "processing 93.0:c__Bathyarchaeia ...\n",
      "100.0:g__UBA233 # children taxa: 2\n",
      "length new taxa list: 78\n",
      "100.0:g__B26-1 # children taxa: 2\n",
      "length new taxa list: 77\n",
      "100.0:g__UBA8941 # children taxa: 2\n",
      "length new taxa list: 76\n",
      "100.0:g__RBG-16-57-9 # children taxa: 2\n",
      "length new taxa list: 75\n",
      "100.0:f__40CM-2-53-6; g__40CM-2-53-6 # children taxa: 5\n",
      "length new taxa list: 71\n",
      "length new taxa: 71\n",
      "71\n"
     ]
    }
   ],
   "source": [
    "to_match = ['c__Thermoplasmata','c__Bathyarchaeia']\n",
    "new_arc = arc_valid.copy()\n",
    "parse_genera_from_tree(arc.seed_node,to_match,new_arc,genomes_per_genera=1)\n",
    "print(len(new_arc))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "processing 91.0:o__Actinomycetales ...\n",
      "100.0:g__Plantibacter # children taxa: 5\n",
      "length new taxa list: 1519\n",
      "100.0:g__Mycetocola_A # children taxa: 2\n",
      "length new taxa list: 1518\n",
      "100.0:g__Okibacterium # children taxa: 2\n",
      "length new taxa list: 1517\n",
      "100.0:g__Microbacterium_A # children taxa: 2\n",
      "length new taxa list: 1516\n",
      "100.0:g__Agromyces # children taxa: 12\n",
      "length new taxa list: 1505\n",
      "100.0:g__Aurantimicrobium # children taxa: 3\n",
      "length new taxa list: 1503\n",
      "100.0:g__Leifsonia # children taxa: 12\n",
      "length new taxa list: 1492\n",
      "100.0:g__Cryobacterium # children taxa: 18\n",
      "length new taxa list: 1475\n",
      "100.0:g__Microterricola # children taxa: 5\n",
      "length new taxa list: 1471\n",
      "100.0:g__Root1293 # children taxa: 3\n",
      "length new taxa list: 1469\n",
      "100.0:g__Rathayibacter # children taxa: 10\n",
      "length new taxa list: 1460\n",
      "100.0:g__Agreia # children taxa: 5\n",
      "length new taxa list: 1456\n",
      "100.0:g__Herbiconiux # children taxa: 2\n",
      "length new taxa list: 1455\n",
      "100.0:g__Clavibacter # children taxa: 11\n",
      "length new taxa list: 1445\n",
      "100.0:g__Salinibacterium # children taxa: 6\n",
      "length new taxa list: 1440\n",
      "93.0:g__Cryobacterium_A # children taxa: 2\n",
      "length new taxa list: 1439\n",
      "100.0:g__Curtobacterium # children taxa: 33\n",
      "length new taxa list: 1407\n",
      "100.0:g__Frigoribacterium # children taxa: 6\n",
      "length new taxa list: 1402\n",
      "100.0:g__Frondihabitans # children taxa: 2\n",
      "length new taxa list: 1401\n",
      "100.0:g__UBA1487 # children taxa: 3\n",
      "length new taxa list: 1399\n",
      "98.0:g__Yonghaparkia # children taxa: 6\n",
      "length new taxa list: 1394\n",
      "100.0:g__Pontimonas # children taxa: 4\n",
      "length new taxa list: 1391\n",
      "100.0:g__Agrococcus # children taxa: 7\n",
      "length new taxa list: 1385\n",
      "100.0:g__Microbacterium # children taxa: 119\n",
      "length new taxa list: 1267\n",
      "100.0:g__Leucobacter # children taxa: 16\n",
      "length new taxa list: 1252\n",
      "83.0:g__Pseudoclavibacter # children taxa: 6\n",
      "length new taxa list: 1247\n",
      "100.0:g__Gulosibacter # children taxa: 4\n",
      "length new taxa list: 1244\n",
      "100.0:g__Aquiluna # children taxa: 7\n",
      "length new taxa list: 1238\n",
      "76.0:g__Rhodoluna # children taxa: 3\n",
      "length new taxa list: 1236\n",
      "99.0:g__Pseudarthrobacter # children taxa: 21\n",
      "length new taxa list: 1216\n",
      "100.0:g__Pseudarthrobacter_A # children taxa: 7\n",
      "length new taxa list: 1210\n",
      "97.0:g__Arthrobacter_I # children taxa: 11\n",
      "length new taxa list: 1200\n",
      "100.0:g__Leaf337 # children taxa: 5\n",
      "length new taxa list: 1196\n",
      "91.0:g__Arthrobacter # children taxa: 7\n",
      "length new taxa list: 1190\n",
      "100.0:g__Paenarthrobacter # children taxa: 11\n",
      "length new taxa list: 1180\n",
      "100.0:g__MA-N2 # children taxa: 3\n",
      "length new taxa list: 1178\n",
      "100.0:g__Arthrobacter_G # children taxa: 2\n",
      "length new taxa list: 1177\n",
      "100.0:g__Sinomonas # children taxa: 4\n",
      "length new taxa list: 1174\n",
      "100.0:g__Arthrobacter_A # children taxa: 13\n",
      "length new taxa list: 1162\n",
      "99.0:g__Renibacterium # children taxa: 2\n",
      "length new taxa list: 1161\n",
      "100.0:g__Arthrobacter_C # children taxa: 2\n",
      "length new taxa list: 1160\n",
      "100.0:g__Tersicoccus # children taxa: 2\n",
      "length new taxa list: 1159\n",
      "100.0:g__Arthrobacter_D # children taxa: 20\n",
      "length new taxa list: 1140\n",
      "100.0:g__Arthrobacter_B # children taxa: 5\n",
      "length new taxa list: 1136\n",
      "100.0:g__Arthrobacter_F # children taxa: 2\n",
      "length new taxa list: 1135\n",
      "100.0:g__Glutamicibacter # children taxa: 12\n",
      "length new taxa list: 1124\n",
      "100.0:g__Paeniglutamicibacter # children taxa: 6\n",
      "length new taxa list: 1119\n",
      "99.0:g__Micrococcus # children taxa: 5\n",
      "length new taxa list: 1115\n",
      "100.0:g__Yaniella # children taxa: 2\n",
      "length new taxa list: 1114\n",
      "100.0:g__Nesterenkonia # children taxa: 7\n",
      "length new taxa list: 1108\n",
      "100.0:g__Pseudoglutamicibacter # children taxa: 2\n",
      "length new taxa list: 1107\n",
      "100.0:g__Kocuria # children taxa: 15\n",
      "length new taxa list: 1093\n",
      "100.0:g__Rothia # children taxa: 13\n",
      "length new taxa list: 1081\n",
      "100.0:f__Brevibacteriaceae; g__Brevibacterium # children taxa: 18\n",
      "length new taxa list: 1064\n",
      "100.0:g__Pauljensenia # children taxa: 27\n",
      "length new taxa list: 1038\n",
      "100.0:g__Gleimia # children taxa: 5\n",
      "length new taxa list: 1034\n",
      "100.0:g__Boudabousia # children taxa: 3\n",
      "length new taxa list: 1032\n",
      "100.0:g__Varibaculum # children taxa: 4\n",
      "length new taxa list: 1029\n",
      "100.0:g__Winkia # children taxa: 2\n",
      "length new taxa list: 1028\n",
      "100.0:g__Mobiluncus # children taxa: 2\n",
      "length new taxa list: 1027\n",
      "100.0:g__Actinomyces # children taxa: 31\n",
      "length new taxa list: 997\n",
      "100.0:g__Actinotignum # children taxa: 4\n",
      "length new taxa list: 994\n",
      "100.0:g__Actinobaculum # children taxa: 2\n",
      "length new taxa list: 993\n",
      "100.0:g__F0332 # children taxa: 4\n",
      "length new taxa list: 990\n",
      "100.0:g__Trueperella # children taxa: 2\n",
      "length new taxa list: 989\n",
      "100.0:g__Arcanobacterium # children taxa: 2\n",
      "length new taxa list: 988\n",
      "100.0:g__Arcanobacterium_A # children taxa: 2\n",
      "length new taxa list: 987\n",
      "100.0:g__UBA1383 # children taxa: 2\n",
      "length new taxa list: 986\n",
      "100.0:g__Bowdeniella # children taxa: 2\n",
      "length new taxa list: 985\n",
      "100.0:g__Serinibacter # children taxa: 4\n",
      "length new taxa list: 982\n",
      "100.0:g__Isoptericola # children taxa: 3\n",
      "length new taxa list: 980\n",
      "100.0:g__Promicromonospora # children taxa: 4\n",
      "length new taxa list: 977\n",
      "100.0:g__Cellulosimicrobium # children taxa: 7\n",
      "length new taxa list: 971\n",
      "100.0:g__Oerskovia # children taxa: 2\n",
      "length new taxa list: 970\n",
      "100.0:g__Timonella # children taxa: 2\n",
      "length new taxa list: 969\n",
      "100.0:g__Jonesia # children taxa: 2\n",
      "length new taxa list: 968\n",
      "100.0:g__Sanguibacter # children taxa: 5\n",
      "length new taxa list: 964\n",
      "100.0:g__Flavimobilis # children taxa: 3\n",
      "length new taxa list: 962\n",
      "100.0:g__Cellulomonas # children taxa: 23\n",
      "length new taxa list: 940\n",
      "100.0:g__Actinotalea # children taxa: 5\n",
      "length new taxa list: 936\n",
      "100.0:f__Demequinaceae; g__Demequina # children taxa: 28\n",
      "length new taxa list: 909\n",
      "100.0:g__Bifidobacterium # children taxa: 85\n",
      "length new taxa list: 825\n",
      "89.0:g__Scardovia # children taxa: 2\n",
      "length new taxa list: 824\n",
      "100.0:g__Alloscardovia # children taxa: 3\n",
      "length new taxa list: 822\n",
      "100.0:g__Pseudoscardovia # children taxa: 3\n",
      "length new taxa list: 820\n",
      "100.0:g__Brachybacterium # children taxa: 16\n",
      "length new taxa list: 805\n",
      "100.0:f__Quadrisphaeraceae; g__Quadrisphaera # children taxa: 2\n",
      "length new taxa list: 804\n",
      "100.0:g__R-H-3 # children taxa: 2\n",
      "length new taxa list: 803\n",
      "100.0:f__Kineococcaceae; g__Kineococcus # children taxa: 3\n",
      "length new taxa list: 801\n",
      "100.0:g__Janibacter # children taxa: 8\n",
      "length new taxa list: 794\n",
      "100.0:g__Tetrasphaera_A # children taxa: 2\n",
      "length new taxa list: 793\n",
      "100.0:g__Knoellia # children taxa: 7\n",
      "length new taxa list: 787\n",
      "88.0:g__Phycicoccus # children taxa: 4\n",
      "length new taxa list: 784\n",
      "100.0:g__Pedococcus # children taxa: 5\n",
      "length new taxa list: 780\n",
      "100.0:g__Terracoccus # children taxa: 2\n",
      "length new taxa list: 779\n",
      "100.0:g__UBA4719 # children taxa: 3\n",
      "length new taxa list: 777\n",
      "100.0:g__Serinicoccus # children taxa: 4\n",
      "length new taxa list: 774\n",
      "97.0:g__Ornithinimicrobium # children taxa: 3\n",
      "length new taxa list: 772\n",
      "100.0:g__Kytococcus # children taxa: 3\n",
      "length new taxa list: 770\n",
      "100.0:g__Branchiibius # children taxa: 2\n",
      "length new taxa list: 769\n",
      "100.0:g__Dermacoccus # children taxa: 2\n",
      "length new taxa list: 768\n",
      "100.0:g__Luteipulveratus # children taxa: 2\n",
      "length new taxa list: 767\n",
      "100.0:g__Mobilicoccus # children taxa: 2\n",
      "length new taxa list: 766\n",
      "100.0:g__Arsenicicoccus # children taxa: 2\n",
      "length new taxa list: 765\n",
      "length new taxa: 765\n",
      "processing 93.0:o__Rhodobacterales ...\n",
      "100.0:g__Tritonibacter # children taxa: 5\n",
      "length new taxa list: 761\n",
      "100.0:g__Phaeobacter # children taxa: 5\n",
      "length new taxa list: 757\n",
      "56.0:g__Pseudophaeobacter # children taxa: 8\n",
      "length new taxa list: 750\n",
      "100.0:g__Leisingera # children taxa: 11\n",
      "length new taxa list: 740\n",
      "100.0:g__56-14-T64 # children taxa: 2\n",
      "length new taxa list: 739\n",
      "100.0:g__Ruegeria_E # children taxa: 2\n",
      "length new taxa list: 738\n",
      "100.0:g__Ruegeria # children taxa: 15\n",
      "length new taxa list: 724\n",
      "100.0:g__Ruegeria_B # children taxa: 2\n",
      "length new taxa list: 723\n",
      "100.0:g__Pseudooceanicola_B # children taxa: 4\n",
      "length new taxa list: 720\n",
      "100.0:g__Marinibacterium # children taxa: 3\n",
      "length new taxa list: 718\n",
      "99.0:g__Sulfitobacter # children taxa: 20\n",
      "length new taxa list: 699\n",
      "100.0:g__Roseobacter # children taxa: 2\n",
      "length new taxa list: 698\n",
      "100.0:g__Sulfitobacter_C # children taxa: 3\n",
      "length new taxa list: 696\n",
      "100.0:g__Ascidiaceihabitans # children taxa: 2\n",
      "length new taxa list: 695\n",
      "96.0:g__Tateyamaria # children taxa: 4\n",
      "length new taxa list: 692\n",
      "99.0:g__Shimia # children taxa: 6\n",
      "length new taxa list: 687\n",
      "100.0:g__Pseudopelagicola # children taxa: 2\n",
      "length new taxa list: 686\n",
      "100.0:g__Cognatishimia # children taxa: 2\n",
      "length new taxa list: 685\n",
      "100.0:g__Planktotalea # children taxa: 3\n",
      "length new taxa list: 683\n",
      "94.0:g__Roseovarius # children taxa: 35\n",
      "length new taxa list: 649\n",
      "100.0:g__HLUCCO07 # children taxa: 2\n",
      "length new taxa list: 648\n",
      "99.0:g__Thalassobius # children taxa: 4\n",
      "length new taxa list: 645\n",
      "100.0:g__EhC02 # children taxa: 2\n",
      "length new taxa list: 644\n",
      "100.0:g__Lutimaribacter # children taxa: 2\n",
      "length new taxa list: 643\n",
      "100.0:g__Pseudooceanicola # children taxa: 9\n",
      "length new taxa list: 635\n",
      "100.0:g__Salipiger # children taxa: 11\n",
      "length new taxa list: 625\n",
      "100.0:g__Antarctobacter # children taxa: 2\n",
      "length new taxa list: 624\n",
      "95.0:g__Sagittula # children taxa: 3\n",
      "length new taxa list: 622\n",
      "100.0:g__Roseivivax # children taxa: 6\n",
      "length new taxa list: 617\n",
      "100.0:g__Marivita # children taxa: 4\n",
      "length new taxa list: 614\n",
      "100.0:g__Puniceibacterium # children taxa: 3\n",
      "length new taxa list: 612\n",
      "100.0:g__Pseudooceanicola_A # children taxa: 2\n",
      "length new taxa list: 611\n",
      "19.0:g__HIMB11 # children taxa: 16\n",
      "length new taxa list: 596\n",
      "100.0:g__Donghicola # children taxa: 2\n",
      "length new taxa list: 595\n",
      "100.0:g__Poseidonocella # children taxa: 2\n",
      "length new taxa list: 594\n",
      "99.0:g__Yoonia # children taxa: 15\n",
      "length new taxa list: 580\n",
      "100.0:g__Loktanella # children taxa: 5\n",
      "length new taxa list: 576\n",
      "100.0:g__Cognatiyoonia # children taxa: 2\n",
      "length new taxa list: 575\n",
      "100.0:g__Limimaricola # children taxa: 5\n",
      "length new taxa list: 571\n",
      "100.0:g__Ketogulonicigenium # children taxa: 2\n",
      "length new taxa list: 570\n",
      "100.0:g__Rubellimicrobium # children taxa: 2\n",
      "length new taxa list: 569\n",
      "100.0:g__Wenxinia # children taxa: 2\n",
      "length new taxa list: 568\n",
      "100.0:g__Marivivens # children taxa: 3\n",
      "length new taxa list: 566\n",
      "95.0:g__Octadecabacter # children taxa: 6\n",
      "length new taxa list: 561\n",
      "100.0:g__HLUCCA08 # children taxa: 3\n",
      "length new taxa list: 559\n",
      "98.0:g__Celeribacter # children taxa: 8\n",
      "length new taxa list: 552\n",
      "100.0:g__Maritimibacter # children taxa: 4\n",
      "length new taxa list: 549\n",
      "100.0:g__HL-12 # children taxa: 3\n",
      "length new taxa list: 547\n",
      "100.0:g__Aliiroseovarius # children taxa: 5\n",
      "length new taxa list: 543\n",
      "100.0:g__Planktomarina # children taxa: 4\n",
      "length new taxa list: 540\n",
      "100.0:g__Litoreibacter # children taxa: 5\n",
      "length new taxa list: 536\n",
      "90.0:g__Marinosulfonomonas # children taxa: 6\n",
      "length new taxa list: 531\n",
      "100.0:g__Halocynthiibacter # children taxa: 2\n",
      "length new taxa list: 530\n",
      "100.0:g__Jannaschia # children taxa: 8\n",
      "length new taxa list: 523\n",
      "100.0:g__EhC01 # children taxa: 2\n",
      "length new taxa list: 522\n",
      "100.0:g__Roseibacterium # children taxa: 2\n",
      "length new taxa list: 521\n",
      "100.0:g__Nioella # children taxa: 4\n",
      "length new taxa list: 518\n",
      "28.0:g__Actibacterium # children taxa: 5\n",
      "length new taxa list: 514\n",
      "100.0:g__Confluentimicrobium # children taxa: 2\n",
      "length new taxa list: 513\n",
      "100.0:g__Rhodovulum # children taxa: 7\n",
      "length new taxa list: 507\n",
      "100.0:g__Palleronia # children taxa: 3\n",
      "length new taxa list: 505\n",
      "100.0:g__Maribius # children taxa: 2\n",
      "length new taxa list: 504\n",
      "100.0:g__Rhodobacter_B # children taxa: 2\n",
      "length new taxa list: 503\n",
      "100.0:g__Pseudoruegeria_C # children taxa: 2\n",
      "length new taxa list: 502\n",
      "99.0:g__Mangrovicoccus # children taxa: 4\n",
      "length new taxa list: 499\n",
      "100.0:g__Paracoccus # children taxa: 43\n",
      "length new taxa list: 457\n",
      "100.0:g__Rhodobacter # children taxa: 5\n",
      "length new taxa list: 453\n",
      "100.0:g__Paenirhodobacter # children taxa: 2\n",
      "length new taxa list: 452\n",
      "100.0:g__Thioclava # children taxa: 11\n",
      "length new taxa list: 442\n",
      "94.0:g__Rhodobacter_D # children taxa: 4\n",
      "length new taxa list: 439\n",
      "100.0:g__Defluviimonas_B # children taxa: 2\n",
      "length new taxa list: 438\n",
      "100.0:g__QY30 # children taxa: 2\n",
      "length new taxa list: 437\n",
      "100.0:g__Defluviimonas # children taxa: 2\n",
      "length new taxa list: 436\n",
      "100.0:g__Tabrizicola # children taxa: 3\n",
      "length new taxa list: 434\n",
      "99.0:g__Rhodobacter_E # children taxa: 3\n",
      "length new taxa list: 432\n",
      "100.0:g__Pseudorhodobacter_A # children taxa: 3\n",
      "length new taxa list: 430\n",
      "100.0:g__XJSP # children taxa: 2\n",
      "length new taxa list: 429\n",
      "100.0:g__TH137 # children taxa: 4\n",
      "length new taxa list: 426\n",
      "100.0:g__Gemmobacter # children taxa: 3\n",
      "length new taxa list: 424\n",
      "100.0:g__Rhodobacter_A # children taxa: 5\n",
      "length new taxa list: 420\n",
      "100.0:g__Pseudorhodobacter # children taxa: 4\n",
      "length new taxa list: 417\n",
      "100.0:g__Gemmobacter_A # children taxa: 2\n",
      "length new taxa list: 416\n",
      "100.0:g__Haematobacter # children taxa: 3\n",
      "length new taxa list: 414\n",
      "100.0:g__Defluviimonas_A # children taxa: 2\n",
      "length new taxa list: 413\n",
      "100.0:g__Pararhodobacter # children taxa: 5\n",
      "length new taxa list: 409\n",
      "100.0:g__GCA-002705045 # children taxa: 6\n",
      "length new taxa list: 404\n",
      "100.0:g__GCA-002712045 # children taxa: 2\n",
      "length new taxa list: 403\n",
      "100.0:g__Bin36 # children taxa: 2\n",
      "length new taxa list: 402\n",
      "100.0:g__Amylibacter_A # children taxa: 5\n",
      "length new taxa list: 398\n",
      "100.0:g__UBA12010 # children taxa: 2\n",
      "length new taxa list: 397\n",
      "100.0:g__UBA3077 # children taxa: 6\n",
      "length new taxa list: 392\n",
      "100.0:g__DSL-40 # children taxa: 2\n",
      "length new taxa list: 391\n",
      "100.0:g__Albimonas # children taxa: 2\n",
      "length new taxa list: 390\n",
      "length new taxa: 390\n",
      "390\n"
     ]
    }
   ],
   "source": [
    "#same for bacteria\n",
    "to_match=['o__Actinomycetales','o__Rhodobacterales']\n",
    "new_bac = bac_valid.copy()\n",
    "parse_genera_from_tree(bac.seed_node,to_match,new_bac,genomes_per_genera=1)\n",
    "print(len(new_bac))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "390 71\n",
      "461\n"
     ]
    }
   ],
   "source": [
    "print(len(new_bac),len(new_arc))\n",
    "print(len(new_bac)+len(new_arc))\n",
    "#that'll be way easier"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'RS_GCF_900177615.1'"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_bac[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False\n",
      "False\n"
     ]
    }
   ],
   "source": [
    "print(new_bac[0][3:] in list(ftp['ncbi_genbank_assembly_accession']))\n",
    "print(new_bac[0][3:] in list(ftp['parsed_genome_accession']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23458\n",
      "390\n",
      "390\n",
      "0\n",
      "23458\n",
      "1248\n",
      "71\n",
      "71\n",
      "0\n",
      "1248\n",
      "461 461\n"
     ]
    }
   ],
   "source": [
    "#create new valdf and save new val list \n",
    "ftp = pd.read_csv(\"GTDBrepGenomes_withftp.csv\")\n",
    "arcftp = pd.read_csv(\"GTDBrepGenomes_arc_withftpsrr.csv\")\n",
    "\n",
    "valdf = pd.DataFrame()\n",
    "start = len(ftp)\n",
    "print(start)\n",
    "for b in new_bac:\n",
    "    if b[3:] in list(ftp['ncbi_genbank_assembly_accession']):\n",
    "        valdf = valdf.append(ftp[ftp['ncbi_genbank_assembly_accession']==b[3:]])\n",
    "        #ftp.drop(ftp[ftp['ncbi_genbank_assembly_accession']==b[3:]].index,inplace=True)\n",
    "    elif b[3:] in list(ftp['parsed_genome_accession']):\n",
    "        valdf = valdf.append(ftp[ftp['parsed_genome_accession']==b[3:]])\n",
    "        #ftp.drop(ftp[ftp['parsed_genome_accession']==b[3:]].index, inplace=True)\n",
    "print(len(valdf))\n",
    "print(len(new_bac))\n",
    "print(start-len(ftp))\n",
    "print(len(ftp))\n",
    "\n",
    "arc_valdf = pd.DataFrame()\n",
    "start = len(arcftp)\n",
    "print(start)\n",
    "for b in new_arc:\n",
    "    if b[3:] in list(arcftp['ncbi_genbank_assembly_accession']):\n",
    "        arc_valdf = arc_valdf.append(arcftp[arcftp['ncbi_genbank_assembly_accession']==b[3:]])\n",
    "        #arcftp.drop(arcftp[arcftp['ncbi_genbank_assembly_accession']==b[3:]].index,inplace=True)\n",
    "    elif b[3:] in list(arcftp['parsed_genome_accession']):\n",
    "        arc_valdf = arc_valdf.append(arcftp[arcftp['parsed_genome_accession']==b[3:]])\n",
    "        #arcftp.drop(arcftp[arcftp['parsed_genome_accession']==b[3:]].index, inplace=True)\n",
    "    elif b in list(arcftp['parsed_genome_accession']): #for when only have SRR for e.g. UBA9915\n",
    "        arc_valdf = arc_valdf.append(arcftp[arcftp['parsed_genome_accession']==b])\n",
    "        #arcftp.drop(arcftp[arcftp['parsed_genome_accession']==b].index, inplace=True)\n",
    "    else:\n",
    "        print(b)\n",
    "print(len(arc_valdf))\n",
    "print(len(new_arc))\n",
    "print(start-len(arcftp))\n",
    "print(len(arcftp))\n",
    "\n",
    "valid = valdf.append(arc_valdf,sort=True)\n",
    "print(len(valdf)+len(arc_valdf),len(valid))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "461 461\n"
     ]
    }
   ],
   "source": [
    "validparsed_filenames = get_filenames(valid)\n",
    "print(len(valid),len(validparsed_filenames))\n",
    "save_filenames('validnewparsed_filenames.txt',validparsed_filenames)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "107\n"
     ]
    }
   ],
   "source": [
    "#leaving this here just to show how to extract subsets of the tree and visualize them, didn't end up using above\n",
    "taxa_to_retain = set([taxon for taxon in arc.taxon_namespace\n",
    "        if taxon.label in arc_valid])\n",
    "print(len(taxa_to_retain))\n",
    "thermo_tree = arc.extract_tree_with_taxa(taxa=taxa_to_retain)\n",
    "#thermo_tree.as_ascii_plot() #this is hard to see b/c it's so big"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# parse train set to keep one genome per genus too, reduce data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23458\n",
      "1248\n"
     ]
    }
   ],
   "source": [
    "allbac = [x.label for x in list(bac.taxon_namespace)]\n",
    "print(len(allbac))\n",
    "allbac[0:15]\n",
    "allarc = [x.label for x in list(arc.taxon_namespace)]\n",
    "print(len(allarc))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "107 131\n",
      "1523 1990\n"
     ]
    }
   ],
   "source": [
    "print(len(arc_valid),len(arc_test))\n",
    "print(len(bac_valid),len(bac_test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19945\n",
      "1010\n",
      "20955\n",
      "19945\n",
      "1010\n",
      "20955\n"
     ]
    }
   ],
   "source": [
    "arc_valid\n",
    "arc_test\n",
    "bac_valid\n",
    "bac_test\n",
    "print(len(allbac) - (len(bac_valid)+len(bac_test)))\n",
    "print(len(allarc)-(len(arc_valid)+len(arc_test)))\n",
    "print(len(allbac) - (len(bac_valid)+len(bac_test)) + len(allarc)-(len(arc_valid)+len(arc_test)))\n",
    "\n",
    "bac_train = [x for x in allbac if x not in bac_valid and x not in bac_test]\n",
    "print(len(bac_train)) #only bac\n",
    "arc_train = [x for x in allarc if x not in arc_valid and x not in arc_test]\n",
    "print(len(arc_train))\n",
    "print(len(bac_train)+len(arc_train))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['RS_GCF_000296615.1',\n",
       " 'RS_GCF_000337335.1',\n",
       " 'RS_GCF_002355655.1',\n",
       " 'RS_GCF_001280455.1',\n",
       " 'RS_GCF_000337035.1']"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "arc_train[0:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'RS_GCF_000296615.1'"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "arc.seed_node.leaf_nodes()[0].taxon.label\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "import random \n",
    "\n",
    "def parse_genera_from_node(node,taxa,genomes_per_genera=1):\n",
    "    for child in node.child_nodes():\n",
    "        #if it's a genus, randomly select one genome from this genus to keep, delete the rest\n",
    "        #(go delete route because there are occasionally unclassified taxa that don't have g__ labels, this will leave them in the list)\n",
    "        if child.label:\n",
    "            if 'g__' in child.label:\n",
    "                leaf_taxa = [leaf.taxon.label for leaf in child.leaf_nodes() if leaf.taxon.label in taxa]\n",
    "                if len(leaf_taxa)>0:\n",
    "                    print(child.label,'# children taxa:',len(leaf_taxa))\n",
    "                    to_keep = random.sample(leaf_taxa,k=genomes_per_genera)\n",
    "                    to_delete = leaf_taxa\n",
    "                    [to_delete.remove(keeper) for keeper in to_keep]\n",
    "                    [taxa.remove(delete) for delete in to_delete]\n",
    "                    print('length new taxa list:',len(taxa))\n",
    "        parse_genera_from_node(child,taxa,genomes_per_genera)\n",
    "        \n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "100.0:g__Halorubrum # children taxa: 28\n",
      "length new taxa list: 983\n",
      "100.0:g__Halopenitus # children taxa: 3\n",
      "length new taxa list: 981\n",
      "96.0:g__Halohasta # children taxa: 2\n",
      "length new taxa list: 980\n",
      "100.0:g__Halolamina # children taxa: 6\n",
      "length new taxa list: 975\n",
      "100.0:g__Halobaculum # children taxa: 3\n",
      "length new taxa list: 973\n",
      "100.0:g__Haloquadratum # children taxa: 4\n",
      "length new taxa list: 970\n",
      "100.0:g__Halobellus # children taxa: 3\n",
      "length new taxa list: 968\n",
      "100.0:g__Halogeometricum # children taxa: 4\n",
      "length new taxa list: 965\n",
      "100.0:g__Halopelagius # children taxa: 2\n",
      "length new taxa list: 964\n",
      "100.0:g__Haloferax # children taxa: 13\n",
      "length new taxa list: 952\n",
      "82.0:g__Halogranum # children taxa: 4\n",
      "length new taxa list: 949\n",
      "100.0:g__Haloplanus # children taxa: 2\n",
      "length new taxa list: 948\n",
      "100.0:g__Halovenus # children taxa: 9\n",
      "length new taxa list: 940\n",
      "100.0:g__Halosimplex # children taxa: 2\n",
      "length new taxa list: 939\n",
      "100.0:g__Halorhabdus # children taxa: 2\n",
      "length new taxa list: 938\n",
      "100.0:g__Haloarcula # children taxa: 9\n",
      "length new taxa list: 930\n",
      "100.0:g__Halomicrobium # children taxa: 4\n",
      "length new taxa list: 927\n",
      "100.0:g__Halorientalis # children taxa: 3\n",
      "length new taxa list: 925\n",
      "100.0:g__QS-1-69-70 # children taxa: 8\n",
      "length new taxa list: 918\n",
      "100.0:g__Natronomonas # children taxa: 2\n",
      "length new taxa list: 917\n",
      "100.0:g__CBA1134 # children taxa: 4\n",
      "length new taxa list: 914\n",
      "100.0:g__Natrinema # children taxa: 14\n",
      "length new taxa list: 901\n",
      "89.0:g__Natronorubrum # children taxa: 7\n",
      "length new taxa list: 895\n",
      "100.0:g__Haloterrigena # children taxa: 2\n",
      "length new taxa list: 894\n",
      "64.0:g__Halobiforma # children taxa: 3\n",
      "length new taxa list: 892\n",
      "100.0:g__B1-Br10-E2g2 # children taxa: 3\n",
      "length new taxa list: 890\n",
      "100.0:g__Natrialba # children taxa: 7\n",
      "length new taxa list: 884\n",
      "100.0:g__Natronococcus # children taxa: 3\n",
      "length new taxa list: 882\n",
      "100.0:g__Halostagnicola # children taxa: 3\n",
      "length new taxa list: 880\n",
      "100.0:g__Tc-Br11 # children taxa: 2\n",
      "length new taxa list: 879\n",
      "100.0:g__Halovivax # children taxa: 2\n",
      "length new taxa list: 878\n",
      "100.0:f__SW-7-71-33; g__SW-7-71-33 # children taxa: 2\n",
      "length new taxa list: 877\n",
      "100.0:g__Halococcus # children taxa: 8\n",
      "length new taxa list: 870\n",
      "100.0:f__Halalkalicoccaceae; g__Halalkalicoccus # children taxa: 2\n",
      "length new taxa list: 869\n",
      "100.0:g__Haladaptatus # children taxa: 4\n",
      "length new taxa list: 866\n",
      "100.0:g__Halobacterium # children taxa: 5\n",
      "length new taxa list: 862\n",
      "100.0:g__Methanoculleus # children taxa: 19\n",
      "length new taxa list: 844\n",
      "100.0:g__Methanolacinia # children taxa: 2\n",
      "length new taxa list: 843\n",
      "100.0:f__Methanofollaceae; g__Methanofollis # children taxa: 3\n",
      "length new taxa list: 841\n",
      "100.0:g__Methanoregula # children taxa: 17\n",
      "length new taxa list: 825\n",
      "100.0:g__Methanolinea # children taxa: 5\n",
      "length new taxa list: 821\n",
      "100.0:g__UBA9949 # children taxa: 2\n",
      "length new taxa list: 820\n",
      "100.0:g__SD8 # children taxa: 2\n",
      "length new taxa list: 819\n",
      "100.0:g__Methanocorpusculum # children taxa: 5\n",
      "length new taxa list: 815\n",
      "100.0:g__Methanocalculus # children taxa: 2\n",
      "length new taxa list: 814\n",
      "100.0:f__Methanospirillaceae; g__Methanospirillum # children taxa: 4\n",
      "length new taxa list: 811\n",
      "100.0:c__Bog-38; o__Bog-38; f__Bog-38; g__Bog-38 # children taxa: 4\n",
      "length new taxa list: 808\n",
      "100.0:g__Methanothrix_A # children taxa: 6\n",
      "length new taxa list: 803\n",
      "100.0:g__Methanothrix # children taxa: 3\n",
      "length new taxa list: 801\n",
      "100.0:g__Methanocella # children taxa: 2\n",
      "length new taxa list: 800\n",
      "100.0:g__Archaeoglobus_A # children taxa: 5\n",
      "length new taxa list: 796\n",
      "100.0:g__Archaeoglobus # children taxa: 2\n",
      "length new taxa list: 795\n",
      "100.0:g__Geoglobus # children taxa: 3\n",
      "length new taxa list: 793\n",
      "100.0:g__MGIIb-O5 # children taxa: 16\n",
      "length new taxa list: 778\n",
      "100.0:g__Thalassarchaeum # children taxa: 10\n",
      "length new taxa list: 769\n",
      "82.0:g__MGIIb-O3 # children taxa: 25\n",
      "length new taxa list: 745\n",
      "100.0:g__MGIIb-O2 # children taxa: 9\n",
      "length new taxa list: 737\n",
      "100.0:g__MGIIb-O1 # children taxa: 13\n",
      "length new taxa list: 725\n",
      "100.0:g__MGIIb-N1 # children taxa: 9\n",
      "length new taxa list: 717\n",
      "100.0:g__MGIIb-N2 # children taxa: 12\n",
      "length new taxa list: 706\n",
      "100.0:g__MGIIb-P # children taxa: 6\n",
      "length new taxa list: 701\n",
      "100.0:g__MGIIb-Q1 # children taxa: 3\n",
      "length new taxa list: 699\n",
      "99.0:g__MGIIa-L1 # children taxa: 27\n",
      "length new taxa list: 673\n",
      "100.0:g__Poseidonia # children taxa: 13\n",
      "length new taxa list: 661\n",
      "100.0:g__MGIIa-L3 # children taxa: 7\n",
      "length new taxa list: 655\n",
      "100.0:g__MGIIa-L4 # children taxa: 2\n",
      "length new taxa list: 654\n",
      "100.0:g__MGIIa-L2 # children taxa: 9\n",
      "length new taxa list: 646\n",
      "100.0:g__MGIIa-K1 # children taxa: 18\n",
      "length new taxa list: 629\n",
      "100.0:g__MGIIa-K2 # children taxa: 2\n",
      "length new taxa list: 628\n",
      "100.0:g__UBA226 # children taxa: 2\n",
      "length new taxa list: 627\n",
      "100.0:g__MGIIa-I # children taxa: 6\n",
      "length new taxa list: 622\n",
      "100.0:g__CG-Epi1 # children taxa: 6\n",
      "length new taxa list: 617\n",
      "100.0:g__UBA8886 # children taxa: 5\n",
      "length new taxa list: 613\n",
      "100.0:f__UBA102; g__UBA102 # children taxa: 2\n",
      "length new taxa list: 612\n",
      "100.0:g__SM1-50 # children taxa: 4\n",
      "length new taxa list: 609\n",
      "100.0:f__UBA9212; g__UBA9212 # children taxa: 2\n",
      "length new taxa list: 608\n",
      "100.0:g__Methanosphaera # children taxa: 13\n",
      "length new taxa list: 596\n",
      "94.0:g__Methanobacterium_C # children taxa: 3\n",
      "length new taxa list: 594\n",
      "100.0:g__Methanobacterium_B # children taxa: 7\n",
      "length new taxa list: 588\n",
      "100.0:g__UBA419 # children taxa: 3\n",
      "length new taxa list: 586\n",
      "100.0:g__Bog-22 # children taxa: 2\n",
      "length new taxa list: 585\n",
      "100.0:g__Methanobacterium # children taxa: 6\n",
      "length new taxa list: 580\n",
      "100.0:g__Methanobacterium_D # children taxa: 5\n",
      "length new taxa list: 576\n",
      "100.0:g__UBA349 # children taxa: 2\n",
      "length new taxa list: 575\n",
      "100.0:g__Methanobrevibacter_A # children taxa: 19\n",
      "length new taxa list: 557\n",
      "100.0:g__Methanobrevibacter_B # children taxa: 4\n",
      "length new taxa list: 554\n",
      "100.0:g__Methanobrevibacter # children taxa: 4\n",
      "length new taxa list: 551\n",
      "100.0:g__Methanobrevibacter_C # children taxa: 3\n",
      "length new taxa list: 549\n",
      "100.0:g__Methanobrevibacter_D # children taxa: 2\n",
      "length new taxa list: 548\n",
      "100.0:f__Methanothermobacteriaceae; g__Methanothermobacter # children taxa: 4\n",
      "length new taxa list: 545\n",
      "100.0:g__Methanococcus # children taxa: 6\n",
      "length new taxa list: 540\n",
      "100.0:g__Methanothermococcus_A # children taxa: 2\n",
      "length new taxa list: 539\n",
      "100.0:g__Methanotorris # children taxa: 2\n",
      "length new taxa list: 538\n",
      "100.0:g__Methanocaldococcus # children taxa: 5\n",
      "length new taxa list: 534\n",
      "100.0:g__Methanocaldococcus_A # children taxa: 2\n",
      "length new taxa list: 533\n",
      "100.0:c__Methanopyri; o__Methanopyrales; f__Methanopyraceae; g__Methanopyrus # children taxa: 2\n",
      "length new taxa list: 532\n",
      "100.0:g__Thermococcus # children taxa: 23\n",
      "length new taxa list: 510\n",
      "100.0:g__Pyrococcus # children taxa: 8\n",
      "length new taxa list: 503\n",
      "100.0:g__Thermococcus_B # children taxa: 2\n",
      "length new taxa list: 502\n",
      "100.0:g__Thermococcus_A # children taxa: 5\n",
      "length new taxa list: 498\n",
      "100.0:o__Methanofastidiosales; f__Methanofastidiosaceae; g__Methanofastidiosum # children taxa: 5\n",
      "length new taxa list: 494\n",
      "100.0:f__BMS3B; g__BMS3B # children taxa: 2\n",
      "length new taxa list: 493\n",
      "93.0:g__Acidilobus # children taxa: 2\n",
      "length new taxa list: 492\n",
      "100.0:g__Caldisphaera # children taxa: 2\n",
      "length new taxa list: 491\n",
      "100.0:g__Aeropyrum # children taxa: 2\n",
      "length new taxa list: 490\n",
      "100.0:g__Pyrodictium # children taxa: 2\n",
      "length new taxa list: 489\n",
      "100.0:g__Desulfurococcus # children taxa: 2\n",
      "length new taxa list: 488\n",
      "100.0:g__Staphylothermus # children taxa: 2\n",
      "length new taxa list: 487\n",
      "89.0:g__Acidianus # children taxa: 6\n",
      "length new taxa list: 482\n",
      "100.0:g__Metallosphaera # children taxa: 4\n",
      "length new taxa list: 479\n",
      "100.0:g__Sulfolobus_B # children taxa: 3\n",
      "length new taxa list: 477\n",
      "100.0:g__Sulfolobus_A # children taxa: 3\n",
      "length new taxa list: 475\n",
      "100.0:g__Sulfolobus # children taxa: 2\n",
      "length new taxa list: 474\n",
      "100.0:g__Vulcanisaeta # children taxa: 12\n",
      "length new taxa list: 463\n",
      "100.0:g__Thermocladium # children taxa: 3\n",
      "length new taxa list: 461\n",
      "100.0:g__Caldivirga # children taxa: 3\n",
      "length new taxa list: 459\n",
      "100.0:g__Pyrobaculum # children taxa: 8\n",
      "length new taxa list: 452\n",
      "100.0:g__Thermoproteus # children taxa: 5\n",
      "length new taxa list: 448\n",
      "100.0:g__Thermofilum_A # children taxa: 2\n",
      "length new taxa list: 447\n",
      "100.0:g__Thermofilum # children taxa: 2\n",
      "length new taxa list: 446\n",
      "100.0:g__Methanomethylicus # children taxa: 2\n",
      "length new taxa list: 445\n",
      "100.0:g__SMTZ1-45 # children taxa: 2\n",
      "length new taxa list: 444\n",
      "100.0:g__MP8T-1 # children taxa: 2\n",
      "length new taxa list: 443\n",
      "100.0:o__UBA460; f__UBA460; g__UBA460 # children taxa: 2\n",
      "length new taxa list: 442\n",
      "100.0:g__UBA525 # children taxa: 2\n",
      "length new taxa list: 441\n",
      "98.0:g__UBA11998 # children taxa: 3\n",
      "length new taxa list: 439\n",
      "100.0:f__ARS1403; g__ARS1403 # children taxa: 2\n",
      "length new taxa list: 438\n",
      "100.0:g__NP1466 # children taxa: 2\n",
      "length new taxa list: 437\n",
      "100.0:o__Parvarchaeales; f__Parvarchaeaceae; g__Parvarchaeum # children taxa: 2\n",
      "length new taxa list: 436\n",
      "100.0:f__Nanopusillaceae; g__Nanopusillus # children taxa: 3\n",
      "length new taxa list: 434\n",
      "100.0:g__B1-Br10-U2g19 # children taxa: 2\n",
      "length new taxa list: 433\n",
      "100.0:g__B1-Br10-U2g21 # children taxa: 3\n",
      "length new taxa list: 431\n",
      "100.0:f__GW2011-AR5; g__GW2011-AR5 # children taxa: 2\n",
      "length new taxa list: 430\n",
      "100.0:f__UBA55; g__UBA55 # children taxa: 2\n",
      "length new taxa list: 429\n",
      "91.0:g__UBA493 # children taxa: 3\n",
      "length new taxa list: 427\n",
      "100.0:f__WOR-SM1-SCG; g__WOR-SM1-SCG # children taxa: 2\n",
      "length new taxa list: 426\n",
      "99.0:o__Altiarchaeales; f__Altiarchaeaceae; g__Altiarchaeum # children taxa: 2\n",
      "length new taxa list: 425\n",
      "425\n"
     ]
    }
   ],
   "source": [
    "parsed_arc_train = arc_train.copy()\n",
    "parse_genera_from_node(arc.seed_node,parsed_arc_train,genomes_per_genera=1)\n",
    "print(len(parsed_arc_train))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "425\n"
     ]
    }
   ],
   "source": [
    "print(len(parsed_arc_train))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## how many if parse at family level?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "24706"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ftp)+len(arcftp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "#these are the train, valid, test sets that were split on taxonomic group; \n",
    "#valid contains all genomes from bacterial groups ['o__Actinomycetales','o__Rhodobacterales'] and archaeal groups ['c__Thermoplasmata','c__Bathyarchaeia'] \n",
    "#test contains all genomes from bacterial groups ['o__Bacteroidales','o__Rhizobiales'] and archaeal groups ['o__Methanosarcinales','o__Nitrososphaerales']\n",
    "#train contains everything else exclusive of these taxa\n",
    "\n",
    "train = pd.read_csv(\"TrainNew_GTDBrepwithftp.csv\")\n",
    "valid = pd.read_csv(\"ValidNew_GTDBrepwithftp.csv\")\n",
    "test = pd.read_csv(\"TestNew_GTDBrepwithftp.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20955 1630 2121\n",
      "24706\n"
     ]
    }
   ],
   "source": [
    "print(len(train),len(valid),len(test))\n",
    "print(len(train)+len(valid)+len(test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23458\n",
      "1248\n",
      "appending 944 labels from node <Node object at 0x7f979bce1cf8: '91.0:o__Actinomycetales' (None)>\n",
      "appending 579 labels from node <Node object at 0x7f979da97048: '93.0:o__Rhodobacterales' (None)>\n",
      "appending 1029 labels from node <Node object at 0x7f979c5edeb8: '91.0:o__Bacteroidales' (None)>\n",
      "appending 961 labels from node <Node object at 0x7f979d832fd0: '20.0:o__Rhizobiales' (None)>\n",
      "1523 1990\n",
      "appending 67 labels from node <Node object at 0x7f979b6e0780: '86.0:c__Thermoplasmata' (None)>\n",
      "appending 40 labels from node <Node object at 0x7f979cd60630: '93.0:c__Bathyarchaeia' (None)>\n",
      "appending 63 labels from node <Node object at 0x7f979cf68198: '83.0:o__Methanosarcinales' (None)>\n",
      "appending 68 labels from node <Node object at 0x7f979b1f8b00: '100.0:o__Nitrososphaerales' (None)>\n",
      "107 131\n",
      "19945\n",
      "1010\n",
      "20955\n"
     ]
    }
   ],
   "source": [
    "#if you shut down your notebook and need to redefine these values:\n",
    "allbac = [x.label for x in list(bac.taxon_namespace)]\n",
    "print(len(allbac))\n",
    "allbac[0:15]\n",
    "allarc = [x.label for x in list(arc.taxon_namespace)]\n",
    "print(len(allarc))\n",
    "\n",
    "bac_valid = get_subset_leaves(bac.seed_node, to_match=['o__Actinomycetales','o__Rhodobacterales'],taxa=[])\n",
    "bac_test = get_subset_leaves(bac.seed_node, to_match=['o__Bacteroidales','o__Rhizobiales'],taxa=[])\n",
    "print(len(bac_valid),len(bac_test))\n",
    "arc_valid = get_subset_leaves(arc.seed_node, to_match=['c__Thermoplasmata','c__Bathyarchaeia'],taxa=[])\n",
    "arc_test = get_subset_leaves(arc.seed_node, to_match=['o__Methanosarcinales','o__Nitrososphaerales'],taxa=[])\n",
    "print(len(arc_valid),len(arc_test))\n",
    "bac_train = [x for x in allbac if x not in bac_valid and x not in bac_test]\n",
    "print(len(bac_train)) #only bac\n",
    "arc_train = [x for x in allarc if x not in arc_valid and x not in arc_test]\n",
    "print(len(arc_train))\n",
    "print(len(bac_train)+len(arc_train)) #should be 20955"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "import random \n",
    "\n",
    "def parse_taxa_from_node(node,taxa,genomes_per_taxa=1,prefix='g__'):\n",
    "    for child in node.child_nodes():\n",
    "        #if it's a genus, randomly select one genome from this genus to keep, delete the rest\n",
    "        #(go delete route because there are occasionally unclassified taxa that don't have g__ labels, this will leave them in the list)\n",
    "        if child.label:\n",
    "            if prefix in child.label:\n",
    "                leaf_taxa = [leaf.taxon.label for leaf in child.leaf_nodes() if leaf.taxon.label in taxa]\n",
    "                if len(leaf_taxa)>0:\n",
    "                    print(child.label,'# children taxa:',len(leaf_taxa))\n",
    "                    to_keep = random.sample(leaf_taxa,k=genomes_per_taxa)\n",
    "                    to_delete = leaf_taxa\n",
    "                    [to_delete.remove(keeper) for keeper in to_keep]\n",
    "                    [taxa.remove(delete) for delete in to_delete]\n",
    "                    print('length new taxa list:',len(taxa))\n",
    "        parse_taxa_from_node(child,taxa,genomes_per_taxa,prefix)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "100.0:f__Haloferacaceae # children taxa: 81\n",
      "length new taxa list: 930\n",
      "100.0:f__Haloarculaceae # children taxa: 56\n",
      "length new taxa list: 875\n",
      "79.0:f__Natrialbaceae # children taxa: 56\n",
      "length new taxa list: 820\n",
      "100.0:f__SW-7-71-33; g__SW-7-71-33 # children taxa: 2\n",
      "length new taxa list: 819\n",
      "100.0:f__Halococcaceae # children taxa: 13\n",
      "length new taxa list: 807\n",
      "100.0:f__Halalkalicoccaceae; g__Halalkalicoccus # children taxa: 2\n",
      "length new taxa list: 806\n",
      "100.0:f__Haloadaptaceae # children taxa: 6\n",
      "length new taxa list: 801\n",
      "100.0:f__Halobacteriaceae # children taxa: 8\n",
      "length new taxa list: 794\n",
      "98.0:f__Methanocullaceae # children taxa: 21\n",
      "length new taxa list: 774\n",
      "100.0:f__Methanomicrobiaceae # children taxa: 6\n",
      "length new taxa list: 769\n",
      "100.0:f__Methanofollaceae; g__Methanofollis # children taxa: 3\n",
      "length new taxa list: 767\n",
      "96.0:f__Methanoregulaceae # children taxa: 27\n",
      "length new taxa list: 741\n",
      "100.0:f__Methanosphaerulaceae # children taxa: 3\n",
      "length new taxa list: 739\n",
      "100.0:f__Methanocorpusculaceae # children taxa: 7\n",
      "length new taxa list: 733\n",
      "100.0:f__Methanospirillaceae; g__Methanospirillum # children taxa: 4\n",
      "length new taxa list: 730\n",
      "100.0:c__Bog-38; o__Bog-38; f__Bog-38; g__Bog-38 # children taxa: 4\n",
      "length new taxa list: 727\n",
      "100.0:o__Methanotrichales; f__Methanotrichaceae # children taxa: 13\n",
      "length new taxa list: 715\n",
      "100.0:f__Methanocellaceae # children taxa: 3\n",
      "length new taxa list: 713\n",
      "100.0:o__ANME-1; f__ANME-1 # children taxa: 3\n",
      "length new taxa list: 711\n",
      "100.0:o__Syntrophoarchaeales; f__Syntrophoarchaeaceae # children taxa: 2\n",
      "length new taxa list: 710\n",
      "100.0:f__Archaeoglobaceae # children taxa: 16\n",
      "length new taxa list: 695\n",
      "100.0:f__Thalassoarchaeaceae # children taxa: 104\n",
      "length new taxa list: 592\n",
      "100.0:f__Poseidoniaceae # children taxa: 87\n",
      "length new taxa list: 506\n",
      "100.0:f__CG-Epi1 # children taxa: 11\n",
      "length new taxa list: 496\n",
      "100.0:f__UBA102; g__UBA102 # children taxa: 2\n",
      "length new taxa list: 495\n",
      "100.0:o__DHVEG-1; f__DHVEG-1 # children taxa: 7\n",
      "length new taxa list: 489\n",
      "100.0:f__UBA9212; g__UBA9212 # children taxa: 2\n",
      "length new taxa list: 488\n",
      "99.0:f__Methanobacteriaceae # children taxa: 77\n",
      "length new taxa list: 412\n",
      "100.0:f__Methanothermobacteriaceae; g__Methanothermobacter # children taxa: 4\n",
      "length new taxa list: 409\n",
      "100.0:f__Methanococcaceae # children taxa: 11\n",
      "length new taxa list: 399\n",
      "100.0:f__Methanocaldococcaceae # children taxa: 7\n",
      "length new taxa list: 393\n",
      "100.0:c__Methanopyri; o__Methanopyrales; f__Methanopyraceae; g__Methanopyrus # children taxa: 2\n",
      "length new taxa list: 392\n",
      "100.0:o__Thermococcales; f__Thermococcaceae # children taxa: 41\n",
      "length new taxa list: 352\n",
      "100.0:o__Methanofastidiosales; f__Methanofastidiosaceae; g__Methanofastidiosum # children taxa: 5\n",
      "length new taxa list: 348\n",
      "100.0:f__BMS3B; g__BMS3B # children taxa: 2\n",
      "length new taxa list: 347\n",
      "100.0:p__Hadarchaeota; c__Hadarchaeia; o__Hadarchaeales; f__Hadarchaeaceae # children taxa: 2\n",
      "length new taxa list: 346\n",
      "100.0:o__UBA164; f__UBA164 # children taxa: 2\n",
      "length new taxa list: 345\n",
      "100.0:f__NZ13-MGT # children taxa: 3\n",
      "length new taxa list: 343\n",
      "100.0:f__Caldiarchaeaceae # children taxa: 2\n",
      "length new taxa list: 342\n",
      "100.0:f__HR02 # children taxa: 3\n",
      "length new taxa list: 340\n",
      "100.0:f__Acidilobaceae # children taxa: 8\n",
      "length new taxa list: 333\n",
      "100.0:f__Pyrodictiaceae # children taxa: 4\n",
      "length new taxa list: 330\n",
      "100.0:f__Desulfurococcaceae # children taxa: 8\n",
      "length new taxa list: 323\n",
      "100.0:f__NBVN01 # children taxa: 2\n",
      "length new taxa list: 322\n",
      "100.0:o__Sulfolobales; f__Sulfolobaceae # children taxa: 21\n",
      "length new taxa list: 302\n",
      "100.0:o__Ignicoccales; f__Ignicoccaceae # children taxa: 2\n",
      "length new taxa list: 301\n",
      "100.0:f__Thermocladiaceae # children taxa: 20\n",
      "length new taxa list: 282\n",
      "100.0:f__Thermoproteaceae # children taxa: 13\n",
      "length new taxa list: 270\n",
      "100.0:f__Thermofilaceae # children taxa: 4\n",
      "length new taxa list: 267\n",
      "100.0:o__Geoarchaeales; f__Geoarchaeaceae # children taxa: 2\n",
      "length new taxa list: 266\n",
      "100.0:f__Marsarchaeaceae # children taxa: 2\n",
      "length new taxa list: 265\n",
      "100.0:c__Methanomethylicia; o__Methanomethyliales; f__Methanomethyliaceae # children taxa: 3\n",
      "length new taxa list: 263\n",
      "100.0:o__Thorarchaeales; f__Thorarchaeaceae # children taxa: 6\n",
      "length new taxa list: 258\n",
      "100.0:o__LC-2; f__LC-2 # children taxa: 2\n",
      "length new taxa list: 257\n",
      "100.0:o__UBA460; f__UBA460; g__UBA460 # children taxa: 2\n",
      "length new taxa list: 256\n",
      "100.0:f__UBA525 # children taxa: 3\n",
      "length new taxa list: 254\n",
      "93.0:f__UBA12501 # children taxa: 2\n",
      "length new taxa list: 253\n",
      "82.0:f__GW2011-AR4 # children taxa: 5\n",
      "length new taxa list: 249\n",
      "100.0:f__UBA10216 # children taxa: 2\n",
      "length new taxa list: 248\n",
      "100.0:f__ARS49 # children taxa: 2\n",
      "length new taxa list: 247\n",
      "97.0:f__GW2011-AR9 # children taxa: 9\n",
      "length new taxa list: 239\n",
      "100.0:f__GW2011-AR17 # children taxa: 2\n",
      "length new taxa list: 238\n",
      "100.0:f__GW2011-AR20 # children taxa: 2\n",
      "length new taxa list: 237\n",
      "100.0:f__ARS1403; g__ARS1403 # children taxa: 2\n",
      "length new taxa list: 236\n",
      "100.0:f__GW2011-AR1 # children taxa: 10\n",
      "length new taxa list: 227\n",
      "100.0:f__UBA73 # children taxa: 8\n",
      "length new taxa list: 220\n",
      "100.0:f__ARS1160 # children taxa: 2\n",
      "length new taxa list: 219\n",
      "100.0:o__UBA583; f__UBA583 # children taxa: 2\n",
      "length new taxa list: 218\n",
      "100.0:o__Parvarchaeales; f__Parvarchaeaceae; g__Parvarchaeum # children taxa: 2\n",
      "length new taxa list: 217\n",
      "100.0:f__Nanopusillaceae; g__Nanopusillus # children taxa: 3\n",
      "length new taxa list: 215\n",
      "100.0:p__Nanohaloarchaeota; c__Nanosalinia; o__Nanosalinales; f__Nanosalinaceae # children taxa: 11\n",
      "length new taxa list: 205\n",
      "100.0:f__CG10238-14 # children taxa: 2\n",
      "length new taxa list: 204\n",
      "100.0:f__GW2011-AR5; g__GW2011-AR5 # children taxa: 2\n",
      "length new taxa list: 203\n",
      "100.0:f__UBA55; g__UBA55 # children taxa: 2\n",
      "length new taxa list: 202\n",
      "88.0:f__UBA10161 # children taxa: 2\n",
      "length new taxa list: 201\n",
      "100.0:o__Micrarchaeales; f__Micrarchaeaceae # children taxa: 4\n",
      "length new taxa list: 198\n",
      "100.0:f__UBA93 # children taxa: 2\n",
      "length new taxa list: 197\n",
      "100.0:f__Iainarchaeaceae # children taxa: 5\n",
      "length new taxa list: 193\n",
      "100.0:f__WOR-SM1-SCG; g__WOR-SM1-SCG # children taxa: 2\n",
      "length new taxa list: 192\n",
      "99.0:o__Altiarchaeales; f__Altiarchaeaceae; g__Altiarchaeum # children taxa: 2\n",
      "length new taxa list: 191\n",
      "191\n"
     ]
    }
   ],
   "source": [
    "parsed_arc_train_family = arc_train.copy()\n",
    "parse_taxa_from_node(arc.seed_node,parsed_arc_train_family,genomes_per_taxa=1,prefix='f__')\n",
    "print(len(parsed_arc_train_family))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1010 191\n"
     ]
    }
   ],
   "source": [
    "print(len(arc_train),len(parsed_arc_train_family))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "95.0:f__Acutalibacteraceae # children taxa: 183\n",
      "length new taxa list: 19763\n",
      "85.0:f__Ruminococcaceae # children taxa: 158\n",
      "length new taxa list: 19606\n",
      "100.0:f__Ethanoligenenaceae # children taxa: 4\n",
      "length new taxa list: 19603\n",
      "74.0:f__Oscillospiraceae # children taxa: 118\n",
      "length new taxa list: 19486\n",
      "100.0:f__Butyricicoccaceae # children taxa: 11\n",
      "length new taxa list: 19476\n",
      "100.0:f__CAG-272 # children taxa: 24\n",
      "length new taxa list: 19453\n",
      "100.0:f__CAG-382 # children taxa: 10\n",
      "length new taxa list: 19444\n",
      "100.0:f__CAG-74 # children taxa: 31\n",
      "length new taxa list: 19414\n",
      "100.0:f__UBA1750 # children taxa: 3\n",
      "length new taxa list: 19412\n",
      "100.0:f__CAG-138 # children taxa: 19\n",
      "length new taxa list: 19394\n",
      "68.0:f__Christensenellaceae # children taxa: 8\n",
      "length new taxa list: 19387\n",
      "95.0:f__UBA1242 # children taxa: 11\n",
      "length new taxa list: 19377\n",
      "100.0:f__UBA4651 # children taxa: 3\n",
      "length new taxa list: 19375\n",
      "100.0:f__CAG-552 # children taxa: 2\n",
      "length new taxa list: 19374\n",
      "100.0:f__CAG-727 # children taxa: 11\n",
      "length new taxa list: 19364\n",
      "100.0:f__CAG-314 # children taxa: 3\n",
      "length new taxa list: 19362\n",
      "100.0:f__CAG-917 # children taxa: 12\n",
      "length new taxa list: 19351\n",
      "100.0:f__CAG-1252; g__CAG-1252 # children taxa: 6\n",
      "length new taxa list: 19346\n",
      "100.0:f__DTU072 # children taxa: 6\n",
      "length new taxa list: 19341\n",
      "100.0:f__Saccharofermentanaceae # children taxa: 20\n",
      "length new taxa list: 19322\n",
      "100.0:f__UBA5734 # children taxa: 4\n",
      "length new taxa list: 19319\n",
      "100.0:f__DTU023 # children taxa: 7\n",
      "length new taxa list: 19313\n",
      "98.0:f__Fastidiosipilaceae # children taxa: 2\n",
      "length new taxa list: 19312\n",
      "100.0:f__UBA1212; g__UBA1212 # children taxa: 2\n",
      "length new taxa list: 19311\n",
      "100.0:f__CAG-508 # children taxa: 28\n",
      "length new taxa list: 19284\n",
      "100.0:f__UBA1234; g__UBA1234 # children taxa: 2\n",
      "length new taxa list: 19283\n",
      "100.0:f__CAG-465; g__CAG-465 # children taxa: 3\n",
      "length new taxa list: 19281\n",
      "100.0:f__Lachnospiraceae # children taxa: 615\n",
      "length new taxa list: 18667\n",
      "100.0:f__UBA1390 # children taxa: 4\n",
      "length new taxa list: 18664\n",
      "100.0:f__Cellulosilyticaceae # children taxa: 15\n",
      "length new taxa list: 18650\n",
      "100.0:f__Vallitaleaceae # children taxa: 9\n",
      "length new taxa list: 18642\n",
      "100.0:f__Anaerotignaceae # children taxa: 11\n",
      "length new taxa list: 18632\n",
      "100.0:f__CAG-274 # children taxa: 2\n",
      "length new taxa list: 18631\n",
      "100.0:f__UBA12420; g__UBA12420 # children taxa: 3\n",
      "length new taxa list: 18629\n",
      "84.0:f__Acetivibrionaceae # children taxa: 21\n",
      "length new taxa list: 18609\n",
      "100.0:f__Ruminiclostridiaceae_A # children taxa: 7\n",
      "length new taxa list: 18603\n",
      "100.0:f__UBA1381 # children taxa: 6\n",
      "length new taxa list: 18598\n",
      "100.0:f__Firm-18 # children taxa: 2\n",
      "length new taxa list: 18597\n",
      "100.0:f__Anaerovoracaceae # children taxa: 55\n",
      "length new taxa list: 18543\n",
      "100.0:f__Fusibacteraceae # children taxa: 6\n",
      "length new taxa list: 18538\n",
      "99.0:f__DTU036 # children taxa: 2\n",
      "length new taxa list: 18537\n",
      "100.0:f__Caminicellaceae # children taxa: 3\n",
      "length new taxa list: 18535\n",
      "100.0:f__Thermotaleaceae # children taxa: 5\n",
      "length new taxa list: 18531\n",
      "98.0:f__Peptostreptococcaceae # children taxa: 39\n",
      "length new taxa list: 18493\n",
      "100.0:f__Peptoclostridiaceae # children taxa: 2\n",
      "length new taxa list: 18492\n",
      "100.0:f__Filifactoraceae # children taxa: 8\n",
      "length new taxa list: 18485\n",
      "100.0:f__Natronincolaceae # children taxa: 10\n",
      "length new taxa list: 18476\n",
      "100.0:f__Tindalliaceae; g__Tindallia # children taxa: 2\n",
      "length new taxa list: 18475\n",
      "100.0:f__Helcococcaceae # children taxa: 41\n",
      "length new taxa list: 18435\n",
      "99.0:f__Peptoniphilaceae # children taxa: 28\n",
      "length new taxa list: 18408\n",
      "100.0:f__Ezakiellaceae # children taxa: 5\n",
      "length new taxa list: 18404\n",
      "94.0:f__Tissierellaceae # children taxa: 7\n",
      "length new taxa list: 18398\n",
      "100.0:f__Tepidimicrobiaceae # children taxa: 4\n",
      "length new taxa list: 18395\n",
      "100.0:f__Sporanaerobacteraceae # children taxa: 3\n",
      "length new taxa list: 18393\n",
      "100.0:f__Proteiniboraceae # children taxa: 3\n",
      "length new taxa list: 18391\n",
      "100.0:f__Gottschalkiaceae # children taxa: 3\n",
      "length new taxa list: 18389\n",
      "100.0:f__Caloranaerobacteraceae; g__Caloranaerobacter # children taxa: 3\n",
      "length new taxa list: 18387\n",
      "100.0:f__Sedimentibacteraceae # children taxa: 19\n",
      "length new taxa list: 18369\n",
      "100.0:f__Dethiosulfatibacteraceae # children taxa: 4\n",
      "length new taxa list: 18366\n",
      "100.0:f__Eubacteriaceae # children taxa: 14\n",
      "length new taxa list: 18353\n",
      "100.0:f__Alkalibacteraceae; g__Alkalibacter # children taxa: 2\n",
      "length new taxa list: 18352\n",
      "100.0:f__UBA1433; g__UBA1433 # children taxa: 2\n",
      "length new taxa list: 18351\n",
      "100.0:o__Lutisporales; f__Lutisporaceae # children taxa: 4\n",
      "length new taxa list: 18348\n",
      "84.0:f__Clostridiaceae # children taxa: 131\n",
      "length new taxa list: 18218\n",
      "100.0:f__Savagellaceae; g__Savagella # children taxa: 3\n",
      "length new taxa list: 18216\n",
      "100.0:f__Caloramatoraceae # children taxa: 9\n",
      "length new taxa list: 18208\n",
      "100.0:f__Oxobacteraceae # children taxa: 2\n",
      "length new taxa list: 18207\n",
      "100.0:f__DTU083 # children taxa: 5\n",
      "length new taxa list: 18203\n",
      "100.0:f__Caldicoprobacteraceae # children taxa: 4\n",
      "length new taxa list: 18200\n",
      "100.0:o__Thermoanaerobacterales; f__Thermoanaerobacteraceae # children taxa: 14\n",
      "length new taxa list: 18187\n",
      "100.0:f__UBA4877; g__UBA4877 # children taxa: 2\n",
      "length new taxa list: 18186\n",
      "100.0:f__Caldanaerobiaceae; g__Caldanaerobius # children taxa: 2\n",
      "length new taxa list: 18185\n",
      "100.0:o__Caldicellulosiruptorales; f__Caldicellulosiruptoraceae; g__Caldicellulosiruptor # children taxa: 10\n",
      "length new taxa list: 18176\n",
      "100.0:f__Tepidanaerobacteraceae # children taxa: 4\n",
      "length new taxa list: 18173\n",
      "100.0:f__Thermovenabulaceae # children taxa: 5\n",
      "length new taxa list: 18169\n",
      "100.0:f__Paenibacillaceae # children taxa: 219\n",
      "length new taxa list: 17951\n",
      "100.0:o__Caldalkalibacillales; f__Caldalkalibacillaceae # children taxa: 2\n",
      "length new taxa list: 17950\n",
      "100.0:o__Brevibacillales; f__Brevibacillaceae # children taxa: 20\n",
      "length new taxa list: 17931\n",
      "100.0:o__Aneurinibacillales; f__Aneurinibacillaceae # children taxa: 7\n",
      "length new taxa list: 17925\n",
      "100.0:o__Tepidibacillales; f__Tepidibacillaceae # children taxa: 3\n",
      "length new taxa list: 17923\n",
      "100.0:f__Thermoactinomycetaceae # children taxa: 21\n",
      "length new taxa list: 17903\n",
      "100.0:f__Novibacillaceae # children taxa: 2\n",
      "length new taxa list: 17902\n",
      "93.0:o__Thermicanales; f__Thermicanaceae # children taxa: 4\n",
      "length new taxa list: 17899\n",
      "100.0:o__Bacillales_F; f__Bacillaceae_M; g__Bacillus_BB # children taxa: 2\n",
      "length new taxa list: 17898\n",
      "100.0:p__Firmicutes_J; c__Desulfuribacillia; o__Desulfuribacillales; f__Desulfuribacillaceae # children taxa: 2\n",
      "length new taxa list: 17897\n",
      "100.0:f__Alicyclobacillaceae # children taxa: 19\n",
      "length new taxa list: 17879\n",
      "100.0:f__Acidibacillaceae # children taxa: 3\n",
      "length new taxa list: 17877\n",
      "100.0:o__Kyrpidiales; f__Kyrpidiaceae; g__Kyrpidia # children taxa: 2\n",
      "length new taxa list: 17876\n",
      "100.0:f__Tumebacillaceae # children taxa: 4\n",
      "length new taxa list: 17873\n",
      "100.0:f__Effusibacillaceae; g__Effusibacillus # children taxa: 2\n",
      "length new taxa list: 17872\n",
      "100.0:o__Dethiobacterales; f__Dethiobacteraceae # children taxa: 4\n",
      "length new taxa list: 17869\n",
      "100.0:f__DTU022 # children taxa: 2\n",
      "length new taxa list: 17868\n",
      "100.0:f__UBA8154 # children taxa: 2\n",
      "length new taxa list: 17867\n",
      "100.0:c__Natranaerobiia; o__Natranaerobiales; f__Natranaerobiaceae # children taxa: 2\n",
      "length new taxa list: 17866\n",
      "100.0:o__UBA4975; f__UBA4975; g__UBA4975 # children taxa: 4\n",
      "length new taxa list: 17863\n",
      "100.0:o__Proteinivoracales; f__Proteinivoraceae; g__Anaerobranca # children taxa: 2\n",
      "length new taxa list: 17862\n",
      "100.0:f__UBA994 # children taxa: 2\n",
      "length new taxa list: 17861\n",
      "100.0:f__DTU010 # children taxa: 7\n",
      "length new taxa list: 17855\n",
      "100.0:f__RSA5 # children taxa: 2\n",
      "length new taxa list: 17854\n",
      "100.0:c__DTU065; o__DTU065; f__DTU065 # children taxa: 2\n",
      "length new taxa list: 17853\n",
      "100.0:c__DTU084; o__DTU084; f__DTU084 # children taxa: 4\n",
      "length new taxa list: 17850\n",
      "100.0:o__UBA4971; f__UBA4971 # children taxa: 4\n",
      "length new taxa list: 17847\n",
      "100.0:o__DTUO25; f__DTU025 # children taxa: 4\n",
      "length new taxa list: 17844\n",
      "100.0:f__UBA10575 # children taxa: 3\n",
      "length new taxa list: 17842\n",
      "100.0:o__UBA8346; f__UBA8346; g__UBA8346 # children taxa: 2\n",
      "length new taxa list: 17841\n",
      "100.0:c__Sulfobacillia; o__Sulfobacillales; f__Sulfobacillaceae # children taxa: 7\n",
      "length new taxa list: 17835\n",
      "100.0:f__Symbiobacteriaceae; g__Symbiobacterium # children taxa: 2\n",
      "length new taxa list: 17834\n",
      "100.0:f__Thermaerobacteraceae; g__Thermaerobacter # children taxa: 3\n",
      "length new taxa list: 17832\n",
      "100.0:o__D8A-2; f__D2 # children taxa: 3\n",
      "length new taxa list: 17830\n",
      "100.0:o__UBA995; f__UBA995; g__UBA995 # children taxa: 2\n",
      "length new taxa list: 17829\n",
      "100.0:o__UBA9673; f__UBA9673; g__UBA9673 # children taxa: 2\n",
      "length new taxa list: 17828\n",
      "95.0:f__Selenomonadaceae # children taxa: 59\n",
      "length new taxa list: 17770\n",
      "100.0:f__Pelosinaceae; g__Pelosinus # children taxa: 5\n",
      "length new taxa list: 17766\n",
      "100.0:f__Propionisporaceae # children taxa: 2\n",
      "length new taxa list: 17765\n",
      "100.0:f__Sporomusaceae # children taxa: 5\n",
      "length new taxa list: 17761\n",
      "100.0:f__Thermosinaceae # children taxa: 2\n",
      "length new taxa list: 17760\n",
      "100.0:f__Acetonemaceae # children taxa: 2\n",
      "length new taxa list: 17759\n",
      "100.0:f__Acidaminococcaceae # children taxa: 21\n",
      "length new taxa list: 17739\n",
      "100.0:f__Veillonellaceae # children taxa: 16\n",
      "length new taxa list: 17724\n",
      "100.0:f__Megasphaeraceae # children taxa: 16\n",
      "length new taxa list: 17709\n",
      "100.0:f__Dialisteraceae # children taxa: 15\n",
      "length new taxa list: 17695\n",
      "100.0:f__Negativicoccaceae; g__Negativicoccus # children taxa: 2\n",
      "length new taxa list: 17694\n",
      "100.0:f__Desulfitobacteriaceae # children taxa: 27\n",
      "length new taxa list: 17668\n",
      "100.0:f__Syntrophobotulaceae # children taxa: 5\n",
      "length new taxa list: 17664\n",
      "99.0:f__DRI-13 # children taxa: 4\n",
      "length new taxa list: 17661\n",
      "100.0:f__DTU073 # children taxa: 3\n",
      "length new taxa list: 17659\n",
      "100.0:f__BICA1-8 # children taxa: 3\n",
      "length new taxa list: 17657\n",
      "98.0:f__Peptococcaceae # children taxa: 2\n",
      "length new taxa list: 17656\n",
      "100.0:o__Moorellales; f__Moorellaceae # children taxa: 9\n",
      "length new taxa list: 17648\n",
      "100.0:o__DTU086; f__DTU086 # children taxa: 2\n",
      "length new taxa list: 17647\n",
      "100.0:o__Desulfitibacterales; f__Desulfitibacteraceae; g__Desulfitibacter # children taxa: 2\n",
      "length new taxa list: 17646\n",
      "100.0:f__UBA5755 # children taxa: 2\n",
      "length new taxa list: 17645\n",
      "100.0:f__UBA4068 # children taxa: 3\n",
      "length new taxa list: 17643\n",
      "100.0:f__Syntrophomonadaceae # children taxa: 17\n",
      "length new taxa list: 17627\n",
      "100.0:f__DTU052; g__DTU052 # children taxa: 2\n",
      "length new taxa list: 17626\n",
      "100.0:f__Thermacetogeniaceae # children taxa: 5\n",
      "length new taxa list: 17622\n",
      "100.0:f__Palsa-1188; g__Palsa-1188 # children taxa: 2\n",
      "length new taxa list: 17621\n",
      "100.0:f__Nap2-2B # children taxa: 10\n",
      "length new taxa list: 17612\n",
      "100.0:f__Desulfovirgulaceae # children taxa: 5\n",
      "length new taxa list: 17608\n",
      "100.0:f__Desulfocucumaceae # children taxa: 4\n",
      "length new taxa list: 17605\n",
      "100.0:f__Desulfotomaculaceae_A # children taxa: 5\n",
      "length new taxa list: 17601\n",
      "100.0:f__Pelotomaculaceae # children taxa: 7\n",
      "length new taxa list: 17595\n",
      "100.0:f__Desulfotomaculaceae # children taxa: 8\n",
      "length new taxa list: 17588\n",
      "100.0:f__Desulfotomaculaceae_B # children taxa: 2\n",
      "length new taxa list: 17587\n",
      "100.0:f__Desulforudaceae # children taxa: 4\n",
      "length new taxa list: 17584\n",
      "100.0:o__Carboxydothermales; f__Carboxydothermaceae; g__Carboxydothermus # children taxa: 3\n",
      "length new taxa list: 17582\n",
      "100.0:f__UBA2595 # children taxa: 3\n",
      "length new taxa list: 17580\n",
      "100.0:f__Halanaerobiaceae; g__Halanaerobium # children taxa: 9\n",
      "length new taxa list: 17572\n",
      "100.0:f__Halobacteroidaceae # children taxa: 5\n",
      "length new taxa list: 17568\n",
      "100.0:f__Acetohalobiaceae # children taxa: 3\n",
      "length new taxa list: 17566\n",
      "100.0:f__Streptococcaceae # children taxa: 245\n",
      "length new taxa list: 17322\n",
      "96.0:f__Enterococcaceae # children taxa: 62\n",
      "length new taxa list: 17261\n",
      "100.0:f__Vagococcaceae # children taxa: 6\n",
      "length new taxa list: 17256\n",
      "100.0:f__Lactobacillaceae # children taxa: 252\n",
      "length new taxa list: 17005\n",
      "100.0:f__Aerococcaceae # children taxa: 48\n",
      "length new taxa list: 16958\n",
      "100.0:f__Carnobacteriaceae # children taxa: 36\n",
      "length new taxa list: 16923\n",
      "100.0:f__Listeriaceae # children taxa: 22\n",
      "length new taxa list: 16902\n",
      "100.0:f__Brochotrichaceae; g__Brochothrix # children taxa: 2\n",
      "length new taxa list: 16901\n",
      "53.0:f__Bacillaceae_A # children taxa: 108\n",
      "length new taxa list: 16794\n",
      "100.0:f__Bacillaceae_B # children taxa: 11\n",
      "length new taxa list: 16784\n",
      "100.0:f__Jeotgalibacillaceae; g__Jeotgalibacillus # children taxa: 5\n",
      "length new taxa list: 16780\n",
      "100.0:f__Domibacillaceae # children taxa: 15\n",
      "length new taxa list: 16766\n",
      "98.0:f__Bacillaceae_C # children taxa: 14\n",
      "length new taxa list: 16753\n",
      "100.0:f__Caldibacillaceae # children taxa: 4\n",
      "length new taxa list: 16750\n",
      "99.0:f__Bacillaceae # children taxa: 44\n",
      "length new taxa list: 16707\n",
      "100.0:f__Anoxybacillaceae # children taxa: 33\n",
      "length new taxa list: 16675\n",
      "100.0:f__Bacillaceae_H # children taxa: 10\n",
      "length new taxa list: 16666\n",
      "100.0:f__Bacillaceae_I # children taxa: 7\n",
      "length new taxa list: 16660\n",
      "100.0:f__Bacillaceae_L # children taxa: 7\n",
      "length new taxa list: 16654\n",
      "100.0:f__Bacillaceae_G # children taxa: 60\n",
      "length new taxa list: 16595\n",
      "100.0:f__Bacillaceae_J # children taxa: 3\n",
      "length new taxa list: 16593\n",
      "97.0:f__Bacillaceae_D # children taxa: 28\n",
      "length new taxa list: 16566\n",
      "99.0:f__Bacillaceae_F # children taxa: 4\n",
      "length new taxa list: 16563\n",
      "100.0:f__Anaerobacillaceae; g__Anaerobacillus # children taxa: 4\n",
      "length new taxa list: 16560\n",
      "100.0:f__Salisediminibacteriaceae # children taxa: 19\n",
      "length new taxa list: 16542\n",
      "100.0:f__Marinococcaceae # children taxa: 16\n",
      "length new taxa list: 16527\n",
      "100.0:f__Fictibacillaceae # children taxa: 10\n",
      "length new taxa list: 16518\n",
      "100.0:f__Anaerobacillaceae_A; g__Anaerobacillus_A # children taxa: 2\n",
      "length new taxa list: 16517\n",
      "100.0:f__Sporolactobacillaceae # children taxa: 9\n",
      "length new taxa list: 16509\n",
      "100.0:f__UBA6769; g__UBA6769 # children taxa: 2\n",
      "length new taxa list: 16508\n",
      "97.0:f__Amphibacillaceae # children taxa: 87\n",
      "length new taxa list: 16422\n",
      "100.0:f__Halobacillaceae # children taxa: 20\n",
      "length new taxa list: 16403\n",
      "100.0:f__Halobacillaceae_A # children taxa: 5\n",
      "length new taxa list: 16399\n",
      "100.0:f__Alkalibacillaceae # children taxa: 8\n",
      "length new taxa list: 16392\n",
      "100.0:o__Exiguobacterales; f__Exiguobacteraceae # children taxa: 18\n",
      "length new taxa list: 16375\n",
      "100.0:o__Bacillales_A; f__Planococcaceae # children taxa: 116\n",
      "length new taxa list: 16260\n",
      "100.0:f__Staphylococcaceae # children taxa: 70\n",
      "length new taxa list: 16191\n",
      "100.0:f__Salinicoccaceae # children taxa: 19\n",
      "length new taxa list: 16173\n",
      "100.0:f__Gemellaceae # children taxa: 10\n",
      "length new taxa list: 16164\n",
      "100.0:f__Metamycoplasmataceae # children taxa: 62\n",
      "length new taxa list: 16103\n",
      "100.0:f__Hepatoplasmataceae # children taxa: 4\n",
      "length new taxa list: 16100\n",
      "100.0:f__Mycoplasmoidaceae # children taxa: 26\n",
      "length new taxa list: 16075\n",
      "100.0:f__Mycoplasmataceae # children taxa: 48\n",
      "length new taxa list: 16028\n",
      "100.0:f__CAG-826 # children taxa: 41\n",
      "length new taxa list: 15988\n",
      "100.0:f__CAG-449 # children taxa: 2\n",
      "length new taxa list: 15987\n",
      "100.0:f__CAG-288 # children taxa: 14\n",
      "length new taxa list: 15974\n",
      "100.0:f__Erysipelotrichaceae # children taxa: 61\n",
      "length new taxa list: 15914\n",
      "100.0:f__Erysipelatoclostridiaceae # children taxa: 24\n",
      "length new taxa list: 15891\n",
      "100.0:f__CAG-611 # children taxa: 15\n",
      "length new taxa list: 15877\n",
      "100.0:f__CAG-433 # children taxa: 3\n",
      "length new taxa list: 15875\n",
      "100.0:f__UBA5348 # children taxa: 2\n",
      "length new taxa list: 15874\n",
      "100.0:f__UBA660; g__UBA660 # children taxa: 2\n",
      "length new taxa list: 15873\n",
      "100.0:f__CAG-302 # children taxa: 7\n",
      "length new taxa list: 15867\n",
      "100.0:f__CAG-822 # children taxa: 9\n",
      "length new taxa list: 15859\n",
      "100.0:f__CAG-1000 # children taxa: 20\n",
      "length new taxa list: 15840\n",
      "100.0:f__Acholeplasmataceae # children taxa: 34\n",
      "length new taxa list: 15807\n",
      "100.0:f__Acholeplasmataceae_A # children taxa: 5\n",
      "length new taxa list: 15803\n",
      "100.0:f__CAG-307 # children taxa: 4\n",
      "length new taxa list: 15800\n",
      "100.0:f__UBA5603 # children taxa: 11\n",
      "length new taxa list: 15790\n",
      "100.0:f__UBA1188; g__UBA1188 # children taxa: 2\n",
      "length new taxa list: 15789\n",
      "100.0:f__Izemoplasmataceae # children taxa: 4\n",
      "length new taxa list: 15786\n",
      "100.0:f__CAG-698 # children taxa: 8\n",
      "length new taxa list: 15779\n",
      "100.0:f__CAG-313 # children taxa: 7\n",
      "length new taxa list: 15773\n",
      "100.0:f__Turicibacteraceae # children taxa: 5\n",
      "length new taxa list: 15769\n",
      "100.0:f__Fusobacteriaceae # children taxa: 31\n",
      "length new taxa list: 15739\n",
      "100.0:f__Leptotrichiaceae # children taxa: 22\n",
      "length new taxa list: 15718\n",
      "100.0:f__Cyanobiaceae # children taxa: 340\n",
      "length new taxa list: 15379\n",
      "100.0:f__Nostocaceae # children taxa: 83\n",
      "length new taxa list: 15297\n",
      "91.0:f__Chroococcidiopsidaceae # children taxa: 9\n",
      "length new taxa list: 15289\n",
      "100.0:f__Phormidiaceae_A # children taxa: 2\n",
      "length new taxa list: 15288\n",
      "95.0:f__Chamaesiphonaceae # children taxa: 4\n",
      "length new taxa list: 15285\n",
      "83.0:f__Microcystaceae # children taxa: 25\n",
      "length new taxa list: 15261\n",
      "100.0:f__Cyanobacteriaceae # children taxa: 6\n",
      "length new taxa list: 15256\n",
      "100.0:f__Xenococcaceae # children taxa: 7\n",
      "length new taxa list: 15250\n",
      "100.0:f__Limnotrichaceae; g__Limnothrix # children taxa: 5\n",
      "length new taxa list: 15246\n",
      "100.0:f__Spirulinaceae # children taxa: 3\n",
      "length new taxa list: 15244\n",
      "100.0:f__Rubidibacteraceae # children taxa: 6\n",
      "length new taxa list: 15239\n",
      "100.0:f__Coleofasciculaceae # children taxa: 6\n",
      "length new taxa list: 15234\n",
      "100.0:f__Desertifilaceae # children taxa: 3\n",
      "length new taxa list: 15232\n",
      "100.0:f__Phormidiaceae # children taxa: 21\n",
      "length new taxa list: 15212\n",
      "82.0:f__Geitlerinemaceae # children taxa: 5\n",
      "length new taxa list: 15208\n",
      "90.0:f__Oscillatoriaceae # children taxa: 3\n",
      "length new taxa list: 15206\n",
      "100.0:o__Leptolyngbyales; f__Leptolyngbyaceae # children taxa: 9\n",
      "length new taxa list: 15198\n",
      "93.0:o__Neosynechococcales; f__Neosynechococcaceae # children taxa: 2\n",
      "length new taxa list: 15197\n",
      "100.0:o__Elainellales; f__Elainellaceae # children taxa: 5\n",
      "length new taxa list: 15193\n",
      "100.0:o__Phormidesmiales; f__Phormidesmiaceae # children taxa: 17\n",
      "length new taxa list: 15177\n",
      "100.0:o__Thermosynechococcales; f__Thermosynechococcaceae # children taxa: 8\n",
      "length new taxa list: 15170\n",
      "100.0:o__CACIAM-69d; f__CACIAM-69d; g__CACIAM-69d # children taxa: 2\n",
      "length new taxa list: 15169\n",
      "100.0:o__Pseudanabaenales; f__Pseudanabaenaceae # children taxa: 15\n",
      "length new taxa list: 15155\n",
      "100.0:f__JA-3-3Ab; g__JA-3-3Ab # children taxa: 2\n",
      "length new taxa list: 15154\n",
      "100.0:o__Gloeobacterales; f__Gloeobacteraceae; g__Gloeobacter # children taxa: 2\n",
      "length new taxa list: 15153\n",
      "100.0:f__Gastranaerophilaceae # children taxa: 36\n",
      "length new taxa list: 15118\n",
      "100.0:f__UBA9579 # children taxa: 5\n",
      "length new taxa list: 15114\n",
      "100.0:o__Vampirovibrionales; f__Vampirovibrionaceae; g__Vampirovibrio # children taxa: 3\n",
      "length new taxa list: 15112\n",
      "100.0:o__Obscuribacterales; f__Obscuribacteraceae # children taxa: 6\n",
      "length new taxa list: 15107\n",
      "100.0:f__2-02-FULL-35-15 # children taxa: 2\n",
      "length new taxa list: 15106\n",
      "100.0:o__UBA7694; f__UBA7694 # children taxa: 2\n",
      "length new taxa list: 15105\n",
      "100.0:f__XYB2-FULL-48-7 # children taxa: 3\n",
      "length new taxa list: 15103\n",
      "100.0:f__O2-12-FULL-45-9 # children taxa: 3\n",
      "length new taxa list: 15101\n",
      "100.0:f__XYC2-FULL-37-10 # children taxa: 2\n",
      "length new taxa list: 15100\n",
      "100.0:f__GCA-002719695 # children taxa: 5\n",
      "length new taxa list: 15096\n",
      "100.0:f__Mycobacteriaceae # children taxa: 611\n",
      "length new taxa list: 14486\n",
      "100.0:f__Pseudonocardiaceae # children taxa: 159\n",
      "length new taxa list: 14328\n",
      "100.0:f__Nakamurellaceae; g__Nakamurella # children taxa: 5\n",
      "length new taxa list: 14324\n",
      "100.0:f__Geodermatophilaceae # children taxa: 46\n",
      "length new taxa list: 14279\n",
      "100.0:f__Antricoccaceae; g__Antricoccus # children taxa: 2\n",
      "length new taxa list: 14278\n",
      "100.0:f__Jatrophihabitantaceae # children taxa: 5\n",
      "length new taxa list: 14274\n",
      "100.0:f__Micromonosporaceae # children taxa: 115\n",
      "length new taxa list: 14160\n",
      "100.0:f__Cryptosporangiaceae; g__Cryptosporangium # children taxa: 2\n",
      "length new taxa list: 14159\n",
      "100.0:o__Frankiales; f__Frankiaceae # children taxa: 19\n",
      "length new taxa list: 14141\n",
      "100.0:o__Streptosporangiales; f__Streptosporangiaceae # children taxa: 102\n",
      "length new taxa list: 14040\n",
      "100.0:o__Acidothermales; f__Acidothermaceae # children taxa: 2\n",
      "length new taxa list: 14039\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "100.0:f__Streptomycetaceae # children taxa: 528\n",
      "length new taxa list: 13512\n",
      "100.0:f__Catenulisporaceae # children taxa: 2\n",
      "length new taxa list: 13511\n",
      "100.0:f__Nocardioidaceae # children taxa: 69\n",
      "length new taxa list: 13443\n",
      "100.0:f__Propionibacteriaceae # children taxa: 54\n",
      "length new taxa list: 13390\n",
      "100.0:f__Kribbellaceae; g__Kribbella # children taxa: 4\n",
      "length new taxa list: 13387\n",
      "100.0:f__Actinopolymorphaceae # children taxa: 4\n",
      "length new taxa list: 13384\n",
      "100.0:o__Jiangellales; f__Jiangellaceae # children taxa: 7\n",
      "length new taxa list: 13378\n",
      "100.0:f__Nanopelagicaceae # children taxa: 29\n",
      "length new taxa list: 13350\n",
      "100.0:f__S36-B12 # children taxa: 10\n",
      "length new taxa list: 13341\n",
      "85.0:f__AcAMD-5 # children taxa: 2\n",
      "length new taxa list: 13340\n",
      "100.0:f__UBA12327 # children taxa: 2\n",
      "length new taxa list: 13339\n",
      "100.0:f__Euzebyaceae; g__Euzebya # children taxa: 2\n",
      "length new taxa list: 13338\n",
      "100.0:f__UBA4738 # children taxa: 2\n",
      "length new taxa list: 13337\n",
      "100.0:f__MedAcidi-G1 # children taxa: 24\n",
      "length new taxa list: 13314\n",
      "100.0:f__UBA11606 # children taxa: 4\n",
      "length new taxa list: 13311\n",
      "100.0:f__UBA8592 # children taxa: 18\n",
      "length new taxa list: 13294\n",
      "100.0:f__Microtrichaceae # children taxa: 4\n",
      "length new taxa list: 13291\n",
      "100.0:f__TK06 # children taxa: 17\n",
      "length new taxa list: 13275\n",
      "100.0:f__Ilumatobacteraceae # children taxa: 17\n",
      "length new taxa list: 13259\n",
      "100.0:f__RAAP-2 # children taxa: 49\n",
      "length new taxa list: 13211\n",
      "100.0:f__UBA8190 # children taxa: 5\n",
      "length new taxa list: 13207\n",
      "100.0:f__Bog-793 # children taxa: 6\n",
      "length new taxa list: 13202\n",
      "100.0:f__Acidimicrobiaceae # children taxa: 4\n",
      "length new taxa list: 13199\n",
      "100.0:f__Palsa-688 # children taxa: 4\n",
      "length new taxa list: 13196\n",
      "100.0:f__UBA5794 # children taxa: 4\n",
      "length new taxa list: 13193\n",
      "100.0:f__UBA11373 # children taxa: 2\n",
      "length new taxa list: 13192\n",
      "100.0:f__UBA4744 # children taxa: 2\n",
      "length new taxa list: 13191\n",
      "100.0:f__ZC4RG35 # children taxa: 2\n",
      "length new taxa list: 13190\n",
      "100.0:o__TMED189; f__TMED189; g__TMED189 # children taxa: 9\n",
      "length new taxa list: 13182\n",
      "100.0:f__Solirubrobacteraceae # children taxa: 19\n",
      "length new taxa list: 13164\n",
      "100.0:f__Thermoleophilaceae; g__Thermoleophilum # children taxa: 2\n",
      "length new taxa list: 13163\n",
      "100.0:f__70-9 # children taxa: 2\n",
      "length new taxa list: 13162\n",
      "100.0:o__Ga0077560; f__Ga0077560 # children taxa: 2\n",
      "length new taxa list: 13161\n",
      "100.0:o__20CM-4-69-9; f__20CM-4-69-9 # children taxa: 13\n",
      "length new taxa list: 13149\n",
      "100.0:o__UBA2241; f__UBA2241 # children taxa: 7\n",
      "length new taxa list: 13143\n",
      "100.0:o__RBG-16-64-13; f__RBG-16-64-13; g__RBG-16-64-13 # children taxa: 2\n",
      "length new taxa list: 13142\n",
      "100.0:c__Rubrobacteria; o__Rubrobacterales; f__Rubrobacteraceae # children taxa: 3\n",
      "length new taxa list: 13140\n",
      "100.0:f__Atopobiaceae # children taxa: 52\n",
      "length new taxa list: 13089\n",
      "100.0:f__Coriobacteriaceae # children taxa: 27\n",
      "length new taxa list: 13063\n",
      "100.0:f__Eggerthellaceae # children taxa: 34\n",
      "length new taxa list: 13030\n",
      "100.0:f__UBA8131 # children taxa: 3\n",
      "length new taxa list: 13028\n",
      "100.0:f__UBA2279 # children taxa: 9\n",
      "length new taxa list: 13020\n",
      "100.0:f__UBA912 # children taxa: 2\n",
      "length new taxa list: 13019\n",
      "100.0:o__UBA10029; f__UBA10029 # children taxa: 2\n",
      "length new taxa list: 13018\n",
      "100.0:c__UBA1414; o__UBA1414; f__UBA1414 # children taxa: 4\n",
      "length new taxa list: 13015\n",
      "100.0:o__Fen-727; f__Fen-727 # children taxa: 3\n",
      "length new taxa list: 13013\n",
      "100.0:o__UBP12; f__UBA5184 # children taxa: 25\n",
      "length new taxa list: 12989\n",
      "100.0:c__UBP9; o__UBA4705; f__UBA4705 # children taxa: 4\n",
      "length new taxa list: 12986\n",
      "100.0:o__Fimbriimonadales; f__Fimbriimonadaceae # children taxa: 13\n",
      "length new taxa list: 12974\n",
      "100.0:o__OS-L; f__GBS-DC # children taxa: 3\n",
      "length new taxa list: 12972\n",
      "98.0:f__UBA2163 # children taxa: 73\n",
      "length new taxa list: 12900\n",
      "100.0:f__GCA-002724975; g__GCA-002724975 # children taxa: 2\n",
      "length new taxa list: 12899\n",
      "100.0:f__UBA12079; g__UBA12079 # children taxa: 2\n",
      "length new taxa list: 12898\n",
      "96.0:f__Zambryskibacteraceae # children taxa: 25\n",
      "length new taxa list: 12874\n",
      "56.0:f__UBA11359_A # children taxa: 14\n",
      "length new taxa list: 12861\n",
      "57.0:f__UBA11359 # children taxa: 8\n",
      "length new taxa list: 12854\n",
      "91.0:f__GWA1-40-21 # children taxa: 2\n",
      "length new taxa list: 12853\n",
      "70.0:f__UBA1539 # children taxa: 14\n",
      "length new taxa list: 12840\n",
      "100.0:f__UBA6899; g__UBA6899 # children taxa: 6\n",
      "length new taxa list: 12835\n",
      "99.0:f__XYD1-FULL-46-19 # children taxa: 4\n",
      "length new taxa list: 12832\n",
      "97.0:f__UBA12445 # children taxa: 2\n",
      "length new taxa list: 12831\n",
      "65.0:f__UBA9973 # children taxa: 43\n",
      "length new taxa list: 12789\n",
      "100.0:f__UBA5272 # children taxa: 2\n",
      "length new taxa list: 12788\n",
      "91.0:f__UBA1006 # children taxa: 2\n",
      "length new taxa list: 12787\n",
      "100.0:f__UBA11713 # children taxa: 5\n",
      "length new taxa list: 12783\n",
      "100.0:f__1-14-0-10-42-19 # children taxa: 2\n",
      "length new taxa list: 12782\n",
      "100.0:f__UBA9983 # children taxa: 2\n",
      "length new taxa list: 12781\n",
      "89.0:f__Tagabacteraceae # children taxa: 2\n",
      "length new taxa list: 12780\n",
      "100.0:o__WOR2-41-13_A; f__2-01-FULL-45-33 # children taxa: 7\n",
      "length new taxa list: 12774\n",
      "100.0:f__GWA2-47-10b # children taxa: 2\n",
      "length new taxa list: 12773\n",
      "100.0:f__2-02-FULL-48-12 # children taxa: 3\n",
      "length new taxa list: 12771\n",
      "100.0:f__GWA2-42-14 # children taxa: 4\n",
      "length new taxa list: 12768\n",
      "100.0:f__HO2-52-23 # children taxa: 4\n",
      "length new taxa list: 12765\n",
      "100.0:f__GWA2-44-9 # children taxa: 17\n",
      "length new taxa list: 12749\n",
      "99.0:f__IGHO2-01-FULL-4-A # children taxa: 2\n",
      "length new taxa list: 12748\n",
      "91.0:f__IGHO2-12-FULL-4 # children taxa: 2\n",
      "length new taxa list: 12747\n",
      "91.0:f__WO2-44-18 # children taxa: 3\n",
      "length new taxa list: 12745\n",
      "100.0:f__02-FULL-45-10c # children taxa: 2\n",
      "length new taxa list: 12744\n",
      "100.0:f__1-14-0-10-44-23 # children taxa: 2\n",
      "length new taxa list: 12743\n",
      "98.0:f__Colwellbacteraceae # children taxa: 7\n",
      "length new taxa list: 12737\n",
      "100.0:f__2-01-FULL-56-20 # children taxa: 2\n",
      "length new taxa list: 12736\n",
      "100.0:f__XYB1-36-10 # children taxa: 2\n",
      "length new taxa list: 12735\n",
      "95.0:f__UBA9933 # children taxa: 9\n",
      "length new taxa list: 12727\n",
      "92.0:f__GWB1-50-10 # children taxa: 8\n",
      "length new taxa list: 12720\n",
      "100.0:f__Brennerbacteraceae # children taxa: 4\n",
      "length new taxa list: 12717\n",
      "100.0:f__HR35 # children taxa: 2\n",
      "length new taxa list: 12716\n",
      "100.0:f__UBA10092 # children taxa: 3\n",
      "length new taxa list: 12714\n",
      "100.0:f__Paceibacteraceae; g__Paceibacter # children taxa: 2\n",
      "length new taxa list: 12713\n",
      "93.0:f__GWA2-38-27 # children taxa: 3\n",
      "length new taxa list: 12711\n",
      "95.0:f__CG1-02-41-26 # children taxa: 4\n",
      "length new taxa list: 12708\n",
      "97.0:f__UBA5633 # children taxa: 6\n",
      "length new taxa list: 12703\n",
      "100.0:f__Staskawiczbacteraceae # children taxa: 11\n",
      "length new taxa list: 12693\n",
      "100.0:f__GWA2-46-15; g__GWA2-46-15 # children taxa: 2\n",
      "length new taxa list: 12692\n",
      "100.0:f__UBA5738; g__UBA5738 # children taxa: 2\n",
      "length new taxa list: 12691\n",
      "100.0:f__UBA10102 # children taxa: 5\n",
      "length new taxa list: 12687\n",
      "100.0:f__GWB1-42-6 # children taxa: 4\n",
      "length new taxa list: 12684\n",
      "100.0:f__UBA2143 # children taxa: 4\n",
      "length new taxa list: 12681\n",
      "100.0:o__Terrybacterales; f__GWB1-40-14 # children taxa: 4\n",
      "length new taxa list: 12678\n",
      "100.0:f__M10-OD1-4 # children taxa: 3\n",
      "length new taxa list: 12676\n",
      "100.0:f__UBA2206 # children taxa: 16\n",
      "length new taxa list: 12661\n",
      "100.0:f__UBA9337; g__UBA9337 # children taxa: 4\n",
      "length new taxa list: 12658\n",
      "100.0:f__2-12-FULL-40-10 # children taxa: 2\n",
      "length new taxa list: 12657\n",
      "100.0:f__GWC2-37-73 # children taxa: 10\n",
      "length new taxa list: 12648\n",
      "100.0:f__UBA1568; g__UBA1568 # children taxa: 6\n",
      "length new taxa list: 12643\n",
      "100.0:f__UBA11712 # children taxa: 4\n",
      "length new taxa list: 12640\n",
      "100.0:f__UBA2193 # children taxa: 3\n",
      "length new taxa list: 12638\n",
      "100.0:f__UBA922 # children taxa: 25\n",
      "length new taxa list: 12614\n",
      "100.0:f__GWA2-37-8 # children taxa: 6\n",
      "length new taxa list: 12609\n",
      "100.0:f__UBA10003 # children taxa: 2\n",
      "length new taxa list: 12608\n",
      "100.0:f__GWF2-40-263 # children taxa: 13\n",
      "length new taxa list: 12596\n",
      "100.0:f__UBA9934 # children taxa: 4\n",
      "length new taxa list: 12593\n",
      "100.0:f__UBA11717 # children taxa: 3\n",
      "length new taxa list: 12591\n",
      "100.0:f__2-12-FULL-60-25 # children taxa: 3\n",
      "length new taxa list: 12589\n",
      "100.0:f__SG8-24 # children taxa: 3\n",
      "length new taxa list: 12587\n",
      "100.0:f__GWA2-47-26; g__GWA2-47-26 # children taxa: 2\n",
      "length new taxa list: 12586\n",
      "100.0:o__GWA2-46-9; f__GWA2-46-9 # children taxa: 3\n",
      "length new taxa list: 12584\n",
      "71.0:f__GWE2-39-37 # children taxa: 6\n",
      "length new taxa list: 12579\n",
      "95.0:f__UBA12075 # children taxa: 4\n",
      "length new taxa list: 12576\n",
      "100.0:f__UBA12465 # children taxa: 7\n",
      "length new taxa list: 12570\n",
      "100.0:f__UBA917 # children taxa: 4\n",
      "length new taxa list: 12567\n",
      "100.0:f__XYC2-FULL-47-12 # children taxa: 5\n",
      "length new taxa list: 12563\n",
      "100.0:f__UBA6539 # children taxa: 5\n",
      "length new taxa list: 12559\n",
      "100.0:f__2-01-FULL-39-33; g__2-01-FULL-39-33 # children taxa: 2\n",
      "length new taxa list: 12558\n",
      "100.0:o__UBA11705; f__UBA11705 # children taxa: 5\n",
      "length new taxa list: 12554\n",
      "100.0:f__UBA6264 # children taxa: 2\n",
      "length new taxa list: 12553\n",
      "100.0:f__GWA2-36-10 # children taxa: 5\n",
      "length new taxa list: 12549\n",
      "100.0:f__UBA10009 # children taxa: 2\n",
      "length new taxa list: 12548\n",
      "100.0:f__UBA1558 # children taxa: 2\n",
      "length new taxa list: 12547\n",
      "100.0:o__Veblenbacterales; f__UBA10138 # children taxa: 2\n",
      "length new taxa list: 12546\n",
      "100.0:f__GWA2-42-15 # children taxa: 4\n",
      "length new taxa list: 12543\n",
      "100.0:f__UBA2196 # children taxa: 2\n",
      "length new taxa list: 12542\n",
      "100.0:o__UBA2169; f__UBA2169 # children taxa: 4\n",
      "length new taxa list: 12539\n",
      "100.0:o__UBA9629; f__UBA9629 # children taxa: 2\n",
      "length new taxa list: 12538\n",
      "100.0:f__XYC2-FULL-38-9 # children taxa: 2\n",
      "length new taxa list: 12537\n",
      "100.0:f__HO2-48-17 # children taxa: 2\n",
      "length new taxa list: 12536\n",
      "100.0:f__UM-FILTER-42-10; g__UM-FILTER-42-10 # children taxa: 2\n",
      "length new taxa list: 12535\n",
      "100.0:f__O2-02-FULL-48-8 # children taxa: 5\n",
      "length new taxa list: 12531\n",
      "100.0:f__UBA920 # children taxa: 2\n",
      "length new taxa list: 12530\n",
      "100.0:f__2-01-FULL-46-14; g__2-01-FULL-46-14 # children taxa: 2\n",
      "length new taxa list: 12529\n",
      "100.0:f__2-01-FULL-50-11 # children taxa: 2\n",
      "length new taxa list: 12528\n",
      "100.0:c__Andersenbacteria; o__UBA10190; f__UBA10190 # children taxa: 5\n",
      "length new taxa list: 12524\n",
      "100.0:o__Peribacterales; f__Peribacteraceae # children taxa: 17\n",
      "length new taxa list: 12508\n",
      "100.0:o__UBA4473; f__UBA4473 # children taxa: 2\n",
      "length new taxa list: 12507\n",
      "100.0:o__UM-FILTER-43-11; f__UM-FILTER-43-11 # children taxa: 2\n",
      "length new taxa list: 12506\n",
      "100.0:f__LOWO2-01-FULL-3 # children taxa: 4\n",
      "length new taxa list: 12503\n",
      "100.0:f__UBA12463; g__UBA12463 # children taxa: 2\n",
      "length new taxa list: 12502\n",
      "100.0:f__UBA1369 # children taxa: 2\n",
      "length new taxa list: 12501\n",
      "100.0:f__UBA12470 # children taxa: 3\n",
      "length new taxa list: 12499\n",
      "100.0:f__UM-FILTER-53-11 # children taxa: 2\n",
      "length new taxa list: 12498\n",
      "100.0:f__GWF2-33-10 # children taxa: 2\n",
      "length new taxa list: 12497\n",
      "100.0:o__BD1-5; f__UBA6164 # children taxa: 14\n",
      "length new taxa list: 12484\n",
      "100.0:o__Absconditabacterales; f__X112 # children taxa: 9\n",
      "length new taxa list: 12476\n",
      "100.0:f__Saccharimonadaceae # children taxa: 49\n",
      "length new taxa list: 12428\n",
      "84.0:f__UBA4665 # children taxa: 11\n",
      "length new taxa list: 12418\n",
      "84.0:f__UBA10212 # children taxa: 6\n",
      "length new taxa list: 12413\n",
      "100.0:o__UBA4664; f__UBA4664 # children taxa: 2\n",
      "length new taxa list: 12412\n",
      "100.0:o__CPR2; f__CPR2 # children taxa: 3\n",
      "length new taxa list: 12410\n",
      "100.0:f__UBA1875 # children taxa: 2\n",
      "length new taxa list: 12409\n",
      "100.0:f__GWA2-35-9; g__GWA2-35-9 # children taxa: 2\n",
      "length new taxa list: 12408\n",
      "100.0:o__XYA2-FULL-43-10; f__XYA2-FULL-43-10 # children taxa: 2\n",
      "length new taxa list: 12407\n",
      "100.0:f__UBA10110 # children taxa: 3\n",
      "length new taxa list: 12405\n",
      "100.0:f__Kazan-3B-28; g__Kazan-3B-28 # children taxa: 2\n",
      "length new taxa list: 12404\n",
      "100.0:f__UBA8517 # children taxa: 27\n",
      "length new taxa list: 12378\n",
      "100.0:f__UBA10165 # children taxa: 4\n",
      "length new taxa list: 12375\n",
      "100.0:f__Amesbacteraceae # children taxa: 8\n",
      "length new taxa list: 12368\n",
      "100.0:f__UBA12405 # children taxa: 12\n",
      "length new taxa list: 12357\n",
      "100.0:f__PJMF01 # children taxa: 18\n",
      "length new taxa list: 12340\n",
      "100.0:f__UBA12028 # children taxa: 3\n",
      "length new taxa list: 12338\n",
      "100.0:f__RBG-16-45-19 # children taxa: 2\n",
      "length new taxa list: 12337\n",
      "100.0:f__MFAQ01 # children taxa: 10\n",
      "length new taxa list: 12328\n",
      "100.0:f__UBA12108 # children taxa: 2\n",
      "length new taxa list: 12327\n",
      "100.0:f__Chisholmbacteraceae # children taxa: 2\n",
      "length new taxa list: 12326\n",
      "100.0:f__CG1-02-47-37 # children taxa: 2\n",
      "length new taxa list: 12325\n",
      "100.0:f__UBA1400 # children taxa: 2\n",
      "length new taxa list: 12324\n",
      "100.0:f__GWC2-37-13 # children taxa: 30\n",
      "length new taxa list: 12295\n",
      "100.0:f__HO2-37-13b # children taxa: 7\n",
      "length new taxa list: 12289\n",
      "100.0:f__UBA1406 # children taxa: 3\n",
      "length new taxa list: 12287\n",
      "100.0:f__UBA12049 # children taxa: 19\n",
      "length new taxa list: 12269\n",
      "100.0:f__GWA1-39-11 # children taxa: 4\n",
      "length new taxa list: 12266\n",
      "100.0:f__UBA927 # children taxa: 8\n",
      "length new taxa list: 12259\n",
      "100.0:f__UBA10105 # children taxa: 2\n",
      "length new taxa list: 12258\n",
      "100.0:o__2-02-FULL-39-11; f__2-02-FULL-39-11 # children taxa: 9\n",
      "length new taxa list: 12250\n",
      "100.0:o__Daviesbacterales; f__UBA10151 # children taxa: 13\n",
      "length new taxa list: 12238\n",
      "100.0:o__Curtissbacterales; f__GWA2-41-24 # children taxa: 10\n",
      "length new taxa list: 12229\n",
      "100.0:f__RBG-16-39-9b # children taxa: 3\n",
      "length new taxa list: 12227\n",
      "100.0:f__UBA10185 # children taxa: 16\n",
      "length new taxa list: 12212\n",
      "100.0:o__2-01-FULL-39-13; f__2-01-FULL-39-13 # children taxa: 2\n",
      "length new taxa list: 12211\n",
      "100.0:f__SC72 # children taxa: 14\n",
      "length new taxa list: 12198\n",
      "100.0:o__B142; f__OLB20 # children taxa: 2\n",
      "length new taxa list: 12197\n",
      "52.0:f__Bin127 # children taxa: 15\n",
      "length new taxa list: 12183\n",
      "80.0:f__TMED-70 # children taxa: 15\n",
      "length new taxa list: 12169\n",
      "98.0:f__UBA1151 # children taxa: 2\n",
      "length new taxa list: 12168\n",
      "100.0:f__UBA826 # children taxa: 12\n",
      "length new taxa list: 12157\n",
      "100.0:f__UBA11138 # children taxa: 5\n",
      "length new taxa list: 12153\n",
      "100.0:f__GCA-002694895; g__GCA-002694895 # children taxa: 4\n",
      "length new taxa list: 12150\n",
      "100.0:f__GCA-002723415; g__GCA-002723415 # children taxa: 2\n",
      "length new taxa list: 12149\n",
      "100.0:f__UBA2962; g__UBA2962 # children taxa: 4\n",
      "length new taxa list: 12146\n",
      "100.0:f__UBA3495 # children taxa: 22\n",
      "length new taxa list: 12125\n",
      "100.0:f__UBA2588; g__UBA2588 # children taxa: 3\n",
      "length new taxa list: 12123\n",
      "100.0:f__UBA2963 # children taxa: 5\n",
      "length new taxa list: 12119\n",
      "97.0:f__Dehalococcoidaceae # children taxa: 8\n",
      "length new taxa list: 12112\n",
      "52.0:f__UBA5627 # children taxa: 3\n",
      "length new taxa list: 12110\n",
      "97.0:f__UBA3254 # children taxa: 4\n",
      "length new taxa list: 12107\n",
      "100.0:f__RBG-16-60-22 # children taxa: 7\n",
      "length new taxa list: 12101\n",
      "43.0:f__UBA5760 # children taxa: 3\n",
      "length new taxa list: 12099\n",
      "95.0:f__UBA2162 # children taxa: 3\n",
      "length new taxa list: 12097\n",
      "66.0:f__UBA5629 # children taxa: 6\n",
      "length new taxa list: 12092\n",
      "99.0:f__UBA5620 # children taxa: 2\n",
      "length new taxa list: 12091\n",
      "100.0:f__AB-539-J10 # children taxa: 5\n",
      "length new taxa list: 12087\n",
      "100.0:f__UBA2979 # children taxa: 7\n",
      "length new taxa list: 12081\n",
      "100.0:f__GCA-002718395; g__GCA-002718395 # children taxa: 3\n",
      "length new taxa list: 12079\n",
      "100.0:o__Bin125; f__Bin125 # children taxa: 2\n",
      "length new taxa list: 12078\n",
      "100.0:o__UBA2991; f__UBA2991 # children taxa: 4\n",
      "length new taxa list: 12075\n",
      "100.0:f__HRbin24 # children taxa: 2\n",
      "length new taxa list: 12074\n",
      "100.0:f__RBG-16-68-14 # children taxa: 2\n",
      "length new taxa list: 12073\n",
      "90.0:f__Anaerolineaceae # children taxa: 58\n",
      "length new taxa list: 12016\n",
      "100.0:f__envOPS12 # children taxa: 19\n",
      "length new taxa list: 11998\n",
      "97.0:f__UBA4823 # children taxa: 4\n",
      "length new taxa list: 11995\n",
      "100.0:f__UBA6092 # children taxa: 2\n",
      "length new taxa list: 11994\n",
      "100.0:f__UBA11579; g__UBA11579 # children taxa: 2\n",
      "length new taxa list: 11993\n",
      "100.0:f__RBG-16-64-43 # children taxa: 2\n",
      "length new taxa list: 11992\n",
      "100.0:f__UBA11657; g__UBA11657 # children taxa: 4\n",
      "length new taxa list: 11989\n",
      "100.0:f__A4b # children taxa: 9\n",
      "length new taxa list: 11981\n",
      "100.0:f__UBA2796 # children taxa: 2\n",
      "length new taxa list: 11980\n",
      "100.0:f__Promineofilaceae # children taxa: 4\n",
      "length new taxa list: 11977\n",
      "100.0:f__Thermoflexaceae; g__Thermoflexus # children taxa: 2\n",
      "length new taxa list: 11976\n",
      "99.0:f__Caldilineaceae # children taxa: 6\n",
      "length new taxa list: 11971\n",
      "100.0:f__CG2-30-64-16; g__CG2-30-64-16 # children taxa: 2\n",
      "length new taxa list: 11970\n",
      "100.0:o__UBA2200; f__UBA2200 # children taxa: 3\n",
      "length new taxa list: 11968\n",
      "100.0:o__UBA1429; f__UBA1429 # children taxa: 2\n",
      "length new taxa list: 11967\n",
      "100.0:o__UBA7937; f__UBA7937; g__UBA7937 # children taxa: 2\n",
      "length new taxa list: 11966\n",
      "100.0:f__Chloroflexaceae # children taxa: 9\n",
      "length new taxa list: 11958\n",
      "100.0:f__Roseiflexaceae # children taxa: 4\n",
      "length new taxa list: 11955\n",
      "72.0:f__Herpetosiphonaceae # children taxa: 4\n",
      "length new taxa list: 11952\n",
      "100.0:f__Thermomicrobiaceae # children taxa: 7\n",
      "length new taxa list: 11946\n",
      "100.0:c__Ktedonobacteria; o__Ktedonobacterales; f__Ktedonobacteraceae # children taxa: 7\n",
      "length new taxa list: 11940\n",
      "100.0:c__UBA5177; o__UBA5177; f__UBA5177 # children taxa: 2\n",
      "length new taxa list: 11939\n",
      "100.0:f__UBA8260 # children taxa: 16\n",
      "length new taxa list: 11924\n",
      "100.0:f__Bog-877; g__Bog-877 # children taxa: 4\n",
      "length new taxa list: 11921\n",
      "100.0:o__Dormibacterales; f__Dormibacteraceae # children taxa: 12\n",
      "length new taxa list: 11910\n",
      "100.0:f__CSP1-4 # children taxa: 25\n",
      "length new taxa list: 11886\n",
      "100.0:o__P2-11E; f__P2-11E # children taxa: 5\n",
      "length new taxa list: 11882\n",
      "100.0:f__Deinococcaceae # children taxa: 36\n",
      "length new taxa list: 11847\n",
      "100.0:f__Trueperaceae # children taxa: 3\n",
      "length new taxa list: 11845\n",
      "100.0:f__Thermaceae # children taxa: 28\n",
      "length new taxa list: 11818\n",
      "99.0:f__Marinithermaceae # children taxa: 2\n",
      "length new taxa list: 11817\n",
      "100.0:f__Fervidobacteriaceae # children taxa: 13\n",
      "length new taxa list: 11805\n",
      "97.0:f__Thermotogaceae # children taxa: 13\n",
      "length new taxa list: 11793\n",
      "100.0:f__Petrotogaceae # children taxa: 14\n",
      "length new taxa list: 11780\n",
      "100.0:f__Kosmotogaceae # children taxa: 11\n",
      "length new taxa list: 11770\n",
      "100.0:o__Mesoaciditogales; f__Mesoaciditogaceae # children taxa: 2\n",
      "length new taxa list: 11769\n",
      "100.0:o__Bipolaricaulales; f__Bipolaricaulaceae # children taxa: 8\n",
      "length new taxa list: 11762\n",
      "100.0:o__UBA7950; f__UBA7950 # children taxa: 2\n",
      "length new taxa list: 11761\n",
      "100.0:o__Caldisericales; f__Caldisericaceae # children taxa: 5\n",
      "length new taxa list: 11757\n",
      "100.0:o__UBA4822; f__UBA4822; g__UBA4822 # children taxa: 4\n",
      "length new taxa list: 11754\n",
      "100.0:p__Coprothermobacterota; c__Coprothermobacteria; o__Coprothermobacterales; f__Coprothermobacteraceae; g__Coprothermobacter # children taxa: 3\n",
      "length new taxa list: 11752\n",
      "100.0:p__Dictyoglomota; c__Dictyoglomia; o__Dictyoglomales; f__Dictyoglomaceae # children taxa: 3\n",
      "length new taxa list: 11750\n",
      "100.0:p__Thermodesulfobiota; c__Thermodesulfobiia; o__Thermodesulfobiales; f__Thermodesulfobiaceae; g__Thermodesulfobium # children taxa: 2\n",
      "length new taxa list: 11749\n",
      "100.0:f__Aminobacteriaceae # children taxa: 7\n",
      "length new taxa list: 11743\n",
      "100.0:f__Dethiosulfovibrionaceae # children taxa: 6\n",
      "length new taxa list: 11738\n",
      "100.0:f__Thermovirgaceae # children taxa: 5\n",
      "length new taxa list: 11734\n",
      "100.0:f__Synergistaceae # children taxa: 11\n",
      "length new taxa list: 11724\n",
      "100.0:f__79-D21; g__79-D21 # children taxa: 3\n",
      "length new taxa list: 11722\n",
      "100.0:f__Aminiphilaceae; g__Aminiphilus # children taxa: 2\n",
      "length new taxa list: 11721\n",
      "100.0:f__Acetomicrobiaceae; g__Acetomicrobium # children taxa: 5\n",
      "length new taxa list: 11717\n",
      "100.0:c__GBS-1; o__GBS-1; f__GBS-1 # children taxa: 2\n",
      "length new taxa list: 11716\n",
      "100.0:f__34-128 # children taxa: 10\n",
      "length new taxa list: 11707\n",
      "99.0:f__Caldatribacteriaceae # children taxa: 3\n",
      "length new taxa list: 11705\n",
      "100.0:p__Riflebacteria; c__UBA8953; o__UBA8953; f__UBA8953 # children taxa: 5\n",
      "length new taxa list: 11701\n",
      "100.0:p__UBP17; c__UBA6191; o__UBA6191; f__UBA6191 # children taxa: 2\n",
      "length new taxa list: 11700\n",
      "100.0:p__Wallbacteria; c__UBA9980; o__UBA9980; f__UBA9980; g__UBA9980 # children taxa: 2\n",
      "length new taxa list: 11699\n",
      "100.0:f__UBA10450 # children taxa: 61\n",
      "length new taxa list: 11639\n",
      "100.0:f__Chthoniobacteraceae # children taxa: 2\n",
      "length new taxa list: 11638\n",
      "100.0:f__Terrimicrobiaceae # children taxa: 5\n",
      "length new taxa list: 11634\n",
      "100.0:f__Xiphinematobacteraceae # children taxa: 2\n",
      "length new taxa list: 11633\n",
      "100.0:f__UBA6821; g__UBA6821 # children taxa: 2\n",
      "length new taxa list: 11632\n",
      "100.0:f__Akkermansiaceae # children taxa: 35\n",
      "length new taxa list: 11598\n",
      "95.0:f__DEV007 # children taxa: 16\n",
      "length new taxa list: 11583\n",
      "100.0:f__Verrucomicrobiaceae # children taxa: 10\n",
      "length new taxa list: 11574\n",
      "100.0:f__Methylacidiphilaceae # children taxa: 5\n",
      "length new taxa list: 11570\n",
      "100.0:f__UBA1321 # children taxa: 3\n",
      "length new taxa list: 11568\n",
      "100.0:f__UBA3015; g__UBA3015 # children taxa: 4\n",
      "length new taxa list: 11565\n",
      "97.0:f__Pedosphaeraceae # children taxa: 18\n",
      "length new taxa list: 11548\n",
      "100.0:f__Palsa-1400; g__Palsa-1400 # children taxa: 2\n",
      "length new taxa list: 11547\n",
      "100.0:f__AV2; g__AV2 # children taxa: 4\n",
      "length new taxa list: 11544\n",
      "100.0:f__UBA9464 # children taxa: 2\n",
      "length new taxa list: 11543\n",
      "100.0:f__UBA1096; g__UBA1096 # children taxa: 8\n",
      "length new taxa list: 11536\n",
      "100.0:f__UBA1100 # children taxa: 7\n",
      "length new taxa list: 11530\n",
      "100.0:f__AAA164-E04 # children taxa: 4\n",
      "length new taxa list: 11527\n",
      "100.0:f__UBA1412; g__UBA1412 # children taxa: 2\n",
      "length new taxa list: 11526\n",
      "100.0:o__Palsa-1439; f__Palsa-1439; g__Palsa-1439 # children taxa: 2\n",
      "length new taxa list: 11525\n",
      "100.0:f__Opitutaceae # children taxa: 39\n",
      "length new taxa list: 11487\n",
      "100.0:f__UBA2987 # children taxa: 2\n",
      "length new taxa list: 11486\n",
      "100.0:f__Verruco-01; g__Verruco-01 # children taxa: 2\n",
      "length new taxa list: 11485\n",
      "100.0:f__MB11C04 # children taxa: 19\n",
      "length new taxa list: 11467\n",
      "100.0:f__Puniceicoccaceae # children taxa: 13\n",
      "length new taxa list: 11455\n",
      "100.0:f__UBA953 # children taxa: 5\n",
      "length new taxa list: 11451\n",
      "100.0:f__CAG-312; g__CAG-312 # children taxa: 4\n",
      "length new taxa list: 11448\n",
      "100.0:f__UBA9783; g__UBA9783 # children taxa: 3\n",
      "length new taxa list: 11446\n",
      "100.0:f__UBA1067 # children taxa: 21\n",
      "length new taxa list: 11426\n",
      "100.0:f__UBA1859 # children taxa: 8\n",
      "length new taxa list: 11419\n",
      "100.0:o__SS1-B-03-39; f__UBA6053 # children taxa: 2\n",
      "length new taxa list: 11418\n",
      "100.0:f__Lenti-01; g__Lenti-01 # children taxa: 3\n",
      "length new taxa list: 11416\n",
      "99.0:f__Victivallaceae # children taxa: 7\n",
      "length new taxa list: 11410\n",
      "100.0:f__UBA1829 # children taxa: 5\n",
      "length new taxa list: 11406\n",
      "100.0:f__GWF2-45-14 # children taxa: 2\n",
      "length new taxa list: 11405\n",
      "100.0:f__GWF2-50-93; g__GWF2-50-93 # children taxa: 2\n",
      "length new taxa list: 11404\n",
      "100.0:f__Lenti-02; g__Lenti-02 # children taxa: 2\n",
      "length new taxa list: 11403\n",
      "100.0:f__UBA1407 # children taxa: 2\n",
      "length new taxa list: 11402\n",
      "100.0:c__UBA3054; o__UBA3054; f__UBA3054; g__UBA3054 # children taxa: 2\n",
      "length new taxa list: 11401\n",
      "92.0:f__Parachlamydiaceae # children taxa: 10\n",
      "length new taxa list: 11392\n",
      "100.0:f__Criblamydiaceae # children taxa: 2\n",
      "length new taxa list: 11391\n",
      "99.0:f__SM23-39 # children taxa: 7\n",
      "length new taxa list: 11385\n",
      "100.0:f__Ga0074140 # children taxa: 5\n",
      "length new taxa list: 11381\n",
      "89.0:f__Simkaniaceae # children taxa: 3\n",
      "length new taxa list: 11379\n",
      "100.0:o__Chlamydiales; f__Chlamydiaceae # children taxa: 15\n",
      "length new taxa list: 11365\n",
      "100.0:f__Pirellulaceae # children taxa: 47\n",
      "length new taxa list: 11319\n",
      "100.0:f__Ga0077529 # children taxa: 4\n",
      "length new taxa list: 11316\n",
      "100.0:f__UBA7805; g__UBA7805 # children taxa: 2\n",
      "length new taxa list: 11315\n",
      "100.0:f__Thermoguttaceae # children taxa: 5\n",
      "length new taxa list: 11311\n",
      "100.0:f__UBA1268 # children taxa: 8\n",
      "length new taxa list: 11304\n",
      "100.0:o__Planctomycetales; f__Planctomycetaceae # children taxa: 18\n",
      "length new taxa list: 11287\n",
      "100.0:f__Gemmataceae # children taxa: 8\n",
      "length new taxa list: 11280\n",
      "100.0:o__Isosphaerales; f__Isosphaeraceae # children taxa: 6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "length new taxa list: 11275\n",
      "100.0:f__SM1A02 # children taxa: 44\n",
      "length new taxa list: 11232\n",
      "99.0:f__Phycisphaeraceae # children taxa: 7\n",
      "length new taxa list: 11226\n",
      "100.0:o__UBA1161; f__UBA1161 # children taxa: 3\n",
      "length new taxa list: 11224\n",
      "75.0:f__SG8-4 # children taxa: 18\n",
      "length new taxa list: 11207\n",
      "100.0:f__SM-Chi-D1 # children taxa: 3\n",
      "length new taxa list: 11205\n",
      "100.0:f__SM23-33 # children taxa: 5\n",
      "length new taxa list: 11201\n",
      "100.0:f__Brocadiaceae # children taxa: 9\n",
      "length new taxa list: 11193\n",
      "100.0:f__Scalinduaceae; g__Scalindua # children taxa: 4\n",
      "length new taxa list: 11190\n",
      "100.0:f__2-02-FULL-50-16-A # children taxa: 2\n",
      "length new taxa list: 11189\n",
      "100.0:f__UBA1135 # children taxa: 6\n",
      "length new taxa list: 11184\n",
      "100.0:f__GCA-002686595 # children taxa: 6\n",
      "length new taxa list: 11179\n",
      "100.0:f__UBA2386 # children taxa: 3\n",
      "length new taxa list: 11177\n",
      "100.0:f__GCA-002748355 # children taxa: 2\n",
      "length new taxa list: 11176\n",
      "100.0:c__GCA-002687715; o__GCA-002687715; f__GCA-002687715 # children taxa: 2\n",
      "length new taxa list: 11175\n",
      "100.0:o__UBA1146; f__UBA1146 # children taxa: 5\n",
      "length new taxa list: 11171\n",
      "100.0:o__UBA8108; f__UBA8108; g__UBA8108 # children taxa: 4\n",
      "length new taxa list: 11168\n",
      "100.0:o__UBA11346; f__UBA11346 # children taxa: 2\n",
      "length new taxa list: 11167\n",
      "100.0:c__SZUA-567; o__SZUA-567; f__SZUA-567 # children taxa: 2\n",
      "length new taxa list: 11166\n",
      "100.0:f__Treponemataceae # children taxa: 65\n",
      "length new taxa list: 11102\n",
      "100.0:f__Treponemataceae_B # children taxa: 7\n",
      "length new taxa list: 11096\n",
      "99.0:f__UBA8932 # children taxa: 10\n",
      "length new taxa list: 11087\n",
      "100.0:f__UBA12059 # children taxa: 5\n",
      "length new taxa list: 11083\n",
      "100.0:f__Spirochaetaceae_B # children taxa: 9\n",
      "length new taxa list: 11075\n",
      "98.0:f__Alkalispirochaetaceae # children taxa: 4\n",
      "length new taxa list: 11072\n",
      "100.0:f__Salinispiraceae # children taxa: 2\n",
      "length new taxa list: 11071\n",
      "100.0:f__Spirochaetaceae_A; g__Spirochaeta_A # children taxa: 2\n",
      "length new taxa list: 11070\n",
      "100.0:f__UBA9216; g__UBA9216 # children taxa: 2\n",
      "length new taxa list: 11069\n",
      "100.0:f__Marispirochaetaceae; g__Marispirochaeta # children taxa: 2\n",
      "length new taxa list: 11068\n",
      "100.0:f__Sediminispirochaetaceae; g__Sediminispirochaeta # children taxa: 2\n",
      "length new taxa list: 11067\n",
      "97.0:f__RBG-16-67-19 # children taxa: 4\n",
      "length new taxa list: 11064\n",
      "100.0:f__Sphaerochaetaceae # children taxa: 21\n",
      "length new taxa list: 11044\n",
      "100.0:o__Borreliales; f__Borreliaceae # children taxa: 20\n",
      "length new taxa list: 11025\n",
      "100.0:c__GWE2-31-10; o__GWE2-31-10; f__GWE2-31-10 # children taxa: 2\n",
      "length new taxa list: 11024\n",
      "100.0:f__GWF1-51-8; g__GWF1-51-8 # children taxa: 2\n",
      "length new taxa list: 11023\n",
      "100.0:c__Brachyspirae; o__Brachyspirales; f__Brachyspiraceae; g__Brachyspira # children taxa: 12\n",
      "length new taxa list: 11012\n",
      "100.0:f__Leptospiraceae # children taxa: 42\n",
      "length new taxa list: 10971\n",
      "100.0:f__Leptonemataceae # children taxa: 4\n",
      "length new taxa list: 10968\n",
      "100.0:f__UBA5550 # children taxa: 7\n",
      "length new taxa list: 10962\n",
      "100.0:p__UBP7; c__UBA6624; o__UBA6624; f__UBA6624; g__UBA6624 # children taxa: 2\n",
      "length new taxa list: 10961\n",
      "100.0:f__Flavobacteriaceae # children taxa: 675\n",
      "length new taxa list: 10287\n",
      "100.0:f__Weeksellaceae # children taxa: 171\n",
      "length new taxa list: 10117\n",
      "100.0:f__Blattabacteriaceae # children taxa: 14\n",
      "length new taxa list: 10104\n",
      "100.0:f__UBA1820; g__UBA1820 # children taxa: 2\n",
      "length new taxa list: 10103\n",
      "100.0:f__Cryomorphaceae # children taxa: 26\n",
      "length new taxa list: 10078\n",
      "77.0:f__TMED113 # children taxa: 13\n",
      "length new taxa list: 10066\n",
      "100.0:f__UBA7430 # children taxa: 6\n",
      "length new taxa list: 10061\n",
      "97.0:f__BACL11 # children taxa: 23\n",
      "length new taxa list: 10039\n",
      "100.0:f__Crocinitomicaceae # children taxa: 35\n",
      "length new taxa list: 10005\n",
      "100.0:f__UBA10066 # children taxa: 11\n",
      "length new taxa list: 9995\n",
      "100.0:f__UA16 # children taxa: 32\n",
      "length new taxa list: 9964\n",
      "100.0:f__PHOS-HE28; g__PHOS-HE28 # children taxa: 4\n",
      "length new taxa list: 9961\n",
      "96.0:f__1G12 # children taxa: 11\n",
      "length new taxa list: 9951\n",
      "100.0:f__GCA-002722245; g__GCA-002722245 # children taxa: 3\n",
      "length new taxa list: 9949\n",
      "100.0:f__koll-22 # children taxa: 12\n",
      "length new taxa list: 9938\n",
      "100.0:f__NORP154 # children taxa: 2\n",
      "length new taxa list: 9937\n",
      "91.0:f__Palsa-965 # children taxa: 3\n",
      "length new taxa list: 9935\n",
      "100.0:f__2-12-FULL-35-15; g__2-12-FULL-35-15 # children taxa: 2\n",
      "length new taxa list: 9934\n",
      "100.0:f__b-17BO # children taxa: 5\n",
      "length new taxa list: 9930\n",
      "100.0:f__OLB10; g__OLB10 # children taxa: 2\n",
      "length new taxa list: 9929\n",
      "100.0:f__2013-40CM-41-45 # children taxa: 2\n",
      "length new taxa list: 9928\n",
      "100.0:o__Sphingobacteriales; f__Sphingobacteriaceae # children taxa: 145\n",
      "length new taxa list: 9784\n",
      "100.0:f__UBA9320 # children taxa: 9\n",
      "length new taxa list: 9776\n",
      "100.0:f__UBA955 # children taxa: 5\n",
      "length new taxa list: 9772\n",
      "100.0:f__UKL13-3 # children taxa: 3\n",
      "length new taxa list: 9770\n",
      "100.0:f__Spirosomaceae # children taxa: 58\n",
      "length new taxa list: 9713\n",
      "100.0:f__Microscillaceae # children taxa: 5\n",
      "length new taxa list: 9709\n",
      "100.0:f__Thermonemataceae # children taxa: 3\n",
      "length new taxa list: 9707\n",
      "100.0:f__Hymenobacteraceae # children taxa: 50\n",
      "length new taxa list: 9658\n",
      "97.0:f__Cytophagaceae # children taxa: 4\n",
      "length new taxa list: 9655\n",
      "99.0:f__Catalimonadaceae # children taxa: 2\n",
      "length new taxa list: 9654\n",
      "89.0:f__Cyclobacteriaceae # children taxa: 102\n",
      "length new taxa list: 9553\n",
      "100.0:f__Flammeovirgaceae # children taxa: 7\n",
      "length new taxa list: 9547\n",
      "99.0:f__Bernardetiaceae # children taxa: 5\n",
      "length new taxa list: 9543\n",
      "100.0:f__Marinoscillaceae # children taxa: 7\n",
      "length new taxa list: 9537\n",
      "100.0:f__Amoebophilaceae # children taxa: 6\n",
      "length new taxa list: 9532\n",
      "100.0:f__Chitinophagaceae # children taxa: 132\n",
      "length new taxa list: 9401\n",
      "100.0:f__Saprospiraceae # children taxa: 24\n",
      "length new taxa list: 9378\n",
      "100.0:f__UBA2359; g__UBA2359 # children taxa: 3\n",
      "length new taxa list: 9376\n",
      "100.0:f__LD1 # children taxa: 5\n",
      "length new taxa list: 9372\n",
      "100.0:f__UBA8649; g__UBA8649 # children taxa: 3\n",
      "length new taxa list: 9370\n",
      "100.0:f__BACL12 # children taxa: 3\n",
      "length new taxa list: 9368\n",
      "100.0:f__Salinibacteraceae # children taxa: 20\n",
      "length new taxa list: 9349\n",
      "100.0:f__Rhodothermaceae; g__Rhodothermus # children taxa: 3\n",
      "length new taxa list: 9347\n",
      "100.0:f__UBA10348 # children taxa: 2\n",
      "length new taxa list: 9346\n",
      "100.0:f__Rubricoccaceae # children taxa: 3\n",
      "length new taxa list: 9344\n",
      "100.0:f__Balneolaceae # children taxa: 23\n",
      "length new taxa list: 9322\n",
      "100.0:f__HLUCCA01 # children taxa: 2\n",
      "length new taxa list: 9321\n",
      "100.0:f__Chlorobiaceae # children taxa: 19\n",
      "length new taxa list: 9303\n",
      "92.0:f__Chloroherpetonaceae # children taxa: 4\n",
      "length new taxa list: 9300\n",
      "96.0:f__Ignavibacteriaceae # children taxa: 15\n",
      "length new taxa list: 9286\n",
      "100.0:f__Melioribacteraceae # children taxa: 12\n",
      "length new taxa list: 9275\n",
      "100.0:f__B-1AR # children taxa: 6\n",
      "length new taxa list: 9270\n",
      "100.0:f__OLB5; g__OLB5 # children taxa: 2\n",
      "length new taxa list: 9269\n",
      "100.0:f__UBA6906 # children taxa: 11\n",
      "length new taxa list: 9259\n",
      "100.0:f__UBA10030 # children taxa: 9\n",
      "length new taxa list: 9251\n",
      "100.0:c__Kryptonia; o__Kryptoniales; f__Kryptoniaceae # children taxa: 6\n",
      "length new taxa list: 9246\n",
      "100.0:f__Kapabacteriaceae # children taxa: 5\n",
      "length new taxa list: 9242\n",
      "100.0:f__NICIL-2 # children taxa: 2\n",
      "length new taxa list: 9241\n",
      "100.0:f__UBA961 # children taxa: 2\n",
      "length new taxa list: 9240\n",
      "100.0:f__UBA2268 # children taxa: 2\n",
      "length new taxa list: 9239\n",
      "65.0:f__GCA-002839825 # children taxa: 2\n",
      "length new taxa list: 9238\n",
      "100.0:o__Palsa-1295; f__Palsa-1295 # children taxa: 2\n",
      "length new taxa list: 9237\n",
      "100.0:f__SZUA-365 # children taxa: 2\n",
      "length new taxa list: 9236\n",
      "100.0:o__AABM5-25-91; f__AABM5-25-91 # children taxa: 2\n",
      "length new taxa list: 9235\n",
      "79.0:f__TCS55 # children taxa: 21\n",
      "length new taxa list: 9215\n",
      "100.0:f__UBA1611 # children taxa: 14\n",
      "length new taxa list: 9202\n",
      "91.0:f__UBA8229 # children taxa: 5\n",
      "length new taxa list: 9198\n",
      "100.0:f__Marinisomataceae; g__Marinisoma # children taxa: 6\n",
      "length new taxa list: 9193\n",
      "100.0:f__S15-B10 # children taxa: 9\n",
      "length new taxa list: 9185\n",
      "100.0:f__UBA2128; g__UBA2128 # children taxa: 2\n",
      "length new taxa list: 9184\n",
      "100.0:f__GCA-002707645 # children taxa: 7\n",
      "length new taxa list: 9178\n",
      "100.0:f__SCGC-AAA003-L08; g__SCGC-AAA003-L08 # children taxa: 2\n",
      "length new taxa list: 9177\n",
      "100.0:f__TMED161; g__TMED161 # children taxa: 4\n",
      "length new taxa list: 9174\n",
      "100.0:f__GCA-002719095; g__GCA-002719095 # children taxa: 2\n",
      "length new taxa list: 9173\n",
      "100.0:f__TMED6; g__TMED6 # children taxa: 4\n",
      "length new taxa list: 9170\n",
      "100.0:f__GCA-002704045; g__GCA-002704045 # children taxa: 2\n",
      "length new taxa list: 9169\n",
      "100.0:f__46-47; g__46-47 # children taxa: 2\n",
      "length new taxa list: 9168\n",
      "100.0:c__UBA2242; o__UBA2242; f__UBA2242 # children taxa: 3\n",
      "length new taxa list: 9166\n",
      "100.0:p__Delongbacteria; c__UBA4055; o__UBA4055; f__UBA4055 # children taxa: 4\n",
      "length new taxa list: 9163\n",
      "100.0:o__Fibrobacterales; f__Fibrobacteraceae # children taxa: 36\n",
      "length new taxa list: 9128\n",
      "99.0:f__Chitinispirillaceae # children taxa: 3\n",
      "length new taxa list: 9126\n",
      "100.0:f__Cloacimonadaceae # children taxa: 16\n",
      "length new taxa list: 9111\n",
      "100.0:f__TCS61 # children taxa: 3\n",
      "length new taxa list: 9109\n",
      "100.0:o__JGIOTU-2; f__JGIOTU-2 # children taxa: 3\n",
      "length new taxa list: 9107\n",
      "100.0:f__GWC2-71-9 # children taxa: 35\n",
      "length new taxa list: 9073\n",
      "100.0:f__Gemmatimonadaceae # children taxa: 21\n",
      "length new taxa list: 9053\n",
      "100.0:o__SG8-23; f__UBA6960 # children taxa: 18\n",
      "length new taxa list: 9036\n",
      "100.0:o__LZORAL124-64-63; f__LZORAL124-64-63 # children taxa: 2\n",
      "length new taxa list: 9035\n",
      "100.0:o__SSS58A; f__SSS58A # children taxa: 3\n",
      "length new taxa list: 9033\n",
      "51.0:f__EtOH8 # children taxa: 3\n",
      "length new taxa list: 9031\n",
      "100.0:f__UBA2968 # children taxa: 3\n",
      "length new taxa list: 9029\n",
      "100.0:f__UBA8231 # children taxa: 3\n",
      "length new taxa list: 9027\n",
      "100.0:f__GCA-002724215 # children taxa: 2\n",
      "length new taxa list: 9026\n",
      "100.0:f__UBA10806; g__UBA10806 # children taxa: 2\n",
      "length new taxa list: 9025\n",
      "100.0:f__SM1-73 # children taxa: 2\n",
      "length new taxa list: 9024\n",
      "100.0:o__UBA1063; f__UBA1063; g__UBA1063 # children taxa: 2\n",
      "length new taxa list: 9023\n",
      "100.0:p__Calescibacterota; c__Calescibacteriia; o__Calescibacteriales; f__Calescibacteriaceae; g__Calescibacterium # children taxa: 2\n",
      "length new taxa list: 9022\n",
      "100.0:o__SM23-42; f__SM23-42 # children taxa: 4\n",
      "length new taxa list: 9019\n",
      "100.0:o__UBA2258; f__UBA2258 # children taxa: 2\n",
      "length new taxa list: 9018\n",
      "100.0:c__UBA3073; o__UBA3073; f__UBA3073; g__UBA3073 # children taxa: 2\n",
      "length new taxa list: 9017\n",
      "100.0:c__32-111; o__32-111; f__32-111 # children taxa: 3\n",
      "length new taxa list: 9015\n",
      "100.0:p__Fermentibacterota; c__Fermentibacteria; o__Fermentibacterales; f__Fermentibacteraceae # children taxa: 6\n",
      "length new taxa list: 9010\n",
      "100.0:f__UBA9662 # children taxa: 6\n",
      "length new taxa list: 9005\n",
      "100.0:f__UBA2224 # children taxa: 2\n",
      "length new taxa list: 9004\n",
      "100.0:o__Sumerlaeales; f__Sumerlaeaceae # children taxa: 2\n",
      "length new taxa list: 9003\n",
      "100.0:f__Pseudomonadaceae # children taxa: 514\n",
      "length new taxa list: 8490\n",
      "100.0:f__Endozoicomonadaceae # children taxa: 11\n",
      "length new taxa list: 8480\n",
      "100.0:f__Halomonadaceae # children taxa: 125\n",
      "length new taxa list: 8356\n",
      "99.0:f__Saccharospirillaceae # children taxa: 25\n",
      "length new taxa list: 8332\n",
      "100.0:f__Marinomonadaceae; g__Marinomonas # children taxa: 20\n",
      "length new taxa list: 8313\n",
      "96.0:f__Nitrincolaceae # children taxa: 38\n",
      "length new taxa list: 8276\n",
      "100.0:f__Oleiphilaceae # children taxa: 66\n",
      "length new taxa list: 8211\n",
      "100.0:f__Porticoccaceae # children taxa: 46\n",
      "length new taxa list: 8166\n",
      "100.0:f__Cellvibrionaceae # children taxa: 50\n",
      "length new taxa list: 8117\n",
      "100.0:f__Halieaceae # children taxa: 37\n",
      "length new taxa list: 8081\n",
      "100.0:f__Spongiibacteraceae # children taxa: 12\n",
      "length new taxa list: 8070\n",
      "100.0:f__UBA7434; g__UBA7434 # children taxa: 5\n",
      "length new taxa list: 8066\n",
      "100.0:f__HTCC2089 # children taxa: 48\n",
      "length new taxa list: 8019\n",
      "100.0:f__Pseudohongiellaceae # children taxa: 36\n",
      "length new taxa list: 7984\n",
      "100.0:f__Litoricolaceae; g__Litoricola # children taxa: 9\n",
      "length new taxa list: 7976\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "100.0:f__Moraxellaceae # children taxa: 190\n",
      "length new taxa list: 7787\n",
      "100.0:f__Alcanivoracaceae # children taxa: 20\n",
      "length new taxa list: 7768\n",
      "100.0:f__Ketobacteraceae # children taxa: 9\n",
      "length new taxa list: 7760\n",
      "100.0:f__UBA3067 # children taxa: 3\n",
      "length new taxa list: 7758\n",
      "78.0:f__Enterobacteriaceae # children taxa: 509\n",
      "length new taxa list: 7250\n",
      "100.0:f__Pasteurellaceae # children taxa: 89\n",
      "length new taxa list: 7162\n",
      "100.0:f__Vibrionaceae # children taxa: 205\n",
      "length new taxa list: 6958\n",
      "84.0:f__Aeromonadaceae # children taxa: 50\n",
      "length new taxa list: 6909\n",
      "51.0:f__Alteromonadaceae # children taxa: 214\n",
      "length new taxa list: 6696\n",
      "100.0:f__Psychromonadaceae # children taxa: 18\n",
      "length new taxa list: 6679\n",
      "100.0:f__Psychrobiaceae; g__Psychrobium # children taxa: 2\n",
      "length new taxa list: 6678\n",
      "100.0:f__Psychromonadaceae_A # children taxa: 4\n",
      "length new taxa list: 6675\n",
      "100.0:f__Shewanellaceae # children taxa: 55\n",
      "length new taxa list: 6621\n",
      "100.0:f__Gallaecimonadaceae; g__Gallaecimonas # children taxa: 2\n",
      "length new taxa list: 6620\n",
      "100.0:f__Succinivibrionaceae # children taxa: 24\n",
      "length new taxa list: 6597\n",
      "96.0:f__Kangiellaceae # children taxa: 9\n",
      "length new taxa list: 6589\n",
      "97.0:f__GCA-002770795 # children taxa: 2\n",
      "length new taxa list: 6588\n",
      "88.0:o__Legionellales; f__Legionellaceae # children taxa: 67\n",
      "length new taxa list: 6522\n",
      "100.0:o__UBA1113; f__UBA1113 # children taxa: 2\n",
      "length new taxa list: 6521\n",
      "100.0:o__UBA5158; f__UBA5158 # children taxa: 9\n",
      "length new taxa list: 6513\n",
      "100.0:o__Diplorickettsiales; f__Diplorickettsiaceae # children taxa: 5\n",
      "length new taxa list: 6509\n",
      "86.0:o__UBA12402; f__UBA12402 # children taxa: 6\n",
      "length new taxa list: 6504\n",
      "100.0:o__Coxiellales; f__Coxiellaceae # children taxa: 13\n",
      "length new taxa list: 6492\n",
      "100.0:o__UBA9339; f__UBA9339 # children taxa: 3\n",
      "length new taxa list: 6490\n",
      "100.0:o__Francisellales; f__Francisellaceae # children taxa: 17\n",
      "length new taxa list: 6474\n",
      "100.0:o__Evansiales; f__Evansiaceae # children taxa: 2\n",
      "length new taxa list: 6473\n",
      "100.0:f__Berkiellaceae # children taxa: 3\n",
      "length new taxa list: 6471\n",
      "100.0:o__Piscirickettsiales; f__Piscirickettsiaceae; g__Piscirickettsia # children taxa: 3\n",
      "length new taxa list: 6469\n",
      "100.0:f__Thiomicrospiraceae # children taxa: 22\n",
      "length new taxa list: 6448\n",
      "100.0:f__Thioglobaceae # children taxa: 19\n",
      "length new taxa list: 6430\n",
      "100.0:f__Wohlfahrtiimonadaceae # children taxa: 9\n",
      "length new taxa list: 6422\n",
      "100.0:f__Cardiobacteriaceae # children taxa: 5\n",
      "length new taxa list: 6418\n",
      "100.0:f__Xanthomonadaceae # children taxa: 194\n",
      "length new taxa list: 6225\n",
      "100.0:f__Rhodanobacteraceae # children taxa: 63\n",
      "length new taxa list: 6163\n",
      "89.0:f__Ahniellaceae # children taxa: 4\n",
      "length new taxa list: 6160\n",
      "100.0:f__Wenzhouxiangellaceae # children taxa: 2\n",
      "length new taxa list: 6159\n",
      "100.0:f__SZUA-36 # children taxa: 2\n",
      "length new taxa list: 6158\n",
      "100.0:f__Marinicellaceae; g__Marinicella # children taxa: 2\n",
      "length new taxa list: 6157\n",
      "100.0:o__Steroidobacterales; f__Steroidobacteraceae # children taxa: 17\n",
      "length new taxa list: 6141\n",
      "100.0:o__Woeseiales; f__Woeseiaceae # children taxa: 8\n",
      "length new taxa list: 6134\n",
      "100.0:o__GCA-2729495; f__GCA-2729495 # children taxa: 3\n",
      "length new taxa list: 6132\n",
      "100.0:o__UBA6522; f__UBA6522 # children taxa: 5\n",
      "length new taxa list: 6128\n",
      "100.0:f__Nevskiaceae # children taxa: 14\n",
      "length new taxa list: 6115\n",
      "100.0:f__Salinisphaeraceae # children taxa: 10\n",
      "length new taxa list: 6106\n",
      "100.0:f__Oceanococcaceae # children taxa: 2\n",
      "length new taxa list: 6105\n",
      "100.0:o__UBA5335; f__UBA5335; g__UBA5335 # children taxa: 3\n",
      "length new taxa list: 6103\n",
      "100.0:f__Nitrococcaceae # children taxa: 6\n",
      "length new taxa list: 6098\n",
      "100.0:f__Halorhodospiraceae # children taxa: 3\n",
      "length new taxa list: 6096\n",
      "100.0:f__Thioalkalivibrionaceae # children taxa: 19\n",
      "length new taxa list: 6078\n",
      "99.0:f__Ectothiorhodospiraceae # children taxa: 12\n",
      "length new taxa list: 6067\n",
      "100.0:f__Acidihalobacteraceae # children taxa: 4\n",
      "length new taxa list: 6064\n",
      "100.0:o__Thiotrichales; f__Thiotrichaceae # children taxa: 13\n",
      "length new taxa list: 6052\n",
      "100.0:o__GCA-001735895; f__GCA-001735895; g__GCA-001735895 # children taxa: 2\n",
      "length new taxa list: 6051\n",
      "100.0:o__UBA4575; f__UBA4575 # children taxa: 2\n",
      "length new taxa list: 6050\n",
      "100.0:o__Halothiobacillales; f__Halothiobacillaceae # children taxa: 6\n",
      "length new taxa list: 6045\n",
      "98.0:o__Granulosicoccales; f__Granulosicoccaceae # children taxa: 5\n",
      "length new taxa list: 6041\n",
      "100.0:o__SZUA-229; f__SZUA-229 # children taxa: 4\n",
      "length new taxa list: 6038\n",
      "100.0:f__Methylomonadaceae # children taxa: 46\n",
      "length new taxa list: 5993\n",
      "100.0:f__UBA2778 # children taxa: 2\n",
      "length new taxa list: 5992\n",
      "48.0:f__Cycloclasticaceae # children taxa: 12\n",
      "length new taxa list: 5981\n",
      "100.0:f__Methylococcaceae # children taxa: 8\n",
      "length new taxa list: 5974\n",
      "100.0:f__Methylothermaceae; g__Methylohalobius # children taxa: 2\n",
      "length new taxa list: 5973\n",
      "81.0:f__Sedimenticolaceae # children taxa: 19\n",
      "length new taxa list: 5955\n",
      "100.0:f__Chromatiaceae # children taxa: 21\n",
      "length new taxa list: 5935\n",
      "100.0:f__Methylophagaceae # children taxa: 20\n",
      "length new taxa list: 5916\n",
      "100.0:f__Nitrosococcaceae # children taxa: 4\n",
      "length new taxa list: 5913\n",
      "100.0:o__Beggiatoales; f__Beggiatoaceae # children taxa: 8\n",
      "length new taxa list: 5906\n",
      "100.0:f__Thiohalomonadaceae # children taxa: 5\n",
      "length new taxa list: 5902\n",
      "72.0:f__SG8-11 # children taxa: 2\n",
      "length new taxa list: 5901\n",
      "100.0:f__UBA6429 # children taxa: 3\n",
      "length new taxa list: 5899\n",
      "100.0:o__SZUA-140; f__SZUA-140 # children taxa: 3\n",
      "length new taxa list: 5897\n",
      "100.0:o__Ga0077554; f__Ga007554; g__Ga0077554 # children taxa: 2\n",
      "length new taxa list: 5896\n",
      "100.0:o__Tenderiales; f__Tenderiaceae # children taxa: 2\n",
      "length new taxa list: 5895\n",
      "100.0:o__Competibacterales; f__Competibacteraceae # children taxa: 10\n",
      "length new taxa list: 5886\n",
      "100.0:o__21-64-14; f__21-64-14 # children taxa: 2\n",
      "length new taxa list: 5885\n",
      "99.0:o__Thiohalobacterales; f__Thiohalobacteraceae # children taxa: 4\n",
      "length new taxa list: 5882\n",
      "100.0:o__SZUA-21; f__SZUA-21 # children taxa: 2\n",
      "length new taxa list: 5881\n",
      "100.0:o__UBA4486; f__UBA4486 # children taxa: 11\n",
      "length new taxa list: 5871\n",
      "42.0:o__Ga0077536; f__Ga0077536 # children taxa: 4\n",
      "length new taxa list: 5868\n",
      "60.0:o__UBA7966; f__UBA7966 # children taxa: 3\n",
      "length new taxa list: 5866\n",
      "100.0:o__Immundisolibacterales; f__Immundisolibacteraceae; g__Immundisolibacter # children taxa: 2\n",
      "length new taxa list: 5865\n",
      "99.0:f__Burkholderiaceae # children taxa: 856\n",
      "length new taxa list: 5010\n",
      "69.0:f__Rhodocyclaceae # children taxa: 95\n",
      "length new taxa list: 4916\n",
      "100.0:f__SG8-39 # children taxa: 13\n",
      "length new taxa list: 4904\n",
      "100.0:f__UKL13-2 # children taxa: 3\n",
      "length new taxa list: 4902\n",
      "100.0:f__Palsa-1005 # children taxa: 3\n",
      "length new taxa list: 4900\n",
      "100.0:f__Neisseriaceae # children taxa: 70\n",
      "length new taxa list: 4831\n",
      "100.0:f__UBA11063; g__UBA11063 # children taxa: 2\n",
      "length new taxa list: 4830\n",
      "100.0:f__Chromobacteriaceae # children taxa: 26\n",
      "length new taxa list: 4805\n",
      "100.0:f__Aquaspirillaceae # children taxa: 5\n",
      "length new taxa list: 4801\n",
      "100.0:f__Chitinibacteraceae # children taxa: 13\n",
      "length new taxa list: 4789\n",
      "100.0:f__Chitinimonadaceae; g__Chitinimonas # children taxa: 3\n",
      "length new taxa list: 4787\n",
      "97.0:f__Methylophilaceae # children taxa: 45\n",
      "length new taxa list: 4743\n",
      "100.0:f__Gallionellaceae # children taxa: 25\n",
      "length new taxa list: 4719\n",
      "100.0:f__Hydrogenophilaceae # children taxa: 27\n",
      "length new taxa list: 4693\n",
      "100.0:f__Sulfuriferulaceae # children taxa: 3\n",
      "length new taxa list: 4691\n",
      "42.0:f__Sulfuricellaceae # children taxa: 7\n",
      "length new taxa list: 4685\n",
      "100.0:f__Nitrosomonadaceae # children taxa: 51\n",
      "length new taxa list: 4635\n",
      "100.0:f__SG8-41 # children taxa: 8\n",
      "length new taxa list: 4628\n",
      "100.0:f__Ga0077523 # children taxa: 3\n",
      "length new taxa list: 4626\n",
      "100.0:f__Ferrovaceae # children taxa: 3\n",
      "length new taxa list: 4624\n",
      "100.0:f__SG8-40 # children taxa: 6\n",
      "length new taxa list: 4619\n",
      "100.0:f__2013Ark19i; g__2013Ark19i # children taxa: 2\n",
      "length new taxa list: 4618\n",
      "100.0:o__Acidithiobacillales; f__Acidithiobacillaceae # children taxa: 9\n",
      "length new taxa list: 4610\n",
      "100.0:f__LS-SOB # children taxa: 19\n",
      "length new taxa list: 4592\n",
      "100.0:f__UBA7415 # children taxa: 2\n",
      "length new taxa list: 4591\n",
      "100.0:f__Sulfurifustaceae # children taxa: 6\n",
      "length new taxa list: 4586\n",
      "100.0:f__Acidiferrobacteraceae; g__Acidiferrobacter # children taxa: 2\n",
      "length new taxa list: 4585\n",
      "100.0:f__SPGG2 # children taxa: 2\n",
      "length new taxa list: 4584\n",
      "100.0:f__D2472 # children taxa: 14\n",
      "length new taxa list: 4571\n",
      "100.0:f__TMED112 # children taxa: 10\n",
      "length new taxa list: 4562\n",
      "100.0:f__SAR86 # children taxa: 4\n",
      "length new taxa list: 4559\n",
      "100.0:f__AG-339-G14 # children taxa: 4\n",
      "length new taxa list: 4556\n",
      "100.0:f__GCA-002716945 # children taxa: 6\n",
      "length new taxa list: 4551\n",
      "100.0:f__GCA-002705445 # children taxa: 2\n",
      "length new taxa list: 4550\n",
      "100.0:o__UBA11654; f__UBA11654; g__UBA11654 # children taxa: 6\n",
      "length new taxa list: 4545\n",
      "100.0:f__UBA7916; g__UBA7916 # children taxa: 3\n",
      "length new taxa list: 4543\n",
      "100.0:f__GCA-002691585; g__GCA-002691585 # children taxa: 2\n",
      "length new taxa list: 4542\n",
      "100.0:o__Mariprofundales; f__Mariprofundaceae # children taxa: 17\n",
      "length new taxa list: 4526\n",
      "100.0:f__Parvibaculaceae # children taxa: 12\n",
      "length new taxa list: 4515\n",
      "100.0:f__RS24 # children taxa: 10\n",
      "length new taxa list: 4506\n",
      "100.0:f__Caulobacteraceae # children taxa: 115\n",
      "length new taxa list: 4392\n",
      "91.0:f__Hyphomonadaceae # children taxa: 55\n",
      "length new taxa list: 4338\n",
      "100.0:f__Maricaulaceae # children taxa: 27\n",
      "length new taxa list: 4312\n",
      "100.0:f__Parvularculaceae # children taxa: 9\n",
      "length new taxa list: 4304\n",
      "100.0:o__UBA1301; f__UBA1301 # children taxa: 10\n",
      "length new taxa list: 4295\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "100.0:f__Sphingomonadaceae # children taxa: 398\n",
      "length new taxa list: 3898\n",
      "100.0:f__Kordiimonadaceae # children taxa: 5\n",
      "length new taxa list: 3894\n",
      "100.0:f__Emcibacteraceae # children taxa: 8\n",
      "length new taxa list: 3887\n",
      "100.0:o__Ferrovibrionales; f__Ferrovibrionaceae; g__Ferrovibrio # children taxa: 2\n",
      "length new taxa list: 3886\n",
      "100.0:o__Zavarziniales; f__Zavarziniaceae; g__Zavarzinia # children taxa: 2\n",
      "length new taxa list: 3885\n",
      "100.0:o__Sneathiellales; f__Sneathiellaceae # children taxa: 7\n",
      "length new taxa list: 3879\n",
      "100.0:o__UBA2966; f__UBA2966 # children taxa: 2\n",
      "length new taxa list: 3878\n",
      "100.0:f__UBA7887 # children taxa: 5\n",
      "length new taxa list: 3874\n",
      "100.0:o__UBA828; f__UBA828 # children taxa: 5\n",
      "length new taxa list: 3870\n",
      "100.0:o__Geminicoccales; f__Geminicoccaceae # children taxa: 4\n",
      "length new taxa list: 3867\n",
      "100.0:o__SP197; f__SP197; g__SP197 # children taxa: 2\n",
      "length new taxa list: 3866\n",
      "100.0:o__Acetobacterales; f__Acetobacteraceae # children taxa: 114\n",
      "length new taxa list: 3753\n",
      "100.0:o__Reyranellales; f__Reyranellaceae # children taxa: 8\n",
      "length new taxa list: 3746\n",
      "98.0:f__Micavibrionaceae # children taxa: 38\n",
      "length new taxa list: 3709\n",
      "78.0:f__GCA-2720935 # children taxa: 2\n",
      "length new taxa list: 3708\n",
      "100.0:f__UBA2020; g__UBA2020 # children taxa: 3\n",
      "length new taxa list: 3706\n",
      "100.0:f__Azospirillaceae # children taxa: 30\n",
      "length new taxa list: 3677\n",
      "100.0:f__Inquilinaceae; g__Inquilinus # children taxa: 2\n",
      "length new taxa list: 3676\n",
      "100.0:f__Elsteraceae; g__Elstera # children taxa: 2\n",
      "length new taxa list: 3675\n",
      "86.0:o__Nisaeales; f__Nisaeaceae # children taxa: 10\n",
      "length new taxa list: 3666\n",
      "100.0:o__TMED26; f__TMED26; g__TMED26 # children taxa: 2\n",
      "length new taxa list: 3665\n",
      "100.0:o__Thalassobaculales; f__Thalassobaculaceae # children taxa: 4\n",
      "length new taxa list: 3662\n",
      "100.0:o__Oceanibaculales; f__Oceanibaculaceae; g__Oceanibaculum # children taxa: 3\n",
      "length new taxa list: 3660\n",
      "100.0:f__Kiloniellaceae # children taxa: 6\n",
      "length new taxa list: 3655\n",
      "100.0:f__Fodinicurvataceae # children taxa: 5\n",
      "length new taxa list: 3651\n",
      "100.0:o__UBA7985; f__UBA7985; g__UBA7985 # children taxa: 5\n",
      "length new taxa list: 3647\n",
      "100.0:f__UBA8366 # children taxa: 8\n",
      "length new taxa list: 3640\n",
      "100.0:f__GCA-2717185 # children taxa: 2\n",
      "length new taxa list: 3639\n",
      "100.0:f__GCA-2696645; g__GCA-2696645 # children taxa: 2\n",
      "length new taxa list: 3638\n",
      "100.0:f__UBA2964 # children taxa: 7\n",
      "length new taxa list: 3632\n",
      "100.0:f__UBA830 # children taxa: 3\n",
      "length new taxa list: 3630\n",
      "100.0:o__UBA6615; f__UBA6615 # children taxa: 6\n",
      "length new taxa list: 3625\n",
      "100.0:o__Bin65; f__Bin65; g__Bin65 # children taxa: 2\n",
      "length new taxa list: 3624\n",
      "91.0:o__UBA11136; f__UBA11136 # children taxa: 2\n",
      "length new taxa list: 3623\n",
      "97.0:f__Casp-alpha2 # children taxa: 5\n",
      "length new taxa list: 3619\n",
      "100.0:f__UBA3470 # children taxa: 4\n",
      "length new taxa list: 3616\n",
      "100.0:f__UBA1479; g__UBA1479 # children taxa: 5\n",
      "length new taxa list: 3612\n",
      "100.0:f__UBA4479; g__UBA4479 # children taxa: 2\n",
      "length new taxa list: 3611\n",
      "96.0:f__2-02-FULL-58-16 # children taxa: 2\n",
      "length new taxa list: 3610\n",
      "100.0:f__Magnetovibrionaceae; g__Magnetovibrio # children taxa: 3\n",
      "length new taxa list: 3608\n",
      "100.0:f__UBA2165 # children taxa: 8\n",
      "length new taxa list: 3601\n",
      "100.0:f__Thalassospiraceae; g__Thalassospira # children taxa: 18\n",
      "length new taxa list: 3584\n",
      "100.0:f__Terasakiellaceae; g__Terasakiella # children taxa: 2\n",
      "length new taxa list: 3583\n",
      "100.0:f__Magnetospirillaceae # children taxa: 16\n",
      "length new taxa list: 3568\n",
      "16.0:o__Rhodospirillales; f__Rhodospirillaceae # children taxa: 15\n",
      "length new taxa list: 3554\n",
      "100.0:o__Rhodospirillales_C; f__Rhodospirillaceae_A # children taxa: 2\n",
      "length new taxa list: 3553\n",
      "100.0:f__CAG-239 # children taxa: 15\n",
      "length new taxa list: 3539\n",
      "100.0:f__CAG-977 # children taxa: 6\n",
      "length new taxa list: 3534\n",
      "100.0:f__UBA3637 # children taxa: 4\n",
      "length new taxa list: 3531\n",
      "100.0:f__Caedimonadaceae # children taxa: 4\n",
      "length new taxa list: 3528\n",
      "100.0:o__UBA9655; f__UBA9655; g__UBA9655 # children taxa: 2\n",
      "length new taxa list: 3527\n",
      "100.0:f__Paracaedibacteraceae # children taxa: 3\n",
      "length new taxa list: 3525\n",
      "100.0:f__UBA11393; g__UBA11393 # children taxa: 2\n",
      "length new taxa list: 3524\n",
      "100.0:f__Holosporaceae # children taxa: 4\n",
      "length new taxa list: 3521\n",
      "100.0:f__UBA5542 # children taxa: 2\n",
      "length new taxa list: 3520\n",
      "100.0:f__UBA7879; g__UBA7879 # children taxa: 2\n",
      "length new taxa list: 3519\n",
      "100.0:f__Puniceispirillaceae # children taxa: 20\n",
      "length new taxa list: 3500\n",
      "100.0:f__UBA1172 # children taxa: 2\n",
      "length new taxa list: 3499\n",
      "100.0:f__AAA536-G10 # children taxa: 15\n",
      "length new taxa list: 3485\n",
      "100.0:f__Pelagibacteraceae # children taxa: 118\n",
      "length new taxa list: 3368\n",
      "100.0:f__AG-422-B15; g__AG-422-B15 # children taxa: 5\n",
      "length new taxa list: 3364\n",
      "100.0:f__GCA-002718135 # children taxa: 9\n",
      "length new taxa list: 3356\n",
      "100.0:f__HIMB59; g__HIMB59 # children taxa: 6\n",
      "length new taxa list: 3351\n",
      "100.0:o__GCA-002720895; f__GCA-002720895 # children taxa: 5\n",
      "length new taxa list: 3347\n",
      "100.0:o__TMED127; f__TMED127 # children taxa: 14\n",
      "length new taxa list: 3334\n",
      "100.0:f__TMED109 # children taxa: 7\n",
      "length new taxa list: 3328\n",
      "100.0:f__TMED131 # children taxa: 6\n",
      "length new taxa list: 3323\n",
      "100.0:o__GCA-2701885; f__GCA-2701885; g__GCA-2701885 # children taxa: 3\n",
      "length new taxa list: 3321\n",
      "100.0:f__Anaplasmataceae # children taxa: 26\n",
      "length new taxa list: 3296\n",
      "100.0:f__UBA1997 # children taxa: 4\n",
      "length new taxa list: 3293\n",
      "100.0:f__Rickettsiaceae # children taxa: 22\n",
      "length new taxa list: 3272\n",
      "100.0:f__Arcanobacteraceae # children taxa: 2\n",
      "length new taxa list: 3271\n",
      "99.0:f__Midichloriaceae # children taxa: 3\n",
      "length new taxa list: 3269\n",
      "100.0:f__UBA6187 # children taxa: 2\n",
      "length new taxa list: 3268\n",
      "100.0:f__UBA3002 # children taxa: 6\n",
      "length new taxa list: 3263\n",
      "100.0:f__UBA998 # children taxa: 3\n",
      "length new taxa list: 3261\n",
      "100.0:f__Rs-D84; g__Rs-D84 # children taxa: 10\n",
      "length new taxa list: 3252\n",
      "100.0:o__IN39; f__IN39; g__IN39 # children taxa: 2\n",
      "length new taxa list: 3251\n",
      "100.0:o__Rickettsiales_A; f__UBA4311; g__UBA4311 # children taxa: 2\n",
      "length new taxa list: 3250\n",
      "100.0:f__UBA2136; g__UBA2136 # children taxa: 2\n",
      "length new taxa list: 3249\n",
      "100.0:f__UBA6156 # children taxa: 3\n",
      "length new taxa list: 3247\n",
      "100.0:f__UBA12223; g__UBA12223 # children taxa: 2\n",
      "length new taxa list: 3246\n",
      "100.0:f__UBA8363 # children taxa: 4\n",
      "length new taxa list: 3243\n",
      "83.0:f__WMHbin3 # children taxa: 4\n",
      "length new taxa list: 3240\n",
      "100.0:f__DC0425bin3 # children taxa: 2\n",
      "length new taxa list: 3239\n",
      "100.0:f__Magnetococcaceae # children taxa: 2\n",
      "length new taxa list: 3238\n",
      "100.0:f__Babeliaceae # children taxa: 7\n",
      "length new taxa list: 3232\n",
      "88.0:f__Vermiphilaceae # children taxa: 4\n",
      "length new taxa list: 3229\n",
      "89.0:f__UBA12409 # children taxa: 2\n",
      "length new taxa list: 3228\n",
      "81.0:f__RVW-14 # children taxa: 5\n",
      "length new taxa list: 3224\n",
      "100.0:p__UBP6; c__UBA1177; o__UBA1177; f__UBA1177; g__UBA1177 # children taxa: 3\n",
      "length new taxa list: 3222\n",
      "100.0:f__NAC60-12 # children taxa: 16\n",
      "length new taxa list: 3207\n",
      "100.0:f__XYD2-FULL-50-16 # children taxa: 3\n",
      "length new taxa list: 3205\n",
      "100.0:f__Campylobacteraceae # children taxa: 73\n",
      "length new taxa list: 3133\n",
      "100.0:f__Sulfurospirillaceae # children taxa: 19\n",
      "length new taxa list: 3115\n",
      "100.0:f__Helicobacteraceae # children taxa: 66\n",
      "length new taxa list: 3050\n",
      "100.0:f__Arcobacteraceae # children taxa: 32\n",
      "length new taxa list: 3019\n",
      "100.0:f__Sulfurimonadaceae # children taxa: 29\n",
      "length new taxa list: 2991\n",
      "100.0:f__Sulfurovaceae # children taxa: 19\n",
      "length new taxa list: 2973\n",
      "100.0:f__Nitratiruptoraceae; g__Nitratiruptor # children taxa: 2\n",
      "length new taxa list: 2972\n",
      "100.0:o__Nautiliales; f__Nautiliaceae # children taxa: 4\n",
      "length new taxa list: 2969\n",
      "100.0:f__Hippeaceae # children taxa: 4\n",
      "length new taxa list: 2966\n",
      "100.0:f__Desulfurellaceae; g__Desulfurella # children taxa: 2\n",
      "length new taxa list: 2965\n",
      "100.0:f__Aquificaceae # children taxa: 8\n",
      "length new taxa list: 2958\n",
      "100.0:f__Hydrogenobaculaceae; g__Hydrogenobaculum # children taxa: 4\n",
      "length new taxa list: 2955\n",
      "100.0:o__Hydrogenothermales; f__Hydrogenothermaceae # children taxa: 9\n",
      "length new taxa list: 2947\n",
      "100.0:c__Desulfurobacteriia; o__Desulfurobacteriales; f__Desulfurobacteriaceae # children taxa: 5\n",
      "length new taxa list: 2943\n",
      "100.0:f__Denitrovibrionaceae # children taxa: 7\n",
      "length new taxa list: 2937\n",
      "100.0:f__Flexistipitaceae; g__Flexistipes # children taxa: 2\n",
      "length new taxa list: 2936\n",
      "100.0:f__Calditerrivibrionaceae; g__Calditerrivibrio # children taxa: 2\n",
      "length new taxa list: 2935\n",
      "100.0:p__Chrysiogenetota; c__Chrysiogenetes; o__Chrysiogenales; f__Chrysiogenaceae # children taxa: 2\n",
      "length new taxa list: 2934\n",
      "100.0:o__UBA1144; f__UBA1144 # children taxa: 9\n",
      "length new taxa list: 2926\n",
      "100.0:f__UBA2774 # children taxa: 3\n",
      "length new taxa list: 2924\n",
      "100.0:f__CSP1-2 # children taxa: 2\n",
      "length new taxa list: 2923\n",
      "99.0:f__UBA2156 # children taxa: 3\n",
      "length new taxa list: 2921\n",
      "88.0:f__BuS5 # children taxa: 4\n",
      "length new taxa list: 2918\n",
      "100.0:f__Desulfatirhabdiaceae # children taxa: 3\n",
      "length new taxa list: 2916\n",
      "100.0:f__Desulfosarcinaceae # children taxa: 8\n",
      "length new taxa list: 2909\n",
      "34.0:f__Desulfosarcinaceae_B # children taxa: 2\n",
      "length new taxa list: 2908\n",
      "100.0:f__Desulfobacteraceae # children taxa: 22\n",
      "length new taxa list: 2887\n",
      "100.0:f__Desulfococcaceae_A; g__Desulfococcus_A # children taxa: 2\n",
      "length new taxa list: 2886\n",
      "100.0:f__Desulfatibacillaceae; g__Desulfatibacillum # children taxa: 2\n",
      "length new taxa list: 2885\n",
      "59.0:f__NaphS2 # children taxa: 3\n",
      "length new taxa list: 2883\n",
      "100.0:f__Syntrophobacteraceae # children taxa: 10\n",
      "length new taxa list: 2874\n",
      "100.0:o__Desulfarculales; f__Desulfarculaceae # children taxa: 4\n",
      "length new taxa list: 2871\n",
      "100.0:f__Desulfobaccaceae # children taxa: 2\n",
      "length new taxa list: 2870\n",
      "100.0:f__Desulfocapsaceae # children taxa: 20\n",
      "length new taxa list: 2851\n",
      "100.0:f__Desulfobulbaceae # children taxa: 12\n",
      "length new taxa list: 2840\n",
      "100.0:f__BM004 # children taxa: 2\n",
      "length new taxa list: 2839\n",
      "100.0:f__Desulfurivibrionaceae # children taxa: 15\n",
      "length new taxa list: 2825\n",
      "100.0:f__Thermodesulfobacteriaceae # children taxa: 10\n",
      "length new taxa list: 2816\n",
      "100.0:f__Thermodesulfatatoraceae; g__Thermodesulfatator # children taxa: 3\n",
      "length new taxa list: 2814\n",
      "100.0:f__UBA3076; g__UBA3076 # children taxa: 3\n",
      "length new taxa list: 2812\n",
      "100.0:f__Smithellaceae # children taxa: 24\n",
      "length new taxa list: 2789\n",
      "100.0:f__Syntrophaceae; g__Syntrophus # children taxa: 2\n",
      "length new taxa list: 2788\n",
      "100.0:f__UBA5619; g__UBA5619 # children taxa: 6\n",
      "length new taxa list: 2783\n",
      "100.0:f__UBA2185 # children taxa: 4\n",
      "length new taxa list: 2780\n",
      "100.0:f__UBA2207 # children taxa: 2\n",
      "length new taxa list: 2779\n",
      "100.0:f__UBA2192; g__UBA2192 # children taxa: 2\n",
      "length new taxa list: 2778\n",
      "100.0:f__UBA2210 # children taxa: 2\n",
      "length new taxa list: 2777\n",
      "100.0:f__UBA1163 # children taxa: 5\n",
      "length new taxa list: 2773\n",
      "100.0:f__RBG-16-54-18 # children taxa: 3\n",
      "length new taxa list: 2771\n",
      "100.0:o__SM23-61; f__SM23-61; g__SM23-61 # children taxa: 2\n",
      "length new taxa list: 2770\n",
      "100.0:f__Syntrophorhabdaceae # children taxa: 11\n",
      "length new taxa list: 2760\n",
      "100.0:f__WCHB1-27 # children taxa: 2\n",
      "length new taxa list: 2759\n",
      "100.0:o__UBA1062; f__UBA1062; g__UBA1062 # children taxa: 3\n",
      "length new taxa list: 2757\n",
      "99.0:f__UBA2197; g__UBA2197 # children taxa: 3\n",
      "length new taxa list: 2755\n",
      "100.0:f__Desulfuromonadaceae_A; g__Desulfuromonas_D # children taxa: 3\n",
      "length new taxa list: 2753\n",
      "100.0:f__Pelobacteraceae_A # children taxa: 7\n",
      "length new taxa list: 2747\n",
      "100.0:f__Geothermobacteraceae # children taxa: 3\n",
      "length new taxa list: 2745\n",
      "100.0:f__SZUA-401; g__SZUA-401 # children taxa: 2\n",
      "length new taxa list: 2744\n",
      "100.0:f__UBA2294 # children taxa: 3\n",
      "length new taxa list: 2742\n",
      "100.0:f__Desulfuromonadaceae # children taxa: 6\n",
      "length new taxa list: 2737\n",
      "100.0:f__Geopsychrobacteraceae # children taxa: 9\n",
      "length new taxa list: 2729\n",
      "100.0:f__Geoalkalibacteraceae # children taxa: 2\n",
      "length new taxa list: 2728\n",
      "88.0:f__Geobacteraceae # children taxa: 19\n",
      "length new taxa list: 2710\n",
      "100.0:f__Pelobacteraceae # children taxa: 17\n",
      "length new taxa list: 2694\n",
      "100.0:f__GWC2-55-46 # children taxa: 5\n",
      "length new taxa list: 2690\n",
      "100.0:f__UBA9637 # children taxa: 2\n",
      "length new taxa list: 2689\n",
      "100.0:f__UBA6898 # children taxa: 7\n",
      "length new taxa list: 2683\n",
      "100.0:f__SM23-35 # children taxa: 3\n",
      "length new taxa list: 2681\n",
      "100.0:f__UBA1546; g__UBA1546 # children taxa: 3\n",
      "length new taxa list: 2679\n",
      "100.0:f__Thermodesulfovibrionaceae; g__Thermodesulfovibrio # children taxa: 6\n",
      "length new taxa list: 2674\n",
      "100.0:f__UBA9935 # children taxa: 5\n",
      "length new taxa list: 2670\n",
      "100.0:f__Magnetobacteriaceae # children taxa: 5\n",
      "length new taxa list: 2666\n",
      "100.0:f__JdFR-85 # children taxa: 4\n",
      "length new taxa list: 2663\n",
      "100.0:o__UBA6902; f__UBA6902 # children taxa: 5\n",
      "length new taxa list: 2659\n",
      "100.0:f__Nitrospiraceae # children taxa: 32\n",
      "length new taxa list: 2628\n",
      "100.0:f__UBA8639 # children taxa: 6\n",
      "length new taxa list: 2623\n",
      "100.0:f__UBA2166; g__UBA2166 # children taxa: 2\n",
      "length new taxa list: 2622\n",
      "100.0:o__HDB-SIOI813; f__HDB-SIOI813 # children taxa: 2\n",
      "length new taxa list: 2621\n",
      "100.0:c__UBA9217; o__UBA9217; f__UBA9217 # children taxa: 3\n",
      "length new taxa list: 2619\n",
      "100.0:o__Rokubacteriales; f__GWA2-73-35 # children taxa: 31\n",
      "length new taxa list: 2589\n",
      "100.0:f__Methylomirabilaceae; g__Methylomirabilis # children taxa: 6\n",
      "length new taxa list: 2584\n",
      "100.0:f__CSP1-5; g__CSP1-5 # children taxa: 2\n",
      "length new taxa list: 2583\n",
      "100.0:p__Moduliflexota; c__Moduliflexia; o__Moduliflexales; f__Moduliflexaceae # children taxa: 3\n",
      "length new taxa list: 2581\n",
      "100.0:c__Nitrospinia; o__Nitrospinales; f__Nitrospinaceae # children taxa: 16\n",
      "length new taxa list: 2566\n",
      "100.0:f__UBA9942 # children taxa: 3\n",
      "length new taxa list: 2564\n",
      "100.0:f__UBA8248 # children taxa: 3\n",
      "length new taxa list: 2562\n",
      "100.0:p__Tectomicrobia; c__Entotheonellia; o__Entotheonellales; f__Entotheonellaceae; g__Entotheonella # children taxa: 2\n",
      "length new taxa list: 2561\n",
      "100.0:p__Schekmanbacteria; c__GWA2-38-11; o__GWA2-38-11; f__GWA2-38-11 # children taxa: 3\n",
      "length new taxa list: 2559\n",
      "100.0:p__BMS3Abin14; c__BMS3Abin14; o__BMS3Abin14; f__BMS3Abin14 # children taxa: 3\n",
      "length new taxa list: 2557\n",
      "100.0:o__Binatales; f__Binataceae; g__Binatus # children taxa: 4\n",
      "length new taxa list: 2554\n",
      "100.0:o__UBA9968; f__UBA9968 # children taxa: 6\n",
      "length new taxa list: 2549\n",
      "100.0:f__BM512; g__BM512 # children taxa: 2\n",
      "length new taxa list: 2548\n",
      "100.0:p__MBNT15; c__MBNT15; o__MBNT15; f__MBNT15 # children taxa: 9\n",
      "length new taxa list: 2540\n",
      "100.0:f__Fen-1088 # children taxa: 13\n",
      "length new taxa list: 2528\n",
      "100.0:f__Palsa-1104 # children taxa: 2\n",
      "length new taxa list: 2527\n",
      "100.0:o__Haliangiales; f__Haliangiaceae # children taxa: 2\n",
      "length new taxa list: 2526\n",
      "100.0:o__Nannocystales; f__Nannocystaceae # children taxa: 6\n",
      "length new taxa list: 2521\n",
      "100.0:o__HGW-17; f__HGW-17 # children taxa: 3\n",
      "length new taxa list: 2519\n",
      "100.0:f__Polyangiaceae # children taxa: 16\n",
      "length new taxa list: 2504\n",
      "100.0:f__SG8-38 # children taxa: 3\n",
      "length new taxa list: 2502\n",
      "100.0:f__Myxococcaceae # children taxa: 20\n",
      "length new taxa list: 2483\n",
      "100.0:f__UBA5297 # children taxa: 2\n",
      "length new taxa list: 2482\n",
      "100.0:f__Anaeromyxobacteraceae; g__Anaeromyxobacter # children taxa: 5\n",
      "length new taxa list: 2478\n",
      "100.0:f__Vulgatibacteraceae # children taxa: 2\n",
      "length new taxa list: 2477\n",
      "100.0:o__UBA4151; f__UBA4151; g__UBA4151 # children taxa: 2\n",
      "length new taxa list: 2476\n",
      "100.0:f__Bradymonadaceae # children taxa: 2\n",
      "length new taxa list: 2475\n",
      "100.0:f__UBA4248 # children taxa: 2\n",
      "length new taxa list: 2474\n",
      "100.0:o__UBA7976; f__UBA1532 # children taxa: 2\n",
      "length new taxa list: 2473\n",
      "100.0:f__UBA4427 # children taxa: 6\n",
      "length new taxa list: 2468\n",
      "99.0:f__UBA6930 # children taxa: 2\n",
      "length new taxa list: 2467\n",
      "93.0:f__UBA9160 # children taxa: 2\n",
      "length new taxa list: 2466\n",
      "100.0:f__UBA796 # children taxa: 5\n",
      "length new taxa list: 2462\n",
      "100.0:o__UBA9615; f__UBA9615 # children taxa: 2\n",
      "length new taxa list: 2461\n",
      "100.0:f__Oligoflexaceae # children taxa: 3\n",
      "length new taxa list: 2459\n",
      "85.0:f__Bog-1112 # children taxa: 2\n",
      "length new taxa list: 2458\n",
      "100.0:o__Bacteriovoracales; f__Bacteriovoracaceae # children taxa: 39\n",
      "length new taxa list: 2420\n",
      "100.0:f__UBA1018 # children taxa: 5\n",
      "length new taxa list: 2416\n",
      "100.0:o__UBA2394; f__UBA2394 # children taxa: 2\n",
      "length new taxa list: 2415\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "76.0:f__Bdellovibrionaceae # children taxa: 21\n",
      "length new taxa list: 2395\n",
      "99.0:f__UBA1609 # children taxa: 4\n",
      "length new taxa list: 2392\n",
      "100.0:f__UBA840 # children taxa: 2\n",
      "length new taxa list: 2391\n",
      "100.0:o__2-02-FULL-40-11; f__2-02-FULL-40-11 # children taxa: 2\n",
      "length new taxa list: 2390\n",
      "100.0:o__2-02-FULL-44-16; f__2-02-FULL-44-16 # children taxa: 2\n",
      "length new taxa list: 2389\n",
      "100.0:f__Desulfovibrionaceae # children taxa: 79\n",
      "length new taxa list: 2311\n",
      "100.0:f__Desulfonatronaceae; g__Desulfonatronum # children taxa: 6\n",
      "length new taxa list: 2306\n",
      "100.0:f__Desulfomicrobiaceae # children taxa: 9\n",
      "length new taxa list: 2298\n",
      "100.0:f__Desulfonauticaceae; g__Desulfonauticus # children taxa: 2\n",
      "length new taxa list: 2297\n",
      "100.0:f__Desulfonatronovibrionaceae # children taxa: 3\n",
      "length new taxa list: 2295\n",
      "91.0:f__Desulfohalobiaceae # children taxa: 3\n",
      "length new taxa list: 2293\n",
      "100.0:p__Nitrospirota_A; c__Leptospirillia; o__Leptospirillales; f__Leptospirillaceae # children taxa: 4\n",
      "length new taxa list: 2290\n",
      "83.0:f__Koribacteraceae # children taxa: 60\n",
      "length new taxa list: 2231\n",
      "100.0:f__Gp1-AA117; g__Gp1-AA17 # children taxa: 2\n",
      "length new taxa list: 2230\n",
      "100.0:f__Gp1-AA112; g__Gp1-AA112 # children taxa: 2\n",
      "length new taxa list: 2229\n",
      "100.0:f__Acidobacteriaceae # children taxa: 65\n",
      "length new taxa list: 2165\n",
      "100.0:f__Bryobacteraceae # children taxa: 32\n",
      "length new taxa list: 2134\n",
      "100.0:o__2-12-FULL-54-10; f__2-12-FULL-54-10 # children taxa: 2\n",
      "length new taxa list: 2133\n",
      "100.0:f__UBA7541 # children taxa: 35\n",
      "length new taxa list: 2099\n",
      "100.0:o__UBA7540; f__UBA7540 # children taxa: 8\n",
      "length new taxa list: 2092\n",
      "100.0:o__20CM-2-55-15; f__20CM-2-55-15; g__20CM-2-55-15 # children taxa: 3\n",
      "length new taxa list: 2090\n",
      "100.0:o__Pyrinomonadales; f__Pyrinomonadaceae # children taxa: 23\n",
      "length new taxa list: 2068\n",
      "100.0:o__UBA7656; f__UBA7656 # children taxa: 2\n",
      "length new taxa list: 2067\n",
      "100.0:o__Chloracidobacteriales; f__Chloracidobacteriaceae; g__Chloracidobacterium # children taxa: 3\n",
      "length new taxa list: 2065\n",
      "100.0:o__HR10; f__HR10; g__HR10 # children taxa: 2\n",
      "length new taxa list: 2064\n",
      "100.0:c__UBA6911; o__UBA6911; f__UBA6911 # children taxa: 2\n",
      "length new taxa list: 2063\n",
      "100.0:f__UBA823 # children taxa: 9\n",
      "length new taxa list: 2055\n",
      "100.0:f__UBA8438; g__UBA8438 # children taxa: 5\n",
      "length new taxa list: 2051\n",
      "100.0:f__Fen-181 # children taxa: 2\n",
      "length new taxa list: 2050\n",
      "62.0:f__UBA2999 # children taxa: 12\n",
      "length new taxa list: 2039\n",
      "100.0:f__2-12-FULL-66-21 # children taxa: 3\n",
      "length new taxa list: 2037\n",
      "100.0:f__Vicinamibacteraceae; g__Luteitalea # children taxa: 2\n",
      "length new taxa list: 2036\n",
      "100.0:f__SCN-69-37; g__SCN-69-37 # children taxa: 2\n",
      "length new taxa list: 2035\n",
      "100.0:o__Fen-336; f__Fen-336 # children taxa: 4\n",
      "length new taxa list: 2032\n",
      "100.0:f__Gp22-AA2 # children taxa: 3\n",
      "length new taxa list: 2030\n",
      "100.0:c__UBA890; o__UBA890; f__UBA890; g__UBA890 # children taxa: 2\n",
      "length new taxa list: 2029\n",
      "100.0:f__Thermoanaerobaculaceae # children taxa: 5\n",
      "length new taxa list: 2025\n",
      "100.0:f__UBA5066 # children taxa: 2\n",
      "length new taxa list: 2024\n",
      "100.0:o__Gp7-AA8; f__Gp7-AA8 # children taxa: 4\n",
      "length new taxa list: 2021\n",
      "100.0:o__UBA5704; f__UBA5704 # children taxa: 2\n",
      "length new taxa list: 2020\n",
      "100.0:c__Holophagae; o__Holophagales; f__Holophagaceae # children taxa: 9\n",
      "length new taxa list: 2012\n",
      "100.0:f__OPB95; g__UBA10528 # children taxa: 4\n",
      "length new taxa list: 2009\n",
      "100.0:f__RBG-16-66-30; g__RBG-16-66-30 # children taxa: 3\n",
      "length new taxa list: 2007\n",
      "100.0:f__UBA4085 # children taxa: 3\n",
      "length new taxa list: 2005\n",
      "99.0:f__Aminicenantaceae # children taxa: 2\n",
      "length new taxa list: 2004\n",
      "100.0:o__UBA2199; f__UBA2199 # children taxa: 5\n",
      "length new taxa list: 2000\n",
      "100.0:f__UBA6249 # children taxa: 10\n",
      "length new taxa list: 1991\n",
      "100.0:f__UBA12090 # children taxa: 4\n",
      "length new taxa list: 1988\n",
      "100.0:o__UBA1572; f__UBA1572 # children taxa: 3\n",
      "length new taxa list: 1986\n",
      "99.0:f__kpj58rc # children taxa: 8\n",
      "length new taxa list: 1979\n",
      "100.0:f__GCA-002753745 # children taxa: 2\n",
      "length new taxa list: 1978\n",
      "100.0:o__4484-171; f__4484-171 # children taxa: 2\n",
      "length new taxa list: 1977\n",
      "100.0:o__UBA6215; f__UBA6215 # children taxa: 3\n",
      "length new taxa list: 1975\n",
      "100.0:o__UBA10183; f__UBA10183; g__UBA10183 # children taxa: 4\n",
      "length new taxa list: 1972\n",
      "100.0:f__2-01-FULL-45-10 # children taxa: 4\n",
      "length new taxa list: 1969\n",
      "100.0:o__UBA9649; f__UBA9649 # children taxa: 3\n",
      "length new taxa list: 1967\n",
      "100.0:o__2-02-FULL-51-18; f__2-02-FULL-51-18 # children taxa: 2\n",
      "length new taxa list: 1966\n",
      "100.0:f__01-FULL-45-10b # children taxa: 3\n",
      "length new taxa list: 1964\n",
      "100.0:f__UBA2337 # children taxa: 4\n",
      "length new taxa list: 1961\n",
      "100.0:f__Palsa-1178 # children taxa: 2\n",
      "length new taxa list: 1960\n",
      "100.0:p__Firestonebacteria; c__D2-FULL-39-29; o__D2-FULL-39-29; f__D2-FULL-39-29; g__D2-FULL-39-29 # children taxa: 2\n",
      "length new taxa list: 1959\n",
      "100.0:p__FCPU426; c__Palsa-1180; o__Palsa-1180; f__Palsa-1180 # children taxa: 2\n",
      "length new taxa list: 1958\n",
      "100.0:f__UBA9959 # children taxa: 19\n",
      "length new taxa list: 1940\n",
      "100.0:f__Elusimicrobiaceae # children taxa: 10\n",
      "length new taxa list: 1931\n",
      "100.0:f__UBA9628 # children taxa: 4\n",
      "length new taxa list: 1928\n",
      "100.0:f__UBA1565 # children taxa: 4\n",
      "length new taxa list: 1925\n",
      "97.0:f__2-01-FULL-60-11 # children taxa: 3\n",
      "length new taxa list: 1923\n",
      "100.0:f__F11 # children taxa: 2\n",
      "length new taxa list: 1922\n",
      "100.0:f__Endomicrobiaceae # children taxa: 5\n",
      "length new taxa list: 1918\n",
      "100.0:f__XYA2-FULL-47-53 # children taxa: 2\n",
      "length new taxa list: 1917\n",
      "100.0:f__XYA2-FULL-39-19 # children taxa: 2\n",
      "length new taxa list: 1916\n",
      "100.0:c__UBA8919; o__UBA8919; f__UBA8919 # children taxa: 4\n",
      "length new taxa list: 1913\n",
      "100.0:p__UBA6262; c__UBA6262; o__UBA6262; f__UBA6262 # children taxa: 2\n",
      "length new taxa list: 1912\n",
      "100.0:p__CG03; c__CG03; o__CG03; f__CG03 # children taxa: 3\n",
      "length new taxa list: 1910\n",
      "1910\n"
     ]
    }
   ],
   "source": [
    "parsed_bac_train_family = bac_train.copy()\n",
    "parse_taxa_from_node(bac.seed_node,parsed_bac_train_family,genomes_per_taxa=1,prefix='f__')\n",
    "print(len(parsed_bac_train_family))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19945 1910\n"
     ]
    }
   ],
   "source": [
    "print(len(bac_train),len(parsed_bac_train_family))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2101\n"
     ]
    }
   ],
   "source": [
    "print(len(parsed_bac_train_family)+len(parsed_arc_train_family))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_df_subset(ftp,arcftp,bac_list,arc_list):\n",
    "    newdf_b = pd.DataFrame()\n",
    "    for b in bac_list:\n",
    "        if b[3:] in list(ftp['ncbi_genbank_assembly_accession']):\n",
    "            newdf_b = newdf_b.append(ftp[ftp['ncbi_genbank_assembly_accession']==b[3:]])\n",
    "        elif b[3:] in list(ftp['parsed_genome_accession']):\n",
    "            newdf_b = newdf_b.append(ftp[ftp['parsed_genome_accession']==b[3:]])\n",
    "        else:\n",
    "            print('couldnt get record:',b)\n",
    "            \n",
    "    newdf_a = pd.DataFrame()\n",
    "    for b in arc_list:\n",
    "        if b[3:] in list(arcftp['ncbi_genbank_assembly_accession']):\n",
    "            newdf_a = newdf_a.append(arcftp[arcftp['ncbi_genbank_assembly_accession']==b[3:]])\n",
    "            #arcftp.drop(arcftp[arcftp['ncbi_genbank_assembly_accession']==b[3:]].index,inplace=True)\n",
    "        elif b[3:] in list(arcftp['parsed_genome_accession']):\n",
    "            newdf_a = newdf_a.append(arcftp[arcftp['parsed_genome_accession']==b[3:]])\n",
    "            #arcftp.drop(arcftp[arcftp['parsed_genome_accession']==b[3:]].index, inplace=True)\n",
    "        elif b in list(arcftp['parsed_genome_accession']): #for when only have SRR for e.g. UBA9915\n",
    "            newdf_a = newdf_a.append(arcftp[arcftp['parsed_genome_accession']==b])\n",
    "            #arcftp.drop(arcftp[arcftp['parsed_genome_accession']==b].index, inplace=True)\n",
    "        else:\n",
    "            print('couldnt get record',b)\n",
    "\n",
    "    assert len(newdf_b)==len(bac_list)\n",
    "    assert len(newdf_a)==len(arc_list)\n",
    "    newdf = newdf_b.append(newdf_a,sort=True)\n",
    "    return newdf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2101\n"
     ]
    }
   ],
   "source": [
    "traindf_family = get_df_subset(ftp,arcftp,parsed_bac_train_family,parsed_arc_train_family)\n",
    "print(len(traindf_family))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2101\n"
     ]
    }
   ],
   "source": [
    "train_fnames_family = get_filenames(traindf_family)\n",
    "print(len(train_fnames_family))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "traindf_family.to_csv('TrainNew_ParsedFamily_GTDBrepwithftp.csv')\n",
    "save_filenames('trainnew_parsedfamily_filenames.txt',train_fnames_family)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## what if parse at order level?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "100.0:c__Halobacteria; o__Halobacteriales # children taxa: 229\n",
      "length new taxa list: 782\n",
      "100.0:c__Methanomicrobia; o__Methanomicrobiales # children taxa: 71\n",
      "length new taxa list: 712\n",
      "100.0:c__Bog-38; o__Bog-38; f__Bog-38; g__Bog-38 # children taxa: 4\n",
      "length new taxa list: 709\n",
      "100.0:o__Methanotrichales; f__Methanotrichaceae # children taxa: 13\n",
      "length new taxa list: 697\n",
      "100.0:c__Methanocellia; o__Methanocellales # children taxa: 4\n",
      "length new taxa list: 694\n",
      "100.0:o__ANME-1; f__ANME-1 # children taxa: 3\n",
      "length new taxa list: 692\n",
      "100.0:o__Syntrophoarchaeales; f__Syntrophoarchaeaceae # children taxa: 2\n",
      "length new taxa list: 691\n",
      "100.0:o__Archaeoglobales # children taxa: 17\n",
      "length new taxa list: 675\n",
      "100.0:c__Methanonatronarchaeia; o__Methanonatronarchaeales # children taxa: 3\n",
      "length new taxa list: 673\n",
      "100.0:o__Poseidoniales # children taxa: 191\n",
      "length new taxa list: 483\n",
      "100.0:o__MGIII # children taxa: 13\n",
      "length new taxa list: 471\n",
      "100.0:o__DHVEG-1; f__DHVEG-1 # children taxa: 7\n",
      "length new taxa list: 465\n",
      "86.0:o__UBA9212 # children taxa: 3\n",
      "length new taxa list: 463\n",
      "100.0:c__Methanobacteria; o__Methanobacteriales # children taxa: 84\n",
      "length new taxa list: 380\n",
      "100.0:c__Methanococci; o__Methanococcales # children taxa: 18\n",
      "length new taxa list: 363\n",
      "100.0:c__Methanopyri; o__Methanopyrales; f__Methanopyraceae; g__Methanopyrus # children taxa: 2\n",
      "length new taxa list: 362\n",
      "100.0:o__Thermococcales; f__Thermococcaceae # children taxa: 41\n",
      "length new taxa list: 322\n",
      "100.0:o__Methanofastidiosales; f__Methanofastidiosaceae; g__Methanofastidiosum # children taxa: 5\n",
      "length new taxa list: 318\n",
      "100.0:p__Hydrothermarchaeota; c__Hydrothermarchaeia; o__Hydrothermarchaeales # children taxa: 5\n",
      "length new taxa list: 314\n",
      "100.0:p__Hadarchaeota; c__Hadarchaeia; o__Hadarchaeales; f__Hadarchaeaceae # children taxa: 2\n",
      "length new taxa list: 313\n",
      "100.0:o__UBA164; f__UBA164 # children taxa: 2\n",
      "length new taxa list: 312\n",
      "100.0:o__Caldiarchaeales # children taxa: 8\n",
      "length new taxa list: 305\n",
      "70.0:o__Desulfurococcales # children taxa: 26\n",
      "length new taxa list: 280\n",
      "100.0:o__Sulfolobales; f__Sulfolobaceae # children taxa: 21\n",
      "length new taxa list: 260\n",
      "100.0:o__Ignicoccales; f__Ignicoccaceae # children taxa: 2\n",
      "length new taxa list: 259\n",
      "100.0:o__Thermoproteales # children taxa: 33\n",
      "length new taxa list: 227\n",
      "86.0:o__Thermofilales # children taxa: 5\n",
      "length new taxa list: 223\n",
      "100.0:o__Geoarchaeales; f__Geoarchaeaceae # children taxa: 2\n",
      "length new taxa list: 222\n",
      "99.0:o__Marsarchaeales # children taxa: 3\n",
      "length new taxa list: 220\n",
      "100.0:c__Methanomethylicia; o__Methanomethyliales; f__Methanomethyliaceae # children taxa: 3\n",
      "length new taxa list: 218\n",
      "100.0:o__Thorarchaeales; f__Thorarchaeaceae # children taxa: 6\n",
      "length new taxa list: 213\n",
      "100.0:o__LC-2; f__LC-2 # children taxa: 2\n",
      "length new taxa list: 212\n",
      "100.0:o__UBA460; f__UBA460; g__UBA460 # children taxa: 2\n",
      "length new taxa list: 211\n",
      "90.0:o__Woesearchaeales # children taxa: 42\n",
      "length new taxa list: 170\n",
      "93.0:o__SCGC-AAA011-G17 # children taxa: 13\n",
      "length new taxa list: 158\n",
      "100.0:o__Pacearchaeales # children taxa: 33\n",
      "length new taxa list: 126\n",
      "100.0:o__UBA10117 # children taxa: 3\n",
      "length new taxa list: 124\n",
      "100.0:o__UBA583; f__UBA583 # children taxa: 2\n",
      "length new taxa list: 123\n",
      "100.0:o__Parvarchaeales; f__Parvarchaeaceae; g__Parvarchaeum # children taxa: 2\n",
      "length new taxa list: 122\n",
      "100.0:o__Nanoarchaeales # children taxa: 4\n",
      "length new taxa list: 119\n",
      "100.0:p__Nanohaloarchaeota; c__Nanosalinia; o__Nanosalinales; f__Nanosalinaceae # children taxa: 11\n",
      "length new taxa list: 109\n",
      "100.0:o__CG10238-14 # children taxa: 3\n",
      "length new taxa list: 107\n",
      "100.0:o__GW2011-AR5 # children taxa: 3\n",
      "length new taxa list: 105\n",
      "100.0:p__UAP2; c__UBA543; o__UBA543 # children taxa: 3\n",
      "length new taxa list: 103\n",
      "100.0:o__UBA10214 # children taxa: 5\n",
      "length new taxa list: 99\n",
      "100.0:o__Micrarchaeales; f__Micrarchaeaceae # children taxa: 4\n",
      "length new taxa list: 96\n",
      "100.0:o__UBA8480 # children taxa: 6\n",
      "length new taxa list: 91\n",
      "100.0:o__Iainarchaeales # children taxa: 11\n",
      "length new taxa list: 81\n",
      "88.0:o__SCGC-AAA252-I15 # children taxa: 3\n",
      "length new taxa list: 79\n",
      "97.0:o__WOR-SM1-SCG # children taxa: 3\n",
      "length new taxa list: 77\n",
      "99.0:o__Altiarchaeales; f__Altiarchaeaceae; g__Altiarchaeum # children taxa: 2\n",
      "length new taxa list: 76\n",
      "76\n",
      "100.0:o__Oscillospirales # children taxa: 511\n",
      "length new taxa list: 19435\n",
      "84.0:o__Christensenellales # children taxa: 68\n",
      "length new taxa list: 19368\n",
      "100.0:o__4C28d-15 # children taxa: 56\n",
      "length new taxa list: 19313\n",
      "100.0:o__Saccharofermentanales # children taxa: 34\n",
      "length new taxa list: 19280\n",
      "100.0:o__UBA1212 # children taxa: 4\n",
      "length new taxa list: 19277\n",
      "100.0:o__TANB77 # children taxa: 33\n",
      "length new taxa list: 19245\n",
      "100.0:o__Lachnospirales # children taxa: 662\n",
      "length new taxa list: 18584\n",
      "71.0:o__Acetivibrionales # children taxa: 30\n",
      "length new taxa list: 18555\n",
      "98.0:o__Monoglobales # children taxa: 10\n",
      "length new taxa list: 18546\n",
      "95.0:o__Peptostreptococcales # children taxa: 134\n",
      "length new taxa list: 18413\n",
      "98.0:o__Tissierellales # children taxa: 124\n",
      "length new taxa list: 18290\n",
      "98.0:o__Eubacteriales # children taxa: 20\n",
      "length new taxa list: 18271\n",
      "100.0:o__Lutisporales; f__Lutisporaceae # children taxa: 4\n",
      "length new taxa list: 18268\n",
      "97.0:o__Clostridiales # children taxa: 145\n",
      "length new taxa list: 18124\n",
      "100.0:o__Caldicoprobacterales # children taxa: 11\n",
      "length new taxa list: 18114\n",
      "100.0:o__Thermoanaerobacterales; f__Thermoanaerobacteraceae # children taxa: 14\n",
      "length new taxa list: 18101\n",
      "99.0:o__Caldanaerobiales # children taxa: 4\n",
      "length new taxa list: 18098\n",
      "100.0:o__Caldicellulosiruptorales; f__Caldicellulosiruptoraceae; g__Caldicellulosiruptor # children taxa: 10\n",
      "length new taxa list: 18089\n",
      "100.0:o__Thermovenabulales # children taxa: 9\n",
      "length new taxa list: 18081\n",
      "100.0:o__Paenibacillales # children taxa: 220\n",
      "length new taxa list: 17862\n",
      "100.0:o__Caldalkalibacillales; f__Caldalkalibacillaceae # children taxa: 2\n",
      "length new taxa list: 17861\n",
      "100.0:o__Brevibacillales; f__Brevibacillaceae # children taxa: 20\n",
      "length new taxa list: 17842\n",
      "100.0:o__Aneurinibacillales; f__Aneurinibacillaceae # children taxa: 7\n",
      "length new taxa list: 17836\n",
      "100.0:o__Tepidibacillales; f__Tepidibacillaceae # children taxa: 3\n",
      "length new taxa list: 17834\n",
      "100.0:o__Thermoactinomycetales # children taxa: 23\n",
      "length new taxa list: 17812\n",
      "93.0:o__Thermicanales; f__Thermicanaceae # children taxa: 4\n",
      "length new taxa list: 17809\n",
      "100.0:o__Bacillales_F; f__Bacillaceae_M; g__Bacillus_BB # children taxa: 2\n",
      "length new taxa list: 17808\n",
      "100.0:p__Firmicutes_J; c__Desulfuribacillia; o__Desulfuribacillales; f__Desulfuribacillaceae # children taxa: 2\n",
      "length new taxa list: 17807\n",
      "100.0:o__Alicyclobacillales # children taxa: 22\n",
      "length new taxa list: 17786\n",
      "100.0:o__Kyrpidiales; f__Kyrpidiaceae; g__Kyrpidia # children taxa: 2\n",
      "length new taxa list: 17785\n",
      "100.0:o__Tumebacillales # children taxa: 6\n",
      "length new taxa list: 17780\n",
      "100.0:o__Dethiobacterales; f__Dethiobacteraceae # children taxa: 4\n",
      "length new taxa list: 17777\n",
      "100.0:o__DTU022 # children taxa: 4\n",
      "length new taxa list: 17774\n",
      "100.0:c__Natranaerobiia; o__Natranaerobiales; f__Natranaerobiaceae # children taxa: 2\n",
      "length new taxa list: 17773\n",
      "100.0:o__UBA4975; f__UBA4975; g__UBA4975 # children taxa: 4\n",
      "length new taxa list: 17770\n",
      "100.0:o__Proteinivoracales; f__Proteinivoraceae; g__Anaerobranca # children taxa: 2\n",
      "length new taxa list: 17769\n",
      "100.0:c__UBA994; o__UBA994 # children taxa: 3\n",
      "length new taxa list: 17767\n",
      "100.0:o__DTU010 # children taxa: 8\n",
      "length new taxa list: 17760\n",
      "100.0:o__DTU080 # children taxa: 3\n",
      "length new taxa list: 17758\n",
      "100.0:o__Limnochordales # children taxa: 2\n",
      "length new taxa list: 17757\n",
      "100.0:c__DTU065; o__DTU065; f__DTU065 # children taxa: 2\n",
      "length new taxa list: 17756\n",
      "100.0:c__DTU084; o__DTU084; f__DTU084 # children taxa: 4\n",
      "length new taxa list: 17753\n",
      "100.0:o__UBA4971; f__UBA4971 # children taxa: 4\n",
      "length new taxa list: 17750\n",
      "100.0:o__DTUO25; f__DTU025 # children taxa: 4\n",
      "length new taxa list: 17747\n",
      "100.0:o__UBA10575 # children taxa: 4\n",
      "length new taxa list: 17744\n",
      "100.0:o__UBA8346; f__UBA8346; g__UBA8346 # children taxa: 2\n",
      "length new taxa list: 17743\n",
      "100.0:c__Sulfobacillia; o__Sulfobacillales; f__Sulfobacillaceae # children taxa: 7\n",
      "length new taxa list: 17737\n",
      "100.0:c__Symbiobacteriia; o__Symbiobacteriales # children taxa: 3\n",
      "length new taxa list: 17735\n",
      "100.0:c__Thermaerobacteria; o__Thermaerobacterales # children taxa: 4\n",
      "length new taxa list: 17732\n",
      "100.0:o__D8A-2; f__D2 # children taxa: 3\n",
      "length new taxa list: 17730\n",
      "100.0:o__UBA995; f__UBA995; g__UBA995 # children taxa: 2\n",
      "length new taxa list: 17729\n",
      "100.0:o__UBA9673; f__UBA9673; g__UBA9673 # children taxa: 2\n",
      "length new taxa list: 17728\n",
      "88.0:o__Selenomonadales # children taxa: 61\n",
      "length new taxa list: 17668\n",
      "100.0:o__Propionisporales # children taxa: 7\n",
      "length new taxa list: 17662\n",
      "78.0:o__Sporomusales # children taxa: 11\n",
      "length new taxa list: 17652\n",
      "100.0:o__Acidaminococcales # children taxa: 22\n",
      "length new taxa list: 17631\n",
      "100.0:o__Veillonellales # children taxa: 49\n",
      "length new taxa list: 17583\n",
      "100.0:o__Desulfitobacteriales # children taxa: 32\n",
      "length new taxa list: 17552\n",
      "100.0:o__DRI-13 # children taxa: 8\n",
      "length new taxa list: 17545\n",
      "98.0:o__Peptococcales # children taxa: 6\n",
      "length new taxa list: 17540\n",
      "100.0:o__Moorellales; f__Moorellaceae # children taxa: 9\n",
      "length new taxa list: 17532\n",
      "100.0:o__PH28-bin88 # children taxa: 2\n",
      "length new taxa list: 17531\n",
      "100.0:o__DTU086; f__DTU086 # children taxa: 2\n",
      "length new taxa list: 17530\n",
      "100.0:o__Desulfitibacterales; f__Desulfitibacteraceae; g__Desulfitibacter # children taxa: 2\n",
      "length new taxa list: 17529\n",
      "100.0:o__UBA4068 # children taxa: 5\n",
      "length new taxa list: 17525\n",
      "100.0:c__TC1; o__TC1 # children taxa: 2\n",
      "length new taxa list: 17524\n",
      "100.0:o__Syntrophomonadales # children taxa: 20\n",
      "length new taxa list: 17505\n",
      "100.0:o__Thermacetogeniales # children taxa: 8\n",
      "length new taxa list: 17498\n",
      "100.0:o__Desulfotomaculales # children taxa: 43\n",
      "length new taxa list: 17456\n",
      "100.0:o__Ammonifexales # children taxa: 6\n",
      "length new taxa list: 17451\n",
      "100.0:o__Carboxydothermales; f__Carboxydothermaceae; g__Carboxydothermus # children taxa: 3\n",
      "length new taxa list: 17449\n",
      "100.0:c__Thermincolia; o__Thermincolales # children taxa: 4\n",
      "length new taxa list: 17446\n",
      "100.0:o__Halanaerobiales # children taxa: 12\n",
      "length new taxa list: 17435\n",
      "100.0:o__Halobacteroidales # children taxa: 8\n",
      "length new taxa list: 17428\n",
      "92.0:o__Lactobacillales # children taxa: 674\n",
      "length new taxa list: 16755\n",
      "49.0:o__Bacillales # children taxa: 543\n",
      "length new taxa list: 16213\n",
      "100.0:o__Exiguobacterales; f__Exiguobacteraceae # children taxa: 18\n",
      "length new taxa list: 16196\n",
      "100.0:o__Bacillales_A; f__Planococcaceae # children taxa: 116\n",
      "length new taxa list: 16081\n",
      "100.0:o__Staphylococcales # children taxa: 100\n",
      "length new taxa list: 15982\n",
      "76.0:o__Mycoplasmatales # children taxa: 143\n",
      "length new taxa list: 15840\n",
      "100.0:o__RFN20 # children taxa: 59\n",
      "length new taxa list: 15782\n",
      "100.0:o__Erysipelotrichales # children taxa: 85\n",
      "length new taxa list: 15698\n",
      "100.0:o__RF39 # children taxa: 58\n",
      "length new taxa list: 15641\n",
      "100.0:o__Acholeplasmatales # children taxa: 43\n",
      "length new taxa list: 15599\n",
      "100.0:o__Izemoplasmatales # children taxa: 17\n",
      "length new taxa list: 15583\n",
      "100.0:o__ML615J-28 # children taxa: 15\n",
      "length new taxa list: 15569\n",
      "99.0:o__Haloplasmatales # children taxa: 6\n",
      "length new taxa list: 15564\n",
      "100.0:o__Fusobacteriales # children taxa: 53\n",
      "length new taxa list: 15512\n",
      "39.0:o__Synechococcales # children taxa: 342\n",
      "length new taxa list: 15171\n",
      "82.0:o__Cyanobacteriales # children taxa: 190\n",
      "length new taxa list: 14982\n",
      "100.0:o__Leptolyngbyales; f__Leptolyngbyaceae # children taxa: 9\n",
      "length new taxa list: 14974\n",
      "93.0:o__Neosynechococcales; f__Neosynechococcaceae # children taxa: 2\n",
      "length new taxa list: 14973\n",
      "100.0:o__Elainellales; f__Elainellaceae # children taxa: 5\n",
      "length new taxa list: 14969\n",
      "100.0:o__Phormidesmiales; f__Phormidesmiaceae # children taxa: 17\n",
      "length new taxa list: 14953\n",
      "100.0:o__Thermosynechococcales; f__Thermosynechococcaceae # children taxa: 8\n",
      "length new taxa list: 14946\n",
      "100.0:o__CACIAM-69d; f__CACIAM-69d; g__CACIAM-69d # children taxa: 2\n",
      "length new taxa list: 14945\n",
      "100.0:o__Pseudanabaenales; f__Pseudanabaenaceae # children taxa: 15\n",
      "length new taxa list: 14931\n",
      "100.0:o__PCC-7336 # children taxa: 3\n",
      "length new taxa list: 14929\n",
      "100.0:o__Gloeobacterales; f__Gloeobacteraceae; g__Gloeobacter # children taxa: 2\n",
      "length new taxa list: 14928\n",
      "100.0:o__Gastranaerophilales # children taxa: 41\n",
      "length new taxa list: 14888\n",
      "100.0:o__Vampirovibrionales; f__Vampirovibrionaceae; g__Vampirovibrio # children taxa: 3\n",
      "length new taxa list: 14886\n",
      "100.0:o__Obscuribacterales; f__Obscuribacteraceae # children taxa: 6\n",
      "length new taxa list: 14881\n",
      "100.0:o__Caenarcaniphilales # children taxa: 3\n",
      "length new taxa list: 14879\n",
      "100.0:o__S15B-MN24 # children taxa: 2\n",
      "length new taxa list: 14878\n",
      "100.0:o__UBA7694; f__UBA7694 # children taxa: 2\n",
      "length new taxa list: 14877\n",
      "100.0:o__O2-12-FULL-45-9 # children taxa: 6\n",
      "length new taxa list: 14872\n",
      "100.0:o__XYC2-FULL-46-14 # children taxa: 3\n",
      "length new taxa list: 14870\n",
      "73.0:o__UBA817 # children taxa: 9\n",
      "length new taxa list: 14862\n",
      "98.0:o__Mycobacteriales # children taxa: 948\n",
      "length new taxa list: 13915\n",
      "100.0:o__Frankiales; f__Frankiaceae # children taxa: 19\n",
      "length new taxa list: 13897\n",
      "100.0:o__Streptosporangiales; f__Streptosporangiaceae # children taxa: 102\n",
      "length new taxa list: 13796\n",
      "100.0:o__Acidothermales; f__Acidothermaceae # children taxa: 2\n",
      "length new taxa list: 13795\n",
      "95.0:o__Streptomycetales # children taxa: 531\n",
      "length new taxa list: 13265\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "100.0:o__Propionibacteriales # children taxa: 131\n",
      "length new taxa list: 13135\n",
      "100.0:o__Jiangellales; f__Jiangellaceae # children taxa: 7\n",
      "length new taxa list: 13129\n",
      "97.0:o__Nanopelagicales # children taxa: 46\n",
      "length new taxa list: 13084\n",
      "100.0:o__Euzebyales # children taxa: 3\n",
      "length new taxa list: 13082\n",
      "100.0:c__UBA4738; o__UBA4738 # children taxa: 3\n",
      "length new taxa list: 13080\n",
      "100.0:o__Microtrichales # children taxa: 88\n",
      "length new taxa list: 12993\n",
      "100.0:o__Acidimicrobiales # children taxa: 70\n",
      "length new taxa list: 12924\n",
      "100.0:o__IMCC26256 # children taxa: 3\n",
      "length new taxa list: 12922\n",
      "100.0:o__UBA5794 # children taxa: 12\n",
      "length new taxa list: 12911\n",
      "100.0:o__TMED189; f__TMED189; g__TMED189 # children taxa: 9\n",
      "length new taxa list: 12903\n",
      "100.0:o__Solirubrobacterales # children taxa: 23\n",
      "length new taxa list: 12881\n",
      "100.0:o__Ga0077560; f__Ga0077560 # children taxa: 2\n",
      "length new taxa list: 12880\n",
      "100.0:o__20CM-4-69-9; f__20CM-4-69-9 # children taxa: 13\n",
      "length new taxa list: 12868\n",
      "100.0:o__UBA2241; f__UBA2241 # children taxa: 7\n",
      "length new taxa list: 12862\n",
      "100.0:o__RBG-16-64-13; f__RBG-16-64-13; g__RBG-16-64-13 # children taxa: 2\n",
      "length new taxa list: 12861\n",
      "100.0:c__Rubrobacteria; o__Rubrobacterales; f__Rubrobacteraceae # children taxa: 3\n",
      "length new taxa list: 12859\n",
      "100.0:o__Coriobacteriales # children taxa: 119\n",
      "length new taxa list: 12741\n",
      "100.0:o__OPB41 # children taxa: 14\n",
      "length new taxa list: 12728\n",
      "100.0:o__UBA10029; f__UBA10029 # children taxa: 2\n",
      "length new taxa list: 12727\n",
      "100.0:o__UBA9087 # children taxa: 3\n",
      "length new taxa list: 12725\n",
      "100.0:c__UBA1414; o__UBA1414; f__UBA1414 # children taxa: 4\n",
      "length new taxa list: 12722\n",
      "100.0:o__Fen-727; f__Fen-727 # children taxa: 3\n",
      "length new taxa list: 12720\n",
      "100.0:o__UBP12; f__UBA5184 # children taxa: 25\n",
      "length new taxa list: 12696\n",
      "100.0:c__UBP9; o__UBA4705; f__UBA4705 # children taxa: 4\n",
      "length new taxa list: 12693\n",
      "100.0:p__Firmicutes_H; c__CSP1-3; o__CSP1-3 # children taxa: 3\n",
      "length new taxa list: 12691\n",
      "100.0:o__Fimbriimonadales; f__Fimbriimonadaceae # children taxa: 13\n",
      "length new taxa list: 12679\n",
      "100.0:o__OS-L; f__GBS-DC # children taxa: 3\n",
      "length new taxa list: 12677\n",
      "91.0:o__CG2-30-59-28 # children taxa: 2\n",
      "length new taxa list: 12676\n",
      "100.0:o__UBA5377 # children taxa: 3\n",
      "length new taxa list: 12674\n",
      "99.0:o__UBA9983_A # children taxa: 207\n",
      "length new taxa list: 12468\n",
      "74.0:o__WO2-41-13 # children taxa: 9\n",
      "length new taxa list: 12460\n",
      "30.0:o__UBA9983 # children taxa: 5\n",
      "length new taxa list: 12456\n",
      "100.0:o__WOR2-41-13_A; f__2-01-FULL-45-33 # children taxa: 7\n",
      "length new taxa list: 12450\n",
      "97.0:o__Ryanbacterales # children taxa: 5\n",
      "length new taxa list: 12446\n",
      "100.0:o__Sungbacterales # children taxa: 8\n",
      "length new taxa list: 12439\n",
      "100.0:o__2-02-FULL-40-12 # children taxa: 22\n",
      "length new taxa list: 12418\n",
      "99.0:o__UBA6257 # children taxa: 48\n",
      "length new taxa list: 12371\n",
      "66.0:o__GWC2-36-17_A # children taxa: 4\n",
      "length new taxa list: 12368\n",
      "87.0:o__Paceibacterales # children taxa: 52\n",
      "length new taxa list: 12317\n",
      "84.0:o__Portnoybacterales # children taxa: 12\n",
      "length new taxa list: 12306\n",
      "100.0:o__Terrybacterales; f__GWB1-40-14 # children taxa: 4\n",
      "length new taxa list: 12303\n",
      "100.0:o__Spechtbacterales # children taxa: 2\n",
      "length new taxa list: 12302\n",
      "100.0:o__GWC2-36-17 # children taxa: 4\n",
      "length new taxa list: 12299\n",
      "100.0:c__Paceibacteria_A; o__Moranbacterales # children taxa: 47\n",
      "length new taxa list: 12253\n",
      "71.0:o__Magasanikbacterales # children taxa: 36\n",
      "length new taxa list: 12218\n",
      "48.0:o__SG8-24 # children taxa: 32\n",
      "length new taxa list: 12187\n",
      "100.0:o__GWA2-46-9; f__GWA2-46-9 # children taxa: 3\n",
      "length new taxa list: 12185\n",
      "100.0:o__BM507 # children taxa: 28\n",
      "length new taxa list: 12158\n",
      "100.0:o__Buchananbacterales # children taxa: 9\n",
      "length new taxa list: 12150\n",
      "100.0:o__UBA11705; f__UBA11705 # children taxa: 5\n",
      "length new taxa list: 12146\n",
      "98.0:o__UBA2591 # children taxa: 5\n",
      "length new taxa list: 12142\n",
      "99.0:o__UBA1558 # children taxa: 10\n",
      "length new taxa list: 12133\n",
      "100.0:o__Veblenbacterales; f__UBA10138 # children taxa: 2\n",
      "length new taxa list: 12132\n",
      "100.0:o__UBA2196 # children taxa: 6\n",
      "length new taxa list: 12127\n",
      "100.0:o__UBA2169; f__UBA2169 # children taxa: 4\n",
      "length new taxa list: 12124\n",
      "100.0:o__UBA9629; f__UBA9629 # children taxa: 2\n",
      "length new taxa list: 12123\n",
      "98.0:o__UBA9570 # children taxa: 5\n",
      "length new taxa list: 12119\n",
      "82.0:o__UBA10025 # children taxa: 4\n",
      "length new taxa list: 12116\n",
      "100.0:c__Doudnabacteria; o__UBA920 # children taxa: 12\n",
      "length new taxa list: 12105\n",
      "100.0:c__Andersenbacteria; o__UBA10190; f__UBA10190 # children taxa: 5\n",
      "length new taxa list: 12101\n",
      "100.0:o__Peribacterales; f__Peribacteraceae # children taxa: 17\n",
      "length new taxa list: 12085\n",
      "100.0:o__UBA4473; f__UBA4473 # children taxa: 2\n",
      "length new taxa list: 12084\n",
      "100.0:o__UM-FILTER-43-11; f__UM-FILTER-43-11 # children taxa: 2\n",
      "length new taxa list: 12083\n",
      "100.0:o__UBA1369 # children taxa: 20\n",
      "length new taxa list: 12064\n",
      "100.0:o__BD1-5; f__UBA6164 # children taxa: 14\n",
      "length new taxa list: 12051\n",
      "100.0:o__Absconditabacterales; f__X112 # children taxa: 9\n",
      "length new taxa list: 12043\n",
      "92.0:o__Saccharimonadales # children taxa: 70\n",
      "length new taxa list: 11974\n",
      "100.0:o__UBA4664; f__UBA4664 # children taxa: 2\n",
      "length new taxa list: 11973\n",
      "100.0:o__CPR2; f__CPR2 # children taxa: 3\n",
      "length new taxa list: 11971\n",
      "100.0:o__UBA1875 # children taxa: 3\n",
      "length new taxa list: 11969\n",
      "100.0:o__GWA2-35-9 # children taxa: 4\n",
      "length new taxa list: 11966\n",
      "99.0:o__UBA12157 # children taxa: 2\n",
      "length new taxa list: 11965\n",
      "100.0:o__XYA2-FULL-43-10; f__XYA2-FULL-43-10 # children taxa: 2\n",
      "length new taxa list: 11964\n",
      "99.0:o__UBA1384 # children taxa: 2\n",
      "length new taxa list: 11963\n",
      "100.0:c__Kazan-3B-28; o__Kazan-3B-28 # children taxa: 5\n",
      "length new taxa list: 11959\n",
      "95.0:o__GWA2-44-7 # children taxa: 40\n",
      "length new taxa list: 11920\n",
      "90.0:o__Shapirobacterales # children taxa: 16\n",
      "length new taxa list: 11905\n",
      "100.0:o__UBA1400 # children taxa: 44\n",
      "length new taxa list: 11862\n",
      "100.0:o__UBA1406 # children taxa: 40\n",
      "length new taxa list: 11823\n",
      "100.0:o__GCA-001788215 # children taxa: 2\n",
      "length new taxa list: 11822\n",
      "100.0:o__Levybacterales # children taxa: 23\n",
      "length new taxa list: 11800\n",
      "100.0:o__UBA10105 # children taxa: 10\n",
      "length new taxa list: 11791\n",
      "100.0:o__2-02-FULL-39-11; f__2-02-FULL-39-11 # children taxa: 9\n",
      "length new taxa list: 11783\n",
      "100.0:o__Daviesbacterales; f__UBA10151 # children taxa: 13\n",
      "length new taxa list: 11771\n",
      "100.0:o__Curtissbacterales; f__GWA2-41-24 # children taxa: 10\n",
      "length new taxa list: 11762\n",
      "100.0:o__Woykebacterales # children taxa: 4\n",
      "length new taxa list: 11759\n",
      "100.0:o__UBA101185 # children taxa: 17\n",
      "length new taxa list: 11743\n",
      "100.0:o__2-01-FULL-39-13; f__2-01-FULL-39-13 # children taxa: 2\n",
      "length new taxa list: 11742\n",
      "100.0:c__CPR3; o__CPR3 # children taxa: 2\n",
      "length new taxa list: 11741\n",
      "100.0:c__4484-211; o__4484-211 # children taxa: 2\n",
      "length new taxa list: 11740\n",
      "100.0:o__SC72 # children taxa: 15\n",
      "length new taxa list: 11726\n",
      "100.0:o__B142; f__OLB20 # children taxa: 2\n",
      "length new taxa list: 11725\n",
      "100.0:o__UBA1151 # children taxa: 32\n",
      "length new taxa list: 11694\n",
      "97.0:o__SAR202 # children taxa: 30\n",
      "length new taxa list: 11665\n",
      "100.0:o__UBA3495 # children taxa: 26\n",
      "length new taxa list: 11640\n",
      "100.0:o__UBA2963 # children taxa: 6\n",
      "length new taxa list: 11635\n",
      "98.0:o__UBA6952 # children taxa: 2\n",
      "length new taxa list: 11634\n",
      "100.0:o__UBA1127 # children taxa: 3\n",
      "length new taxa list: 11632\n",
      "99.0:o__Dehalococcoidales # children taxa: 29\n",
      "length new taxa list: 11604\n",
      "100.0:o__GIF9 # children taxa: 13\n",
      "length new taxa list: 11592\n",
      "97.0:o__UBA2777 # children taxa: 2\n",
      "length new taxa list: 11591\n",
      "100.0:o__UBA2979 # children taxa: 10\n",
      "length new taxa list: 11582\n",
      "100.0:o__Bin125; f__Bin125 # children taxa: 2\n",
      "length new taxa list: 11581\n",
      "100.0:o__UBA2991; f__UBA2991 # children taxa: 4\n",
      "length new taxa list: 11578\n",
      "94.0:o__SM23-28-2 # children taxa: 6\n",
      "length new taxa list: 11573\n",
      "100.0:o__Anaerolineales # children taxa: 95\n",
      "length new taxa list: 11479\n",
      "100.0:o__SBR1031 # children taxa: 13\n",
      "length new taxa list: 11467\n",
      "100.0:o__Promineofilales # children taxa: 5\n",
      "length new taxa list: 11463\n",
      "97.0:o__Thermoflexales # children taxa: 3\n",
      "length new taxa list: 11461\n",
      "100.0:o__Caldilineales # children taxa: 7\n",
      "length new taxa list: 11455\n",
      "84.0:o__UBA3071 # children taxa: 3\n",
      "length new taxa list: 11453\n",
      "100.0:o__4572-78 # children taxa: 2\n",
      "length new taxa list: 11452\n",
      "100.0:o__UBA2200; f__UBA2200 # children taxa: 3\n",
      "length new taxa list: 11450\n",
      "100.0:o__UBA1429; f__UBA1429 # children taxa: 2\n",
      "length new taxa list: 11449\n",
      "100.0:o__UBA7937; f__UBA7937; g__UBA7937 # children taxa: 2\n",
      "length new taxa list: 11448\n",
      "100.0:o__Chloroflexales # children taxa: 17\n",
      "length new taxa list: 11432\n",
      "100.0:o__54-19 # children taxa: 2\n",
      "length new taxa list: 11431\n",
      "100.0:o__Thermomicrobiales # children taxa: 8\n",
      "length new taxa list: 11424\n",
      "100.0:c__Ktedonobacteria; o__Ktedonobacterales; f__Ktedonobacteraceae # children taxa: 7\n",
      "length new taxa list: 11418\n",
      "100.0:c__UBA5177; o__UBA5177; f__UBA5177 # children taxa: 2\n",
      "length new taxa list: 11417\n",
      "100.0:o__UBA8260 # children taxa: 20\n",
      "length new taxa list: 11398\n",
      "100.0:o__Dormibacterales; f__Dormibacteraceae # children taxa: 12\n",
      "length new taxa list: 11387\n",
      "100.0:o__CSP1-4 # children taxa: 26\n",
      "length new taxa list: 11362\n",
      "100.0:o__P2-11E; f__P2-11E # children taxa: 5\n",
      "length new taxa list: 11358\n",
      "100.0:p__Deinococcota; c__Deinococci; o__Deinococcales # children taxa: 69\n",
      "length new taxa list: 11290\n",
      "100.0:o__Thermotogales # children taxa: 26\n",
      "length new taxa list: 11265\n",
      "99.0:o__Petrotogales # children taxa: 25\n",
      "length new taxa list: 11241\n",
      "100.0:o__Mesoaciditogales; f__Mesoaciditogaceae # children taxa: 2\n",
      "length new taxa list: 11240\n",
      "100.0:o__Bipolaricaulales; f__Bipolaricaulaceae # children taxa: 8\n",
      "length new taxa list: 11233\n",
      "100.0:o__UBA7950; f__UBA7950 # children taxa: 2\n",
      "length new taxa list: 11232\n",
      "100.0:o__Caldisericales; f__Caldisericaceae # children taxa: 5\n",
      "length new taxa list: 11228\n",
      "100.0:o__UBA4822; f__UBA4822; g__UBA4822 # children taxa: 4\n",
      "length new taxa list: 11225\n",
      "100.0:p__Coprothermobacterota; c__Coprothermobacteria; o__Coprothermobacterales; f__Coprothermobacteraceae; g__Coprothermobacter # children taxa: 3\n",
      "length new taxa list: 11223\n",
      "100.0:p__Dictyoglomota; c__Dictyoglomia; o__Dictyoglomales; f__Dictyoglomaceae # children taxa: 3\n",
      "length new taxa list: 11221\n",
      "100.0:p__Thermodesulfobiota; c__Thermodesulfobiia; o__Thermodesulfobiales; f__Thermodesulfobiaceae; g__Thermodesulfobium # children taxa: 2\n",
      "length new taxa list: 11220\n",
      "100.0:c__Synergistia; o__Synergistales # children taxa: 40\n",
      "length new taxa list: 11181\n",
      "100.0:c__GBS-1; o__GBS-1; f__GBS-1 # children taxa: 2\n",
      "length new taxa list: 11180\n",
      "100.0:c__JS1; o__SB-45 # children taxa: 11\n",
      "length new taxa list: 11170\n",
      "100.0:c__Caldatribacteriia; o__Caldatribacteriales # children taxa: 4\n",
      "length new taxa list: 11167\n",
      "100.0:p__Riflebacteria; c__UBA8953; o__UBA8953; f__UBA8953 # children taxa: 5\n",
      "length new taxa list: 11163\n",
      "100.0:p__UBP17; c__UBA6191; o__UBA6191; f__UBA6191 # children taxa: 2\n",
      "length new taxa list: 11162\n",
      "100.0:p__Wallbacteria; c__UBA9980; o__UBA9980; f__UBA9980; g__UBA9980 # children taxa: 2\n",
      "length new taxa list: 11161\n",
      "100.0:o__Chthoniobacterales # children taxa: 72\n",
      "length new taxa list: 11090\n",
      "100.0:o__Verrucomicrobiales # children taxa: 61\n",
      "length new taxa list: 11030\n",
      "100.0:o__Methylacidiphilales # children taxa: 13\n",
      "length new taxa list: 11018\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "100.0:o__Pedosphaerales # children taxa: 50\n",
      "length new taxa list: 10969\n",
      "100.0:o__Palsa-1439; f__Palsa-1439; g__Palsa-1439 # children taxa: 2\n",
      "length new taxa list: 10968\n",
      "100.0:o__Opitutales # children taxa: 94\n",
      "length new taxa list: 10875\n",
      "100.0:o__RFP12 # children taxa: 22\n",
      "length new taxa list: 10854\n",
      "99.0:o__UBA8416 # children taxa: 3\n",
      "length new taxa list: 10852\n",
      "100.0:o__Kiritimatiellales # children taxa: 9\n",
      "length new taxa list: 10844\n",
      "100.0:o__SS1-B-03-39; f__UBA6053 # children taxa: 2\n",
      "length new taxa list: 10843\n",
      "100.0:o__LD1-PB3 # children taxa: 4\n",
      "length new taxa list: 10840\n",
      "100.0:o__Victivallales # children taxa: 18\n",
      "length new taxa list: 10823\n",
      "100.0:o__UBA1407 # children taxa: 5\n",
      "length new taxa list: 10819\n",
      "100.0:c__UBA3054; o__UBA3054; f__UBA3054; g__UBA3054 # children taxa: 2\n",
      "length new taxa list: 10818\n",
      "60.0:o__Parachlamydiales # children taxa: 30\n",
      "length new taxa list: 10789\n",
      "100.0:o__Chlamydiales; f__Chlamydiaceae # children taxa: 15\n",
      "length new taxa list: 10775\n",
      "100.0:o__Pirellulales # children taxa: 68\n",
      "length new taxa list: 10708\n",
      "100.0:o__Planctomycetales; f__Planctomycetaceae # children taxa: 18\n",
      "length new taxa list: 10691\n",
      "100.0:o__Gemmatales # children taxa: 9\n",
      "length new taxa list: 10683\n",
      "100.0:o__Isosphaerales; f__Isosphaeraceae # children taxa: 6\n",
      "length new taxa list: 10678\n",
      "100.0:o__Phycisphaerales # children taxa: 51\n",
      "length new taxa list: 10628\n",
      "100.0:o__UBA1161; f__UBA1161 # children taxa: 3\n",
      "length new taxa list: 10626\n",
      "100.0:o__SG8-4 # children taxa: 22\n",
      "length new taxa list: 10605\n",
      "100.0:o__SM23-33 # children taxa: 6\n",
      "length new taxa list: 10600\n",
      "100.0:o__UBA1845 # children taxa: 3\n",
      "length new taxa list: 10598\n",
      "100.0:o__Brocadiales # children taxa: 15\n",
      "length new taxa list: 10584\n",
      "99.0:o__UBA1135 # children taxa: 12\n",
      "length new taxa list: 10573\n",
      "100.0:o__UBA2386 # children taxa: 5\n",
      "length new taxa list: 10569\n",
      "100.0:c__GCA-002687715; o__GCA-002687715; f__GCA-002687715 # children taxa: 2\n",
      "length new taxa list: 10568\n",
      "100.0:o__UBA1146; f__UBA1146 # children taxa: 5\n",
      "length new taxa list: 10564\n",
      "100.0:o__UBA8108; f__UBA8108; g__UBA8108 # children taxa: 4\n",
      "length new taxa list: 10561\n",
      "100.0:o__UBA11346; f__UBA11346 # children taxa: 2\n",
      "length new taxa list: 10560\n",
      "100.0:c__SZUA-567; o__SZUA-567; f__SZUA-567 # children taxa: 2\n",
      "length new taxa list: 10559\n",
      "100.0:o__Treponematales # children taxa: 88\n",
      "length new taxa list: 10472\n",
      "45.0:o__Spirochaetales # children taxa: 29\n",
      "length new taxa list: 10444\n",
      "100.0:o__Sphaerochaetales # children taxa: 22\n",
      "length new taxa list: 10423\n",
      "100.0:o__Borreliales; f__Borreliaceae # children taxa: 20\n",
      "length new taxa list: 10404\n",
      "100.0:c__GWE2-31-10; o__GWE2-31-10; f__GWE2-31-10 # children taxa: 2\n",
      "length new taxa list: 10403\n",
      "100.0:c__Brevinematia; o__Brevinematales # children taxa: 3\n",
      "length new taxa list: 10401\n",
      "100.0:c__Brachyspirae; o__Brachyspirales; f__Brachyspiraceae; g__Brachyspira # children taxa: 12\n",
      "length new taxa list: 10390\n",
      "100.0:o__Leptospirales # children taxa: 46\n",
      "length new taxa list: 10345\n",
      "100.0:c__UBA4802; o__UBA4802 # children taxa: 9\n",
      "length new taxa list: 10337\n",
      "100.0:p__UBP7; c__UBA6624; o__UBA6624; f__UBA6624; g__UBA6624 # children taxa: 2\n",
      "length new taxa list: 10336\n",
      "88.0:o__Flavobacteriales # children taxa: 1045\n",
      "length new taxa list: 9292\n",
      "98.0:o__AKYH767 # children taxa: 12\n",
      "length new taxa list: 9281\n",
      "100.0:o__AKYH767-A # children taxa: 4\n",
      "length new taxa list: 9278\n",
      "100.0:o__Sphingobacteriales; f__Sphingobacteriaceae # children taxa: 145\n",
      "length new taxa list: 9134\n",
      "99.0:o__NS11-12g # children taxa: 18\n",
      "length new taxa list: 9117\n",
      "99.0:o__Cytophagales # children taxa: 250\n",
      "length new taxa list: 8868\n",
      "100.0:o__Chitinophagales # children taxa: 172\n",
      "length new taxa list: 8697\n",
      "100.0:o__Rhodothermales # children taxa: 31\n",
      "length new taxa list: 8667\n",
      "100.0:o__Balneolales # children taxa: 25\n",
      "length new taxa list: 8643\n",
      "100.0:c__Chlorobia; o__Chlorobiales # children taxa: 23\n",
      "length new taxa list: 8621\n",
      "100.0:o__Ignavibacteriales # children taxa: 28\n",
      "length new taxa list: 8594\n",
      "100.0:o__SJA-28 # children taxa: 8\n",
      "length new taxa list: 8587\n",
      "99.0:o__UBA10030 # children taxa: 22\n",
      "length new taxa list: 8566\n",
      "100.0:c__Kryptonia; o__Kryptoniales; f__Kryptoniaceae # children taxa: 6\n",
      "length new taxa list: 8561\n",
      "100.0:o__Kapabacteriales # children taxa: 16\n",
      "length new taxa list: 8546\n",
      "100.0:o__Palsa-1295; f__Palsa-1295 # children taxa: 2\n",
      "length new taxa list: 8545\n",
      "100.0:c__SZUA-365; o__SZUA-365 # children taxa: 3\n",
      "length new taxa list: 8543\n",
      "100.0:o__JdFR-76 # children taxa: 3\n",
      "length new taxa list: 8541\n",
      "100.0:o__AABM5-25-91; f__AABM5-25-91 # children taxa: 2\n",
      "length new taxa list: 8540\n",
      "100.0:o__Marinisomatales # children taxa: 58\n",
      "length new taxa list: 8483\n",
      "100.0:o__SCGC-AAA003-L08 # children taxa: 23\n",
      "length new taxa list: 8461\n",
      "100.0:c__AB16; o__AB16 # children taxa: 3\n",
      "length new taxa list: 8459\n",
      "100.0:c__UBA2242; o__UBA2242; f__UBA2242 # children taxa: 3\n",
      "length new taxa list: 8457\n",
      "100.0:p__Delongbacteria; c__UBA4055; o__UBA4055; f__UBA4055 # children taxa: 4\n",
      "length new taxa list: 8454\n",
      "100.0:o__Fibrobacterales; f__Fibrobacteraceae # children taxa: 36\n",
      "length new taxa list: 8419\n",
      "100.0:c__OXYB2-FULL-49-7; o__OXYB2-FULL-49-7 # children taxa: 2\n",
      "length new taxa list: 8418\n",
      "100.0:c__Chitinivibrionia; o__Chitinivibrionales # children taxa: 5\n",
      "length new taxa list: 8414\n",
      "100.0:o__Cloacimonadales # children taxa: 20\n",
      "length new taxa list: 8395\n",
      "100.0:o__JGIOTU-2; f__JGIOTU-2 # children taxa: 3\n",
      "length new taxa list: 8393\n",
      "100.0:o__Gemmatimonadales # children taxa: 56\n",
      "length new taxa list: 8338\n",
      "100.0:o__SG8-23; f__UBA6960 # children taxa: 18\n",
      "length new taxa list: 8321\n",
      "100.0:o__LZORAL124-64-63; f__LZORAL124-64-63 # children taxa: 2\n",
      "length new taxa list: 8320\n",
      "100.0:o__SSS58A; f__SSS58A # children taxa: 3\n",
      "length new taxa list: 8318\n",
      "100.0:p__Edwardsbacteria; c__AC1; o__AC1 # children taxa: 4\n",
      "length new taxa list: 8315\n",
      "100.0:o__UBA2968 # children taxa: 4\n",
      "length new taxa list: 8312\n",
      "100.0:o__UBA8231 # children taxa: 5\n",
      "length new taxa list: 8308\n",
      "100.0:o__UBA10806 # children taxa: 3\n",
      "length new taxa list: 8306\n",
      "100.0:o__GN15 # children taxa: 2\n",
      "length new taxa list: 8305\n",
      "100.0:o__MSB-5A5 # children taxa: 5\n",
      "length new taxa list: 8301\n",
      "100.0:o__UBA1063; f__UBA1063; g__UBA1063 # children taxa: 2\n",
      "length new taxa list: 8300\n",
      "100.0:p__Calescibacterota; c__Calescibacteriia; o__Calescibacteriales; f__Calescibacteriaceae; g__Calescibacterium # children taxa: 2\n",
      "length new taxa list: 8299\n",
      "100.0:o__SM23-42; f__SM23-42 # children taxa: 4\n",
      "length new taxa list: 8296\n",
      "100.0:o__UBA2258; f__UBA2258 # children taxa: 2\n",
      "length new taxa list: 8295\n",
      "100.0:c__UBA3073; o__UBA3073; f__UBA3073; g__UBA3073 # children taxa: 2\n",
      "length new taxa list: 8294\n",
      "100.0:c__32-111; o__32-111; f__32-111 # children taxa: 3\n",
      "length new taxa list: 8292\n",
      "100.0:p__Fermentibacterota; c__Fermentibacteria; o__Fermentibacterales; f__Fermentibacteraceae # children taxa: 6\n",
      "length new taxa list: 8287\n",
      "100.0:o__WGA-4E # children taxa: 7\n",
      "length new taxa list: 8281\n",
      "100.0:p__Hydrogenedentota; c__Hydrogenedentia; o__Hydrogenedentiales # children taxa: 4\n",
      "length new taxa list: 8278\n",
      "100.0:o__Sumerlaeales; f__Sumerlaeaceae # children taxa: 2\n",
      "length new taxa list: 8277\n",
      "90.0:o__Pseudomonadales # children taxa: 1270\n",
      "length new taxa list: 7008\n",
      "61.0:o__Enterobacterales # children taxa: 1186\n",
      "length new taxa list: 5823\n",
      "88.0:o__Legionellales; f__Legionellaceae # children taxa: 67\n",
      "length new taxa list: 5757\n",
      "100.0:o__UBA1113; f__UBA1113 # children taxa: 2\n",
      "length new taxa list: 5756\n",
      "100.0:o__UBA5158; f__UBA5158 # children taxa: 9\n",
      "length new taxa list: 5748\n",
      "100.0:o__Diplorickettsiales; f__Diplorickettsiaceae # children taxa: 5\n",
      "length new taxa list: 5744\n",
      "86.0:o__UBA12402; f__UBA12402 # children taxa: 6\n",
      "length new taxa list: 5739\n",
      "100.0:o__Coxiellales; f__Coxiellaceae # children taxa: 13\n",
      "length new taxa list: 5727\n",
      "100.0:o__UBA9339; f__UBA9339 # children taxa: 3\n",
      "length new taxa list: 5725\n",
      "100.0:o__Francisellales; f__Francisellaceae # children taxa: 17\n",
      "length new taxa list: 5709\n",
      "100.0:o__Evansiales; f__Evansiaceae # children taxa: 2\n",
      "length new taxa list: 5708\n",
      "100.0:o__Berkiellales # children taxa: 4\n",
      "length new taxa list: 5705\n",
      "100.0:o__Piscirickettsiales; f__Piscirickettsiaceae; g__Piscirickettsia # children taxa: 3\n",
      "length new taxa list: 5703\n",
      "54.0:o__Thiomicrospirales # children taxa: 43\n",
      "length new taxa list: 5661\n",
      "100.0:o__Cardiobacteriales # children taxa: 14\n",
      "length new taxa list: 5648\n",
      "100.0:o__Xanthomonadales # children taxa: 270\n",
      "length new taxa list: 5379\n",
      "100.0:o__Steroidobacterales; f__Steroidobacteraceae # children taxa: 17\n",
      "length new taxa list: 5363\n",
      "100.0:o__Woeseiales; f__Woeseiaceae # children taxa: 8\n",
      "length new taxa list: 5356\n",
      "100.0:o__GCA-2729495; f__GCA-2729495 # children taxa: 3\n",
      "length new taxa list: 5354\n",
      "100.0:o__UBA6522; f__UBA6522 # children taxa: 5\n",
      "length new taxa list: 5350\n",
      "100.0:o__Nevskiales # children taxa: 27\n",
      "length new taxa list: 5324\n",
      "100.0:o__UBA5335; f__UBA5335; g__UBA5335 # children taxa: 3\n",
      "length new taxa list: 5322\n",
      "100.0:o__Nitrococcales # children taxa: 10\n",
      "length new taxa list: 5313\n",
      "100.0:o__Ectothiorhodospirales # children taxa: 35\n",
      "length new taxa list: 5279\n",
      "100.0:o__Thiotrichales; f__Thiotrichaceae # children taxa: 13\n",
      "length new taxa list: 5267\n",
      "100.0:o__GCA-001735895; f__GCA-001735895; g__GCA-001735895 # children taxa: 2\n",
      "length new taxa list: 5266\n",
      "100.0:o__UBA4575; f__UBA4575 # children taxa: 2\n",
      "length new taxa list: 5265\n",
      "100.0:o__Halothiobacillales; f__Halothiobacillaceae # children taxa: 6\n",
      "length new taxa list: 5260\n",
      "98.0:o__Granulosicoccales; f__Granulosicoccaceae # children taxa: 5\n",
      "length new taxa list: 5256\n",
      "100.0:o__SZUA-229; f__SZUA-229 # children taxa: 4\n",
      "length new taxa list: 5253\n",
      "100.0:o__Methylococcales # children taxa: 70\n",
      "length new taxa list: 5184\n",
      "100.0:o__Chromatiales # children taxa: 40\n",
      "length new taxa list: 5145\n",
      "100.0:o__Nitrosococcales # children taxa: 24\n",
      "length new taxa list: 5122\n",
      "100.0:o__Beggiatoales; f__Beggiatoaceae # children taxa: 8\n",
      "length new taxa list: 5115\n",
      "53.0:o__Thiohalomonadales # children taxa: 13\n",
      "length new taxa list: 5103\n",
      "100.0:o__SZUA-140; f__SZUA-140 # children taxa: 3\n",
      "length new taxa list: 5101\n",
      "100.0:o__Ga0077554; f__Ga007554; g__Ga0077554 # children taxa: 2\n",
      "length new taxa list: 5100\n",
      "100.0:o__Tenderiales; f__Tenderiaceae # children taxa: 2\n",
      "length new taxa list: 5099\n",
      "100.0:o__Competibacterales; f__Competibacteraceae # children taxa: 10\n",
      "length new taxa list: 5090\n",
      "100.0:o__21-64-14; f__21-64-14 # children taxa: 2\n",
      "length new taxa list: 5089\n",
      "99.0:o__Thiohalobacterales; f__Thiohalobacteraceae # children taxa: 4\n",
      "length new taxa list: 5086\n",
      "100.0:o__SZUA-21; f__SZUA-21 # children taxa: 2\n",
      "length new taxa list: 5085\n",
      "100.0:o__UBA4486; f__UBA4486 # children taxa: 11\n",
      "length new taxa list: 5075\n",
      "42.0:o__Ga0077536; f__Ga0077536 # children taxa: 4\n",
      "length new taxa list: 5072\n",
      "60.0:o__UBA7966; f__UBA7966 # children taxa: 3\n",
      "length new taxa list: 5070\n",
      "100.0:o__Immundisolibacterales; f__Immundisolibacteraceae; g__Immundisolibacter # children taxa: 2\n",
      "length new taxa list: 5069\n",
      "97.0:o__Burkholderiales # children taxa: 1273\n",
      "length new taxa list: 3797\n",
      "100.0:o__Acidithiobacillales; f__Acidithiobacillaceae # children taxa: 9\n",
      "length new taxa list: 3789\n",
      "100.0:o__UBA10353 # children taxa: 23\n",
      "length new taxa list: 3767\n",
      "100.0:o__Acidiferrobacterales # children taxa: 12\n",
      "length new taxa list: 3756\n",
      "100.0:o__SAR86 # children taxa: 32\n",
      "length new taxa list: 3725\n",
      "100.0:o__GCA-002705445 # children taxa: 8\n",
      "length new taxa list: 3718\n",
      "100.0:o__UBA11654; f__UBA11654; g__UBA11654 # children taxa: 6\n",
      "length new taxa list: 3713\n",
      "100.0:o__UBA7916 # children taxa: 8\n",
      "length new taxa list: 3706\n",
      "100.0:o__Mariprofundales; f__Mariprofundaceae # children taxa: 17\n",
      "length new taxa list: 3690\n",
      "98.0:o__Parvibaculales # children taxa: 23\n",
      "length new taxa list: 3668\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "100.0:o__Caulobacterales # children taxa: 206\n",
      "length new taxa list: 3463\n",
      "100.0:o__UBA1301; f__UBA1301 # children taxa: 10\n",
      "length new taxa list: 3454\n",
      "94.0:o__Sphingomonadales # children taxa: 411\n",
      "length new taxa list: 3044\n",
      "100.0:o__Ferrovibrionales; f__Ferrovibrionaceae; g__Ferrovibrio # children taxa: 2\n",
      "length new taxa list: 3043\n",
      "100.0:o__Zavarziniales; f__Zavarziniaceae; g__Zavarzinia # children taxa: 2\n",
      "length new taxa list: 3042\n",
      "100.0:o__Sneathiellales; f__Sneathiellaceae # children taxa: 7\n",
      "length new taxa list: 3036\n",
      "100.0:o__UBA2966; f__UBA2966 # children taxa: 2\n",
      "length new taxa list: 3035\n",
      "99.0:o__UBA7887 # children taxa: 6\n",
      "length new taxa list: 3030\n",
      "100.0:o__UBA828; f__UBA828 # children taxa: 5\n",
      "length new taxa list: 3026\n",
      "100.0:o__Geminicoccales; f__Geminicoccaceae # children taxa: 4\n",
      "length new taxa list: 3023\n",
      "100.0:o__SP197; f__SP197; g__SP197 # children taxa: 2\n",
      "length new taxa list: 3022\n",
      "100.0:o__Acetobacterales; f__Acetobacteraceae # children taxa: 114\n",
      "length new taxa list: 2909\n",
      "100.0:o__Reyranellales; f__Reyranellaceae # children taxa: 8\n",
      "length new taxa list: 2902\n",
      "46.0:o__Micavibrionales # children taxa: 49\n",
      "length new taxa list: 2854\n",
      "72.0:o__Azospirillales # children taxa: 33\n",
      "length new taxa list: 2822\n",
      "59.0:o__Elsterales # children taxa: 3\n",
      "length new taxa list: 2820\n",
      "86.0:o__Nisaeales; f__Nisaeaceae # children taxa: 10\n",
      "length new taxa list: 2811\n",
      "100.0:o__TMED26; f__TMED26; g__TMED26 # children taxa: 2\n",
      "length new taxa list: 2810\n",
      "100.0:o__Thalassobaculales; f__Thalassobaculaceae # children taxa: 4\n",
      "length new taxa list: 2807\n",
      "100.0:o__Oceanibaculales; f__Oceanibaculaceae; g__Oceanibaculum # children taxa: 3\n",
      "length new taxa list: 2805\n",
      "100.0:o__Kiloniellales # children taxa: 11\n",
      "length new taxa list: 2795\n",
      "100.0:o__UBA7985; f__UBA7985; g__UBA7985 # children taxa: 5\n",
      "length new taxa list: 2791\n",
      "100.0:o__UBA8366 # children taxa: 12\n",
      "length new taxa list: 2780\n",
      "100.0:o__Defluviicoccales # children taxa: 11\n",
      "length new taxa list: 2770\n",
      "100.0:o__UBA6615; f__UBA6615 # children taxa: 6\n",
      "length new taxa list: 2765\n",
      "100.0:o__Bin65; f__Bin65; g__Bin65 # children taxa: 2\n",
      "length new taxa list: 2764\n",
      "91.0:o__UBA11136; f__UBA11136 # children taxa: 2\n",
      "length new taxa list: 2763\n",
      "66.0:o__Rhodospirillales_A # children taxa: 53\n",
      "length new taxa list: 2711\n",
      "97.0:o__Rhodospirillales_B # children taxa: 17\n",
      "length new taxa list: 2695\n",
      "16.0:o__Rhodospirillales; f__Rhodospirillaceae # children taxa: 15\n",
      "length new taxa list: 2681\n",
      "100.0:o__Rhodospirillales_C; f__Rhodospirillaceae_A # children taxa: 2\n",
      "length new taxa list: 2680\n",
      "99.0:o__RF32 # children taxa: 27\n",
      "length new taxa list: 2654\n",
      "100.0:o__Caedimonadales # children taxa: 7\n",
      "length new taxa list: 2648\n",
      "100.0:o__UBA9655; f__UBA9655; g__UBA9655 # children taxa: 2\n",
      "length new taxa list: 2647\n",
      "100.0:o__Paracaedibacterales # children taxa: 5\n",
      "length new taxa list: 2643\n",
      "97.0:o__Holosporales # children taxa: 5\n",
      "length new taxa list: 2639\n",
      "93.0:o__UBA7879 # children taxa: 4\n",
      "length new taxa list: 2636\n",
      "96.0:o__Puniceispirillales # children taxa: 38\n",
      "length new taxa list: 2599\n",
      "100.0:o__Pelagibacterales # children taxa: 123\n",
      "length new taxa list: 2477\n",
      "100.0:o__HIMB59 # children taxa: 15\n",
      "length new taxa list: 2463\n",
      "100.0:o__GCA-002720895; f__GCA-002720895 # children taxa: 5\n",
      "length new taxa list: 2459\n",
      "100.0:o__TMED127; f__TMED127 # children taxa: 14\n",
      "length new taxa list: 2446\n",
      "100.0:o__TMED109 # children taxa: 13\n",
      "length new taxa list: 2434\n",
      "100.0:o__GCA-2701885; f__GCA-2701885; g__GCA-2701885 # children taxa: 3\n",
      "length new taxa list: 2432\n",
      "23.0:o__Rickettsiales # children taxa: 62\n",
      "length new taxa list: 2371\n",
      "100.0:o__UBA998 # children taxa: 9\n",
      "length new taxa list: 2363\n",
      "100.0:o__Rs-D84 # children taxa: 12\n",
      "length new taxa list: 2352\n",
      "100.0:o__IN39; f__IN39; g__IN39 # children taxa: 2\n",
      "length new taxa list: 2351\n",
      "100.0:o__Rickettsiales_A; f__UBA4311; g__UBA4311 # children taxa: 2\n",
      "length new taxa list: 2350\n",
      "100.0:o__UBA1280 # children taxa: 8\n",
      "length new taxa list: 2343\n",
      "100.0:c__Magnetococcia; o__Magnetococcales # children taxa: 12\n",
      "length new taxa list: 2332\n",
      "100.0:p__Dependentiae; c__Babeliae; o__Babeliales # children taxa: 22\n",
      "length new taxa list: 2311\n",
      "100.0:p__AB1-6; c__AB1-6; o__AB1-6 # children taxa: 2\n",
      "length new taxa list: 2310\n",
      "100.0:p__UBP6; c__UBA1177; o__UBA1177; f__UBA1177; g__UBA1177 # children taxa: 3\n",
      "length new taxa list: 2308\n",
      "100.0:o__SAR324 # children taxa: 17\n",
      "length new taxa list: 2292\n",
      "100.0:o__XYD2-FULL-50-16 # children taxa: 4\n",
      "length new taxa list: 2289\n",
      "95.0:o__Campylobacterales # children taxa: 244\n",
      "length new taxa list: 2046\n",
      "100.0:o__Nautiliales; f__Nautiliaceae # children taxa: 4\n",
      "length new taxa list: 2043\n",
      "100.0:c__Desulfurellia; o__Desulfurellales # children taxa: 6\n",
      "length new taxa list: 2038\n",
      "100.0:o__Aquificales # children taxa: 12\n",
      "length new taxa list: 2027\n",
      "100.0:o__Hydrogenothermales; f__Hydrogenothermaceae # children taxa: 9\n",
      "length new taxa list: 2019\n",
      "100.0:c__Desulfurobacteriia; o__Desulfurobacteriales; f__Desulfurobacteriaceae # children taxa: 5\n",
      "length new taxa list: 2015\n",
      "95.0:o__Deferribacterales # children taxa: 13\n",
      "length new taxa list: 2003\n",
      "100.0:p__Chrysiogenetota; c__Chrysiogenetes; o__Chrysiogenales; f__Chrysiogenaceae # children taxa: 2\n",
      "length new taxa list: 2002\n",
      "100.0:o__UBA1144; f__UBA1144 # children taxa: 9\n",
      "length new taxa list: 1994\n",
      "100.0:o__UBA2774 # children taxa: 5\n",
      "length new taxa list: 1990\n",
      "100.0:o__Desulfobacterales # children taxa: 54\n",
      "length new taxa list: 1937\n",
      "97.0:o__C00003060 # children taxa: 3\n",
      "length new taxa list: 1935\n",
      "100.0:o__Desulfatiglandales # children taxa: 6\n",
      "length new taxa list: 1930\n",
      "100.0:o__Syntrophobacterales # children taxa: 11\n",
      "length new taxa list: 1920\n",
      "100.0:o__Desulfarculales; f__Desulfarculaceae # children taxa: 4\n",
      "length new taxa list: 1917\n",
      "100.0:c__Desulfobaccia; o__Desulfobaccales # children taxa: 3\n",
      "length new taxa list: 1915\n",
      "100.0:c__Desulfobulbia; o__Desulfobulbales # children taxa: 49\n",
      "length new taxa list: 1867\n",
      "100.0:c__Thermodesulfobacteria; o__Thermodesulfobacteriales # children taxa: 13\n",
      "length new taxa list: 1855\n",
      "100.0:c__Dissulfuribacteria; o__Dissulfuribacterales # children taxa: 5\n",
      "length new taxa list: 1851\n",
      "100.0:c__Desulfofervidia; o__Desulfofervidales # children taxa: 2\n",
      "length new taxa list: 1850\n",
      "100.0:c__Syntrophia; o__Syntrophales # children taxa: 53\n",
      "length new taxa list: 1798\n",
      "99.0:o__BSN033 # children taxa: 8\n",
      "length new taxa list: 1791\n",
      "100.0:o__SM23-61; f__SM23-61; g__SM23-61 # children taxa: 2\n",
      "length new taxa list: 1790\n",
      "100.0:c__Syntrophorhabdia; o__Syntrophorhabdales # children taxa: 13\n",
      "length new taxa list: 1778\n",
      "100.0:o__UBA1062; f__UBA1062; g__UBA1062 # children taxa: 3\n",
      "length new taxa list: 1776\n",
      "100.0:o__Desulfuromonadales # children taxa: 45\n",
      "length new taxa list: 1732\n",
      "100.0:o__Geobacterales # children taxa: 36\n",
      "length new taxa list: 1697\n",
      "100.0:p__GWC2-55-46; c__GWC2-55-46; o__GWC2-55-46 # children taxa: 8\n",
      "length new taxa list: 1690\n",
      "100.0:o__Thermodesulfovibrionales # children taxa: 37\n",
      "length new taxa list: 1654\n",
      "100.0:o__UBA6902; f__UBA6902 # children taxa: 5\n",
      "length new taxa list: 1650\n",
      "100.0:o__Nitrospirales # children taxa: 40\n",
      "length new taxa list: 1611\n",
      "100.0:o__HDB-SIOI813; f__HDB-SIOI813 # children taxa: 2\n",
      "length new taxa list: 1610\n",
      "100.0:c__UBA9217; o__UBA9217; f__UBA9217 # children taxa: 3\n",
      "length new taxa list: 1608\n",
      "100.0:o__Rokubacteriales; f__GWA2-73-35 # children taxa: 31\n",
      "length new taxa list: 1578\n",
      "100.0:o__Methylomirabilales # children taxa: 9\n",
      "length new taxa list: 1570\n",
      "100.0:p__Moduliflexota; c__Moduliflexia; o__Moduliflexales; f__Moduliflexaceae # children taxa: 3\n",
      "length new taxa list: 1568\n",
      "100.0:p__SZUA-182; c__SZUA-182; o__SZUA-182 # children taxa: 2\n",
      "length new taxa list: 1567\n",
      "100.0:c__Nitrospinia; o__Nitrospinales; f__Nitrospinaceae # children taxa: 16\n",
      "length new taxa list: 1552\n",
      "100.0:c__UBA9942; o__UBA9942 # children taxa: 4\n",
      "length new taxa list: 1549\n",
      "100.0:p__UBA8248; c__UBA8248; o__UBA8248 # children taxa: 4\n",
      "length new taxa list: 1546\n",
      "100.0:p__Tectomicrobia; c__Entotheonellia; o__Entotheonellales; f__Entotheonellaceae; g__Entotheonella # children taxa: 2\n",
      "length new taxa list: 1545\n",
      "100.0:p__Schekmanbacteria; c__GWA2-38-11; o__GWA2-38-11; f__GWA2-38-11 # children taxa: 3\n",
      "length new taxa list: 1543\n",
      "100.0:p__BMS3Abin14; c__BMS3Abin14; o__BMS3Abin14; f__BMS3Abin14 # children taxa: 3\n",
      "length new taxa list: 1541\n",
      "100.0:o__Binatales; f__Binataceae; g__Binatus # children taxa: 4\n",
      "length new taxa list: 1538\n",
      "100.0:o__UBA9968; f__UBA9968 # children taxa: 6\n",
      "length new taxa list: 1533\n",
      "100.0:p__Deferrisomatota; c__Defferisomatia; o__Defferisomatales # children taxa: 3\n",
      "length new taxa list: 1531\n",
      "100.0:p__MBNT15; c__MBNT15; o__MBNT15; f__MBNT15 # children taxa: 9\n",
      "length new taxa list: 1523\n",
      "99.0:o__Palsa-1104 # children taxa: 15\n",
      "length new taxa list: 1509\n",
      "100.0:o__Haliangiales; f__Haliangiaceae # children taxa: 2\n",
      "length new taxa list: 1508\n",
      "100.0:o__Nannocystales; f__Nannocystaceae # children taxa: 6\n",
      "length new taxa list: 1503\n",
      "100.0:o__HGW-17; f__HGW-17 # children taxa: 3\n",
      "length new taxa list: 1501\n",
      "100.0:o__Polyangiales # children taxa: 21\n",
      "length new taxa list: 1481\n",
      "100.0:c__Myxococcia; o__Myxococcales # children taxa: 30\n",
      "length new taxa list: 1452\n",
      "100.0:o__UBA4151; f__UBA4151; g__UBA4151 # children taxa: 2\n",
      "length new taxa list: 1451\n",
      "100.0:o__Bradymonadales # children taxa: 5\n",
      "length new taxa list: 1447\n",
      "100.0:o__UBA7976; f__UBA1532 # children taxa: 2\n",
      "length new taxa list: 1446\n",
      "100.0:o__UBA9160 # children taxa: 10\n",
      "length new taxa list: 1437\n",
      "100.0:o__UBA796 # children taxa: 6\n",
      "length new taxa list: 1432\n",
      "100.0:o__UBA9615; f__UBA9615 # children taxa: 2\n",
      "length new taxa list: 1431\n",
      "100.0:o__Oligoflexales # children taxa: 6\n",
      "length new taxa list: 1426\n",
      "100.0:o__Bacteriovoracales; f__Bacteriovoracaceae # children taxa: 39\n",
      "length new taxa list: 1388\n",
      "100.0:o__UBA1018 # children taxa: 7\n",
      "length new taxa list: 1382\n",
      "100.0:o__UBA2394; f__UBA2394 # children taxa: 2\n",
      "length new taxa list: 1381\n",
      "100.0:c__Bdellovibrionia; o__Bdellovibrionales # children taxa: 29\n",
      "length new taxa list: 1353\n",
      "100.0:o__2-02-FULL-40-11; f__2-02-FULL-40-11 # children taxa: 2\n",
      "length new taxa list: 1352\n",
      "100.0:o__2-02-FULL-44-16; f__2-02-FULL-44-16 # children taxa: 2\n",
      "length new taxa list: 1351\n",
      "100.0:o__GCA-002796325 # children taxa: 2\n",
      "length new taxa list: 1350\n",
      "100.0:o__UBA10199 # children taxa: 2\n",
      "length new taxa list: 1349\n",
      "100.0:p__Desulfobacterota_A; c__Desulfovibrionia; o__Desulfovibrionales # children taxa: 103\n",
      "length new taxa list: 1247\n",
      "100.0:p__Nitrospirota_A; c__Leptospirillia; o__Leptospirillales; f__Leptospirillaceae # children taxa: 4\n",
      "length new taxa list: 1244\n",
      "100.0:o__Acidobacteriales # children taxa: 130\n",
      "length new taxa list: 1115\n",
      "100.0:o__Bryobacterales # children taxa: 33\n",
      "length new taxa list: 1083\n",
      "100.0:o__2-12-FULL-54-10; f__2-12-FULL-54-10 # children taxa: 2\n",
      "length new taxa list: 1082\n",
      "100.0:o__UBA7541 # children taxa: 36\n",
      "length new taxa list: 1047\n",
      "100.0:o__UBA7540; f__UBA7540 # children taxa: 8\n",
      "length new taxa list: 1040\n",
      "100.0:o__20CM-2-55-15; f__20CM-2-55-15; g__20CM-2-55-15 # children taxa: 3\n",
      "length new taxa list: 1038\n",
      "100.0:o__Pyrinomonadales; f__Pyrinomonadaceae # children taxa: 23\n",
      "length new taxa list: 1016\n",
      "100.0:o__UBA7656; f__UBA7656 # children taxa: 2\n",
      "length new taxa list: 1015\n",
      "100.0:o__Chloracidobacteriales; f__Chloracidobacteriaceae; g__Chloracidobacterium # children taxa: 3\n",
      "length new taxa list: 1013\n",
      "100.0:o__HR10; f__HR10; g__HR10 # children taxa: 2\n",
      "length new taxa list: 1012\n",
      "100.0:c__UBA6911; o__UBA6911; f__UBA6911 # children taxa: 2\n",
      "length new taxa list: 1011\n",
      "100.0:o__Vicinamibacterales # children taxa: 35\n",
      "length new taxa list: 977\n",
      "100.0:o__Fen-336; f__Fen-336 # children taxa: 4\n",
      "length new taxa list: 974\n",
      "100.0:o__Gp22-AA2 # children taxa: 4\n",
      "length new taxa list: 971\n",
      "100.0:c__UBA890; o__UBA890; f__UBA890; g__UBA890 # children taxa: 2\n",
      "length new taxa list: 970\n",
      "100.0:o__Thermoanaerobaculales # children taxa: 6\n",
      "length new taxa list: 965\n",
      "100.0:o__UBA5066 # children taxa: 3\n",
      "length new taxa list: 963\n",
      "100.0:o__Gp7-AA8; f__Gp7-AA8 # children taxa: 4\n",
      "length new taxa list: 960\n",
      "100.0:o__UBA5704; f__UBA5704 # children taxa: 2\n",
      "length new taxa list: 959\n",
      "100.0:c__Holophagae; o__Holophagales; f__Holophagaceae # children taxa: 9\n",
      "length new taxa list: 951\n",
      "100.0:o__Aminicenantales # children taxa: 13\n",
      "length new taxa list: 939\n",
      "100.0:o__UBA2199; f__UBA2199 # children taxa: 5\n",
      "length new taxa list: 935\n",
      "81.0:o__GIF10 # children taxa: 15\n",
      "length new taxa list: 921\n",
      "100.0:o__UBA1572; f__UBA1572 # children taxa: 3\n",
      "length new taxa list: 919\n",
      "100.0:o__UBA10015 # children taxa: 10\n",
      "length new taxa list: 910\n",
      "100.0:o__4484-171; f__4484-171 # children taxa: 2\n",
      "length new taxa list: 909\n",
      "100.0:o__UBA6215; f__UBA6215 # children taxa: 3\n",
      "length new taxa list: 907\n",
      "100.0:o__UBA10183; f__UBA10183; g__UBA10183 # children taxa: 4\n",
      "length new taxa list: 904\n",
      "96.0:o__UBA1560 # children taxa: 7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "length new taxa list: 898\n",
      "100.0:o__UBA9649; f__UBA9649 # children taxa: 3\n",
      "length new taxa list: 896\n",
      "100.0:o__2-02-FULL-51-18; f__2-02-FULL-51-18 # children taxa: 2\n",
      "length new taxa list: 895\n",
      "100.0:c__Omnitrophia; o__Omnitrophales # children taxa: 11\n",
      "length new taxa list: 885\n",
      "100.0:p__Goldbacteria; c__PGYV01; o__PGYV01 # children taxa: 3\n",
      "length new taxa list: 883\n",
      "100.0:p__Firestonebacteria; c__D2-FULL-39-29; o__D2-FULL-39-29; f__D2-FULL-39-29; g__D2-FULL-39-29 # children taxa: 2\n",
      "length new taxa list: 882\n",
      "100.0:p__FCPU426; c__Palsa-1180; o__Palsa-1180; f__Palsa-1180 # children taxa: 2\n",
      "length new taxa list: 881\n",
      "100.0:p__Desantisbacteria; c__UBA1551; o__UBA1551 # children taxa: 2\n",
      "length new taxa list: 880\n",
      "100.0:o__Elusimicrobiales # children taxa: 29\n",
      "length new taxa list: 852\n",
      "78.0:o__UBA1565 # children taxa: 9\n",
      "length new taxa list: 844\n",
      "100.0:o__2-02-FULL-39-32 # children taxa: 4\n",
      "length new taxa list: 841\n",
      "100.0:o__F11 # children taxa: 3\n",
      "length new taxa list: 839\n",
      "100.0:o__Endomicrobiales # children taxa: 7\n",
      "length new taxa list: 833\n",
      "100.0:o__CG1-02-37-114 # children taxa: 3\n",
      "length new taxa list: 831\n",
      "100.0:c__UBA8919; o__UBA8919; f__UBA8919 # children taxa: 4\n",
      "length new taxa list: 828\n",
      "100.0:p__UBA6262; c__UBA6262; o__UBA6262; f__UBA6262 # children taxa: 2\n",
      "length new taxa list: 827\n",
      "100.0:p__CG03; c__CG03; o__CG03; f__CG03 # children taxa: 3\n",
      "length new taxa list: 825\n",
      "825\n"
     ]
    }
   ],
   "source": [
    "parsed_arc_train_order = arc_train.copy()\n",
    "parse_taxa_from_node(arc.seed_node,parsed_arc_train_order,genomes_per_taxa=1,prefix='o__')\n",
    "print(len(parsed_arc_train_order))\n",
    "\n",
    "parsed_bac_train_order = bac_train.copy()\n",
    "parse_taxa_from_node(bac.seed_node,parsed_bac_train_order,genomes_per_taxa=1,prefix='o__')\n",
    "print(len(parsed_bac_train_order))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19945 825\n",
      "1010 76\n",
      "901\n"
     ]
    }
   ],
   "source": [
    "print(len(bac_train),len(parsed_bac_train_order))\n",
    "print(len(arc_train),len(parsed_arc_train_order))\n",
    "print(len(parsed_bac_train_order)+len(parsed_arc_train_order))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "901\n",
      "901\n"
     ]
    }
   ],
   "source": [
    "traindf_order = get_df_subset(ftp,arcftp,parsed_bac_train_order,parsed_arc_train_order)\n",
    "print(len(traindf_order))\n",
    "\n",
    "train_fnames_order = get_filenames(traindf_order)\n",
    "print(len(train_fnames_order))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "traindf_order.to_csv('TrainNew_ParsedOrder_GTDBrepwithftp.csv')\n",
    "save_filenames('trainnew_parsedorder_filenames.txt',train_fnames_order)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## aaaaand class level?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "100.0:c__Halobacteria; o__Halobacteriales # children taxa: 229\n",
      "length new taxa list: 782\n",
      "100.0:c__Methanomicrobia; o__Methanomicrobiales # children taxa: 71\n",
      "length new taxa list: 712\n",
      "100.0:c__Bog-38; o__Bog-38; f__Bog-38; g__Bog-38 # children taxa: 4\n",
      "length new taxa list: 709\n",
      "97.0:c__Methanosarcinia # children taxa: 13\n",
      "length new taxa list: 697\n",
      "100.0:c__Methanocellia; o__Methanocellales # children taxa: 4\n",
      "length new taxa list: 694\n",
      "100.0:c__Syntrophoarchaeia # children taxa: 5\n",
      "length new taxa list: 690\n",
      "85.0:c__Archaeoglobi # children taxa: 18\n",
      "length new taxa list: 673\n",
      "100.0:c__Methanonatronarchaeia; o__Methanonatronarchaeales # children taxa: 3\n",
      "length new taxa list: 671\n",
      "95.0:c__Poseidoniia # children taxa: 205\n",
      "length new taxa list: 467\n",
      "100.0:c__E2 # children taxa: 12\n",
      "length new taxa list: 456\n",
      "92.0:c__UBA186 # children taxa: 2\n",
      "length new taxa list: 455\n",
      "100.0:c__Methanobacteria; o__Methanobacteriales # children taxa: 84\n",
      "length new taxa list: 372\n",
      "100.0:c__Methanococci; o__Methanococcales # children taxa: 18\n",
      "length new taxa list: 355\n",
      "100.0:c__Methanopyri; o__Methanopyrales; f__Methanopyraceae; g__Methanopyrus # children taxa: 2\n",
      "length new taxa list: 354\n",
      "100.0:c__Thermococci # children taxa: 46\n",
      "length new taxa list: 309\n",
      "100.0:p__Hydrothermarchaeota; c__Hydrothermarchaeia; o__Hydrothermarchaeales # children taxa: 5\n",
      "length new taxa list: 305\n",
      "100.0:p__Hadarchaeota; c__Hadarchaeia; o__Hadarchaeales; f__Hadarchaeaceae # children taxa: 2\n",
      "length new taxa list: 304\n",
      "80.0:c__Nitrososphaeria # children taxa: 13\n",
      "length new taxa list: 292\n",
      "79.0:c__Thermoprotei # children taxa: 93\n",
      "length new taxa list: 200\n",
      "100.0:c__Methanomethylicia; o__Methanomethyliales; f__Methanomethyliaceae # children taxa: 3\n",
      "length new taxa list: 198\n",
      "70.0:c__Lokiarchaeia # children taxa: 8\n",
      "length new taxa list: 191\n",
      "100.0:c__Heimdallarchaeia # children taxa: 6\n",
      "length new taxa list: 186\n",
      "77.0:p__Nanoarchaeota; c__Nanoarchaeia # children taxa: 101\n",
      "length new taxa list: 86\n",
      "100.0:p__Nanohaloarchaeota; c__Nanosalinia; o__Nanosalinales; f__Nanosalinaceae # children taxa: 11\n",
      "length new taxa list: 76\n",
      "100.0:p__Aenigmarchaeota; c__Aenigmarchaeia # children taxa: 7\n",
      "length new taxa list: 70\n",
      "100.0:p__UAP2; c__UBA543; o__UBA543 # children taxa: 3\n",
      "length new taxa list: 68\n",
      "100.0:p__Micrarchaeota; c__Micrarchaeia # children taxa: 15\n",
      "length new taxa list: 54\n",
      "100.0:p__Iainarchaeota; c__Iainarchaeia # children taxa: 12\n",
      "length new taxa list: 43\n",
      "100.0:p__Altiarchaeota; c__Altiarchaeia # children taxa: 10\n",
      "length new taxa list: 34\n",
      "34\n",
      "32.0:c__Clostridia # children taxa: 1836\n",
      "length new taxa list: 18110\n",
      "87.0:c__Mahellia # children taxa: 12\n",
      "length new taxa list: 18099\n",
      "97.0:c__Thermoanaerobacteria # children taxa: 28\n",
      "length new taxa list: 18072\n",
      "100.0:c__Thermovenabulia # children taxa: 10\n",
      "length new taxa list: 18063\n",
      "46.0:p__Firmicutes_I; c__Bacilli_A # children taxa: 283\n",
      "length new taxa list: 17781\n",
      "100.0:p__Firmicutes_J; c__Desulfuribacillia; o__Desulfuribacillales; f__Desulfuribacillaceae # children taxa: 2\n",
      "length new taxa list: 17780\n",
      "100.0:p__Firmicutes_K; c__Alicyclobacillia # children taxa: 30\n",
      "length new taxa list: 17751\n",
      "100.0:c__Dethiobacteria # children taxa: 8\n",
      "length new taxa list: 17744\n",
      "100.0:c__Natranaerobiia; o__Natranaerobiales; f__Natranaerobiaceae # children taxa: 2\n",
      "length new taxa list: 17743\n",
      "100.0:c__Proteinivoracia # children taxa: 6\n",
      "length new taxa list: 17738\n",
      "100.0:c__UBA994; o__UBA994 # children taxa: 3\n",
      "length new taxa list: 17736\n",
      "98.0:c__Limnochordia # children taxa: 14\n",
      "length new taxa list: 17723\n",
      "100.0:c__DTU065; o__DTU065; f__DTU065 # children taxa: 2\n",
      "length new taxa list: 17722\n",
      "100.0:c__DTU084; o__DTU084; f__DTU084 # children taxa: 4\n",
      "length new taxa list: 17719\n",
      "79.0:c__SHA-98 # children taxa: 8\n",
      "length new taxa list: 17712\n",
      "100.0:c__UBA4882 # children taxa: 7\n",
      "length new taxa list: 17706\n",
      "100.0:c__Sulfobacillia; o__Sulfobacillales; f__Sulfobacillaceae # children taxa: 7\n",
      "length new taxa list: 17700\n",
      "100.0:c__Symbiobacteriia; o__Symbiobacteriales # children taxa: 3\n",
      "length new taxa list: 17698\n",
      "100.0:c__Thermaerobacteria; o__Thermaerobacterales # children taxa: 4\n",
      "length new taxa list: 17695\n",
      "57.0:c__DTU015 # children taxa: 9\n",
      "length new taxa list: 17687\n",
      "93.0:c__UBA3569 # children taxa: 2\n",
      "length new taxa list: 17686\n",
      "100.0:p__Firmicutes_C; c__Negativicutes # children taxa: 153\n",
      "length new taxa list: 17534\n",
      "74.0:c__Desulfitobacteriia # children taxa: 33\n",
      "length new taxa list: 17502\n",
      "98.0:c__Peptococcia # children taxa: 14\n",
      "length new taxa list: 17489\n",
      "71.0:c__Moorellia # children taxa: 17\n",
      "length new taxa list: 17473\n",
      "100.0:c__Dehalobacteriia # children taxa: 7\n",
      "length new taxa list: 17467\n",
      "100.0:c__TC1; o__TC1 # children taxa: 2\n",
      "length new taxa list: 17466\n",
      "78.0:c__Syntrophomonadia # children taxa: 28\n",
      "length new taxa list: 17439\n",
      "94.0:c__Desulfotomaculia # children taxa: 52\n",
      "length new taxa list: 17388\n",
      "100.0:c__Thermincolia; o__Thermincolales # children taxa: 4\n",
      "length new taxa list: 17385\n",
      "100.0:p__Firmicutes_F; c__Halanaerobiia # children taxa: 20\n",
      "length new taxa list: 17366\n",
      "60.0:p__Firmicutes; c__Bacilli # children taxa: 1878\n",
      "length new taxa list: 15489\n",
      "100.0:p__Fusobacteriota; c__Fusobacteriia # children taxa: 54\n",
      "length new taxa list: 15436\n",
      "100.0:c__Cyanobacteriia # children taxa: 597\n",
      "length new taxa list: 14840\n",
      "100.0:c__Vampirovibrionia # children taxa: 53\n",
      "length new taxa list: 14788\n",
      "100.0:c__Sericytochromatia # children taxa: 4\n",
      "length new taxa list: 14785\n",
      "100.0:c__WOR-1 # children taxa: 9\n",
      "length new taxa list: 14777\n",
      "100.0:c__ZB3 # children taxa: 10\n",
      "length new taxa list: 14768\n",
      "99.0:c__GWF2-35-9 # children taxa: 2\n",
      "length new taxa list: 14767\n",
      "99.0:c__Actinobacteria # children taxa: 1791\n",
      "length new taxa list: 12977\n",
      "100.0:c__UBA4738; o__UBA4738 # children taxa: 3\n",
      "length new taxa list: 12975\n",
      "99.0:c__Acidimicrobiia # children taxa: 183\n",
      "length new taxa list: 12793\n",
      "100.0:c__Thermoleophilia # children taxa: 48\n",
      "length new taxa list: 12746\n",
      "100.0:c__Rubrobacteria; o__Rubrobacterales; f__Rubrobacteraceae # children taxa: 3\n",
      "length new taxa list: 12744\n",
      "100.0:c__Coriobacteriia # children taxa: 133\n",
      "length new taxa list: 12612\n",
      "100.0:c__UBA9087 # children taxa: 5\n",
      "length new taxa list: 12608\n",
      "100.0:c__UBA1414; o__UBA1414; f__UBA1414 # children taxa: 4\n",
      "length new taxa list: 12605\n",
      "100.0:c__RBG-13-55-18 # children taxa: 4\n",
      "length new taxa list: 12602\n",
      "100.0:c__Eremiobacteria # children taxa: 26\n",
      "length new taxa list: 12577\n",
      "100.0:c__UBP9; o__UBA4705; f__UBA4705 # children taxa: 4\n",
      "length new taxa list: 12574\n",
      "100.0:p__Firmicutes_H; c__CSP1-3; o__CSP1-3 # children taxa: 3\n",
      "length new taxa list: 12572\n",
      "100.0:c__Fimbriimonadia # children taxa: 16\n",
      "length new taxa list: 12557\n",
      "100.0:c__Abditibacteria # children taxa: 3\n",
      "length new taxa list: 12555\n",
      "93.0:c__UBA5377 # children taxa: 4\n",
      "length new taxa list: 12552\n",
      "79.0:c__Paceibacteria # children taxa: 392\n",
      "length new taxa list: 12161\n",
      "100.0:c__Paceibacteria_A; o__Moranbacterales # children taxa: 47\n",
      "length new taxa list: 12115\n",
      "100.0:c__ABY1 # children taxa: 156\n",
      "length new taxa list: 11960\n",
      "100.0:c__Doudnabacteria; o__UBA920 # children taxa: 12\n",
      "length new taxa list: 11949\n",
      "100.0:c__Andersenbacteria; o__UBA10190; f__UBA10190 # children taxa: 5\n",
      "length new taxa list: 11945\n",
      "100.0:c__Gracilibacteria # children taxa: 67\n",
      "length new taxa list: 11879\n",
      "100.0:c__Saccharimonadia # children taxa: 72\n",
      "length new taxa list: 11808\n",
      "67.0:c__CPR2 # children taxa: 6\n",
      "length new taxa list: 11803\n",
      "100.0:c__UBA1384 # children taxa: 11\n",
      "length new taxa list: 11793\n",
      "100.0:c__Kazan-3B-28; o__Kazan-3B-28 # children taxa: 5\n",
      "length new taxa list: 11789\n",
      "99.0:c__Microgenomatia # children taxa: 211\n",
      "length new taxa list: 11579\n",
      "92.0:c__WWE3 # children taxa: 23\n",
      "length new taxa list: 11557\n",
      "100.0:c__CPR3; o__CPR3 # children taxa: 2\n",
      "length new taxa list: 11556\n",
      "100.0:c__4484-211; o__4484-211 # children taxa: 2\n",
      "length new taxa list: 11555\n",
      "100.0:c__Dojkabacteria # children taxa: 17\n",
      "length new taxa list: 11539\n",
      "100.0:c__Dehalococcoidia # children taxa: 173\n",
      "length new taxa list: 11367\n",
      "100.0:c__Anaerolineae # children taxa: 137\n",
      "length new taxa list: 11231\n",
      "100.0:c__Chloroflexia # children taxa: 28\n",
      "length new taxa list: 11204\n",
      "100.0:c__Ktedonobacteria; o__Ktedonobacterales; f__Ktedonobacteraceae # children taxa: 7\n",
      "length new taxa list: 11198\n",
      "100.0:c__UBA5177; o__UBA5177; f__UBA5177 # children taxa: 2\n",
      "length new taxa list: 11197\n",
      "100.0:p__Dormibacterota; c__Dormibacteria # children taxa: 32\n",
      "length new taxa list: 11166\n",
      "100.0:p__Chloroflexota_A; c__Ellin6529 # children taxa: 32\n",
      "length new taxa list: 11135\n",
      "100.0:p__Deinococcota; c__Deinococci; o__Deinococcales # children taxa: 69\n",
      "length new taxa list: 11067\n",
      "100.0:p__Thermotogota; c__Thermotogae # children taxa: 54\n",
      "length new taxa list: 11014\n",
      "100.0:p__Bipolaricaulota; c__Bipolaricaulia # children taxa: 11\n",
      "length new taxa list: 11004\n",
      "100.0:p__Caldisericota; c__Caldisericia # children taxa: 10\n",
      "length new taxa list: 10995\n",
      "100.0:p__Coprothermobacterota; c__Coprothermobacteria; o__Coprothermobacterales; f__Coprothermobacteraceae; g__Coprothermobacter # children taxa: 3\n",
      "length new taxa list: 10993\n",
      "100.0:p__Dictyoglomota; c__Dictyoglomia; o__Dictyoglomales; f__Dictyoglomaceae # children taxa: 3\n",
      "length new taxa list: 10991\n",
      "100.0:p__Thermodesulfobiota; c__Thermodesulfobiia; o__Thermodesulfobiales; f__Thermodesulfobiaceae; g__Thermodesulfobium # children taxa: 2\n",
      "length new taxa list: 10990\n",
      "100.0:c__Synergistia; o__Synergistales # children taxa: 40\n",
      "length new taxa list: 10951\n",
      "100.0:c__GBS-1; o__GBS-1; f__GBS-1 # children taxa: 2\n",
      "length new taxa list: 10950\n",
      "100.0:c__JS1; o__SB-45 # children taxa: 11\n",
      "length new taxa list: 10940\n",
      "100.0:c__Caldatribacteriia; o__Caldatribacteriales # children taxa: 4\n",
      "length new taxa list: 10937\n",
      "100.0:p__Riflebacteria; c__UBA8953; o__UBA8953; f__UBA8953 # children taxa: 5\n",
      "length new taxa list: 10933\n",
      "100.0:p__UBP17; c__UBA6191; o__UBA6191; f__UBA6191 # children taxa: 2\n",
      "length new taxa list: 10932\n",
      "100.0:p__Wallbacteria; c__UBA9980; o__UBA9980; f__UBA9980; g__UBA9980 # children taxa: 2\n",
      "length new taxa list: 10931\n",
      "100.0:c__Verrucomicrobiae # children taxa: 292\n",
      "length new taxa list: 10640\n",
      "100.0:c__Kiritimatiellae # children taxa: 40\n",
      "length new taxa list: 10601\n",
      "100.0:c__Lentisphaeria # children taxa: 26\n",
      "length new taxa list: 10576\n",
      "100.0:c__UBA3054; o__UBA3054; f__UBA3054; g__UBA3054 # children taxa: 2\n",
      "length new taxa list: 10575\n",
      "63.0:c__UBA1439 # children taxa: 2\n",
      "length new taxa list: 10574\n",
      "100.0:p__Verrucomicrobiota_A; c__Chlamydiia # children taxa: 46\n",
      "length new taxa list: 10529\n",
      "100.0:c__Planctomycetes # children taxa: 101\n",
      "length new taxa list: 10429\n",
      "100.0:c__Phycisphaerae # children taxa: 85\n",
      "length new taxa list: 10345\n",
      "55.0:c__Brocadiae # children taxa: 18\n",
      "length new taxa list: 10328\n",
      "100.0:c__UBA1135 # children taxa: 17\n",
      "length new taxa list: 10312\n",
      "100.0:c__GCA-002687715; o__GCA-002687715; f__GCA-002687715 # children taxa: 2\n",
      "length new taxa list: 10311\n",
      "100.0:c__UBA8108 # children taxa: 10\n",
      "length new taxa list: 10302\n",
      "67.0:c__UBA11346 # children taxa: 3\n",
      "length new taxa list: 10300\n",
      "100.0:c__SZUA-567; o__SZUA-567; f__SZUA-567 # children taxa: 2\n",
      "length new taxa list: 10299\n",
      "100.0:c__Spirochaetia # children taxa: 159\n",
      "length new taxa list: 10141\n",
      "100.0:c__GWE2-31-10; o__GWE2-31-10; f__GWE2-31-10 # children taxa: 2\n",
      "length new taxa list: 10140\n",
      "92.0:c__UBA6919 # children taxa: 2\n",
      "length new taxa list: 10139\n",
      "100.0:c__Brevinematia; o__Brevinematales # children taxa: 3\n",
      "length new taxa list: 10137\n",
      "100.0:c__Brachyspirae; o__Brachyspirales; f__Brachyspiraceae; g__Brachyspira # children taxa: 12\n",
      "length new taxa list: 10126\n",
      "100.0:c__Leptospirae # children taxa: 47\n",
      "length new taxa list: 10080\n",
      "100.0:c__UBA4802; o__UBA4802 # children taxa: 9\n",
      "length new taxa list: 10072\n",
      "100.0:p__UBP7; c__UBA6624; o__UBA6624; f__UBA6624; g__UBA6624 # children taxa: 2\n",
      "length new taxa list: 10071\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "100.0:c__Bacteroidia # children taxa: 1648\n",
      "length new taxa list: 8424\n",
      "100.0:c__Rhodothermia # children taxa: 56\n",
      "length new taxa list: 8369\n",
      "100.0:c__Chlorobia; o__Chlorobiales # children taxa: 23\n",
      "length new taxa list: 8347\n",
      "96.0:c__Ignavibacteria # children taxa: 36\n",
      "length new taxa list: 8312\n",
      "100.0:c__UBA10030 # children taxa: 23\n",
      "length new taxa list: 8290\n",
      "100.0:c__Kryptonia; o__Kryptoniales; f__Kryptoniaceae # children taxa: 6\n",
      "length new taxa list: 8285\n",
      "100.0:c__Kapabacteria # children taxa: 19\n",
      "length new taxa list: 8267\n",
      "100.0:c__SZUA-365; o__SZUA-365 # children taxa: 3\n",
      "length new taxa list: 8265\n",
      "51.0:p__KSB1; c__UBA2214 # children taxa: 6\n",
      "length new taxa list: 8260\n",
      "100.0:p__Calditrichota; c__Calditrichia # children taxa: 2\n",
      "length new taxa list: 8259\n",
      "100.0:c__Marinisomatia # children taxa: 81\n",
      "length new taxa list: 8179\n",
      "100.0:c__AB16; o__AB16 # children taxa: 3\n",
      "length new taxa list: 8177\n",
      "100.0:c__UBA2242; o__UBA2242; f__UBA2242 # children taxa: 3\n",
      "length new taxa list: 8175\n",
      "100.0:p__Delongbacteria; c__UBA4055; o__UBA4055; f__UBA4055 # children taxa: 4\n",
      "length new taxa list: 8172\n",
      "100.0:p__AABM5-125-24; c__AABM5-125-24 # children taxa: 2\n",
      "length new taxa list: 8171\n",
      "100.0:c__Fibrobacteria # children taxa: 38\n",
      "length new taxa list: 8134\n",
      "100.0:c__OXYB2-FULL-49-7; o__OXYB2-FULL-49-7 # children taxa: 2\n",
      "length new taxa list: 8133\n",
      "100.0:c__Chitinivibrionia; o__Chitinivibrionales # children taxa: 5\n",
      "length new taxa list: 8129\n",
      "100.0:p__Cloacimonadota; c__Cloacimonadia # children taxa: 23\n",
      "length new taxa list: 8107\n",
      "100.0:c__Gemmatimonadetes # children taxa: 75\n",
      "length new taxa list: 8033\n",
      "95.0:p__Krumholzibacteriota; c__Krumholzibacteria # children taxa: 5\n",
      "length new taxa list: 8029\n",
      "100.0:p__Eisenbacteria; c__RBG-16-71-46 # children taxa: 2\n",
      "length new taxa list: 8028\n",
      "100.0:p__Edwardsbacteria; c__AC1; o__AC1 # children taxa: 4\n",
      "length new taxa list: 8025\n",
      "100.0:c__UBA2968 # children taxa: 9\n",
      "length new taxa list: 8017\n",
      "100.0:p__Zixibacteria; c__MSB-5A5 # children taxa: 12\n",
      "length new taxa list: 8006\n",
      "100.0:p__UBP14; c__UBA6098 # children taxa: 2\n",
      "length new taxa list: 8005\n",
      "99.0:p__WOR-3_A; c__Hydrothermia # children taxa: 4\n",
      "length new taxa list: 8002\n",
      "100.0:p__Calescibacterota; c__Calescibacteriia; o__Calescibacteriales; f__Calescibacteriaceae; g__Calescibacterium # children taxa: 2\n",
      "length new taxa list: 8001\n",
      "100.0:p__WOR-3; c__WOR-3 # children taxa: 6\n",
      "length new taxa list: 7996\n",
      "100.0:c__UBA3073; o__UBA3073; f__UBA3073; g__UBA3073 # children taxa: 2\n",
      "length new taxa list: 7995\n",
      "100.0:c__32-111; o__32-111; f__32-111 # children taxa: 3\n",
      "length new taxa list: 7993\n",
      "100.0:p__Fermentibacterota; c__Fermentibacteria; o__Fermentibacterales; f__Fermentibacteraceae # children taxa: 6\n",
      "length new taxa list: 7988\n",
      "100.0:p__Poribacteria; c__WGA-4E # children taxa: 8\n",
      "length new taxa list: 7981\n",
      "100.0:p__Hydrogenedentota; c__Hydrogenedentia; o__Hydrogenedentiales # children taxa: 4\n",
      "length new taxa list: 7978\n",
      "100.0:p__Sumerlaeota; c__Sumerlaeia # children taxa: 3\n",
      "length new taxa list: 7976\n",
      "99.0:c__Gammaproteobacteria # children taxa: 4645\n",
      "length new taxa list: 3332\n",
      "96.0:c__Zetaproteobacteria # children taxa: 18\n",
      "length new taxa list: 3315\n",
      "42.0:c__Alphaproteobacteria # children taxa: 1415\n",
      "length new taxa list: 1901\n",
      "100.0:c__Magnetococcia; o__Magnetococcales # children taxa: 12\n",
      "length new taxa list: 1890\n",
      "100.0:p__Dependentiae; c__Babeliae; o__Babeliales # children taxa: 22\n",
      "length new taxa list: 1869\n",
      "100.0:p__AB1-6; c__AB1-6; o__AB1-6 # children taxa: 2\n",
      "length new taxa list: 1868\n",
      "100.0:p__UBP6; c__UBA1177; o__UBA1177; f__UBA1177; g__UBA1177 # children taxa: 3\n",
      "length new taxa list: 1866\n",
      "100.0:p__SAR324; c__SAR324 # children taxa: 21\n",
      "length new taxa list: 1846\n",
      "100.0:c__Campylobacteria # children taxa: 248\n",
      "length new taxa list: 1599\n",
      "100.0:c__Desulfurellia; o__Desulfurellales # children taxa: 6\n",
      "length new taxa list: 1594\n",
      "100.0:c__Aquificae # children taxa: 21\n",
      "length new taxa list: 1574\n",
      "100.0:c__Desulfurobacteriia; o__Desulfurobacteriales; f__Desulfurobacteriaceae # children taxa: 5\n",
      "length new taxa list: 1570\n",
      "100.0:p__Deferribacterota; c__Deferribacteres # children taxa: 14\n",
      "length new taxa list: 1557\n",
      "100.0:p__Chrysiogenetota; c__Chrysiogenetes; o__Chrysiogenales; f__Chrysiogenaceae # children taxa: 2\n",
      "length new taxa list: 1556\n",
      "100.0:p__Dadabacteria; c__UBA1144 # children taxa: 14\n",
      "length new taxa list: 1543\n",
      "31.0:c__Desulfobacteria # children taxa: 63\n",
      "length new taxa list: 1481\n",
      "100.0:c__Syntrophobacteria # children taxa: 12\n",
      "length new taxa list: 1470\n",
      "28.0:c__Desulfarculia # children taxa: 5\n",
      "length new taxa list: 1466\n",
      "100.0:c__Desulfobaccia; o__Desulfobaccales # children taxa: 3\n",
      "length new taxa list: 1464\n",
      "100.0:c__Desulfobulbia; o__Desulfobulbales # children taxa: 49\n",
      "length new taxa list: 1416\n",
      "100.0:c__Thermodesulfobacteria; o__Thermodesulfobacteriales # children taxa: 13\n",
      "length new taxa list: 1404\n",
      "100.0:c__Dissulfuribacteria; o__Dissulfuribacterales # children taxa: 5\n",
      "length new taxa list: 1400\n",
      "100.0:c__Desulfofervidia; o__Desulfofervidales # children taxa: 2\n",
      "length new taxa list: 1399\n",
      "100.0:c__Syntrophia; o__Syntrophales # children taxa: 53\n",
      "length new taxa list: 1347\n",
      "83.0:c__BSN033 # children taxa: 11\n",
      "length new taxa list: 1337\n",
      "100.0:c__Syntrophorhabdia; o__Syntrophorhabdales # children taxa: 13\n",
      "length new taxa list: 1325\n",
      "100.0:c__Desulfomonilia # children taxa: 4\n",
      "length new taxa list: 1322\n",
      "100.0:p__Desulfuromonadota; c__Desulfuromonadia # children taxa: 81\n",
      "length new taxa list: 1242\n",
      "100.0:p__GWC2-55-46; c__GWC2-55-46; o__GWC2-55-46 # children taxa: 8\n",
      "length new taxa list: 1235\n",
      "100.0:c__Thermodesulfovibrionia # children taxa: 43\n",
      "length new taxa list: 1193\n",
      "93.0:c__Nitrospiria # children taxa: 44\n",
      "length new taxa list: 1150\n",
      "100.0:c__UBA9217; o__UBA9217; f__UBA9217 # children taxa: 3\n",
      "length new taxa list: 1148\n",
      "100.0:p__Methylomirabilota; c__Methylomirabilia # children taxa: 40\n",
      "length new taxa list: 1109\n",
      "100.0:p__Moduliflexota; c__Moduliflexia; o__Moduliflexales; f__Moduliflexaceae # children taxa: 3\n",
      "length new taxa list: 1107\n",
      "100.0:p__SZUA-182; c__SZUA-182; o__SZUA-182 # children taxa: 2\n",
      "length new taxa list: 1106\n",
      "100.0:c__Nitrospinia; o__Nitrospinales; f__Nitrospinaceae # children taxa: 16\n",
      "length new taxa list: 1091\n",
      "100.0:c__UBA9942; o__UBA9942 # children taxa: 4\n",
      "length new taxa list: 1088\n",
      "100.0:p__UBA8248; c__UBA8248; o__UBA8248 # children taxa: 4\n",
      "length new taxa list: 1085\n",
      "100.0:p__Tectomicrobia; c__Entotheonellia; o__Entotheonellales; f__Entotheonellaceae; g__Entotheonella # children taxa: 2\n",
      "length new taxa list: 1084\n",
      "100.0:p__Schekmanbacteria; c__GWA2-38-11; o__GWA2-38-11; f__GWA2-38-11 # children taxa: 3\n",
      "length new taxa list: 1082\n",
      "100.0:p__BMS3Abin14; c__BMS3Abin14; o__BMS3Abin14; f__BMS3Abin14 # children taxa: 3\n",
      "length new taxa list: 1080\n",
      "100.0:p__Binatota; c__Binatia # children taxa: 15\n",
      "length new taxa list: 1066\n",
      "100.0:p__Deferrisomatota; c__Defferisomatia; o__Defferisomatales # children taxa: 3\n",
      "length new taxa list: 1064\n",
      "100.0:p__MBNT15; c__MBNT15; o__MBNT15; f__MBNT15 # children taxa: 9\n",
      "length new taxa list: 1056\n",
      "79.0:c__Polyangia # children taxa: 48\n",
      "length new taxa list: 1009\n",
      "100.0:c__Myxococcia; o__Myxococcales # children taxa: 30\n",
      "length new taxa list: 980\n",
      "94.0:c__UBA727 # children taxa: 3\n",
      "length new taxa list: 978\n",
      "63.0:c__Bradimonadia # children taxa: 7\n",
      "length new taxa list: 972\n",
      "100.0:c__UBA9042 # children taxa: 4\n",
      "length new taxa list: 969\n",
      "100.0:c__UBA9160 # children taxa: 11\n",
      "length new taxa list: 959\n",
      "100.0:c__UBA796 # children taxa: 8\n",
      "length new taxa list: 952\n",
      "100.0:c__Oligoflexia # children taxa: 7\n",
      "length new taxa list: 946\n",
      "100.0:c__UBA2361 # children taxa: 2\n",
      "length new taxa list: 945\n",
      "75.0:c__Bacteriovoracia # children taxa: 46\n",
      "length new taxa list: 900\n",
      "59.0:c__UBA2394 # children taxa: 3\n",
      "length new taxa list: 898\n",
      "100.0:c__Bdellovibrionia; o__Bdellovibrionales # children taxa: 29\n",
      "length new taxa list: 870\n",
      "84.0:p__O2-12-FULL-43-9; c__O2-12-FULL-43-9 # children taxa: 3\n",
      "length new taxa list: 868\n",
      "100.0:p__UBA10199; c__UBA10199 # children taxa: 8\n",
      "length new taxa list: 861\n",
      "100.0:p__Desulfobacterota_A; c__Desulfovibrionia; o__Desulfovibrionales # children taxa: 103\n",
      "length new taxa list: 759\n",
      "100.0:p__Nitrospirota_A; c__Leptospirillia; o__Leptospirillales; f__Leptospirillaceae # children taxa: 4\n",
      "length new taxa list: 756\n",
      "100.0:c__Acidobacteriae # children taxa: 212\n",
      "length new taxa list: 545\n",
      "100.0:c__Blastocatellia # children taxa: 30\n",
      "length new taxa list: 516\n",
      "100.0:c__UBA6911; o__UBA6911; f__UBA6911 # children taxa: 2\n",
      "length new taxa list: 515\n",
      "96.0:c__Vicinamibacteria # children taxa: 39\n",
      "length new taxa list: 477\n",
      "97.0:c__Mor1 # children taxa: 6\n",
      "length new taxa list: 472\n",
      "100.0:c__UBA890; o__UBA890; f__UBA890; g__UBA890 # children taxa: 2\n",
      "length new taxa list: 471\n",
      "97.0:c__Thermoanaerobaculia # children taxa: 16\n",
      "length new taxa list: 456\n",
      "100.0:c__Holophagae; o__Holophagales; f__Holophagaceae # children taxa: 9\n",
      "length new taxa list: 448\n",
      "100.0:c__Aminicenantia # children taxa: 19\n",
      "length new taxa list: 430\n",
      "98.0:c__koll11 # children taxa: 53\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "length new taxa list: 378\n",
      "100.0:c__Omnitrophia; o__Omnitrophales # children taxa: 11\n",
      "length new taxa list: 368\n",
      "100.0:p__Goldbacteria; c__PGYV01; o__PGYV01 # children taxa: 3\n",
      "length new taxa list: 366\n",
      "100.0:p__Firestonebacteria; c__D2-FULL-39-29; o__D2-FULL-39-29; f__D2-FULL-39-29; g__D2-FULL-39-29 # children taxa: 2\n",
      "length new taxa list: 365\n",
      "100.0:p__FCPU426; c__Palsa-1180; o__Palsa-1180; f__Palsa-1180 # children taxa: 2\n",
      "length new taxa list: 364\n",
      "100.0:p__Desantisbacteria; c__UBA1551; o__UBA1551 # children taxa: 2\n",
      "length new taxa list: 363\n",
      "100.0:c__Elusimicrobia # children taxa: 46\n",
      "length new taxa list: 318\n",
      "72.0:c__Endomicrobia # children taxa: 11\n",
      "length new taxa list: 308\n",
      "100.0:c__UBA8919; o__UBA8919; f__UBA8919 # children taxa: 4\n",
      "length new taxa list: 305\n",
      "100.0:p__UBA6262; c__UBA6262; o__UBA6262; f__UBA6262 # children taxa: 2\n",
      "length new taxa list: 304\n",
      "100.0:p__CG03; c__CG03; o__CG03; f__CG03 # children taxa: 3\n",
      "length new taxa list: 302\n",
      "302\n"
     ]
    }
   ],
   "source": [
    "parsed_arc_train_class = arc_train.copy()\n",
    "parse_taxa_from_node(arc.seed_node,parsed_arc_train_class,genomes_per_taxa=1,prefix='c__')\n",
    "print(len(parsed_arc_train_class))\n",
    "\n",
    "parsed_bac_train_class = bac_train.copy()\n",
    "parse_taxa_from_node(bac.seed_node,parsed_bac_train_class,genomes_per_taxa=1,prefix='c__')\n",
    "print(len(parsed_bac_train_class))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "34 302\n",
      "336\n"
     ]
    }
   ],
   "source": [
    "print(len(parsed_arc_train_class),len(parsed_bac_train_class))\n",
    "print(len(parsed_arc_train_class)+len(parsed_bac_train_class))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "336\n",
      "336\n"
     ]
    }
   ],
   "source": [
    "traindf_class = get_df_subset(ftp,arcftp,parsed_bac_train_class,parsed_arc_train_class)\n",
    "print(len(traindf_class))\n",
    "\n",
    "train_fnames_class = get_filenames(traindf_class)\n",
    "print(len(train_fnames_class))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "traindf_class.to_csv('TrainNew_ParsedClass_GTDBrepwithftp.csv')\n",
    "save_filenames('trainnew_parsedclass_filenames.txt',train_fnames_class)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## do the same with valid at family and order level"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "100.0:f__Methanomethylophilaceae # children taxa: 21\n",
      "length new taxa list: 87\n",
      "100.0:f__Methanomassiliicoccaceae # children taxa: 7\n",
      "length new taxa list: 81\n",
      "100.0:f__UBA472; g__UBA472 # children taxa: 2\n",
      "length new taxa list: 80\n",
      "100.0:o__UBA10834; f__UBA10834 # children taxa: 4\n",
      "length new taxa list: 77\n",
      "100.0:o__SG8-5; f__SG8-5 # children taxa: 3\n",
      "length new taxa list: 75\n",
      "100.0:o__RBG-16-68-12; f__RBG-16-68-12; g__RBG-16-68-12 # children taxa: 3\n",
      "length new taxa list: 73\n",
      "100.0:o__UBA184; f__UBA184; g__UBA184 # children taxa: 2\n",
      "length new taxa list: 72\n",
      "100.0:f__Thermoplasmataceae # children taxa: 16\n",
      "length new taxa list: 57\n",
      "100.0:f__Aciduliprofundaceae; g__Aciduliprofundum # children taxa: 2\n",
      "length new taxa list: 56\n",
      "100.0:f__UBA233 # children taxa: 12\n",
      "length new taxa list: 45\n",
      "100.0:f__BA1 # children taxa: 4\n",
      "length new taxa list: 42\n",
      "100.0:f__B26-1 # children taxa: 3\n",
      "length new taxa list: 40\n",
      "100.0:o__TCS64; f__TCS64 # children taxa: 5\n",
      "length new taxa list: 36\n",
      "100.0:f__40CM-2-53-6; g__40CM-2-53-6 # children taxa: 5\n",
      "length new taxa list: 32\n",
      "100.0:o__B25; f__B25 # children taxa: 3\n",
      "length new taxa list: 30\n",
      "97.0:f__Microbacteriaceae # children taxa: 338\n",
      "length new taxa list: 1186\n",
      "100.0:f__Micrococcaceae # children taxa: 188\n",
      "length new taxa list: 999\n",
      "100.0:f__Brevibacteriaceae; g__Brevibacterium # children taxa: 18\n",
      "length new taxa list: 982\n",
      "100.0:f__Actinomycetaceae # children taxa: 102\n",
      "length new taxa list: 881\n",
      "100.0:f__Beutenbergiaceae # children taxa: 7\n",
      "length new taxa list: 875\n",
      "100.0:f__Cellulomonadaceae # children taxa: 66\n",
      "length new taxa list: 810\n",
      "100.0:f__Demequinaceae; g__Demequina # children taxa: 28\n",
      "length new taxa list: 783\n",
      "100.0:f__Bifidobacteriaceae # children taxa: 97\n",
      "length new taxa list: 687\n",
      "100.0:f__Dermabacteraceae # children taxa: 20\n",
      "length new taxa list: 668\n",
      "100.0:f__Quadrisphaeraceae; g__Quadrisphaera # children taxa: 2\n",
      "length new taxa list: 667\n",
      "95.0:f__Kineosporiaceae # children taxa: 3\n",
      "length new taxa list: 665\n",
      "100.0:f__Kineococcaceae; g__Kineococcus # children taxa: 3\n",
      "length new taxa list: 663\n",
      "100.0:f__Dermatophilaceae # children taxa: 70\n",
      "length new taxa list: 594\n",
      "100.0:f__Rhodobacteraceae # children taxa: 578\n",
      "length new taxa list: 17\n",
      "30\n",
      "17\n",
      "47\n"
     ]
    }
   ],
   "source": [
    "parsed_arc_valid_family = arc_valid.copy()\n",
    "parse_taxa_from_node(arc.seed_node,parsed_arc_valid_family,genomes_per_taxa=1,prefix='f__')\n",
    "\n",
    "parsed_bac_valid_family = bac_valid.copy()\n",
    "parse_taxa_from_node(bac.seed_node,parsed_bac_valid_family,genomes_per_taxa=1,prefix='f__')\n",
    "\n",
    "print(len(parsed_arc_valid_family))\n",
    "print(len(parsed_bac_valid_family))\n",
    "print(len(parsed_arc_valid_family)+len(parsed_bac_valid_family))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "47\n",
      "47\n"
     ]
    }
   ],
   "source": [
    "valdf_family = get_df_subset(ftp,arcftp,parsed_bac_valid_family,parsed_arc_valid_family)\n",
    "print(len(valdf_family))\n",
    "\n",
    "valid_fnames_family = get_filenames(valdf_family)\n",
    "print(len(valid_fnames_family))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "valdf_family.to_csv('ValidNew_ParsedFamily_GTDBrepwithftp.csv')\n",
    "save_filenames('validnew_parsedfamily_filenames.txt',valid_fnames_family)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "100.0:o__Methanomassiliicoccales # children taxa: 32\n",
      "length new taxa list: 76\n",
      "100.0:o__UBA10834; f__UBA10834 # children taxa: 4\n",
      "length new taxa list: 73\n",
      "100.0:o__SG8-5; f__SG8-5 # children taxa: 3\n",
      "length new taxa list: 71\n",
      "100.0:o__RBG-16-68-12; f__RBG-16-68-12; g__RBG-16-68-12 # children taxa: 3\n",
      "length new taxa list: 69\n",
      "100.0:o__UBA184; f__UBA184; g__UBA184 # children taxa: 2\n",
      "length new taxa list: 68\n",
      "100.0:o__Thermoplasmatales # children taxa: 17\n",
      "length new taxa list: 52\n",
      "99.0:o__Acidiprofundales # children taxa: 3\n",
      "length new taxa list: 50\n",
      "100.0:o__B26-1 # children taxa: 20\n",
      "length new taxa list: 31\n",
      "100.0:o__TCS64; f__TCS64 # children taxa: 5\n",
      "length new taxa list: 27\n",
      "100.0:o__40CM-2-53-6 # children taxa: 8\n",
      "length new taxa list: 20\n",
      "100.0:o__B25; f__B25 # children taxa: 3\n",
      "length new taxa list: 18\n",
      "91.0:o__Actinomycetales # children taxa: 944\n",
      "length new taxa list: 588\n",
      "93.0:o__Rhodobacterales # children taxa: 579\n",
      "length new taxa list: 18\n",
      "18\n",
      "18\n",
      "36\n"
     ]
    }
   ],
   "source": [
    "parsed_arc_valid_order = arc_valid.copy()\n",
    "parse_taxa_from_node(arc.seed_node,parsed_arc_valid_order,genomes_per_taxa=1,prefix='o__')\n",
    "\n",
    "parsed_bac_valid_order = bac_valid.copy()\n",
    "parse_taxa_from_node(bac.seed_node,parsed_bac_valid_order,genomes_per_taxa=9,prefix='o__')\n",
    "\n",
    "print(len(parsed_arc_valid_order))\n",
    "print(len(parsed_bac_valid_order))\n",
    "print(len(parsed_arc_valid_order)+len(parsed_bac_valid_order))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "36\n",
      "36\n"
     ]
    }
   ],
   "source": [
    "valdf_order = get_df_subset(ftp,arcftp,parsed_bac_valid_order,parsed_arc_valid_order)\n",
    "print(len(valdf_order))\n",
    "\n",
    "valid_fnames_order = get_filenames(valdf_order)\n",
    "print(len(valid_fnames_order))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [],
   "source": [
    "valdf_order.to_csv('ValidNew_ParsedOrder_GTDBrepwithftp.csv')\n",
    "save_filenames('validnew_parsedorder_filenames.txt',valid_fnames_order)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "83.0:o__Methanosarcinales # children taxa: 63\n",
      "length new taxa list: 77\n",
      "100.0:o__Nitrososphaerales # children taxa: 68\n",
      "length new taxa list: 18\n",
      "91.0:o__Actinomycetales # children taxa: 944\n",
      "length new taxa list: 588\n",
      "93.0:o__Rhodobacterales # children taxa: 579\n",
      "length new taxa list: 18\n",
      "18\n",
      "18\n",
      "36\n"
     ]
    }
   ],
   "source": [
    "#test for order level\n",
    "parsed_arc_test_order = arc_test.copy()\n",
    "parse_taxa_from_node(arc.seed_node,parsed_arc_test_order,genomes_per_taxa=9,prefix='o__')\n",
    "\n",
    "parsed_bac_test_order = bac_valid.copy()\n",
    "parse_taxa_from_node(bac.seed_node,parsed_bac_test_order,genomes_per_taxa=9,prefix='o__')\n",
    "\n",
    "print(len(parsed_arc_test_order))\n",
    "print(len(parsed_bac_test_order))\n",
    "print(len(parsed_arc_test_order)+len(parsed_bac_test_order))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "36\n",
      "36\n"
     ]
    }
   ],
   "source": [
    "testdf_order = get_df_subset(ftp,arcftp,parsed_bac_test_order,parsed_arc_test_order)\n",
    "print(len(testdf_order))\n",
    "\n",
    "test_fnames_order = get_filenames(testdf_order)\n",
    "print(len(test_fnames_order))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "testdf_order.to_csv('TestNew_ParsedOrder_GTDBrepwithftp.csv')\n",
    "save_filenames('testnew_parsedorder_filenames.txt',test_fnames_order)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th># assembly_accession</th>\n",
       "      <th>asm_name</th>\n",
       "      <th>ftp_path</th>\n",
       "      <th>index</th>\n",
       "      <th>ncbi_assembly_level</th>\n",
       "      <th>ncbi_assembly_name</th>\n",
       "      <th>ncbi_bioproject</th>\n",
       "      <th>ncbi_biosample</th>\n",
       "      <th>ncbi_genbank_assembly_accession</th>\n",
       "      <th>ncbi_genome_representation</th>\n",
       "      <th>ncbi_organism_name</th>\n",
       "      <th>ncbi_refseq_category</th>\n",
       "      <th>parsed_genome_accession</th>\n",
       "      <th>srr_accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>12683</td>\n",
       "      <td>GCA_002350705.1</td>\n",
       "      <td>ASM235070v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/002...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>ASM235070v1</td>\n",
       "      <td>PRJNA348753</td>\n",
       "      <td>SAMN06456497</td>\n",
       "      <td>GCA_002350705.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Ruminococcaceae bacterium UBA2857</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_002350705.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>22768</td>\n",
       "      <td>GCA_900322155.1</td>\n",
       "      <td>Rumen_uncultured_genome_RUG763</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>Rumen uncultured genome RUG763</td>\n",
       "      <td>PRJEB21624</td>\n",
       "      <td>SAMEA104666710</td>\n",
       "      <td>GCA_900322155.1</td>\n",
       "      <td>full</td>\n",
       "      <td>uncultured Clostridiales bacterium</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_900322155.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14522</td>\n",
       "      <td>GCA_002477805.1</td>\n",
       "      <td>ASM247780v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/002...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>ASM247780v1</td>\n",
       "      <td>PRJNA348753</td>\n",
       "      <td>SAMN06454830</td>\n",
       "      <td>GCA_002477805.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Catabacter sp. UBA7584</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_002477805.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>23447</td>\n",
       "      <td>GCA_003543635.1</td>\n",
       "      <td>ASM354363v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/003...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>GCA_003543635.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>12002</td>\n",
       "      <td>GCA_002308795.1</td>\n",
       "      <td>ASM230879v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/002...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>ASM230879v1</td>\n",
       "      <td>PRJNA348753</td>\n",
       "      <td>SAMN06457806</td>\n",
       "      <td>GCA_002308795.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Mageeibacillus sp. UBA1255</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_002308795.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0 # assembly_accession                        asm_name  \\\n",
       "0       12683      GCA_002350705.1                     ASM235070v1   \n",
       "1       22768      GCA_900322155.1  Rumen_uncultured_genome_RUG763   \n",
       "2       14522      GCA_002477805.1                     ASM247780v1   \n",
       "3       23447      GCA_003543635.1                     ASM354363v1   \n",
       "4       12002      GCA_002308795.1                     ASM230879v1   \n",
       "\n",
       "                                            ftp_path index  \\\n",
       "0  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/002...   NaN   \n",
       "1  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...   NaN   \n",
       "2  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/002...   NaN   \n",
       "3  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/003...   NaN   \n",
       "4  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/002...   NaN   \n",
       "\n",
       "  ncbi_assembly_level              ncbi_assembly_name ncbi_bioproject  \\\n",
       "0            Scaffold                     ASM235070v1     PRJNA348753   \n",
       "1            Scaffold  Rumen uncultured genome RUG763      PRJEB21624   \n",
       "2            Scaffold                     ASM247780v1     PRJNA348753   \n",
       "3                none                            none            none   \n",
       "4            Scaffold                     ASM230879v1     PRJNA348753   \n",
       "\n",
       "   ncbi_biosample ncbi_genbank_assembly_accession ncbi_genome_representation  \\\n",
       "0    SAMN06456497                 GCA_002350705.1                       full   \n",
       "1  SAMEA104666710                 GCA_900322155.1                       full   \n",
       "2    SAMN06454830                 GCA_002477805.1                       full   \n",
       "3            none                            none                       none   \n",
       "4    SAMN06457806                 GCA_002308795.1                       full   \n",
       "\n",
       "                   ncbi_organism_name ncbi_refseq_category  \\\n",
       "0   Ruminococcaceae bacterium UBA2857                   na   \n",
       "1  uncultured Clostridiales bacterium                   na   \n",
       "2              Catabacter sp. UBA7584                   na   \n",
       "3                                none                 none   \n",
       "4          Mageeibacillus sp. UBA1255                   na   \n",
       "\n",
       "  parsed_genome_accession srr_accession  \n",
       "0         GCA_002350705.1           NaN  \n",
       "1         GCA_900322155.1           NaN  \n",
       "2         GCA_002477805.1           NaN  \n",
       "3         GCA_003543635.1           NaN  \n",
       "4         GCA_002308795.1           NaN  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#link for paper\n",
    "#saving train/valid/test tables for paper\n",
    "train = pd.read_csv('TrainNew_ParsedOrder_GTDBrepwithftp.csv')\n",
    "valid = pd.read_csv('ValidNew_ParsedOrder_GTDBrepwithftp.csv')\n",
    "test = pd.read_csv('TestNew_ParsedOrder_GTDBrepwithftp.csv')\n",
    "train.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/adrienne/anaconda3/envs/tree/lib/python3.5/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/Users/adrienne/anaconda3/envs/tree/lib/python3.5/site-packages/ipykernel_launcher.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"\n",
      "/Users/adrienne/anaconda3/envs/tree/lib/python3.5/site-packages/ipykernel_launcher.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ncbi_genbank_assembly_accession</th>\n",
       "      <th>ncbi_assembly_name</th>\n",
       "      <th>ncbi_biosample</th>\n",
       "      <th>ncbi_organism_name</th>\n",
       "      <th>parsed_genome_accession</th>\n",
       "      <th>dataset</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_002350705.1</td>\n",
       "      <td>ASM235070v1</td>\n",
       "      <td>SAMN06456497</td>\n",
       "      <td>Ruminococcaceae bacterium UBA2857</td>\n",
       "      <td>GCA_002350705.1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_900322155.1</td>\n",
       "      <td>Rumen uncultured genome RUG763</td>\n",
       "      <td>SAMEA104666710</td>\n",
       "      <td>uncultured Clostridiales bacterium</td>\n",
       "      <td>GCA_900322155.1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_002477805.1</td>\n",
       "      <td>ASM247780v1</td>\n",
       "      <td>SAMN06454830</td>\n",
       "      <td>Catabacter sp. UBA7584</td>\n",
       "      <td>GCA_002477805.1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>GCA_003543635.1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_002308795.1</td>\n",
       "      <td>ASM230879v1</td>\n",
       "      <td>SAMN06457806</td>\n",
       "      <td>Mageeibacillus sp. UBA1255</td>\n",
       "      <td>GCA_002308795.1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  ncbi_genbank_assembly_accession              ncbi_assembly_name  \\\n",
       "0                 GCA_002350705.1                     ASM235070v1   \n",
       "1                 GCA_900322155.1  Rumen uncultured genome RUG763   \n",
       "2                 GCA_002477805.1                     ASM247780v1   \n",
       "3                            none                            none   \n",
       "4                 GCA_002308795.1                     ASM230879v1   \n",
       "\n",
       "   ncbi_biosample                  ncbi_organism_name parsed_genome_accession  \\\n",
       "0    SAMN06456497   Ruminococcaceae bacterium UBA2857         GCA_002350705.1   \n",
       "1  SAMEA104666710  uncultured Clostridiales bacterium         GCA_900322155.1   \n",
       "2    SAMN06454830              Catabacter sp. UBA7584         GCA_002477805.1   \n",
       "3            none                                none         GCA_003543635.1   \n",
       "4    SAMN06457806          Mageeibacillus sp. UBA1255         GCA_002308795.1   \n",
       "\n",
       "  dataset  \n",
       "0   train  \n",
       "1   train  \n",
       "2   train  \n",
       "3   train  \n",
       "4   train  "
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subt = train[['ncbi_genbank_assembly_accession','ncbi_assembly_name','ncbi_biosample','ncbi_organism_name','parsed_genome_accession']]\n",
    "subt['dataset'] = 'train'\n",
    "\n",
    "subv = valid[['ncbi_genbank_assembly_accession','ncbi_assembly_name','ncbi_biosample','ncbi_organism_name','parsed_genome_accession']]\n",
    "subv['dataset'] = 'valid'\n",
    "\n",
    "subtt = test[['ncbi_genbank_assembly_accession','ncbi_assembly_name','ncbi_biosample','ncbi_organism_name','parsed_genome_accession']]\n",
    "subtt['dataset'] = 'test'\n",
    "\n",
    "sub = subt.append(subv).append(subtt)\n",
    "sub.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "sub.to_csv('../GetGTDBGenomes/GenomeIDs_parsedorder.csv',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## subset at class level in train and valid\n",
    "\n",
    "(need to define new valid/test groups because some of them were at the order level)\n",
    "\n",
    "new groups:\n",
    "* bacteria: valid = c__Actinobacteria and c__Alphaproteobacteria; test = c__Bacteroidia and c__Clostridia\n",
    "* archaea: valid = c_Thermoplasmata and c_Bathyarchaeia; test = c__Methanosarcinia and c__Nitrososphaeria"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23458\n",
      "1248\n",
      "appending 2735 labels from node <Node object at 0x7fec75906240: '99.0:c__Actinobacteria' (None)>\n",
      "appending 2955 labels from node <Node object at 0x7fec77698908: '42.0:c__Alphaproteobacteria' (None)>\n",
      "appending 1836 labels from node <Node object at 0x7fec74d571d0: '32.0:c__Clostridia' (None)>\n",
      "appending 2677 labels from node <Node object at 0x7fec76552da0: '100.0:c__Bacteroidia' (None)>\n",
      "5690 4513\n",
      "appending 67 labels from node <Node object at 0x7fec750c5208: '86.0:c__Thermoplasmata' (None)>\n",
      "appending 40 labels from node <Node object at 0x7fec754caf60: '93.0:c__Bathyarchaeia' (None)>\n",
      "appending 76 labels from node <Node object at 0x7fec759b03c8: '97.0:c__Methanosarcinia' (None)>\n",
      "appending 81 labels from node <Node object at 0x7fec7575e208: '80.0:c__Nitrososphaeria' (None)>\n",
      "107 157\n",
      "13255\n",
      "984\n",
      "14239\n"
     ]
    }
   ],
   "source": [
    "#if you shut down your notebook and need to redefine these values:\n",
    "allbac = [x.label for x in list(bac.taxon_namespace)]\n",
    "print(len(allbac))\n",
    "allbac[0:15]\n",
    "allarc = [x.label for x in list(arc.taxon_namespace)]\n",
    "print(len(allarc))\n",
    "\n",
    "bac_valid_class = get_subset_leaves(bac.seed_node, to_match=['c__Actinobacteria','c__Alphaproteobacteria'],taxa=[])\n",
    "bac_test_class = get_subset_leaves(bac.seed_node, to_match=['c__Bacteroidia','c__Clostridia'],taxa=[])\n",
    "print(len(bac_valid_class),len(bac_test_class))\n",
    "arc_valid_class = get_subset_leaves(arc.seed_node, to_match=['c__Thermoplasmata','c__Bathyarchaeia'],taxa=[])\n",
    "arc_test_class = get_subset_leaves(arc.seed_node, to_match=['c__Methanosarcinia','c__Nitrososphaeria'],taxa=[])\n",
    "print(len(arc_valid_class),len(arc_test_class))\n",
    "bac_train_class = [x for x in allbac if x not in bac_valid_class and x not in bac_test_class]\n",
    "print(len(bac_train_class)) #only bac\n",
    "arc_train_class = [x for x in allarc if x not in arc_valid_class and x not in arc_test_class]\n",
    "print(len(arc_train_class))\n",
    "print(len(bac_train_class)+len(arc_train_class)) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24706\n",
      "24706\n"
     ]
    }
   ],
   "source": [
    "print(len(bac_train_class)+len(arc_train_class)+len(bac_valid_class)+len(bac_test_class)+len(arc_valid_class)+len(arc_test_class))\n",
    "print(len(allbac)+len(allarc))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "100.0:c__Halobacteria; o__Halobacteriales # children taxa: 229\n",
      "length new taxa list: 756\n",
      "100.0:c__Methanomicrobia; o__Methanomicrobiales # children taxa: 71\n",
      "length new taxa list: 686\n",
      "100.0:c__Bog-38; o__Bog-38; f__Bog-38; g__Bog-38 # children taxa: 4\n",
      "length new taxa list: 683\n",
      "100.0:c__Methanocellia; o__Methanocellales # children taxa: 4\n",
      "length new taxa list: 680\n",
      "100.0:c__Syntrophoarchaeia # children taxa: 5\n",
      "length new taxa list: 676\n",
      "85.0:c__Archaeoglobi # children taxa: 18\n",
      "length new taxa list: 659\n",
      "100.0:c__Methanonatronarchaeia; o__Methanonatronarchaeales # children taxa: 3\n",
      "length new taxa list: 657\n",
      "95.0:c__Poseidoniia # children taxa: 205\n",
      "length new taxa list: 453\n",
      "100.0:c__E2 # children taxa: 12\n",
      "length new taxa list: 442\n",
      "92.0:c__UBA186 # children taxa: 2\n",
      "length new taxa list: 441\n",
      "100.0:c__Methanobacteria; o__Methanobacteriales # children taxa: 84\n",
      "length new taxa list: 358\n",
      "100.0:c__Methanococci; o__Methanococcales # children taxa: 18\n",
      "length new taxa list: 341\n",
      "100.0:c__Methanopyri; o__Methanopyrales; f__Methanopyraceae; g__Methanopyrus # children taxa: 2\n",
      "length new taxa list: 340\n",
      "100.0:c__Thermococci # children taxa: 46\n",
      "length new taxa list: 295\n",
      "100.0:p__Hydrothermarchaeota; c__Hydrothermarchaeia; o__Hydrothermarchaeales # children taxa: 5\n",
      "length new taxa list: 291\n",
      "100.0:p__Hadarchaeota; c__Hadarchaeia; o__Hadarchaeales; f__Hadarchaeaceae # children taxa: 2\n",
      "length new taxa list: 290\n",
      "79.0:c__Thermoprotei # children taxa: 93\n",
      "length new taxa list: 198\n",
      "100.0:c__Methanomethylicia; o__Methanomethyliales; f__Methanomethyliaceae # children taxa: 3\n",
      "length new taxa list: 196\n",
      "70.0:c__Lokiarchaeia # children taxa: 8\n",
      "length new taxa list: 189\n",
      "100.0:c__Heimdallarchaeia # children taxa: 6\n",
      "length new taxa list: 184\n",
      "77.0:p__Nanoarchaeota; c__Nanoarchaeia # children taxa: 101\n",
      "length new taxa list: 84\n",
      "100.0:p__Nanohaloarchaeota; c__Nanosalinia; o__Nanosalinales; f__Nanosalinaceae # children taxa: 11\n",
      "length new taxa list: 74\n",
      "100.0:p__Aenigmarchaeota; c__Aenigmarchaeia # children taxa: 7\n",
      "length new taxa list: 68\n",
      "100.0:p__UAP2; c__UBA543; o__UBA543 # children taxa: 3\n",
      "length new taxa list: 66\n",
      "100.0:p__Micrarchaeota; c__Micrarchaeia # children taxa: 15\n",
      "length new taxa list: 52\n",
      "100.0:p__Iainarchaeota; c__Iainarchaeia # children taxa: 12\n",
      "length new taxa list: 41\n",
      "100.0:p__Altiarchaeota; c__Altiarchaeia # children taxa: 10\n",
      "length new taxa list: 32\n",
      "32\n",
      "87.0:c__Mahellia # children taxa: 12\n",
      "length new taxa list: 13244\n",
      "97.0:c__Thermoanaerobacteria # children taxa: 28\n",
      "length new taxa list: 13217\n",
      "100.0:c__Thermovenabulia # children taxa: 10\n",
      "length new taxa list: 13208\n",
      "46.0:p__Firmicutes_I; c__Bacilli_A # children taxa: 283\n",
      "length new taxa list: 12926\n",
      "100.0:p__Firmicutes_J; c__Desulfuribacillia; o__Desulfuribacillales; f__Desulfuribacillaceae # children taxa: 2\n",
      "length new taxa list: 12925\n",
      "100.0:p__Firmicutes_K; c__Alicyclobacillia # children taxa: 30\n",
      "length new taxa list: 12896\n",
      "100.0:c__Dethiobacteria # children taxa: 8\n",
      "length new taxa list: 12889\n",
      "100.0:c__Natranaerobiia; o__Natranaerobiales; f__Natranaerobiaceae # children taxa: 2\n",
      "length new taxa list: 12888\n",
      "100.0:c__Proteinivoracia # children taxa: 6\n",
      "length new taxa list: 12883\n",
      "100.0:c__UBA994; o__UBA994 # children taxa: 3\n",
      "length new taxa list: 12881\n",
      "98.0:c__Limnochordia # children taxa: 14\n",
      "length new taxa list: 12868\n",
      "100.0:c__DTU065; o__DTU065; f__DTU065 # children taxa: 2\n",
      "length new taxa list: 12867\n",
      "100.0:c__DTU084; o__DTU084; f__DTU084 # children taxa: 4\n",
      "length new taxa list: 12864\n",
      "79.0:c__SHA-98 # children taxa: 8\n",
      "length new taxa list: 12857\n",
      "100.0:c__UBA4882 # children taxa: 7\n",
      "length new taxa list: 12851\n",
      "100.0:c__Sulfobacillia; o__Sulfobacillales; f__Sulfobacillaceae # children taxa: 7\n",
      "length new taxa list: 12845\n",
      "100.0:c__Symbiobacteriia; o__Symbiobacteriales # children taxa: 3\n",
      "length new taxa list: 12843\n",
      "100.0:c__Thermaerobacteria; o__Thermaerobacterales # children taxa: 4\n",
      "length new taxa list: 12840\n",
      "57.0:c__DTU015 # children taxa: 9\n",
      "length new taxa list: 12832\n",
      "93.0:c__UBA3569 # children taxa: 2\n",
      "length new taxa list: 12831\n",
      "100.0:p__Firmicutes_C; c__Negativicutes # children taxa: 153\n",
      "length new taxa list: 12679\n",
      "74.0:c__Desulfitobacteriia # children taxa: 33\n",
      "length new taxa list: 12647\n",
      "98.0:c__Peptococcia # children taxa: 14\n",
      "length new taxa list: 12634\n",
      "71.0:c__Moorellia # children taxa: 17\n",
      "length new taxa list: 12618\n",
      "100.0:c__Dehalobacteriia # children taxa: 7\n",
      "length new taxa list: 12612\n",
      "100.0:c__TC1; o__TC1 # children taxa: 2\n",
      "length new taxa list: 12611\n",
      "78.0:c__Syntrophomonadia # children taxa: 28\n",
      "length new taxa list: 12584\n",
      "94.0:c__Desulfotomaculia # children taxa: 52\n",
      "length new taxa list: 12533\n",
      "100.0:c__Thermincolia; o__Thermincolales # children taxa: 4\n",
      "length new taxa list: 12530\n",
      "100.0:p__Firmicutes_F; c__Halanaerobiia # children taxa: 20\n",
      "length new taxa list: 12511\n",
      "60.0:p__Firmicutes; c__Bacilli # children taxa: 1878\n",
      "length new taxa list: 10634\n",
      "100.0:p__Fusobacteriota; c__Fusobacteriia # children taxa: 54\n",
      "length new taxa list: 10581\n",
      "100.0:c__Cyanobacteriia # children taxa: 597\n",
      "length new taxa list: 9985\n",
      "100.0:c__Vampirovibrionia # children taxa: 53\n",
      "length new taxa list: 9933\n",
      "100.0:c__Sericytochromatia # children taxa: 4\n",
      "length new taxa list: 9930\n",
      "100.0:c__WOR-1 # children taxa: 9\n",
      "length new taxa list: 9922\n",
      "100.0:c__ZB3 # children taxa: 10\n",
      "length new taxa list: 9913\n",
      "99.0:c__GWF2-35-9 # children taxa: 2\n",
      "length new taxa list: 9912\n",
      "100.0:c__UBA4738; o__UBA4738 # children taxa: 3\n",
      "length new taxa list: 9910\n",
      "99.0:c__Acidimicrobiia # children taxa: 183\n",
      "length new taxa list: 9728\n",
      "100.0:c__Thermoleophilia # children taxa: 48\n",
      "length new taxa list: 9681\n",
      "100.0:c__Rubrobacteria; o__Rubrobacterales; f__Rubrobacteraceae # children taxa: 3\n",
      "length new taxa list: 9679\n",
      "100.0:c__Coriobacteriia # children taxa: 133\n",
      "length new taxa list: 9547\n",
      "100.0:c__UBA9087 # children taxa: 5\n",
      "length new taxa list: 9543\n",
      "100.0:c__UBA1414; o__UBA1414; f__UBA1414 # children taxa: 4\n",
      "length new taxa list: 9540\n",
      "100.0:c__RBG-13-55-18 # children taxa: 4\n",
      "length new taxa list: 9537\n",
      "100.0:c__Eremiobacteria # children taxa: 26\n",
      "length new taxa list: 9512\n",
      "100.0:c__UBP9; o__UBA4705; f__UBA4705 # children taxa: 4\n",
      "length new taxa list: 9509\n",
      "100.0:p__Firmicutes_H; c__CSP1-3; o__CSP1-3 # children taxa: 3\n",
      "length new taxa list: 9507\n",
      "100.0:c__Fimbriimonadia # children taxa: 16\n",
      "length new taxa list: 9492\n",
      "100.0:c__Abditibacteria # children taxa: 3\n",
      "length new taxa list: 9490\n",
      "93.0:c__UBA5377 # children taxa: 4\n",
      "length new taxa list: 9487\n",
      "79.0:c__Paceibacteria # children taxa: 392\n",
      "length new taxa list: 9096\n",
      "100.0:c__Paceibacteria_A; o__Moranbacterales # children taxa: 47\n",
      "length new taxa list: 9050\n",
      "100.0:c__ABY1 # children taxa: 156\n",
      "length new taxa list: 8895\n",
      "100.0:c__Doudnabacteria; o__UBA920 # children taxa: 12\n",
      "length new taxa list: 8884\n",
      "100.0:c__Andersenbacteria; o__UBA10190; f__UBA10190 # children taxa: 5\n",
      "length new taxa list: 8880\n",
      "100.0:c__Gracilibacteria # children taxa: 67\n",
      "length new taxa list: 8814\n",
      "100.0:c__Saccharimonadia # children taxa: 72\n",
      "length new taxa list: 8743\n",
      "67.0:c__CPR2 # children taxa: 6\n",
      "length new taxa list: 8738\n",
      "100.0:c__UBA1384 # children taxa: 11\n",
      "length new taxa list: 8728\n",
      "100.0:c__Kazan-3B-28; o__Kazan-3B-28 # children taxa: 5\n",
      "length new taxa list: 8724\n",
      "99.0:c__Microgenomatia # children taxa: 211\n",
      "length new taxa list: 8514\n",
      "92.0:c__WWE3 # children taxa: 23\n",
      "length new taxa list: 8492\n",
      "100.0:c__CPR3; o__CPR3 # children taxa: 2\n",
      "length new taxa list: 8491\n",
      "100.0:c__4484-211; o__4484-211 # children taxa: 2\n",
      "length new taxa list: 8490\n",
      "100.0:c__Dojkabacteria # children taxa: 17\n",
      "length new taxa list: 8474\n",
      "100.0:c__Dehalococcoidia # children taxa: 173\n",
      "length new taxa list: 8302\n",
      "100.0:c__Anaerolineae # children taxa: 137\n",
      "length new taxa list: 8166\n",
      "100.0:c__Chloroflexia # children taxa: 28\n",
      "length new taxa list: 8139\n",
      "100.0:c__Ktedonobacteria; o__Ktedonobacterales; f__Ktedonobacteraceae # children taxa: 7\n",
      "length new taxa list: 8133\n",
      "100.0:c__UBA5177; o__UBA5177; f__UBA5177 # children taxa: 2\n",
      "length new taxa list: 8132\n",
      "100.0:p__Dormibacterota; c__Dormibacteria # children taxa: 32\n",
      "length new taxa list: 8101\n",
      "100.0:p__Chloroflexota_A; c__Ellin6529 # children taxa: 32\n",
      "length new taxa list: 8070\n",
      "100.0:p__Deinococcota; c__Deinococci; o__Deinococcales # children taxa: 69\n",
      "length new taxa list: 8002\n",
      "100.0:p__Thermotogota; c__Thermotogae # children taxa: 54\n",
      "length new taxa list: 7949\n",
      "100.0:p__Bipolaricaulota; c__Bipolaricaulia # children taxa: 11\n",
      "length new taxa list: 7939\n",
      "100.0:p__Caldisericota; c__Caldisericia # children taxa: 10\n",
      "length new taxa list: 7930\n",
      "100.0:p__Coprothermobacterota; c__Coprothermobacteria; o__Coprothermobacterales; f__Coprothermobacteraceae; g__Coprothermobacter # children taxa: 3\n",
      "length new taxa list: 7928\n",
      "100.0:p__Dictyoglomota; c__Dictyoglomia; o__Dictyoglomales; f__Dictyoglomaceae # children taxa: 3\n",
      "length new taxa list: 7926\n",
      "100.0:p__Thermodesulfobiota; c__Thermodesulfobiia; o__Thermodesulfobiales; f__Thermodesulfobiaceae; g__Thermodesulfobium # children taxa: 2\n",
      "length new taxa list: 7925\n",
      "100.0:c__Synergistia; o__Synergistales # children taxa: 40\n",
      "length new taxa list: 7886\n",
      "100.0:c__GBS-1; o__GBS-1; f__GBS-1 # children taxa: 2\n",
      "length new taxa list: 7885\n",
      "100.0:c__JS1; o__SB-45 # children taxa: 11\n",
      "length new taxa list: 7875\n",
      "100.0:c__Caldatribacteriia; o__Caldatribacteriales # children taxa: 4\n",
      "length new taxa list: 7872\n",
      "100.0:p__Riflebacteria; c__UBA8953; o__UBA8953; f__UBA8953 # children taxa: 5\n",
      "length new taxa list: 7868\n",
      "100.0:p__UBP17; c__UBA6191; o__UBA6191; f__UBA6191 # children taxa: 2\n",
      "length new taxa list: 7867\n",
      "100.0:p__Wallbacteria; c__UBA9980; o__UBA9980; f__UBA9980; g__UBA9980 # children taxa: 2\n",
      "length new taxa list: 7866\n",
      "100.0:c__Verrucomicrobiae # children taxa: 292\n",
      "length new taxa list: 7575\n",
      "100.0:c__Kiritimatiellae # children taxa: 40\n",
      "length new taxa list: 7536\n",
      "100.0:c__Lentisphaeria # children taxa: 26\n",
      "length new taxa list: 7511\n",
      "100.0:c__UBA3054; o__UBA3054; f__UBA3054; g__UBA3054 # children taxa: 2\n",
      "length new taxa list: 7510\n",
      "63.0:c__UBA1439 # children taxa: 2\n",
      "length new taxa list: 7509\n",
      "100.0:p__Verrucomicrobiota_A; c__Chlamydiia # children taxa: 46\n",
      "length new taxa list: 7464\n",
      "100.0:c__Planctomycetes # children taxa: 101\n",
      "length new taxa list: 7364\n",
      "100.0:c__Phycisphaerae # children taxa: 85\n",
      "length new taxa list: 7280\n",
      "55.0:c__Brocadiae # children taxa: 18\n",
      "length new taxa list: 7263\n",
      "100.0:c__UBA1135 # children taxa: 17\n",
      "length new taxa list: 7247\n",
      "100.0:c__GCA-002687715; o__GCA-002687715; f__GCA-002687715 # children taxa: 2\n",
      "length new taxa list: 7246\n",
      "100.0:c__UBA8108 # children taxa: 10\n",
      "length new taxa list: 7237\n",
      "67.0:c__UBA11346 # children taxa: 3\n",
      "length new taxa list: 7235\n",
      "100.0:c__SZUA-567; o__SZUA-567; f__SZUA-567 # children taxa: 2\n",
      "length new taxa list: 7234\n",
      "100.0:c__Spirochaetia # children taxa: 159\n",
      "length new taxa list: 7076\n",
      "100.0:c__GWE2-31-10; o__GWE2-31-10; f__GWE2-31-10 # children taxa: 2\n",
      "length new taxa list: 7075\n",
      "92.0:c__UBA6919 # children taxa: 2\n",
      "length new taxa list: 7074\n",
      "100.0:c__Brevinematia; o__Brevinematales # children taxa: 3\n",
      "length new taxa list: 7072\n",
      "100.0:c__Brachyspirae; o__Brachyspirales; f__Brachyspiraceae; g__Brachyspira # children taxa: 12\n",
      "length new taxa list: 7061\n",
      "100.0:c__Leptospirae # children taxa: 47\n",
      "length new taxa list: 7015\n",
      "100.0:c__UBA4802; o__UBA4802 # children taxa: 9\n",
      "length new taxa list: 7007\n",
      "100.0:p__UBP7; c__UBA6624; o__UBA6624; f__UBA6624; g__UBA6624 # children taxa: 2\n",
      "length new taxa list: 7006\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "100.0:c__Rhodothermia # children taxa: 56\n",
      "length new taxa list: 6951\n",
      "100.0:c__Chlorobia; o__Chlorobiales # children taxa: 23\n",
      "length new taxa list: 6929\n",
      "96.0:c__Ignavibacteria # children taxa: 36\n",
      "length new taxa list: 6894\n",
      "100.0:c__UBA10030 # children taxa: 23\n",
      "length new taxa list: 6872\n",
      "100.0:c__Kryptonia; o__Kryptoniales; f__Kryptoniaceae # children taxa: 6\n",
      "length new taxa list: 6867\n",
      "100.0:c__Kapabacteria # children taxa: 19\n",
      "length new taxa list: 6849\n",
      "100.0:c__SZUA-365; o__SZUA-365 # children taxa: 3\n",
      "length new taxa list: 6847\n",
      "51.0:p__KSB1; c__UBA2214 # children taxa: 6\n",
      "length new taxa list: 6842\n",
      "100.0:p__Calditrichota; c__Calditrichia # children taxa: 2\n",
      "length new taxa list: 6841\n",
      "100.0:c__Marinisomatia # children taxa: 81\n",
      "length new taxa list: 6761\n",
      "100.0:c__AB16; o__AB16 # children taxa: 3\n",
      "length new taxa list: 6759\n",
      "100.0:c__UBA2242; o__UBA2242; f__UBA2242 # children taxa: 3\n",
      "length new taxa list: 6757\n",
      "100.0:p__Delongbacteria; c__UBA4055; o__UBA4055; f__UBA4055 # children taxa: 4\n",
      "length new taxa list: 6754\n",
      "100.0:p__AABM5-125-24; c__AABM5-125-24 # children taxa: 2\n",
      "length new taxa list: 6753\n",
      "100.0:c__Fibrobacteria # children taxa: 38\n",
      "length new taxa list: 6716\n",
      "100.0:c__OXYB2-FULL-49-7; o__OXYB2-FULL-49-7 # children taxa: 2\n",
      "length new taxa list: 6715\n",
      "100.0:c__Chitinivibrionia; o__Chitinivibrionales # children taxa: 5\n",
      "length new taxa list: 6711\n",
      "100.0:p__Cloacimonadota; c__Cloacimonadia # children taxa: 23\n",
      "length new taxa list: 6689\n",
      "100.0:c__Gemmatimonadetes # children taxa: 75\n",
      "length new taxa list: 6615\n",
      "95.0:p__Krumholzibacteriota; c__Krumholzibacteria # children taxa: 5\n",
      "length new taxa list: 6611\n",
      "100.0:p__Eisenbacteria; c__RBG-16-71-46 # children taxa: 2\n",
      "length new taxa list: 6610\n",
      "100.0:p__Edwardsbacteria; c__AC1; o__AC1 # children taxa: 4\n",
      "length new taxa list: 6607\n",
      "100.0:c__UBA2968 # children taxa: 9\n",
      "length new taxa list: 6599\n",
      "100.0:p__Zixibacteria; c__MSB-5A5 # children taxa: 12\n",
      "length new taxa list: 6588\n",
      "100.0:p__UBP14; c__UBA6098 # children taxa: 2\n",
      "length new taxa list: 6587\n",
      "99.0:p__WOR-3_A; c__Hydrothermia # children taxa: 4\n",
      "length new taxa list: 6584\n",
      "100.0:p__Calescibacterota; c__Calescibacteriia; o__Calescibacteriales; f__Calescibacteriaceae; g__Calescibacterium # children taxa: 2\n",
      "length new taxa list: 6583\n",
      "100.0:p__WOR-3; c__WOR-3 # children taxa: 6\n",
      "length new taxa list: 6578\n",
      "100.0:c__UBA3073; o__UBA3073; f__UBA3073; g__UBA3073 # children taxa: 2\n",
      "length new taxa list: 6577\n",
      "100.0:c__32-111; o__32-111; f__32-111 # children taxa: 3\n",
      "length new taxa list: 6575\n",
      "100.0:p__Fermentibacterota; c__Fermentibacteria; o__Fermentibacterales; f__Fermentibacteraceae # children taxa: 6\n",
      "length new taxa list: 6570\n",
      "100.0:p__Poribacteria; c__WGA-4E # children taxa: 8\n",
      "length new taxa list: 6563\n",
      "100.0:p__Hydrogenedentota; c__Hydrogenedentia; o__Hydrogenedentiales # children taxa: 4\n",
      "length new taxa list: 6560\n",
      "100.0:p__Sumerlaeota; c__Sumerlaeia # children taxa: 3\n",
      "length new taxa list: 6558\n",
      "99.0:c__Gammaproteobacteria # children taxa: 4645\n",
      "length new taxa list: 1914\n",
      "96.0:c__Zetaproteobacteria # children taxa: 18\n",
      "length new taxa list: 1897\n",
      "100.0:c__Magnetococcia; o__Magnetococcales # children taxa: 12\n",
      "length new taxa list: 1886\n",
      "100.0:p__Dependentiae; c__Babeliae; o__Babeliales # children taxa: 22\n",
      "length new taxa list: 1865\n",
      "100.0:p__AB1-6; c__AB1-6; o__AB1-6 # children taxa: 2\n",
      "length new taxa list: 1864\n",
      "100.0:p__UBP6; c__UBA1177; o__UBA1177; f__UBA1177; g__UBA1177 # children taxa: 3\n",
      "length new taxa list: 1862\n",
      "100.0:p__SAR324; c__SAR324 # children taxa: 21\n",
      "length new taxa list: 1842\n",
      "100.0:c__Campylobacteria # children taxa: 248\n",
      "length new taxa list: 1595\n",
      "100.0:c__Desulfurellia; o__Desulfurellales # children taxa: 6\n",
      "length new taxa list: 1590\n",
      "100.0:c__Aquificae # children taxa: 21\n",
      "length new taxa list: 1570\n",
      "100.0:c__Desulfurobacteriia; o__Desulfurobacteriales; f__Desulfurobacteriaceae # children taxa: 5\n",
      "length new taxa list: 1566\n",
      "100.0:p__Deferribacterota; c__Deferribacteres # children taxa: 14\n",
      "length new taxa list: 1553\n",
      "100.0:p__Chrysiogenetota; c__Chrysiogenetes; o__Chrysiogenales; f__Chrysiogenaceae # children taxa: 2\n",
      "length new taxa list: 1552\n",
      "100.0:p__Dadabacteria; c__UBA1144 # children taxa: 14\n",
      "length new taxa list: 1539\n",
      "31.0:c__Desulfobacteria # children taxa: 63\n",
      "length new taxa list: 1477\n",
      "100.0:c__Syntrophobacteria # children taxa: 12\n",
      "length new taxa list: 1466\n",
      "28.0:c__Desulfarculia # children taxa: 5\n",
      "length new taxa list: 1462\n",
      "100.0:c__Desulfobaccia; o__Desulfobaccales # children taxa: 3\n",
      "length new taxa list: 1460\n",
      "100.0:c__Desulfobulbia; o__Desulfobulbales # children taxa: 49\n",
      "length new taxa list: 1412\n",
      "100.0:c__Thermodesulfobacteria; o__Thermodesulfobacteriales # children taxa: 13\n",
      "length new taxa list: 1400\n",
      "100.0:c__Dissulfuribacteria; o__Dissulfuribacterales # children taxa: 5\n",
      "length new taxa list: 1396\n",
      "100.0:c__Desulfofervidia; o__Desulfofervidales # children taxa: 2\n",
      "length new taxa list: 1395\n",
      "100.0:c__Syntrophia; o__Syntrophales # children taxa: 53\n",
      "length new taxa list: 1343\n",
      "83.0:c__BSN033 # children taxa: 11\n",
      "length new taxa list: 1333\n",
      "100.0:c__Syntrophorhabdia; o__Syntrophorhabdales # children taxa: 13\n",
      "length new taxa list: 1321\n",
      "100.0:c__Desulfomonilia # children taxa: 4\n",
      "length new taxa list: 1318\n",
      "100.0:p__Desulfuromonadota; c__Desulfuromonadia # children taxa: 81\n",
      "length new taxa list: 1238\n",
      "100.0:p__GWC2-55-46; c__GWC2-55-46; o__GWC2-55-46 # children taxa: 8\n",
      "length new taxa list: 1231\n",
      "100.0:c__Thermodesulfovibrionia # children taxa: 43\n",
      "length new taxa list: 1189\n",
      "93.0:c__Nitrospiria # children taxa: 44\n",
      "length new taxa list: 1146\n",
      "100.0:c__UBA9217; o__UBA9217; f__UBA9217 # children taxa: 3\n",
      "length new taxa list: 1144\n",
      "100.0:p__Methylomirabilota; c__Methylomirabilia # children taxa: 40\n",
      "length new taxa list: 1105\n",
      "100.0:p__Moduliflexota; c__Moduliflexia; o__Moduliflexales; f__Moduliflexaceae # children taxa: 3\n",
      "length new taxa list: 1103\n",
      "100.0:p__SZUA-182; c__SZUA-182; o__SZUA-182 # children taxa: 2\n",
      "length new taxa list: 1102\n",
      "100.0:c__Nitrospinia; o__Nitrospinales; f__Nitrospinaceae # children taxa: 16\n",
      "length new taxa list: 1087\n",
      "100.0:c__UBA9942; o__UBA9942 # children taxa: 4\n",
      "length new taxa list: 1084\n",
      "100.0:p__UBA8248; c__UBA8248; o__UBA8248 # children taxa: 4\n",
      "length new taxa list: 1081\n",
      "100.0:p__Tectomicrobia; c__Entotheonellia; o__Entotheonellales; f__Entotheonellaceae; g__Entotheonella # children taxa: 2\n",
      "length new taxa list: 1080\n",
      "100.0:p__Schekmanbacteria; c__GWA2-38-11; o__GWA2-38-11; f__GWA2-38-11 # children taxa: 3\n",
      "length new taxa list: 1078\n",
      "100.0:p__BMS3Abin14; c__BMS3Abin14; o__BMS3Abin14; f__BMS3Abin14 # children taxa: 3\n",
      "length new taxa list: 1076\n",
      "100.0:p__Binatota; c__Binatia # children taxa: 15\n",
      "length new taxa list: 1062\n",
      "100.0:p__Deferrisomatota; c__Defferisomatia; o__Defferisomatales # children taxa: 3\n",
      "length new taxa list: 1060\n",
      "100.0:p__MBNT15; c__MBNT15; o__MBNT15; f__MBNT15 # children taxa: 9\n",
      "length new taxa list: 1052\n",
      "79.0:c__Polyangia # children taxa: 48\n",
      "length new taxa list: 1005\n",
      "100.0:c__Myxococcia; o__Myxococcales # children taxa: 30\n",
      "length new taxa list: 976\n",
      "94.0:c__UBA727 # children taxa: 3\n",
      "length new taxa list: 974\n",
      "63.0:c__Bradimonadia # children taxa: 7\n",
      "length new taxa list: 968\n",
      "100.0:c__UBA9042 # children taxa: 4\n",
      "length new taxa list: 965\n",
      "100.0:c__UBA9160 # children taxa: 11\n",
      "length new taxa list: 955\n",
      "100.0:c__UBA796 # children taxa: 8\n",
      "length new taxa list: 948\n",
      "100.0:c__Oligoflexia # children taxa: 7\n",
      "length new taxa list: 942\n",
      "100.0:c__UBA2361 # children taxa: 2\n",
      "length new taxa list: 941\n",
      "75.0:c__Bacteriovoracia # children taxa: 46\n",
      "length new taxa list: 896\n",
      "59.0:c__UBA2394 # children taxa: 3\n",
      "length new taxa list: 894\n",
      "100.0:c__Bdellovibrionia; o__Bdellovibrionales # children taxa: 29\n",
      "length new taxa list: 866\n",
      "84.0:p__O2-12-FULL-43-9; c__O2-12-FULL-43-9 # children taxa: 3\n",
      "length new taxa list: 864\n",
      "100.0:p__UBA10199; c__UBA10199 # children taxa: 8\n",
      "length new taxa list: 857\n",
      "100.0:p__Desulfobacterota_A; c__Desulfovibrionia; o__Desulfovibrionales # children taxa: 103\n",
      "length new taxa list: 755\n",
      "100.0:p__Nitrospirota_A; c__Leptospirillia; o__Leptospirillales; f__Leptospirillaceae # children taxa: 4\n",
      "length new taxa list: 752\n",
      "100.0:c__Acidobacteriae # children taxa: 212\n",
      "length new taxa list: 541\n",
      "100.0:c__Blastocatellia # children taxa: 30\n",
      "length new taxa list: 512\n",
      "100.0:c__UBA6911; o__UBA6911; f__UBA6911 # children taxa: 2\n",
      "length new taxa list: 511\n",
      "96.0:c__Vicinamibacteria # children taxa: 39\n",
      "length new taxa list: 473\n",
      "97.0:c__Mor1 # children taxa: 6\n",
      "length new taxa list: 468\n",
      "100.0:c__UBA890; o__UBA890; f__UBA890; g__UBA890 # children taxa: 2\n",
      "length new taxa list: 467\n",
      "97.0:c__Thermoanaerobaculia # children taxa: 16\n",
      "length new taxa list: 452\n",
      "100.0:c__Holophagae; o__Holophagales; f__Holophagaceae # children taxa: 9\n",
      "length new taxa list: 444\n",
      "100.0:c__Aminicenantia # children taxa: 19\n",
      "length new taxa list: 426\n",
      "98.0:c__koll11 # children taxa: 53\n",
      "length new taxa list: 374\n",
      "100.0:c__Omnitrophia; o__Omnitrophales # children taxa: 11\n",
      "length new taxa list: 364\n",
      "100.0:p__Goldbacteria; c__PGYV01; o__PGYV01 # children taxa: 3\n",
      "length new taxa list: 362\n",
      "100.0:p__Firestonebacteria; c__D2-FULL-39-29; o__D2-FULL-39-29; f__D2-FULL-39-29; g__D2-FULL-39-29 # children taxa: 2\n",
      "length new taxa list: 361\n",
      "100.0:p__FCPU426; c__Palsa-1180; o__Palsa-1180; f__Palsa-1180 # children taxa: 2\n",
      "length new taxa list: 360\n",
      "100.0:p__Desantisbacteria; c__UBA1551; o__UBA1551 # children taxa: 2\n",
      "length new taxa list: 359\n",
      "100.0:c__Elusimicrobia # children taxa: 46\n",
      "length new taxa list: 314\n",
      "72.0:c__Endomicrobia # children taxa: 11\n",
      "length new taxa list: 304\n",
      "100.0:c__UBA8919; o__UBA8919; f__UBA8919 # children taxa: 4\n",
      "length new taxa list: 301\n",
      "100.0:p__UBA6262; c__UBA6262; o__UBA6262; f__UBA6262 # children taxa: 2\n",
      "length new taxa list: 300\n",
      "100.0:p__CG03; c__CG03; o__CG03; f__CG03 # children taxa: 3\n",
      "length new taxa list: 298\n",
      "298\n"
     ]
    }
   ],
   "source": [
    "parsed_arc_train_class = arc_train_class.copy()\n",
    "parse_taxa_from_node(arc.seed_node,parsed_arc_train_class,genomes_per_taxa=1,prefix='c__')\n",
    "print(len(parsed_arc_train_class))\n",
    "\n",
    "parsed_bac_train_class = bac_train_class.copy()\n",
    "parse_taxa_from_node(bac.seed_node,parsed_bac_train_class,genomes_per_taxa=1,prefix='c__')\n",
    "print(len(parsed_bac_train_class))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32 298\n",
      "330\n"
     ]
    }
   ],
   "source": [
    "print(len(parsed_arc_train_class),len(parsed_bac_train_class))\n",
    "print(len(parsed_arc_train_class)+len(parsed_bac_train_class))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "330\n",
      "330\n"
     ]
    }
   ],
   "source": [
    "traindf_class = get_df_subset(ftp,arcftp,parsed_bac_train_class,parsed_arc_train_class)\n",
    "print(len(traindf_class))\n",
    "\n",
    "train_fnames_class = get_filenames(traindf_class)\n",
    "print(len(train_fnames_class))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "traindf_class.to_csv('TrainNew_ParsedClass_GTDBrepwithftp.csv')\n",
    "save_filenames('trainnew_parsedclass_filenames.txt',train_fnames_class)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "86.0:c__Thermoplasmata # children taxa: 67\n",
      "length new taxa list: 48\n",
      "93.0:c__Bathyarchaeia # children taxa: 40\n",
      "length new taxa list: 16\n",
      "99.0:c__Actinobacteria # children taxa: 2735\n",
      "length new taxa list: 2963\n",
      "42.0:c__Alphaproteobacteria # children taxa: 2955\n",
      "length new taxa list: 16\n",
      "16\n",
      "16\n",
      "32\n"
     ]
    }
   ],
   "source": [
    "#now in valid - doing 8 per class for a total of 32 genomes in valid set (10% size of train set)\n",
    "parsed_arc_valid_class = arc_valid_class.copy()\n",
    "parse_taxa_from_node(arc.seed_node,parsed_arc_valid_class,genomes_per_taxa=8,prefix='c__')\n",
    "\n",
    "parsed_bac_valid_class = bac_valid_class.copy()\n",
    "parse_taxa_from_node(bac.seed_node,parsed_bac_valid_class,genomes_per_taxa=8,prefix='c__')\n",
    "\n",
    "print(len(parsed_arc_valid_class))\n",
    "print(len(parsed_bac_valid_class))\n",
    "print(len(parsed_arc_valid_class)+len(parsed_bac_valid_class))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32\n",
      "32\n"
     ]
    }
   ],
   "source": [
    "valdf_class = get_df_subset(ftp,arcftp,parsed_bac_valid_class,parsed_arc_valid_class)\n",
    "print(len(valdf_class))\n",
    "\n",
    "valid_fnames_class = get_filenames(valdf_class)\n",
    "print(len(valid_fnames_class))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "valdf_class.to_csv('ValidNew_ParsedClass_GTDBrepwithftp.csv')\n",
    "save_filenames('validnew_parsedclass_filenames.txt',valid_fnames_class)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "97.0:c__Methanosarcinia # children taxa: 76\n",
      "length new taxa list: 89\n",
      "80.0:c__Nitrososphaeria # children taxa: 81\n",
      "length new taxa list: 16\n",
      "16\n",
      "32.0:c__Clostridia # children taxa: 1836\n",
      "length new taxa list: 2685\n",
      "100.0:c__Bacteroidia # children taxa: 2677\n",
      "length new taxa list: 16\n",
      "16\n"
     ]
    }
   ],
   "source": [
    "#now in test - doing 8 per class for total of 32 genomes in test set (~10% train size)\n",
    "parsed_arc_test_class = arc_test_class.copy()\n",
    "parse_taxa_from_node(arc.seed_node,parsed_arc_test_class,genomes_per_taxa=8,prefix='c__')\n",
    "print(len(parsed_arc_test_class))\n",
    "\n",
    "parsed_bac_test_class = bac_test_class.copy()\n",
    "parse_taxa_from_node(bac.seed_node,parsed_bac_test_class,genomes_per_taxa=8,prefix='c__')\n",
    "print(len(parsed_bac_test_class))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32\n",
      "32\n"
     ]
    }
   ],
   "source": [
    "testdf_class = get_df_subset(ftp,arcftp,parsed_bac_test_class,parsed_arc_test_class)\n",
    "print(len(testdf_class))\n",
    "\n",
    "test_fnames_class = get_filenames(testdf_class)\n",
    "print(len(test_fnames_class))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "testdf_class.to_csv('TestNew_ParsedClass_GTDBrepwithftp.csv')\n",
    "save_filenames('testnew_parsedclass_filenames.txt',test_fnames_class)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th># assembly_accession</th>\n",
       "      <th>asm_name</th>\n",
       "      <th>ftp_path</th>\n",
       "      <th>index</th>\n",
       "      <th>ncbi_assembly_level</th>\n",
       "      <th>ncbi_assembly_name</th>\n",
       "      <th>ncbi_bioproject</th>\n",
       "      <th>ncbi_biosample</th>\n",
       "      <th>ncbi_genbank_assembly_accession</th>\n",
       "      <th>ncbi_genome_representation</th>\n",
       "      <th>ncbi_organism_name</th>\n",
       "      <th>ncbi_refseq_category</th>\n",
       "      <th>parsed_genome_accession</th>\n",
       "      <th>srr_accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3714</td>\n",
       "      <td>GCA_000526435.1</td>\n",
       "      <td>ASM52643v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Contig</td>\n",
       "      <td>ASM52643v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02584936</td>\n",
       "      <td>GCA_000526435.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Caldicoprobacter oshimai DSM 21659</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000526435.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>755</td>\n",
       "      <td>GCA_000166775.1</td>\n",
       "      <td>ASM16677v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM16677v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN00713565</td>\n",
       "      <td>GCA_000166775.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Caldicellulosiruptor kronotskyensis 2002</td>\n",
       "      <td>na</td>\n",
       "      <td>GCF_000166775.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1961</td>\n",
       "      <td>GCA_000328765.2</td>\n",
       "      <td>ASM32876v2</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM32876v2</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMEA2272437</td>\n",
       "      <td>GCA_000328765.2</td>\n",
       "      <td>full</td>\n",
       "      <td>Tepidanaerobacter acetatoxydans Re1</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000328765.2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3491</td>\n",
       "      <td>GCA_000499205.1</td>\n",
       "      <td>MAEPY2_1.0</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Contig</td>\n",
       "      <td>MAEPY2 1.0</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02343193</td>\n",
       "      <td>GCA_000499205.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Paenibacillus sp. MAEPY2</td>\n",
       "      <td>na</td>\n",
       "      <td>GCF_000499205.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8636</td>\n",
       "      <td>GCA_001730225.1</td>\n",
       "      <td>ASM173022v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/001...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Contig</td>\n",
       "      <td>ASM173022v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN05731213</td>\n",
       "      <td>GCA_001730225.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Desulfuribacillus alkaliarsenatis</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_001730225.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0 # assembly_accession     asm_name  \\\n",
       "0        3714      GCA_000526435.1   ASM52643v1   \n",
       "1         755      GCA_000166775.1   ASM16677v1   \n",
       "2        1961      GCA_000328765.2   ASM32876v2   \n",
       "3        3491      GCA_000499205.1   MAEPY2_1.0   \n",
       "4        8636      GCA_001730225.1  ASM173022v1   \n",
       "\n",
       "                                            ftp_path index  \\\n",
       "0  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...   NaN   \n",
       "1  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...   NaN   \n",
       "2  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...   NaN   \n",
       "3  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...   NaN   \n",
       "4  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/001...   NaN   \n",
       "\n",
       "  ncbi_assembly_level ncbi_assembly_name ncbi_bioproject ncbi_biosample  \\\n",
       "0              Contig         ASM52643v1     PRJNA224116   SAMN02584936   \n",
       "1     Complete Genome         ASM16677v1     PRJNA224116   SAMN00713565   \n",
       "2     Complete Genome         ASM32876v2     PRJNA224116   SAMEA2272437   \n",
       "3              Contig         MAEPY2 1.0     PRJNA224116   SAMN02343193   \n",
       "4              Contig        ASM173022v1     PRJNA224116   SAMN05731213   \n",
       "\n",
       "  ncbi_genbank_assembly_accession ncbi_genome_representation  \\\n",
       "0                 GCA_000526435.1                       full   \n",
       "1                 GCA_000166775.1                       full   \n",
       "2                 GCA_000328765.2                       full   \n",
       "3                 GCA_000499205.1                       full   \n",
       "4                 GCA_001730225.1                       full   \n",
       "\n",
       "                         ncbi_organism_name   ncbi_refseq_category  \\\n",
       "0        Caldicoprobacter oshimai DSM 21659  representative genome   \n",
       "1  Caldicellulosiruptor kronotskyensis 2002                     na   \n",
       "2       Tepidanaerobacter acetatoxydans Re1  representative genome   \n",
       "3                  Paenibacillus sp. MAEPY2                     na   \n",
       "4         Desulfuribacillus alkaliarsenatis  representative genome   \n",
       "\n",
       "  parsed_genome_accession srr_accession  \n",
       "0         GCF_000526435.1           NaN  \n",
       "1         GCF_000166775.1           NaN  \n",
       "2         GCF_000328765.2           NaN  \n",
       "3         GCF_000499205.1           NaN  \n",
       "4         GCF_001730225.1           NaN  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#saving train/valid/test tables for paper\n",
    "train = pd.read_csv('TrainNew_ParsedClass_GTDBrepwithftp.csv')\n",
    "valid = pd.read_csv('ValidNew_ParsedClass_GTDBrepwithftp.csv')\n",
    "test = pd.read_csv('TestNew_ParsedClass_GTDBrepwithftp.csv')\n",
    "train.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/adrienne/anaconda3/envs/fastai/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ncbi_genbank_assembly_accession</th>\n",
       "      <th>ncbi_assembly_name</th>\n",
       "      <th>ncbi_biosample</th>\n",
       "      <th>ncbi_organism_name</th>\n",
       "      <th>parsed_genome_accession</th>\n",
       "      <th>dataset</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000526435.1</td>\n",
       "      <td>ASM52643v1</td>\n",
       "      <td>SAMN02584936</td>\n",
       "      <td>Caldicoprobacter oshimai DSM 21659</td>\n",
       "      <td>GCF_000526435.1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000166775.1</td>\n",
       "      <td>ASM16677v1</td>\n",
       "      <td>SAMN00713565</td>\n",
       "      <td>Caldicellulosiruptor kronotskyensis 2002</td>\n",
       "      <td>GCF_000166775.1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000328765.2</td>\n",
       "      <td>ASM32876v2</td>\n",
       "      <td>SAMEA2272437</td>\n",
       "      <td>Tepidanaerobacter acetatoxydans Re1</td>\n",
       "      <td>GCF_000328765.2</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000499205.1</td>\n",
       "      <td>MAEPY2 1.0</td>\n",
       "      <td>SAMN02343193</td>\n",
       "      <td>Paenibacillus sp. MAEPY2</td>\n",
       "      <td>GCF_000499205.1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_001730225.1</td>\n",
       "      <td>ASM173022v1</td>\n",
       "      <td>SAMN05731213</td>\n",
       "      <td>Desulfuribacillus alkaliarsenatis</td>\n",
       "      <td>GCF_001730225.1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  ncbi_genbank_assembly_accession ncbi_assembly_name ncbi_biosample  \\\n",
       "0                 GCA_000526435.1         ASM52643v1   SAMN02584936   \n",
       "1                 GCA_000166775.1         ASM16677v1   SAMN00713565   \n",
       "2                 GCA_000328765.2         ASM32876v2   SAMEA2272437   \n",
       "3                 GCA_000499205.1         MAEPY2 1.0   SAMN02343193   \n",
       "4                 GCA_001730225.1        ASM173022v1   SAMN05731213   \n",
       "\n",
       "                         ncbi_organism_name parsed_genome_accession dataset  \n",
       "0        Caldicoprobacter oshimai DSM 21659         GCF_000526435.1   train  \n",
       "1  Caldicellulosiruptor kronotskyensis 2002         GCF_000166775.1   train  \n",
       "2       Tepidanaerobacter acetatoxydans Re1         GCF_000328765.2   train  \n",
       "3                  Paenibacillus sp. MAEPY2         GCF_000499205.1   train  \n",
       "4         Desulfuribacillus alkaliarsenatis         GCF_001730225.1   train  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subt = train[['ncbi_genbank_assembly_accession','ncbi_assembly_name','ncbi_biosample','ncbi_organism_name','parsed_genome_accession']]\n",
    "subt['dataset'] = 'train'\n",
    "subt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/adrienne/anaconda3/envs/fastai/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/Users/adrienne/anaconda3/envs/fastai/lib/python3.7/site-packages/ipykernel_launcher.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ncbi_genbank_assembly_accession</th>\n",
       "      <th>ncbi_assembly_name</th>\n",
       "      <th>ncbi_biosample</th>\n",
       "      <th>ncbi_organism_name</th>\n",
       "      <th>parsed_genome_accession</th>\n",
       "      <th>dataset</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000526435.1</td>\n",
       "      <td>ASM52643v1</td>\n",
       "      <td>SAMN02584936</td>\n",
       "      <td>Caldicoprobacter oshimai DSM 21659</td>\n",
       "      <td>GCF_000526435.1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000166775.1</td>\n",
       "      <td>ASM16677v1</td>\n",
       "      <td>SAMN00713565</td>\n",
       "      <td>Caldicellulosiruptor kronotskyensis 2002</td>\n",
       "      <td>GCF_000166775.1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000328765.2</td>\n",
       "      <td>ASM32876v2</td>\n",
       "      <td>SAMEA2272437</td>\n",
       "      <td>Tepidanaerobacter acetatoxydans Re1</td>\n",
       "      <td>GCF_000328765.2</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000499205.1</td>\n",
       "      <td>MAEPY2 1.0</td>\n",
       "      <td>SAMN02343193</td>\n",
       "      <td>Paenibacillus sp. MAEPY2</td>\n",
       "      <td>GCF_000499205.1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_001730225.1</td>\n",
       "      <td>ASM173022v1</td>\n",
       "      <td>SAMN05731213</td>\n",
       "      <td>Desulfuribacillus alkaliarsenatis</td>\n",
       "      <td>GCF_001730225.1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  ncbi_genbank_assembly_accession ncbi_assembly_name ncbi_biosample  \\\n",
       "0                 GCA_000526435.1         ASM52643v1   SAMN02584936   \n",
       "1                 GCA_000166775.1         ASM16677v1   SAMN00713565   \n",
       "2                 GCA_000328765.2         ASM32876v2   SAMEA2272437   \n",
       "3                 GCA_000499205.1         MAEPY2 1.0   SAMN02343193   \n",
       "4                 GCA_001730225.1        ASM173022v1   SAMN05731213   \n",
       "\n",
       "                         ncbi_organism_name parsed_genome_accession dataset  \n",
       "0        Caldicoprobacter oshimai DSM 21659         GCF_000526435.1   train  \n",
       "1  Caldicellulosiruptor kronotskyensis 2002         GCF_000166775.1   train  \n",
       "2       Tepidanaerobacter acetatoxydans Re1         GCF_000328765.2   train  \n",
       "3                  Paenibacillus sp. MAEPY2         GCF_000499205.1   train  \n",
       "4         Desulfuribacillus alkaliarsenatis         GCF_001730225.1   train  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subv = valid[['ncbi_genbank_assembly_accession','ncbi_assembly_name','ncbi_biosample','ncbi_organism_name','parsed_genome_accession']]\n",
    "subv['dataset'] = 'valid'\n",
    "\n",
    "subtt = test[['ncbi_genbank_assembly_accession','ncbi_assembly_name','ncbi_biosample','ncbi_organism_name','parsed_genome_accession']]\n",
    "subtt['dataset'] = 'test'\n",
    "\n",
    "sub = subt.append(subv).append(subtt)\n",
    "sub.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "sub.to_csv('../GetGTDBGenomes/GenomeIDs_parsedclass.csv',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/adrienne/Projects/LanguageofLife/GetGTDBGenomes_old'"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "394"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(sub)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "330 32 32\n"
     ]
    }
   ],
   "source": [
    "print(len(train),len(valid),len(test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "cl = pd.read_csv('GenomeIDs_parsedclass.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ncbi_genbank_assembly_accession</th>\n",
       "      <th>ncbi_assembly_name</th>\n",
       "      <th>ncbi_biosample</th>\n",
       "      <th>ncbi_organism_name</th>\n",
       "      <th>parsed_genome_accession</th>\n",
       "      <th>dataset</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000526435.1</td>\n",
       "      <td>ASM52643v1</td>\n",
       "      <td>SAMN02584936</td>\n",
       "      <td>Caldicoprobacter oshimai DSM 21659</td>\n",
       "      <td>GCF_000526435.1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000166775.1</td>\n",
       "      <td>ASM16677v1</td>\n",
       "      <td>SAMN00713565</td>\n",
       "      <td>Caldicellulosiruptor kronotskyensis 2002</td>\n",
       "      <td>GCF_000166775.1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000328765.2</td>\n",
       "      <td>ASM32876v2</td>\n",
       "      <td>SAMEA2272437</td>\n",
       "      <td>Tepidanaerobacter acetatoxydans Re1</td>\n",
       "      <td>GCF_000328765.2</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000499205.1</td>\n",
       "      <td>MAEPY2 1.0</td>\n",
       "      <td>SAMN02343193</td>\n",
       "      <td>Paenibacillus sp. MAEPY2</td>\n",
       "      <td>GCF_000499205.1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_001730225.1</td>\n",
       "      <td>ASM173022v1</td>\n",
       "      <td>SAMN05731213</td>\n",
       "      <td>Desulfuribacillus alkaliarsenatis</td>\n",
       "      <td>GCF_001730225.1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  ncbi_genbank_assembly_accession ncbi_assembly_name ncbi_biosample  \\\n",
       "0                 GCA_000526435.1         ASM52643v1   SAMN02584936   \n",
       "1                 GCA_000166775.1         ASM16677v1   SAMN00713565   \n",
       "2                 GCA_000328765.2         ASM32876v2   SAMEA2272437   \n",
       "3                 GCA_000499205.1         MAEPY2 1.0   SAMN02343193   \n",
       "4                 GCA_001730225.1        ASM173022v1   SAMN05731213   \n",
       "\n",
       "                         ncbi_organism_name parsed_genome_accession dataset  \n",
       "0        Caldicoprobacter oshimai DSM 21659         GCF_000526435.1   train  \n",
       "1  Caldicellulosiruptor kronotskyensis 2002         GCF_000166775.1   train  \n",
       "2       Tepidanaerobacter acetatoxydans Re1         GCF_000328765.2   train  \n",
       "3                  Paenibacillus sp. MAEPY2         GCF_000499205.1   train  \n",
       "4         Desulfuribacillus alkaliarsenatis         GCF_001730225.1   train  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cl.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>metaseek_environmental_package</th>\n",
       "      <th>sra_run_accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>sediment</td>\n",
       "      <td>SRR4069407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>sediment</td>\n",
       "      <td>ERR1474613</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>water</td>\n",
       "      <td>SRR4343439</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>built environment</td>\n",
       "      <td>ERR1332600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>human-gut</td>\n",
       "      <td>ERR209626</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  metaseek_environmental_package sra_run_accession\n",
       "0                       sediment        SRR4069407\n",
       "1                       sediment        ERR1474613\n",
       "2                          water        SRR4343439\n",
       "3              built environment        ERR1332600\n",
       "4                      human-gut         ERR209626"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "env = pd.read_csv('EvenEnv100_metagenomes.csv')\n",
    "env.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "394"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(cl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "96"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(env)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Counter({'sediment': 10,\n",
       "         'water': 10,\n",
       "         'built environment': 9,\n",
       "         'human-gut': 9,\n",
       "         'wastewater/sludge': 10,\n",
       "         'microbial mat/biofilm': 10,\n",
       "         'human-skin': 1,\n",
       "         'host-associated': 7,\n",
       "         'miscellaneous': 10,\n",
       "         'soil': 10,\n",
       "         'plant-associated': 10})"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from collections import Counter\n",
    "Counter(env['metaseek_environmental_package'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# parse ftp file to just class-parsed genome IDs "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th># assembly_accession</th>\n",
       "      <th>asm_name</th>\n",
       "      <th>ftp_path</th>\n",
       "      <th>index</th>\n",
       "      <th>ncbi_assembly_level</th>\n",
       "      <th>ncbi_assembly_name</th>\n",
       "      <th>ncbi_bioproject</th>\n",
       "      <th>ncbi_biosample</th>\n",
       "      <th>ncbi_genbank_assembly_accession</th>\n",
       "      <th>ncbi_genome_representation</th>\n",
       "      <th>ncbi_organism_name</th>\n",
       "      <th>ncbi_refseq_category</th>\n",
       "      <th>parsed_genome_accession</th>\n",
       "      <th>srr_accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000006175.2</td>\n",
       "      <td>ASM617v2</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>GCA_000006175.2</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM617v2</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN00000040</td>\n",
       "      <td>GCA_000006175.2</td>\n",
       "      <td>full</td>\n",
       "      <td>Methanococcus voltae A3</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000006175.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000006805.1</td>\n",
       "      <td>ASM680v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>GCA_000006805.1</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM680v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02604216</td>\n",
       "      <td>GCA_000006805.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Halobacterium salinarum NRC-1</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000006805.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000007185.1</td>\n",
       "      <td>ASM718v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>GCA_000007185.1</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM718v1</td>\n",
       "      <td>PRJNA294</td>\n",
       "      <td>SAMN02603235</td>\n",
       "      <td>GCA_000007185.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Methanopyrus kandleri AV19</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_000007185.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000007225.1</td>\n",
       "      <td>ASM722v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>GCA_000007225.1</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM722v1</td>\n",
       "      <td>PRJNA172</td>\n",
       "      <td>SAMN02604075</td>\n",
       "      <td>GCA_000007225.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Pyrobaculum aerophilum str. IM2</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_000007225.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000007305.1</td>\n",
       "      <td>ASM730v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>GCA_000007305.1</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM730v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02604284</td>\n",
       "      <td>GCA_000007305.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Pyrococcus furiosus DSM 3638</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000007305.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  # assembly_accession  asm_name  \\\n",
       "0      GCA_000006175.2  ASM617v2   \n",
       "1      GCA_000006805.1  ASM680v1   \n",
       "2      GCA_000007185.1  ASM718v1   \n",
       "3      GCA_000007225.1  ASM722v1   \n",
       "4      GCA_000007305.1  ASM730v1   \n",
       "\n",
       "                                            ftp_path            index  \\\n",
       "0  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...  GCA_000006175.2   \n",
       "1  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...  GCA_000006805.1   \n",
       "2  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...  GCA_000007185.1   \n",
       "3  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...  GCA_000007225.1   \n",
       "4  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...  GCA_000007305.1   \n",
       "\n",
       "  ncbi_assembly_level ncbi_assembly_name ncbi_bioproject ncbi_biosample  \\\n",
       "0     Complete Genome           ASM617v2     PRJNA224116   SAMN00000040   \n",
       "1     Complete Genome           ASM680v1     PRJNA224116   SAMN02604216   \n",
       "2     Complete Genome           ASM718v1        PRJNA294   SAMN02603235   \n",
       "3     Complete Genome           ASM722v1        PRJNA172   SAMN02604075   \n",
       "4     Complete Genome           ASM730v1     PRJNA224116   SAMN02604284   \n",
       "\n",
       "  ncbi_genbank_assembly_accession ncbi_genome_representation  \\\n",
       "0                 GCA_000006175.2                       full   \n",
       "1                 GCA_000006805.1                       full   \n",
       "2                 GCA_000007185.1                       full   \n",
       "3                 GCA_000007225.1                       full   \n",
       "4                 GCA_000007305.1                       full   \n",
       "\n",
       "                ncbi_organism_name   ncbi_refseq_category  \\\n",
       "0          Methanococcus voltae A3  representative genome   \n",
       "1    Halobacterium salinarum NRC-1  representative genome   \n",
       "2       Methanopyrus kandleri AV19                     na   \n",
       "3  Pyrobaculum aerophilum str. IM2                     na   \n",
       "4     Pyrococcus furiosus DSM 3638  representative genome   \n",
       "\n",
       "  parsed_genome_accession srr_accession  \n",
       "0         GCF_000006175.1           NaN  \n",
       "1         GCF_000006805.1           NaN  \n",
       "2         GCA_000007185.1           NaN  \n",
       "3         GCA_000007225.1           NaN  \n",
       "4         GCF_000007305.1           NaN  "
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bac = pd.read_csv('GTDBrepGenomes_withftp.csv')\n",
    "arc = pd.read_csv('GTDBrepGenomes_arc_withftpsrr.csv')\n",
    "arc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ncbi_assembly_name</th>\n",
       "      <th>ncbi_genbank_assembly_accession</th>\n",
       "      <th>ncbi_bioproject</th>\n",
       "      <th>ncbi_biosample</th>\n",
       "      <th>ncbi_assembly_level</th>\n",
       "      <th>ncbi_organism_name</th>\n",
       "      <th>ncbi_refseq_category</th>\n",
       "      <th>ncbi_genome_representation</th>\n",
       "      <th>parsed_genome_accession</th>\n",
       "      <th># assembly_accession</th>\n",
       "      <th>ftp_path</th>\n",
       "      <th>asm_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ASM582v2</td>\n",
       "      <td>GCA_000005825.2</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02603086</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>Bacillus pseudofirmus OF4</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>full</td>\n",
       "      <td>GCF_000005825.2</td>\n",
       "      <td>GCA_000005825.2</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>ASM582v2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ASM668v1</td>\n",
       "      <td>GCA_000006685.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02603994</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>Chlamydia muridarum str. Nigg</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>full</td>\n",
       "      <td>GCF_000006685.1</td>\n",
       "      <td>GCA_000006685.1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>ASM668v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ASM694v2</td>\n",
       "      <td>GCA_000006945.2</td>\n",
       "      <td>PRJNA57799</td>\n",
       "      <td>SAMN02604315</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>Salmonella enterica subsp. enterica serovar Ty...</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>full</td>\n",
       "      <td>GCF_000006945.2</td>\n",
       "      <td>GCA_000006945.2</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>ASM694v2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ASM696v1</td>\n",
       "      <td>GCA_000006965.1</td>\n",
       "      <td>PRJNA57603</td>\n",
       "      <td>SAMEA3283068</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>Sinorhizobium meliloti 1021</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>full</td>\n",
       "      <td>GCF_000006965.1</td>\n",
       "      <td>GCA_000006965.1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>ASM696v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ASM698v1</td>\n",
       "      <td>GCA_000006985.1</td>\n",
       "      <td>PRJNA57897</td>\n",
       "      <td>SAMN02604006</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>Chlorobaculum tepidum TLS</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>full</td>\n",
       "      <td>GCF_000006985.1</td>\n",
       "      <td>GCA_000006985.1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>ASM698v1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  ncbi_assembly_name ncbi_genbank_assembly_accession ncbi_bioproject  \\\n",
       "0           ASM582v2                 GCA_000005825.2     PRJNA224116   \n",
       "1           ASM668v1                 GCA_000006685.1     PRJNA224116   \n",
       "2           ASM694v2                 GCA_000006945.2      PRJNA57799   \n",
       "3           ASM696v1                 GCA_000006965.1      PRJNA57603   \n",
       "4           ASM698v1                 GCA_000006985.1      PRJNA57897   \n",
       "\n",
       "  ncbi_biosample ncbi_assembly_level  \\\n",
       "0   SAMN02603086     Complete Genome   \n",
       "1   SAMN02603994     Complete Genome   \n",
       "2   SAMN02604315     Complete Genome   \n",
       "3   SAMEA3283068     Complete Genome   \n",
       "4   SAMN02604006     Complete Genome   \n",
       "\n",
       "                                  ncbi_organism_name   ncbi_refseq_category  \\\n",
       "0                          Bacillus pseudofirmus OF4  representative genome   \n",
       "1                      Chlamydia muridarum str. Nigg  representative genome   \n",
       "2  Salmonella enterica subsp. enterica serovar Ty...       reference genome   \n",
       "3                        Sinorhizobium meliloti 1021       reference genome   \n",
       "4                          Chlorobaculum tepidum TLS       reference genome   \n",
       "\n",
       "  ncbi_genome_representation parsed_genome_accession # assembly_accession  \\\n",
       "0                       full         GCF_000005825.2      GCA_000005825.2   \n",
       "1                       full         GCF_000006685.1      GCA_000006685.1   \n",
       "2                       full         GCF_000006945.2      GCA_000006945.2   \n",
       "3                       full         GCF_000006965.1      GCA_000006965.1   \n",
       "4                       full         GCF_000006985.1      GCA_000006985.1   \n",
       "\n",
       "                                            ftp_path  asm_name  \n",
       "0  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...  ASM582v2  \n",
       "1  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...  ASM668v1  \n",
       "2  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...  ASM694v2  \n",
       "3  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...  ASM696v1  \n",
       "4  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...  ASM698v1  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bac.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23458\n",
      "1248\n",
      "24706\n"
     ]
    }
   ],
   "source": [
    "print(len(bac))\n",
    "print(len(arc))\n",
    "print(len(bac)+len(arc))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "394\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ncbi_genbank_assembly_accession</th>\n",
       "      <th>ncbi_assembly_name</th>\n",
       "      <th>ncbi_biosample</th>\n",
       "      <th>ncbi_organism_name</th>\n",
       "      <th>parsed_genome_accession</th>\n",
       "      <th>dataset</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000526435.1</td>\n",
       "      <td>ASM52643v1</td>\n",
       "      <td>SAMN02584936</td>\n",
       "      <td>Caldicoprobacter oshimai DSM 21659</td>\n",
       "      <td>GCF_000526435.1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000166775.1</td>\n",
       "      <td>ASM16677v1</td>\n",
       "      <td>SAMN00713565</td>\n",
       "      <td>Caldicellulosiruptor kronotskyensis 2002</td>\n",
       "      <td>GCF_000166775.1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000328765.2</td>\n",
       "      <td>ASM32876v2</td>\n",
       "      <td>SAMEA2272437</td>\n",
       "      <td>Tepidanaerobacter acetatoxydans Re1</td>\n",
       "      <td>GCF_000328765.2</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000499205.1</td>\n",
       "      <td>MAEPY2 1.0</td>\n",
       "      <td>SAMN02343193</td>\n",
       "      <td>Paenibacillus sp. MAEPY2</td>\n",
       "      <td>GCF_000499205.1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_001730225.1</td>\n",
       "      <td>ASM173022v1</td>\n",
       "      <td>SAMN05731213</td>\n",
       "      <td>Desulfuribacillus alkaliarsenatis</td>\n",
       "      <td>GCF_001730225.1</td>\n",
       "      <td>train</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  ncbi_genbank_assembly_accession ncbi_assembly_name ncbi_biosample  \\\n",
       "0                 GCA_000526435.1         ASM52643v1   SAMN02584936   \n",
       "1                 GCA_000166775.1         ASM16677v1   SAMN00713565   \n",
       "2                 GCA_000328765.2         ASM32876v2   SAMEA2272437   \n",
       "3                 GCA_000499205.1         MAEPY2 1.0   SAMN02343193   \n",
       "4                 GCA_001730225.1        ASM173022v1   SAMN05731213   \n",
       "\n",
       "                         ncbi_organism_name parsed_genome_accession dataset  \n",
       "0        Caldicoprobacter oshimai DSM 21659         GCF_000526435.1   train  \n",
       "1  Caldicellulosiruptor kronotskyensis 2002         GCF_000166775.1   train  \n",
       "2       Tepidanaerobacter acetatoxydans Re1         GCF_000328765.2   train  \n",
       "3                  Paenibacillus sp. MAEPY2         GCF_000499205.1   train  \n",
       "4         Desulfuribacillus alkaliarsenatis         GCF_001730225.1   train  "
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gclass = pd.read_csv('GenomeIDs_parsedclass.csv')\n",
    "print(len(gclass))\n",
    "gclass.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24706\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/adrienne/anaconda3/envs/fastai/lib/python3.7/site-packages/pandas/core/frame.py:6692: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  sort=sort)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th># assembly_accession</th>\n",
       "      <th>asm_name</th>\n",
       "      <th>ftp_path</th>\n",
       "      <th>index</th>\n",
       "      <th>ncbi_assembly_level</th>\n",
       "      <th>ncbi_assembly_name</th>\n",
       "      <th>ncbi_bioproject</th>\n",
       "      <th>ncbi_biosample</th>\n",
       "      <th>ncbi_genbank_assembly_accession</th>\n",
       "      <th>ncbi_genome_representation</th>\n",
       "      <th>ncbi_organism_name</th>\n",
       "      <th>ncbi_refseq_category</th>\n",
       "      <th>parsed_genome_accession</th>\n",
       "      <th>srr_accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000005825.2</td>\n",
       "      <td>ASM582v2</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM582v2</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02603086</td>\n",
       "      <td>GCA_000005825.2</td>\n",
       "      <td>full</td>\n",
       "      <td>Bacillus pseudofirmus OF4</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000005825.2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000006685.1</td>\n",
       "      <td>ASM668v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM668v1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02603994</td>\n",
       "      <td>GCA_000006685.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Chlamydia muridarum str. Nigg</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>GCF_000006685.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000006945.2</td>\n",
       "      <td>ASM694v2</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM694v2</td>\n",
       "      <td>PRJNA57799</td>\n",
       "      <td>SAMN02604315</td>\n",
       "      <td>GCA_000006945.2</td>\n",
       "      <td>full</td>\n",
       "      <td>Salmonella enterica subsp. enterica serovar Ty...</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>GCF_000006945.2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000006965.1</td>\n",
       "      <td>ASM696v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM696v1</td>\n",
       "      <td>PRJNA57603</td>\n",
       "      <td>SAMEA3283068</td>\n",
       "      <td>GCA_000006965.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Sinorhizobium meliloti 1021</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>GCF_000006965.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000006985.1</td>\n",
       "      <td>ASM698v1</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM698v1</td>\n",
       "      <td>PRJNA57897</td>\n",
       "      <td>SAMN02604006</td>\n",
       "      <td>GCA_000006985.1</td>\n",
       "      <td>full</td>\n",
       "      <td>Chlorobaculum tepidum TLS</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>GCF_000006985.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  # assembly_accession  asm_name  \\\n",
       "0      GCA_000005825.2  ASM582v2   \n",
       "1      GCA_000006685.1  ASM668v1   \n",
       "2      GCA_000006945.2  ASM694v2   \n",
       "3      GCA_000006965.1  ASM696v1   \n",
       "4      GCA_000006985.1  ASM698v1   \n",
       "\n",
       "                                            ftp_path index  \\\n",
       "0  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...   NaN   \n",
       "1  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...   NaN   \n",
       "2  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...   NaN   \n",
       "3  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...   NaN   \n",
       "4  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...   NaN   \n",
       "\n",
       "  ncbi_assembly_level ncbi_assembly_name ncbi_bioproject ncbi_biosample  \\\n",
       "0     Complete Genome           ASM582v2     PRJNA224116   SAMN02603086   \n",
       "1     Complete Genome           ASM668v1     PRJNA224116   SAMN02603994   \n",
       "2     Complete Genome           ASM694v2      PRJNA57799   SAMN02604315   \n",
       "3     Complete Genome           ASM696v1      PRJNA57603   SAMEA3283068   \n",
       "4     Complete Genome           ASM698v1      PRJNA57897   SAMN02604006   \n",
       "\n",
       "  ncbi_genbank_assembly_accession ncbi_genome_representation  \\\n",
       "0                 GCA_000005825.2                       full   \n",
       "1                 GCA_000006685.1                       full   \n",
       "2                 GCA_000006945.2                       full   \n",
       "3                 GCA_000006965.1                       full   \n",
       "4                 GCA_000006985.1                       full   \n",
       "\n",
       "                                  ncbi_organism_name   ncbi_refseq_category  \\\n",
       "0                          Bacillus pseudofirmus OF4  representative genome   \n",
       "1                      Chlamydia muridarum str. Nigg  representative genome   \n",
       "2  Salmonella enterica subsp. enterica serovar Ty...       reference genome   \n",
       "3                        Sinorhizobium meliloti 1021       reference genome   \n",
       "4                          Chlorobaculum tepidum TLS       reference genome   \n",
       "\n",
       "  parsed_genome_accession srr_accession  \n",
       "0         GCF_000005825.2           NaN  \n",
       "1         GCF_000006685.1           NaN  \n",
       "2         GCF_000006945.2           NaN  \n",
       "3         GCF_000006965.1           NaN  \n",
       "4         GCF_000006985.1           NaN  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gall = bac.append(arc)\n",
    "print(len(gall))\n",
    "gall.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "gclass1 = gclass.merge(gall[['parsed_genome_accession','ftp_path','srr_accession']],on='parsed_genome_accession')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "394\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ncbi_genbank_assembly_accession</th>\n",
       "      <th>ncbi_assembly_name</th>\n",
       "      <th>ncbi_biosample</th>\n",
       "      <th>ncbi_organism_name</th>\n",
       "      <th>parsed_genome_accession</th>\n",
       "      <th>dataset</th>\n",
       "      <th>ftp_path</th>\n",
       "      <th>srr_accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000526435.1</td>\n",
       "      <td>ASM52643v1</td>\n",
       "      <td>SAMN02584936</td>\n",
       "      <td>Caldicoprobacter oshimai DSM 21659</td>\n",
       "      <td>GCF_000526435.1</td>\n",
       "      <td>train</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000166775.1</td>\n",
       "      <td>ASM16677v1</td>\n",
       "      <td>SAMN00713565</td>\n",
       "      <td>Caldicellulosiruptor kronotskyensis 2002</td>\n",
       "      <td>GCF_000166775.1</td>\n",
       "      <td>train</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000328765.2</td>\n",
       "      <td>ASM32876v2</td>\n",
       "      <td>SAMEA2272437</td>\n",
       "      <td>Tepidanaerobacter acetatoxydans Re1</td>\n",
       "      <td>GCF_000328765.2</td>\n",
       "      <td>train</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000499205.1</td>\n",
       "      <td>MAEPY2 1.0</td>\n",
       "      <td>SAMN02343193</td>\n",
       "      <td>Paenibacillus sp. MAEPY2</td>\n",
       "      <td>GCF_000499205.1</td>\n",
       "      <td>train</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_001730225.1</td>\n",
       "      <td>ASM173022v1</td>\n",
       "      <td>SAMN05731213</td>\n",
       "      <td>Desulfuribacillus alkaliarsenatis</td>\n",
       "      <td>GCF_001730225.1</td>\n",
       "      <td>train</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/001...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  ncbi_genbank_assembly_accession ncbi_assembly_name ncbi_biosample  \\\n",
       "0                 GCA_000526435.1         ASM52643v1   SAMN02584936   \n",
       "1                 GCA_000166775.1         ASM16677v1   SAMN00713565   \n",
       "2                 GCA_000328765.2         ASM32876v2   SAMEA2272437   \n",
       "3                 GCA_000499205.1         MAEPY2 1.0   SAMN02343193   \n",
       "4                 GCA_001730225.1        ASM173022v1   SAMN05731213   \n",
       "\n",
       "                         ncbi_organism_name parsed_genome_accession dataset  \\\n",
       "0        Caldicoprobacter oshimai DSM 21659         GCF_000526435.1   train   \n",
       "1  Caldicellulosiruptor kronotskyensis 2002         GCF_000166775.1   train   \n",
       "2       Tepidanaerobacter acetatoxydans Re1         GCF_000328765.2   train   \n",
       "3                  Paenibacillus sp. MAEPY2         GCF_000499205.1   train   \n",
       "4         Desulfuribacillus alkaliarsenatis         GCF_001730225.1   train   \n",
       "\n",
       "                                            ftp_path srr_accession  \n",
       "0  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...           NaN  \n",
       "1  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...           NaN  \n",
       "2  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...           NaN  \n",
       "3  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...           NaN  \n",
       "4  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/001...           NaN  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(gclass1))\n",
    "gclass1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/526/435/GCA_000526435.1_ASM52643v1'"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gclass1['ftp_path'][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ncbi_genbank_assembly_accession</th>\n",
       "      <th>ncbi_assembly_name</th>\n",
       "      <th>ncbi_biosample</th>\n",
       "      <th>ncbi_organism_name</th>\n",
       "      <th>parsed_genome_accession</th>\n",
       "      <th>dataset</th>\n",
       "      <th>ftp_path</th>\n",
       "      <th>srr_accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>304</th>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA8915</td>\n",
       "      <td>train</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR10597681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>325</th>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10154</td>\n",
       "      <td>train</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR10597694</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328</th>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10219</td>\n",
       "      <td>train</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR10597634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>383</th>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10536</td>\n",
       "      <td>test</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR10597626</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    ncbi_genbank_assembly_accession ncbi_assembly_name ncbi_biosample  \\\n",
       "304                            none               none           none   \n",
       "325                            none               none           none   \n",
       "328                            none               none           none   \n",
       "383                            none               none           none   \n",
       "\n",
       "    ncbi_organism_name parsed_genome_accession dataset ftp_path srr_accession  \n",
       "304               none                 UBA8915   train      NaN   SRR10597681  \n",
       "325               none                UBA10154   train      NaN   SRR10597694  \n",
       "328               none                UBA10219   train      NaN   SRR10597634  \n",
       "383               none                UBA10536    test      NaN   SRR10597626  "
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gclass1[gclass1['ftp_path'].isna()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/499/205/GCA_000499205.1_MAEPY2_1.0 ['MAEPY2', '1.0']\n",
      "8 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/111/575/GCA_900111575.1_IMG-taxon_2602042032_annotated_assembly ['IMG-taxon', '2602042032', 'annotated', 'assembly']\n",
      "15 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/016/865/GCA_900016865.1_Clostridia_bin_genome_5 ['Clostridia', 'bin', 'genome', '5']\n",
      "24 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/104/055/GCA_900104055.1_IMG-taxon_2623620517_annotated_assembly ['IMG-taxon', '2623620517', 'annotated', 'assembly']\n",
      "34 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/111/505/GCA_900111505.1_IMG-taxon_2642422559_annotated_assembly ['IMG-taxon', '2642422559', 'annotated', 'assembly']\n",
      "38 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/313/115/GCA_900313115.1_Rumen_uncultured_genome_RUG770 ['Rumen', 'uncultured', 'genome', 'RUG770']\n",
      "105 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/353/875/GCA_000353875.1__Ca._Caldatribacterium_californiense_ ['\"Ca.', 'Caldatribacterium', 'californiense\"']\n",
      "110 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/105/685/GCA_900105685.1_IMG-taxon_2690315640_annotated_assembly ['IMG-taxon', '2690315640', 'annotated', 'assembly']\n",
      "111 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/320/225/GCA_900320225.1_Rumen_uncultured_genome_RUG572 ['Rumen', 'uncultured', 'genome', 'RUG572']\n",
      "129 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/112/165/GCA_900112165.1_IMG-taxon_2597490362_annotated_assembly ['IMG-taxon', '2597490362', 'annotated', 'assembly']\n",
      "139 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/001/536/065/GCA_001536065.1_JGI_assembly ['JGI', 'assembly']\n",
      "151 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/177/705/GCA_900177705.1_IMG-taxon_2718218506_annotated_assembly ['IMG-taxon', '2718218506', 'annotated', 'assembly']\n",
      "172 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/494/225/GCA_000494225.1_Calescamantes_bacterium_OTU_1 ['Calescamantes', 'bacterium', 'OTU', '1']\n",
      "198 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/469/585/GCA_000469585.1_Chrysiogenes_arsenatis ['Chrysiogenes', 'arsenatis']\n",
      "296 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/321/865/GCA_900321865.1_Rumen_uncultured_genome_RUG730 ['Rumen', 'uncultured', 'genome', 'RUG730']\n",
      "332 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/101/685/GCA_900101685.1_IMG-taxon_2651870117_annotated_assembly ['IMG-taxon', '2651870117', 'annotated', 'assembly']\n",
      "334 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/091/955/GCA_900091955.1_IMG-taxon_2657245721_annotated_assembly ['IMG-taxon', '2657245721', 'annotated', 'assembly']\n",
      "338 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/094/555/GCA_900094555.1_IMG-taxon_2615840609_annotated_assembly ['IMG-taxon', '2615840609', 'annotated', 'assembly']\n",
      "342 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/001/464/655/GCA_001464655.1_Rhizobiales_bacterium_Ga0077555 ['Rhizobiales', 'bacterium', 'Ga0077555']\n",
      "371 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/316/005/GCA_900316005.1_Rumen_uncultured_genome_RUG152 ['Rumen', 'uncultured', 'genome', 'RUG152']\n",
      "377 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/102/545/GCA_900102545.1_IMG-taxon_2634166296_annotated_assembly ['IMG-taxon', '2634166296', 'annotated', 'assembly']\n",
      "380 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/114/835/GCA_900114835.1_IMG-taxon_2642422540_annotated_assembly ['IMG-taxon', '2642422540', 'annotated', 'assembly']\n",
      "390 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/143/675/GCA_900143675.1_Candidatus_Nitrosotalea_sinensis_Nd2_whole_genome ['Candidatus', 'Nitrosotalea', 'sinensis', 'Nd2', 'whole', 'genome']\n"
     ]
    }
   ],
   "source": [
    "for ix, asm in gclass1.iterrows():\n",
    "    split = str(asm['ncbi_assembly_name']).split(' ')\n",
    "    if len(split) > 1:\n",
    "        print(ix, asm['ftp_path'], split)\n",
    "        #join by an underscore instead\n",
    "        gclass1.at[ix,'ncbi_assembly_name'] = '_'.join(split)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/003/456/095/GCA_003456095.1_ASM345609v1'"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gclass1[gclass1['ncbi_genbank_assembly_accession']=='none']['ftp_path'][6]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "#gclass1[gclass1['ncbi_genbank_assembly_accession']=='none']\n",
    "for ix, row in gclass1[gclass1['ncbi_genbank_assembly_accession']=='none'].iterrows():\n",
    "    gclass1.at[ix,'ncbi_genbank_assembly_accession'] = row['parsed_genome_accession']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'float' object has no attribute 'split'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-67-1743e8416f01>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mix\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrow\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mgclass1\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mgclass1\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'ncbi_assembly_name'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m==\u001b[0m\u001b[0;34m'none'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miterrows\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 4\u001b[0;31m     \u001b[0mgclass1\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mat\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mix\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'ncbi_assembly_name'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'ftp_path'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'_'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m: 'float' object has no attribute 'split'"
     ]
    }
   ],
   "source": [
    "gclass1.at[365,'ncbi_assembly_name'] = gclass1['ftp_path'][365].split('_')[-1]\n",
    "\n",
    "for ix, row in gclass1[gclass1['ncbi_assembly_name']=='none'].iterrows():\n",
    "    gclass1.at[ix,'ncbi_assembly_name'] = row['ftp_path'].split('_')[-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ncbi_genbank_assembly_accession</th>\n",
       "      <th>ncbi_assembly_name</th>\n",
       "      <th>ncbi_biosample</th>\n",
       "      <th>ncbi_organism_name</th>\n",
       "      <th>parsed_genome_accession</th>\n",
       "      <th>dataset</th>\n",
       "      <th>ftp_path</th>\n",
       "      <th>srr_accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>304</th>\n",
       "      <td>UBA8915</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA8915</td>\n",
       "      <td>train</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR10597681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>325</th>\n",
       "      <td>UBA10154</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10154</td>\n",
       "      <td>train</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR10597694</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328</th>\n",
       "      <td>UBA10219</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10219</td>\n",
       "      <td>train</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR10597634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>383</th>\n",
       "      <td>UBA10536</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>UBA10536</td>\n",
       "      <td>test</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRR10597626</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    ncbi_genbank_assembly_accession ncbi_assembly_name ncbi_biosample  \\\n",
       "304                         UBA8915               none           none   \n",
       "325                        UBA10154               none           none   \n",
       "328                        UBA10219               none           none   \n",
       "383                        UBA10536               none           none   \n",
       "\n",
       "    ncbi_organism_name parsed_genome_accession dataset ftp_path srr_accession  \n",
       "304               none                 UBA8915   train      NaN   SRR10597681  \n",
       "325               none                UBA10154   train      NaN   SRR10597694  \n",
       "328               none                UBA10219   train      NaN   SRR10597634  \n",
       "383               none                UBA10536    test      NaN   SRR10597626  "
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gclass1[gclass1['ncbi_assembly_name']=='none']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#errors\n",
    "105 , ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/353/875/GCA_000353875.1__Ca._Caldatribacterium_californiense_/GCA_000353875.1_\"Ca._Caldatribacterium_californiense\"_genomic.fna.gz\n",
    "368 , ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/066/385/GCA_900066385.1_13414_6_69/GCA_900066385.1_13414_6#69_genomic.fna.gz\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "gclass1.at[105,'ncbi_assembly_name'] = '_Ca._Caldatribacterium_californiense_'\n",
    "gclass1.at[368,'ncbi_assembly_name'] = '13414_6_69'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "gclass1.to_csv('GenomeIDs_parsedclass_withftp.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# create filename files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>ncbi_genbank_assembly_accession</th>\n",
       "      <th>ncbi_assembly_name</th>\n",
       "      <th>ncbi_biosample</th>\n",
       "      <th>ncbi_organism_name</th>\n",
       "      <th>parsed_genome_accession</th>\n",
       "      <th>dataset</th>\n",
       "      <th>ftp_path</th>\n",
       "      <th>srr_accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>GCA_000526435.1</td>\n",
       "      <td>ASM52643v1</td>\n",
       "      <td>SAMN02584936</td>\n",
       "      <td>Caldicoprobacter oshimai DSM 21659</td>\n",
       "      <td>GCF_000526435.1</td>\n",
       "      <td>train</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>GCA_000166775.1</td>\n",
       "      <td>ASM16677v1</td>\n",
       "      <td>SAMN00713565</td>\n",
       "      <td>Caldicellulosiruptor kronotskyensis 2002</td>\n",
       "      <td>GCF_000166775.1</td>\n",
       "      <td>train</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>GCA_000328765.2</td>\n",
       "      <td>ASM32876v2</td>\n",
       "      <td>SAMEA2272437</td>\n",
       "      <td>Tepidanaerobacter acetatoxydans Re1</td>\n",
       "      <td>GCF_000328765.2</td>\n",
       "      <td>train</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>GCA_000499205.1</td>\n",
       "      <td>MAEPY2_1.0</td>\n",
       "      <td>SAMN02343193</td>\n",
       "      <td>Paenibacillus sp. MAEPY2</td>\n",
       "      <td>GCF_000499205.1</td>\n",
       "      <td>train</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>GCA_001730225.1</td>\n",
       "      <td>ASM173022v1</td>\n",
       "      <td>SAMN05731213</td>\n",
       "      <td>Desulfuribacillus alkaliarsenatis</td>\n",
       "      <td>GCF_001730225.1</td>\n",
       "      <td>train</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/001...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0 ncbi_genbank_assembly_accession ncbi_assembly_name  \\\n",
       "0           0                 GCA_000526435.1         ASM52643v1   \n",
       "1           1                 GCA_000166775.1         ASM16677v1   \n",
       "2           2                 GCA_000328765.2         ASM32876v2   \n",
       "3           3                 GCA_000499205.1         MAEPY2_1.0   \n",
       "4           4                 GCA_001730225.1        ASM173022v1   \n",
       "\n",
       "  ncbi_biosample                        ncbi_organism_name  \\\n",
       "0   SAMN02584936        Caldicoprobacter oshimai DSM 21659   \n",
       "1   SAMN00713565  Caldicellulosiruptor kronotskyensis 2002   \n",
       "2   SAMEA2272437       Tepidanaerobacter acetatoxydans Re1   \n",
       "3   SAMN02343193                  Paenibacillus sp. MAEPY2   \n",
       "4   SAMN05731213         Desulfuribacillus alkaliarsenatis   \n",
       "\n",
       "  parsed_genome_accession dataset  \\\n",
       "0         GCF_000526435.1   train   \n",
       "1         GCF_000166775.1   train   \n",
       "2         GCF_000328765.2   train   \n",
       "3         GCF_000499205.1   train   \n",
       "4         GCF_001730225.1   train   \n",
       "\n",
       "                                            ftp_path srr_accession  \n",
       "0  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...           NaN  \n",
       "1  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...           NaN  \n",
       "2  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...           NaN  \n",
       "3  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...           NaN  \n",
       "4  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/001...           NaN  "
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ids = pd.read_csv('GenomeIDs_parsedclass_withftp.csv')\n",
    "ids.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "train = ids[ids['dataset']=='train']\n",
    "valid = ids[ids['dataset']=='valid']\n",
    "test = ids[ids['dataset']=='test']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32 32\n",
      "32 32\n",
      "330 330\n"
     ]
    }
   ],
   "source": [
    "test_filenames = get_filenames(test)\n",
    "print(len(test), len(test_filenames))\n",
    "valid_filenames = get_filenames(valid)\n",
    "print(len(valid),len(valid_filenames))\n",
    "train_filenames = get_filenames(train)\n",
    "print(len(train),len(train_filenames))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "save_filenames('trainnew_parsedclass_filenames.txt',train_filenames)\n",
    "save_filenames('validnew_parsedclass_filenames.txt',valid_filenames)\n",
    "save_filenames('testnew_parsedclass_filenames.txt',test_filenames)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[PosixPath('/Users/adrienne/Projects/LanguageofLife/GetGTDBGenomes/bac120_r89.tree'),\n",
       " PosixPath('/Users/adrienne/Projects/LanguageofLife/GetGTDBGenomes/ar122_r89.tree'),\n",
       " PosixPath('/Users/adrienne/Projects/LanguageofLife/GetGTDBGenomes/.DS_Store'),\n",
       " PosixPath('/Users/adrienne/Projects/LanguageofLife/GetGTDBGenomes/GTDBrepGenomes_withftp.csv'),\n",
       " PosixPath('/Users/adrienne/Projects/LanguageofLife/GetGTDBGenomes/GTDBrepGenomes_arc_withftpsrr.csv'),\n",
       " PosixPath('/Users/adrienne/Projects/LanguageofLife/GetGTDBGenomes/trainnew_parsedclass_filenames.txt'),\n",
       " PosixPath('/Users/adrienne/Projects/LanguageofLife/GetGTDBGenomes/testnew_parsedclass_filenames.txt'),\n",
       " PosixPath('/Users/adrienne/Projects/LanguageofLife/GetGTDBGenomes/validnew_parsedclass_filenames.txt'),\n",
       " PosixPath('/Users/adrienne/Projects/LanguageofLife/GetGTDBGenomes/bac120_metadata.tsv'),\n",
       " PosixPath('/Users/adrienne/Projects/LanguageofLife/GetGTDBGenomes/EvenEnv100_metagenomes.csv'),\n",
       " PosixPath('/Users/adrienne/Projects/LanguageofLife/GetGTDBGenomes/GetTrainValidTestSplit_GTDBrepGenomes.ipynb'),\n",
       " PosixPath('/Users/adrienne/Projects/LanguageofLife/GetGTDBGenomes/GetGTDBGenomes.ipynb'),\n",
       " PosixPath('/Users/adrienne/Projects/LanguageofLife/GetGTDBGenomes/GenomeIDs_parsedclass_withftp.csv'),\n",
       " PosixPath('/Users/adrienne/Projects/LanguageofLife/GetGTDBGenomes/ar122_metadata_r89.tsv'),\n",
       " PosixPath('/Users/adrienne/Projects/LanguageofLife/GetGTDBGenomes/.ipynb_checkpoints'),\n",
       " PosixPath('/Users/adrienne/Projects/LanguageofLife/GetGTDBGenomes/GenomeIDs_parsedclass.csv'),\n",
       " PosixPath('/Users/adrienne/Projects/LanguageofLife/GetGTDBGenomes/GenomeIDs_parsedorder.csv')]"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pathlib import Path\n",
    "files = [x for x in Path('.').resolve().iterdir()]\n",
    "files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'bac' in files[0].name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'bac120'"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files[0].name.split('_r89.tree')[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['bac120_r89', 'tree']"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files[0].name.split('.')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
